
            <p>Gordon Smyth<br />
            August 16, 2005</p>

            <p>Walter and Eliza Hall Institute of
            Medical Research</p>

            <p><img name="_x0000_i1026" src="images/wehilogo.jpg" height="62" width="58" /></p>

            <h2>Aims</h2>

            <p>This laboratory explores some of the features of the
            limma package for assessing differential expression in
            microarray experiments. Examples are included of cDNA
            two-color microarrays and Affymetrix one-channel
            microarrays. Some pre-processing issues are also
            discussed for two-color arrays.</p>

            <h2>Lab exercises</h2>

            <p><br /></p>

            <table>
              <tbody>
                <tr>
                  <td><b>Exercise</b></td>

                  <td><b>Platform</b></td>

                  <td><b>Design</b></td>

                  <td><b>Topics covered</b></td>
                </tr>

                <tr>
                  <td><a href="ApoAI/">apoAI data</a></td>

                  <td>cDNA</td>

                  <td>Two group comparison with common
                  reference</td>

                  <td>Introduction to linear models. Obtaining
                  empirical Bayes statistics. Getting lists of
                  differentially expressed genes.</td>
                </tr>

                <tr>
                  <td><a href="beta7/">integrin beta7
                  data</a></td>

                  <td>cDNA</td>

                  <td>Direct comparisons with dye-swaps</td>

                  <td>Data entry for two color data. Highlighting
                  control probes. Exploring different background
                  correction methods. Allowing for genewise dye
                  effects.</td>
                </tr>

                <tr>
                  <td><a href="estrogen/">Estrogen
                  data</a></td>

                  <td>Affymetrix</td>

                  <td>2x2 Factorial</td>

                  <td>More on linear models. Use of contrasts. Venn
                  diagrams. Linking gene lists to annotation
                  information on the internet. Gene set tests.</td>
                </tr>

                <tr>
                  <td><a href="drosEmbryo/">Drosophila
                  embryogenesis dataset</a></td>

                  <td>Affymetrix</td>

                  <td>Time course with series-level
                  replication</td>

                  <td>Time course analysis using linear models and
                  moderated F-statistics.</td>
                </tr>
              </tbody>
            </table>

            <h2>Datasets used in the exercises</h2>

            <p>Please check whether you already have the
            "Drosophila Embryo" and "Estrogen" packages from the
            Required Software. (These data sets are stored within R
            packages.)</p>

            <ul>
              <li><a href="Data/apoai.zip">ApoAI Knockout Data</a>
              (1.2 MB)</li>

              <li><a href="Data/integrinbeta7.zip">Integrin beta7
              data</a> (14.6 MB)</li>

              <li><a href="Data/estrogen.zip">Estrogen Data</a>
              (23.6 MB)</li>

              <li>Drosophila Embryo Data (as an R package):
              <a href="Data/drosEmbryo_1.0.zip">Windows</a> -
              <a href="Data/drosEmbryo_1.0.tar.gz">Source (Mac or
              Linux)</a> (3.5 MB)</li>
            </ul>

            <h2>Lab Handouts<br /></h2>

            <p>The handouts in pdf can be accessed here: <a href="printing"><b>lab
            handouts</b></a><br /></p>

            <h2>Required R packages</h2>

            <p>Please install these packages before attempting to
            repeat the lab exercises. Note: if you don't have write
            permission to the system library directory of your R
            installation, you can use the .libPaths() function with
            something like <code>.libPaths("C:/mylibdir")</code>
            before you run <code>install.packages()</code> (or
            equivalent) to install the packages in a customized
            directory location.</p>

            <p>A local cache of packages is here: Local Package Cache</p>

            <table>
              <tbody>
                <tr>
                  <td>
                    <p>Package</p>
                  </td>

                  <td>
                    <p>Windows</p>
                  </td>

                  <td>
                    <p>MacOS X</p>
                  </td>

                  <td>
                    <p>Source</p>
                  </td>
                </tr>

                <tr>
                  <td>
                    <p>limma_2.0.4</p>
                  </td>

                  <td>
                    <p><a href="http://cran.at.r-project.org/bin/windows/contrib/r-release/limma_2.0.4.zip">
                    limma_2.0.4.zip</a></p>
                  </td>

                  <td>
                    <p><a href="http://cran.at.r-project.org/src/contrib/limma_2.0.4.tar.gz">
                    limma_2.0.4.tar.gz</a></p>
                  </td>

                  <td>
                    <p><a href="http://cran.at.r-project.org/src/contrib/limma_2.0.4.tar.gz">
                    limma_2.0.4.tar.gz</a></p>
                  </td>
                </tr>

                <tr>
                  <td>
                    <p>statmod_1.2.0</p>
                  </td>

                  <td>
                    <p><a href="http://cran.at.r-project.org/bin/windows/contrib/r-release/statmod_1.2.0.zip">
                    statmod_1.2.0.zip</a></p>
                  </td>

                  <td>
                    <p><a href="http://cran.at.r-project.org/src/contrib/statmod_1.2.0.tar.gz">
                    statmod_1.2.0.tar.gz</a></p>
                  </td>

                  <td>
                    <p><a href="http://cran.at.r-project.org/src/contrib/statmod_1.2.0.tar.gz">
                    statmod_1.2.0.tar.gz</a></p>
                  </td>
                </tr>

                <tr>
                  <td>
                    <p>affy_1.6.7</p>
                  </td>

                  <td>
                    <p><a href="/packages/bioc/stable/src/contrib/Win32/affy_1.6.7.zip">
                    affy_1.6.7.zip</a></p>
                  </td>

                  <td>
                    <p><a href="http://bioconductor.org/bin/macosx/2.1/affy_1.6.7.tgz">
                    affy_1.6.7.tgz</a></p>
                  </td>

                  <td>
                    <p><a href="/packages/bioc/stable/src/contrib/Source/affy_1.6.7.tar.gz">
                    affy_1.6.7.tar.gz</a></p>
                  </td>
                </tr>

                <tr>
                  <td>
                    <p>Biobase_1.5.12</p>
                  </td>

                  <td>
                    <p><a href="/packages/bioc/stable/src/contrib/Win32/Biobase_1.5.12.zip">
                    Biobase_1.5.12.zip</a></p>
                  </td>

                  <td>
                    <p><a href="http://bioconductor.org/bin/macosx/2.1/Biobase_1.5.12.tgz">
                    Biobase_1.5.12.tgz</a></p>
                  </td>

                  <td>
                    <p><a href="/packages/bioc/stable/src/contrib/Source/Biobase_1.5.12.tar.gz">
                    Biobase_1.5.12.tar.gz</a></p>
                  </td>
                </tr>

                <tr>
                  <td>
                    <p>hgu95av2_1.8.4</p>
                  </td>

                  <td>
                    <p><a href="/packages/data/annotation/stable/src/contrib/Win32/hgu95av2_1.8.4.zip">
                    hgu95av2_1.8.4.zip</a></p>
                  </td>

                  <td>
                    <p><a href="/packages/data/annotation/stable/src/contrib/Source/hgu95av2_1.8.4.tar.gz">
                    hgu95av2_1.8.4.tar.gz</a></p>
                  </td>

                  <td>
                    <p><a href="/packages/data/annotation/stable/src/contrib/Source/hgu95av2_1.8.4.tar.gz">
                    hgu95av2_1.8.4.tar.gz</a></p>
                  </td>
                </tr>

                <tr>
                  <td>
                    <p>hgu95av2cdf_1.5.1</p>
                  </td>

                  <td>
                    <p><a href="/packages/data/annotation/stable/src/contrib/Win32/hgu95av2cdf_1.5.1.zip">
                    hgu95av2cdf_1.5.1.zip</a></p>
                  </td>

                  <td>
                    <p><a href="/packages/data/annotation/stable/src/contrib/Source/hgu95av2cdf_1.5.1.tar.gz">
                    hgu95av2cdf_1.5.1.tar.gz</a></p>
                  </td>

                  <td>
                    <p><a href="/packages/data/annotation/stable/src/contrib/Source/hgu95av2cdf_1.5.1.tar.gz">
                    hgu95av2cdf_1.5.1.tar.gz</a></p>
                  </td>
                </tr>

                <tr>
                  <td>
                    <p>xtable_1.2-5</p>
                  </td>

                  <td>
                    <p><a href="/packages/bioc/stable/src/contrib/Win32/xtable_1.2-5.zip">
                    xtable_1.2-5.zip</a></p>
                  </td>

                  <td>
                    <p><a href="http://bioconductor.org/bin/macosx/2.1/xtable_1.2-5.tgz">
                    xtable_1.2-5.tgz</a></p>
                  </td>

                  <td>
                    <p><a href="/packages/bioc/stable/src/contrib/Source/xtable_1.2-5.tar.gz">
                    xtable_1.2-5.tar.gz</a></p>
                  </td>
                </tr>
              </tbody>
            </table>

            <h2>Getting started</h2>

            <p>You should be running R 2.1.0 or 2.1.1 and limma
            2.0.4. A good way to get started is to open up the
            Limma User's Guide:</p>
            <pre>
 library(limma)
</pre>

            <p>If you're using Windows, just use the drop-down menu
            "Vignettes". Otherwise, type</p>
            <pre>
 limmaUsersGuide()
</pre>

            <h2>References</h2>

            <ol start="1" type="1">
              <li>Smyth, G. K., Thorne, N. P. and Wettenhall J.
              (2005) limma: Linear Models for Microarray Data
              User's Guide. <a href="http://bioinf.wehi.edu.au/limma">http://bioinf.wehi.edu.au/limma</a>
              (Included as part of the limma package.)</li>

              <li>Smyth, G. K. (2005). Limma: linear models for
              microarray data. In: <i>Bioinformatics and
              Computational Biology Solutions using R and
              Bioconductor</i>, R. Gentleman, V. Carey, S. Dudoit,
              R. Irizarry, W. Huber (eds.), Springer, New
              York.</li>

              <li>Smyth, G. K. (2004). Linear models and empirical
              Bayes methods for assessing differential expression
              in microarray experiments. <i>Statistical
              Applications in Genetics and Molecular Biology</i>
              <b>3</b>, No. 1, Article 3. <a href="http://www.bepress.com/sagmb/vol3/iss1/art3/">http://www.bepress.com/sagmb/vol3/iss1/art3/</a></li>
            </ol>
