### Morning Talks ###

* [The Human Microbiome](TheHumanMicrobiome.pdf) - David Fredricks,
  Fred Hutchinson Cancer Research Center.
* [Better Maps of Disease: Building a commons for evolving generative models of
  disease](Friend0729.pdf) - Stephen Friend, SAGE Bionetworks.
* [Next Generation Mendelian Genetics by Exome Sequencing](
  Shendure_20100729.pdf) - Jay Shendure, University of Washington.
* Exploring Genomic Medicine Through Integrative Bioinformatics - 
  Atul Butte, Stanford Center for Biomedical Informatics Research.
* [Sparse partitioning: a tool for analyzing  regression problems with many
  tertiary predictors](Tavare-BioC2010.pdf) - Simon Tavare, University of
  Southern California.
* [Generation gap: How existing bioinformatics resources are adapting to
  high-throughput sequencing](Flicek_20100730.pdf) - Paul Flicek, European
  Bioinformatics Institute.
* Statistical Challenges in transcriptome sequencing - Lior Pachter,
  University of California, Berkeley.
* [Differential expression analysis for sequencing count data](
  2010-07-29_DESeq_Bioc-Meeting_Seattle.pdf) - Simon Anders,
  European Bioinformatics Institute.

### Afternoon Labs ###

#### Introduction to R ####
Instructor: Chao-Jen Wong

* [Introduction to R](IntroToR.pdf)
* [First Steps with R](First_Steps_With_R_SOLUTIONS.pdf)
* [Getting Started with Lattice](Basic_Lattice_SOLUTIONS.pdf)

Install command:

`source("http://bioconductor.org/course-packages/install-HTSTrack.R")`

#### Efficient R Programming ####
Instructor: Martin Morgan

* [Efficient R Programming](EfficientRProgramming.pdf)
* [Efficient R Programming Exercises](EfficientRProgrammingExercises.pdf)

Install command:

`source("http://bioconductor.org/course-packages/install-EfficientR.R")`

#### Using limma for differential expression ####
Instructor: James W. MacDonald

* [Using limma for differential expression](limma2.pdf)

#### Data mining with biomaRt ####
Instructor: Steffen Durinck

* [Database mining with biomaRt](BioC2010_biomaRt.pdf)

#### ChIPpeakAnno Practical ####
Instructor: Lihua Julie Zhu

* [ChIPpeakAnno Practical](BioC2010_ChIPpeakAnno.pdf)

#### Transcript centric annotations and high-throughput sequencing ####
Instructors: Marc Carlson / Herve Pages

* [Using the GenomicFeatures package](GenomicFeatures.pdf)
* [Genomic Features and Sequences in Bioconductor](GenomicFeaturesAndSequences.pdf)

Install command:

`source("http://bioconductor.org/course-packages/install-HTSTrack.R")`

#### Bioconductor tools for input and quality assessment of high-throughput sequence data ####
Instructors: Patrick Aboyoun / Martin Morgan

* [Four exercises using Bioconductor Sequence Infrastructure](FourSeqExercises.pdf)
* [Exercises: Reading and Manipulating Short Reads](Exercises-ShortRead.pdf)
* [Exercises: An Introduction to Rsamtools](Exercises-Rsamtools.pdf)
* [Exercises: A Simple ChIP-Seq Workflow](Exercises-SimpleChIPSeq.pdf)
* [Exercises: A Simple RNA-seq Use Case](Exercises-SimpleRNAseqUseCase.pdf)

Install command:

`source("http://bioconductor.org/course-packages/install-HTSTrack.R")`

#### Analyzing and Visualizing ChIP-seq Data ####
Instructor: Michael Lawrence

* [Some Basic Analysis of ChIP-Seq Data](Workflow.pdf)

#### Analysis of genome-scale count data in Bioconductor ###
Instructors: Mark Robinson / Davis McCarthy

* [Analysis of genome-scale count data in Bioconductor](robinson_mccarthy_bioc2010.pdf)
* [Robinson_McCarthy_BioC2010.zip](Robinson_McCarthy_BioC2010.zip)

#### Using image data for statistical analysis and modeling ####
Instructor: Wolfgang Huber

* [Using image data for statistical analysis and modeling (slides)](100730-seattle-huber.pdf)
* [Using image data for statistical analysis and modeling (R script)](EBImage-seglab.R)
* [Image of actin in cells](001-02-C03-actin.tif)
* [Image of DNA in cells](001-02-C03-dna.tif)
* [Image of tubulin in cells](001-02-C03-tubulin.tif)

#### Genetics of gene expression: comprehensive eQTL search ####
Instructor: Vince Carey

* [Genetics of gene expression (lab)](lab2010.pdf)
* [Genetics of gene expression (R script)](lab2010.R)

#### Gene centric annotations ####
Instructor: Marc Carlson

* [Using Annotations in Bioconductor](ClassicAnnotations.pdf)

Install command:

`source("http://bioconductor.org/course-packages/install-HTSTrack.R")`

#### Analyzing flow cytometry data in Bioconductor ####
Instructors:  Nishant Gopalakrishnan / Chao-Jen Wong

* [Brief Intro to R for Flow Packages Users](intro.pdf)
* [Analyzing flow cytometry data in Bioconductor](seqGating.pdf)
* [Flow cytometry analysis Lab](AnalyzingFlowData.pdf)
* [Flow cytometry analysis Lab Solutions](FlowCytometry_SOLUTIONS.pdf)

Install command:

`source("http://bioconductor.org/course-packages/install-flowTrack.R")`

#### Automated Gating and Metaclustering of High Content Flow Cytometry Data ####
Instructor: Greg Finak

* [Automated Gating and Metaclustering for Flow Cytometry Data](FlowClustFlowMergeSlides.pdf)
* [Automated Gating and Metaclustering for Flow Cytometry Data (white paper)](FlowClustFlowMerge.pdf)

Install command:

`source("http://bioconductor.org/course-packages/install-flowTrack.R")`
