# BioC 2016: Where Software and Biology Connect

June 25-26, 2016 (Developer Day: June 24)<br />
Stanford University, Stanford, CA<br />

This conference highlights current developments within and beyond
_Bioconductor_. Morning scientific talks and afternoon practicals
provide conference participants with insights and tools required for
the analysis and comprehension of high-throughput genomic
data. 'Developer Day' (optional <font color="red">developer day
closed</font>; registrations received after 2 June do not include
developer day) precedes the main conference on June 24, providing
developers and would-be developers an opportunity to gain insights
into project direction and software development best practices.

Speaker, workshop, and additional information coming soon. Travel and
accommodation are participant's responsibilities. For more
information, see [useR! 2016 Lodging][] for possible options.

Scholarship applications are now closed; successful applicants will be
notified on May 18.

## Important Dates

- May 31: Early Registration closes; [Registration][] available
  (optional <font color="red">developer day closed</font>;
  registrations received after 2 June do not include developer day).
- June 1: Developer Day Workshop and Lightning Talk Submission Opens
- June 24: Developer Day!
- June 25 - 26: Main Conference

## Registration

<font color="red">Limited Space Remaining</font> for registration.

[Registration][]! Fees are as follows (optional <font
color="red">developer day closed</font>; registrations received after
2 June do not include developer day):

- _Bioconductor_ contributor / student / limited resources: $250
- Early-bird general attendee (before June 1): $600
- General attendee (after June 1): $800

Attendees are responsible for travel and accommodation. Questions?
Mail to workshop at bioconductor.org.

## Afternoon Workshops

Afternoon Workshop Proposal have been received; successful applicants
will be announced shortly.  Workshops are two-hour, user-oriented
sessions during the main conference.

## Tentative Schedule

Friday June 24 -- Developer Day (optional)

- Project updates
- Short developer-oriented workshops, e.g., 'My First Package', 'Using
  and Extending S4', 'Debugging C Code', 'Using Internet Resources in
  Packages', 'Unit Tests and other Programming Best Practices'.
- Contributed 'lightning' talks highlighting innovative and technical
  aspects of new packages
- Social activities
- Developer Day Keynote Address

    - Dr. Robert Gentleman, 23andMe. Distributed Content Generation
      and Creating Pipelines for Genomic Analysis.

Saturday, June 25

- Morning Speakers (45 minute presentations)

  - Sandrine Dudoit, Stanford University. Identification of Novel
    Cell Types in the Brain Using Single-Cell Transcriptome Sequencing.
  - Susan Holmes. Multicomponent data integration for the Human
    Microbiome.
  - Jenny Bryan, University of British Columbia. Spreadsheets: the Data
    Format we Love to Hate.

- Morning Community Speakers (15 minute presentations)

  - Michael Love, Harvard TH Chan School of Public Health. Bioconductor
    Workflows Following Fast, Lightweight RNA Transcript Quantifiers.
  - Ramnath Vaidya, Alteryx. HtmlWidgets: The Power of Javascript in R!
  - Benjamin Haibe-Kains, Princess Margaret Cancer Center / University
    of Toronto. Data Sharing and Research Reproducibility: Why and How.

- Afternoon workshops, session 1: 1:00 - 2:50

  - Getting to know _R_ and _Bioconductor_. Valerie Obenchain, Lori
    Shepherd (Roswell Park Cancer Institute). Beginner.
  - Building and running automated NGS analysis workflows. Thomas
    Girke (UC Riverside). Intermediate.
  - Introduction to [ImmuneSpaceR][]. Renan Sauteraud, Lev Dashevskiy,
    Greg Finak, Raphael Gottardo (Fred Hutchinson Cancer Research
    Center). Intermediate.
  - Analysis of single-cell RNA-seq data with R and
    Bioconductor. Davide Risso, Kelly Street, Michael Cole (UC
    Berkeley). Intermediate.

- Afternoon workshops, session 2: 3:10 - 5:00

  - Annotating high throughput data using _Bioconductor_
    resources. James MacDonald (University of Washington)
    Intermediate.
  - Interactive visualization with _epiviz_. H&eacute;ctor Corrada
    Bravo, Jayaram Kancherla, Justin Wagner, Deok Park (University of
    Maryland, College Park). Intermediate.
  - Single Cell Differential Expression and Gene Set Enrichment with
    MAST. Andrew McDavid, Raphael Gottardo, Greg Finak (Fred
    Hutchinson Cancer Research). Intermediate.
  - Managing big biological sequence data with [Biostrings][] and
    [DECIPHER][]. Erik Wright (UW, Madison). Intermediate.

Sunday, June 26

- Morning Speakers (45 minute presentations)

  - Rob Tibshirani, Stanford University. Some Recent Advances in
    Post-selection Statistical Inference.
  - William Greenleaf, Department of Genetics and, by courtesy, Applied
    Physics, Stanford University.  ATAC-ing Open Chromatin Data
    Analysis.
  - Stephen Montgomery, Stanford University. Identifying
    Gene-by-Environment Variants in Studies of Gene Expression.

- Morning Community Speakers (15 minute presentations)

  - Jim Hester, RStudio. Using Devtools, Travis and Git for Bioconductor
    Package Development.
  - Johannes Rainer, Center for Biomedicine, European Academy of
    Bozen/Bolzano (EURAC), Bolzano, Italy. Building and Using Ensembl
    Based Annotation Packages with ensembldb.
  - Xiuwen Zheng, University of Washington, Seattle, WA, USA. SeqArray:
    Data Management of Whole-Genome Sequence Variant Calls with
    Thousands of Individuals.

- Afternoon workshops, session 3: 1:00 - 2:50

  - Low-level exploratory data analysis and methods development for
    RNA-seq. Michael Love (Harvard TH Chan School of Public Health).
    Intermediate.
  - Making your packages accessible to non-programmer collaborators
    using the VisRseq platform. Hamid Younesy (Simon Fraser
    University), Torsten M&ouml;ller (University of Vienna) Mohammad
    M. Karimi (University of British Columbia). Beginner.
  - The [MultiAssayExperiment][] class for analysis of multi-omics
    experiments. Levi Waldron, Marcel Ramos (presenter) (CUNY School
    of Public Health), Vince Carey (Brigham and Women's), Kasper
    Hansen (Johns Hopkins University), Martin Morgan (Roswell Park
    Cancer Institute). Intermediate.
  - Analyzing splice events from RNA-seq data with [SGSeq][]. Leonard
    Goldstein (Genentech, Inc.). Intermediate.

- Afternoon workshops, session 4: 3:10 - 5:00

  - Wrapping Your _R_ tools to Analyze National-Scale Cancer Genomics
    in the Cloud. Tengfei Yin, Nan Xiao (Seven Bridges Genomics).
    Beginner.
  - Analysing DNA methylation data with _Bioconductor_. Peter Hickey,
    Kasper Hansen (Department of Biostatistics, Johns Hopkins
    Bloomberg School of Public Health). Intermediate.
  - Introduction to Bayesian Inference using [STAN][] with
    Applications to Cancer Genomics. Jacqueline Buros (Icahn School of
    Medicine at Mount Sinai), Benjamin Goodrich (Columbia University),
    Eric Novik (Stan Group Inc.).  Intermediate.
  - _R_ / _Bioconductor_ tools for mass spectrometry-based
    proteomics. Laurent Gatto (University of Cambridge). Intermediate.

[Registration]: https://cmetracker.net/RP/Login?formname=RegLoginLive&EventID=33505
[useR! 2016]: http://user2016.org/
[useR! 2016 Lodging]: http://user2016.org/#lodging

[ImmuneSpaceR]: https://bioconductor.org/packages/ImmuneSpaceR
[MAST]: https://bioconductor.org/packages/MAST
[Biostrings]: https://bioconductor.org/packages/Biostrings
[DECIPHER]: https://bioconductor.org/packages/DECIPHER
[MultiAssayExperiment]: https://bioconductor.org/packages/MultiAssayExperiment
[SGSeq]: https://bioconductor.org/packages/SGSeq
[STAN]: https://bioconductor.org/packages/STAN
