
            <p><strong>Characterizing co-expression
            networks in liver, adipose, and brain tissues in
            segregating mouse populations</strong></p>

            <p>Recent advances in gene expression microarray
            technology have provided the ability to monitor
            transcriptional changes on a genome-wide scale,
            allowing for the more general characterization of
            interaction networks among gene expression traits. Here
            I discuss the correlation structure among gene
            expression traits for a comprehensive set of genes
            (&gt;20k) in mouse liver, adipose and brain samples
            using oligonucleotide microarrays. This analysis
            reveals interesting properties of the gene-gene
            correlation structures that obtain in specific tissues
            in segregating mouse populations, as well as
            highlighting differences among the tissues and between
            the sexes. Specifically, there are a moderate number of
            gene expression traits (hubs) that have a high degree
            of connectivity to other gene expression traits,
            whereas the majority of gene expression traits are
            found to have limited connectivity to other gene
            expression traits. The degree of connectivity in the
            context of expression quantitative trait loci (eQTL) is
            also characterized, highlighting associations between
            highly interconnected groups of gene and common genetic
            control, where the shared genetic control enhances the
            ability to infer causal associations among gene
            expression traits and between gene expression and
            clinically relevant traits. In defining links (edges)
            between any two genes if they are found to be strongly
            interacting (i.e., above some absolute correlation
            threshold or p-value cutoff), the resulting gene-gene
            interaction network is shown to exhibit scale-free
            network properties and a hierarchical structure that
            tends to reflect common functional properties of genes.
            The degree distributions of these networks are shown to
            follow an inverse power law with exponents between -1
            to -1.3, while the cluster coefficient of some of the
            networks decrease as an exponential function instead of
            a power law. Both properties are distinct from most
            published biological networks. Furthermore, we show
            that strongly correlated genes appear to group into
            distinct sub-clusters enriched for a diversity of
            functional pathways that are further enriched for eQTL
            hot spot regions in the genome. Our analysis provides a
            simple and systematic approach to study the functional
            organization of gene expression interaction networks
            that can be used to functionally characterize gene
            groups and identify key control points in the network
            associated with complex traits such as common human
            diseases.<br />
            <br /></p>
          