<div class="shaded_box">

  <table cellspacing="0" cellpadding="0" summary=""><!-- table used for precise positioning -->
    <tr valign="top">
      <td style="border-right:1px silver solid;width:266px;">
        <h1 style="border:0;margin:0;background-image:none">About<br/>Bioconductor</h1>
        <p style="line-height:1.5em;padding-right:5em">
		Bioconductor provides tools for the analysis and comprehension of
		high-throughput genomic data. Bioconductor uses the R statistical
		programming language, and is open source and open development. It
		has two releases each year, 
		<a href="/packages/release/BiocViews.html#___Software">610 software packages</a>, 
		and an active user community. Bioconductor is also 
		available as an <a href="/help/bioconductor-cloud-ami/">
		Amazon Machine Image (AMI)</a>.</p>
		
		<p>
		    <a href="http://twitter.com/bioconductor" target="bioc_twitter"
		    title="Bioconductor on Twitter">
		    <img border="0" src="/images/twitter_newbird_boxed_blueonwhite.png" height="32" width="32">
		    </a>
		</p>
      </td>
      <td style="padding-left:35px">        
        <p class="olive" style="font-size:large">Use Bioconductor for...</p>
        <ul class="big_arrow_list" style="float:left;width:43%;">
          <li><span style="font-size:120%;font-weight:bold">
            <a href="/help/workflows/oligo-arrays/">Microarrays</a></span>
            <p>
              Import Affymetrix, Illumina, Nimblegen, Agilent, and
              other platforms.  Perform quality assessment,
              normalization, differential expression, clustering,
              classification, gene set enrichment, genetical genomics
              and other workflows for expression, exon, copy number,
              SNP, methylation and other assays.  Access GEO,
              ArrayExpress, Biomart, UCSC, and other community
              resources.
            </p>
          </li>
          <li><span style="font-size:120%;font-weight:bold">
            <a href="/help/workflows/variants/">Variants</a></span>
            <p>
	      Read and write VCF files. Identify structural location of
	      variants and compute amino acid coding changes for
	      non-synonymous variants. Use SIFT and PolyPhen database
	      packages to predict consequence of amino acid coding
	      changes.
            </p>
          </li>

        </ul>
        <ul class="big_arrow_list" style="float:right;width:43%;">
          <li><span style="font-size:120%;font-weight:bold">
            <a href="/help/workflows/high-throughput-sequencing/">
            Sequence Data</a></span>
            <p>
              Import fasta, fastq, ELAND, MAQ, BWA, Bowtie, BAM, gff,
              bed, wig, and other sequence formats.  Trim, transform,
              align, and manipulate sequences. Perform quality
              assessment, ChIP-seq, differential expression, RNA-seq,
              and other workflows.  Access the Sequence Read Archive.
            </p>
          </li>
          <li><span style="font-size:120%;font-weight:bold">
            <a href="/help/workflows/annotation-data/">Annotation</a></span>
            <p>
              Use microarray probe, gene, pathway, gene ontology,
              homology and other annotations.  Access GO, KEGG, NCBI,
              Biomart, UCSC, vendor, and other sources.
            </p>
          </li>
          <li><span style="font-size:120%;font-weight:bold">
            <a href="/help/workflows/high-throughput-assays/">
            High Throughput Assays</a></span>
            <p>
              Import, transform, edit, analyze and visualize flow
              cytometric, mass spec, HTqPCR, cell-based, and other
              assays.
            </p>
           </li>


        </ul>
      </td>
    </tr>
  </table>
</div>
<!--
<div class="hp_alert_box">
  <img src="/images/icons/conference.gif" align="left" alt="" style="margin-right:12px;"/>
  <span style="font-size:larger"><b><a href="https://secure.bioconductor.org/BioC2010/">Bioc 2010</a>: Where Software and Biology Connect</b></span><br/>
  The annual Bioconductor conference is July 28 (developer day), 29, and 30.<br clear="all"/>
</div>
-->
<div class="hp_options_box">

  <div style="float:left;width:33%;">
    <%= render('/_mlwidget/', :ml_item => find_item(@items, '/mailing-list-posts/')) %>
  </div>

  <div style="float:left;width:33%;">
    <div style="padding:0 0 0 12px">
      <h2><img src="/images/icons/calendar.gif" alt=""/><a href="/help/events/">Events</a></h2>
      
      
      <%= render('/_upcoming_events/', :events => find_item(@items, '/help/events/')) %>
      <p style="font-weight:bold"><a href="/help/events/">See all events</a>&nbsp;
	  <span class="blue">&raquo;</span></p>
    </div>
  </div>

  <div style="float:right;width:33%;">
    <div style="padding:0 0 0 12px">
      <h2 style="min-height:40px;">
        <img src="/images/icons/blog.gif" alt=""/>News
      </h2>


      <p><a href="/news/bioc_2_11_release/"><b>Bioconductor 2.11
      released</b></a><br/>  
      Following the usual 6-month cycle, the Bioconductor community
      released Bioconductor 2.11 on October 3rd, 2012. This release
      comprises 610 software packages and more than 650 up-to-date
      annotation packages. It has been expressly designed to work with
      R 2.15.</p>



      <p><a href="/news/bioc_2_10_release/"><b>Bioconductor 2.10
	    released</b></a><br/>  
      Following the usual 6-month cycle, the Bioconductor community
      released Bioconductor 2.10 on April 2nd, 2012. This release
      comprises 554 software packages and more than 600 up-to-date
      annotation packages. It has been expressly designed to work with
      R 2.15.</p>


      <p><a href="/news/bioc_2_9_release/"><b>Bioconductor 2.9
	    released</b></a><br/>  
      Following the usual 6-month cycle, the Bioconductor community
      released Bioconductor 2.9 on November 1st, 2011. This release
      comprises 517 software packages and more than 500 up-to-date
      annotation packages. It has been expressly designed to work with
      R 2.14.</p>



    </div>
  </div>

</div>
<br clear="all"/>
