April 14, 2011

Bioconductors:

We are pleased to announce Bioconductor 2.8, consisting of 466
software packages and more than 500 up-to-date annotation packages.
There are 48 new software packages, and many updates and improvements
to existing packages. Two software packages that were in the previous 
version have been removed. Bioconductor 2.8 is compatible with
R 2.13.0, and is supported on Linux, 32- and 64-bit Windows, and Mac
OS.  Visit

[http://bioconductor.org](http://bioconductor.org)

for details and downloads.

Contents
========

* Getting Started with Bioconductor 2.8
* New Software Packages
* Using Bioconductor in the cloud

Getting Started with Bioconductor 2.8
=====================================

To install Bioconductor 2.8:

1. Install R 2.13.0.  Bioconductor 2.8 has been designed expressly for
this version of R.

2. Follow the instructions here:

[http://bioconductor.org/install/](http://bioconductor.org/install/)

Please visit http://bioconductor.org for details and downloads.

New Software Packages
=====================

There are 48 new packages in this release of Bioconductor.

a4

> Automated Affymetrix Array Analysis Umbrella Package

a4Base

> Automated Affymetrix Array Analysis Base Package

a4Classif

> Automated Affymetrix Array Analysis Classification Package

a4Core

> Automated Affymetrix Array Analysis Core Package

a4Preproc

> Automated Affymetrix Array Analysis Preprocessing Package

a4Reporting

> Automated Affymetrix Array Analysis Reporting Package

AnnotationFuncs

> Annotation translation functions

anota

> ANalysis Of Translational Activity

chopsticks

> The snp.matrix and X.snp.matrix classes

Clonality

> Clonality testing

clst

> Classification by local similarity threshold

clstutils

> Tools for performing taxonomic assignment

clusterProfiler

> statistical analysis and visulization of
> functional profiles for genes and gene clusters

cn.farms

> Factor Analysis for copy number estimation

ENVISIONQuery

> Retrieval from the ENVISION bioinformatics data portal into R

ExiMiR

> R functions for the normalization of Exiqon miRNA array data

flowPhyto

> Methods for Continuous Flow Cytometry

flowPlots

> analysis plots and data class for gated flow cytometry data

gaia

> An R package for genomic analysis of significant
> chromosomal aberrations

genefu

> Relevant Functions for Gene Expression Analysis,
> Especially in Breast Cancer

genoset

> Provides classes similar to ExpressionSet for copy number analysis

GSVA

> Gene Set Variation Analysis

ibh

> Interaction Based Homogeneity for Evaluating Gene Lists

inveRsion

> Inversions in genotype data

IPPD

> Isotopic peak pattern deconvolution for Protein Mass
> Spectrometry by template matching

joda

> JODA algorithm for quantifying gene deregulation using knowledge

lol

> Lots Of Lasso

mcaGUI

> Microbial Community Analysis GUI

mgsa

> Model-based gene set analysis

MLP

> Mean Log P Analysis

mosaics

> MOdel-based one and two Sample Analysis and Inference for ChIP-Seq

MSnbase

> Base Functions and Classes for MS-based Proteomics

NCIgraph

> Pathways from the NCI Pathways Database

phenoDist

> Phenotypic distance measures

phenoTest

> Tools to test correlation between gene expression and phenotype

procoil

> Prediction of Oligomerization of Coiled Coil Proteins

pvac

> PCA-based gene filtering for Affymetrix arrays

qrqc

> Quick Read Quality Control

RNAinteract

> Estimate Pairwise Interactions from multidimensional features

Rsubread

> a super fast, sensitive and accurate read aligner for mapping
> next-generation sequencing reads

seqbias

> Estimation of per-position bias in high-throughput sequencing data

snm

> Supervised Normalization of Microarrays

snpStats

> SnpMatrix and XSnpMatrix classes and methods

survcomp

> Performance Assessment and Comparison for Survival Analysis

TDARACNE

> Network reverse engineering from time course data

TEQC

> Quality control for target capture experiments

TurboNorm

> A fast scatterplot smoother suitable for microarray normalization

Vega

> An R package for copy number data segmentation


Using Bioconductor in the cloud
===============================

This release features the Bioconductor Amazon Machine
Image (AMI), which allows easy access to R and Bioconductor
within the Elastic Compute Cloud (EC2). It's easy to run 
parallelizable tasks on MPI clusters, run R from within
your web browser using RStudio Server, and more. No
installation required. Information available at:

[http://bioconductor.org/help/bioconductor-cloud-ami/](http://bioconductor.org/help/bioconductor-cloud-ami/)
