April 28, 2020

Bioconductors:

We are pleased to announce Bioconductor 3.11, consisting of 1903
software packages, 391 experiment data packages, 961 annotation
packages, and 27 workflows.

There are 98 new software packages, 10 new data experiment packages,
5 new annotation packages, 1 new workflow, and many updates and improvements
to existing packages; Bioconductor 3.11 is compatible with R 4.0.0,
and is supported on Linux, 32- and 64-bit Windows, and macOS 10.13 High Sierra
or higher.  This release will include an updated Bioconductor [Amazon Machine
Image][1] and [Docker containers][2].

Thank you to everyone for your contribution to Bioconductor

Visit [Bioconductor BiocViews][3]
for details and downloads.

[1]: /help/bioconductor-cloud-ami/
[2]: /help/docker/
[3]: /packages/release/BiocViews.html

Contents
--------

* [Getting Started with Bioconductor 3.11](#getting-started-with-bioconductor-311)
* [New Software Packages](#new-software-packages)
* [New Data Experiment Packages](#new-data-experiment-packages)
* [New Annotation Packages](#new-annotation-packages)
* [New Workflow](#new-workflow-packages)
* [NEWS from new and existing software packages](#news-from-new-and-existing-software-packages)
* [NEWS from new and existing data experiment packages](#news-from-new-and-existing-data-experiment-packages)
* [NEWS from new and existing workflows](#news-from-new-and-existing-workflows)
* [Deprecated and Defunct Packages](#deprecated-and-defunct-packages)

Getting Started with Bioconductor 3.11
======================================

To update to or install Bioconductor 3.11:

1. Install R 4.0.0. Bioconductor 3.11 has been designed expressly for
   this version of R.

2. Follow the instructions at
   [Installing Bioconductor](/install/).

New Software Packages
=====================

There are 98 new software packages in this release of Bioconductor.

- [basilisk](/packages/basilisk) Installs a self-contained Python
  instance that is managed by the R installation. This aims to
  provide a consistent Python version that can be used reliably by
  Bioconductor packages. Module versions are also controlled to
  guarantee consistent behavior on different user systems.

- [basilisk.utils](/packages/basilisk.utils) Implements utilities for
  installation of the basilisk package, primarily to avoid re-writing
  the same R code in both the configure script (for centrally
  administered R installations) and in the lazy installation
  mechanism (for distributed binaries). It is highly unlikely that
  developers - or, heaven forbid, end-users! - will need to interact
  with this package directly; they should be using the basilisk
  package instead.

- [BgeeCall](/packages/BgeeCall) Reference intergenic regions are
  generated by the Bgee RNA-Seq pipeline. These intergenic regions
  are used to generate all Bgee RNA-Seq present/absent expression
  calls. BgeeCall now allows to generate present/absent calls for any
  RNA-Seq library as long as reference intergenic sequences have been
  generated for the corresponding species. The threshold of
  present/absent expression is no longer arbitrary defined but is
  calculated based on expression of all RNA-Seq libraries integrated
  in Bgee.

- [biobtreeR](/packages/biobtreeR) The biobtreeR package provides an
  interface to [biobtree](https://github.com/tamerh/biobtree) tool
  which covers large set of bioinformatics datasets and allows search
  and chain mappings functionalities.

- [BiocDockerManager](/packages/BiocDockerManager) Package works
  analogous to BiocManager but for docker images. Use the
  BiocDockerManager package to install and manage docker images
  provided by the Bioconductor project. A convenient package to
  install images, update images and find which Bioconductor based
  docker images are available.

- [BRGenomics](/packages/BRGenomics) This package provides useful and
  efficient utilites for the analysis of high-resolution genomic data
  using standard Bioconductor methods and classes. BRGenomics is
  feature-rich and simplifies a number of post-alignment processing
  steps and data handling. Emphasis is on efficient analysis of
  multiple datasets, with support for normalization and blacklisting.
  Included are functions for: spike-in normalizing data; generating
  basepair-resolution readcounts and coverage data (e.g. for
  heatmaps); importing and processing bam files (e.g. for conversion
  to bigWig files); generating metaplots/metaprofiles (bootstrapped
  mean profiles) with confidence intervals; conveniently calling
  DESeq2 without using sample-blind estimates of genewise dispersion;
  among other features.

- [CARNIVAL](/packages/CARNIVAL) An upgraded causal reasoning tool
  from Melas et al in R with updated assignments of TFs' weights from
  PROGENy scores. Optimization parameters can be freely adjusted and
  multiple solutions can be obtained and aggregated.

- [ceRNAnetsim](/packages/ceRNAnetsim) This package simulates
  regulations of ceRNA (Competing Endogenous) expression levels after
  a expression level change in one or more miRNA/mRNAs. The
  methodolgy adopted by the package has potential to incorparate any
  ceRNA (circRNA, lincRNA, etc.) into miRNA:target interaction
  network.  The package basically distributes miRNA expression over
  available ceRNAs where each ceRNA attracks miRNAs proportional to
  its amount. But, the package can utilize multiple parameters that
  modify miRNA effect on its target (seed type, binding energy,
  binding location, etc.).  The functions handle the given dataset as
  graph object and the processes progress via edge and node
  variables.

- [CeTF](/packages/CeTF) This package provides the necessary
  functions for performing the Partial Correlation coefficient with
  Information Theory (PCIT) (Reverter and Chan 2008) and Regulatory
  Impact Factors (RIF) (Reverter et al. 2010) algorithm. The PCIT
  algorithm identifies meaningful correlations to define edges in a
  weighted network and can be applied to any correlation-based
  network including but not limited to gene co-expression networks,
  while the RIF algorithm identify critical Transcription Factors
  (TF) from gene expression data. These two algorithms when combined
  provide a very relevant layer of information for gene expression
  studies (Microarray, RNA-seq and single-cell RNA-seq data).

- [CiteFuse](/packages/CiteFuse) CiteFuse pacakage implements a suite
  of methods and tools for CITE-seq data from pre-processing to
  integrative analytics, including doublet detection, network-based
  modality integration, cell type clustering, differential RNA and
  protein expression analysis, ADT evaluation, ligand-receptor
  interaction analysis, and interactive web-based visualisation of
  the analyses.

- [clustifyr](/packages/clustifyr) Package designed to aid in
  classifying cells from single-cell RNA sequencing data using
  external reference data (e.g., bulk RNA-seq, scRNA-seq, microarray,
  gene lists). A variety of correlation based methods and gene list
  enrichment methods are provided to assist cell type assignment.

- [cmapR](/packages/cmapR) The Connectivity Map (CMap) is a massive
  resource of perturbational gene expression profiles built by
  researchers at the Broad Institute and funded by the NIH Library of
  Integrated Network-Based Cellular Signatures (LINCS) program.
  Please visit https://clue.io for more information. The cmapR
  package implements methods to parse, manipulate, and write common
  (CMap) data objects, such as annotated matrices and collections of
  gene sets.

- [combi](/packages/combi) Combine quasi-likelihood estimation,
  compositional regression models and latent variable models for
  integrative visualization of several omics datasets. Both
  unconstrained and constrained integration is available, the results
  are shown as interpretable multiplots.

- [CoreGx](/packages/CoreGx) A collection of functions and classes
  which serve as the foundation for our lab's suite of R packages,
  such as 'PharmacoGx' and 'RadioGx'. This package was created to
  abstract shared functionality from other lab package releases to
  increase ease of maintainability and reduce code repetition in
  current and future 'Gx' suite programs. Major features include a
  'CoreSet' class, from which 'RadioSet' and 'PharmaSet' are derived,
  along with get and set methods for each respective slot. Additional
  functions related to fitting and plotting dose response curves,
  quantifying statistical correlation and calculating area under the
  curve (AUC) or survival fraction (SF) are included. For more
  details please see the included documentation, as well as: Smirnov,
  P., Safikhani, Z., El-Hachem, N., Wang, D., She, A., Olsen, C.,
  Freeman, M., Selby, H., Gendoo, D., Grossman, P., Beck, A., Aerts,
  H., Lupien, M., Goldenberg, A. (2015)
  <doi:10.1093/bioinformatics/btv723>. Manem, V., Labie, M., Smirnov,
  P., Kofia, V., Freeman, M., Koritzinksy, M., Abazeed, M.,
  Haibe-Kains, B., Bratman, S. (2018) <doi:10.1101/449793>.

- [CSSQ](/packages/CSSQ) This package is desgined to perform
  statistical analysis to identify statistically significant
  differentially bound regions between multiple groups of ChIP-seq
  dataset.

- [ctgGEM](/packages/ctgGEM) Cell Tree Generator for Gene Expression
  Matrices (ctgGEM) streamlines the building of cell-state
  hierarchies from single-cell gene expression data across multiple
  existing tools for improved comparability and reproducibility. It
  supports pseudotemporal ordering algorithms and visualization tools
  from monocle, cellTree, TSCAN, sincell, and destiny, and provides a
  unified output format for integration with downstream data analysis
  workflows and Cytoscape.

- [cytomapper](/packages/cytomapper) Highly multiplexed imaging
  cytometry acquires the single-cell expression of selected proteins
  in a spatially-resolved fashion. These measurements can be
  visualized across multiple length-scales. First, pixel-level
  intensities represent the spatial distributions of feature
  expression with highest resolution. Second, after segmentation,
  expression values or cell-level metadata (e.g. cell-type
  information) can be visualized on segmented cell areas. This
  package contains functions for the visualization of multiplexed
  read-outs and cell-level information obtained by multiplexed
  imaging cytometry. The main functions of this package allow 1. the
  visualization of pixel-level information across multiple channels
  and 2. the display of cell-level information (expression and/or
  metadata) on segmentation masks.

- [DAMEfinder](/packages/DAMEfinder) 'DAMEfinder' offers
  functionality for taking methtuple or bismark outputs to calculate
  ASM scores and compute DAMEs. It also offers nice visualization of
  methyl-circle plots.

- [dearseq](/packages/dearseq) Differential Expression Analysis
  RNA-seq data with variance component score test accounting for data
  heteroscedasticity through precision weights. Perform both
  gene-wise and gene set analyses, and can deal with repeated or
  longitudinal data. Methods are detailed in: Agniel D & Hejblum BP (2017) Variance 
  component score test for time-course gene set
  analysis of longitudinal RNA-seq data, Biostatistics,
  18(4):589-604. and Gauthier M, Agniel D, Thiébaut R & Hejblum BP
  (2019). dearseq: a variance component score test for RNA-Seq
  differential analysis that effectively controls the false discovery
  rate, *bioRxiv* 635714.

- [DIAlignR](/packages/DIAlignR) To obtain unbiased proteome coverage
  from a biological sample, mass-spectrometer is operated in Data
  Independent Acquisition (DIA) mode. Alignment of these DIA runs
  establishes consistency and less missing values in complete
  data-matrix. This package implements dynamic programming with
  affine gap penalty based approach for pair-wise alignment of
  analytes. A hybrid approach of global alignment (through MS2
  features) and local alignment (with MS2 chromatograms) is
  implemented in this tool.

- [distinct](/packages/distinct) distinct is a statistical method to
  perform differential testing between two or more groups of
  distributions; differential testing is performed via hierarchical
  non-parametric permutation tests on the cumulative distribution
  functions (cdfs) of each sample. While most methods for
  differential expression target differences in the mean abundance
  between conditions, distinct, by comparing full cdfs, identifies,
  both, differential patterns involving changes in the mean, as well
  as more subtle variations that do not involve the mean (e.g.,
  unimodal vs. bi-modal distributions with the same mean). distinct
  is a general and flexible tool: due to its fully non-parametric
  nature, which makes no assumptions on how the data was generated,
  it can be applied to a variety of datasets. It is particularly
  suitable to perform differential state analyses on single cell data
  (i.e., differential analyses within sub-populations of cells), such
  as single cell RNA sequencing (scRNA-seq) and high-dimensional flow
  or mass cytometry (HDCyto) data. To use distinct one needs data
  from two or more groups of samples (i.e., experimental conditions),
  with at least 2 samples (i.e., biological replicates) per group.

- [dittoSeq](/packages/dittoSeq) A universal, user friendly,
  single-cell and bulk RNA sequencing visualization toolkit that
  allows highly customizable creation of color blindness friendly,
  publication-quality figures. dittoSeq accepts both
  SingleCellExperiment (SCE) and Seurat objects, as well as the
  import and usage, via conversion to an SCE, of SummarizedExperiment
  or DGEList bulk data. Visualizations include dimensionality
  reduction plots, heatmaps, scatterplots, percent composition or
  expression across groups, and more. Customizations range from size
  and title adjustments to automatic generation of annotations for
  heatmaps, overlay of trajectory analysis onto any dimensionality
  reduciton plot, hidden data overlay upon cursor hovering via
  ggplotly conversion, and many more. All with simple, discrete
  inputs. Color blindness friendliness is powered by legend
  adjustments (enlarged keys), and by allowing the use of shapes or
  letter-overlay in addition to the carefully selected dittoColors().

- [dpeak](/packages/dpeak) dPeak is a statistical framework for the
  high resolution identification of protein-DNA interaction sites
  using PET and SET ChIP-Seq and ChIP-exo data. It provides
  computationally efficient and user friendly interface to process
  ChIP-seq and ChIP-exo data, implement exploratory analysis, fit
  dPeak model, and export list of predicted binding sites for
  downstream analysis.

- [Dune](/packages/Dune) Given a set of clustering labels, Dune
  merges pairs of clusters to increase mean ARI between labels,
  improving replicability.

- [easyreporting](/packages/easyreporting) An S4 class for
  facilitating the automated creation of rmarkdown files inside other
  packages/software, even without knowing rmarkdown language. Best if
  implemented in functions as "recursive" style programming.

- [eisaR](/packages/eisaR) Exon-intron split analysis (EISA) uses
  ordinary RNA-seq data to measure changes in mature RNA and pre-mRNA
  reads across different experimental conditions to quantify
  transcriptional and post-transcriptional regulation of gene
  expression. For details see Gaidatzis et al., Nat Biotechnol 2015.
  doi: 10.1038/nbt.3269. eisaR implements the major steps of EISA in
  R.

- [EnMCB](/packages/EnMCB) Creation of the correlated blocks using
  DNA methylation profiles. A stacked ensemble of machine learning
  models, which combined the support vector machine and elastic-net
  regression model, can be constructed to predict disease
  progression.

- [EpiTxDb](/packages/EpiTxDb) EpiTxDb facilitates the storage of
  epitranscriptomic information. More specifically, it can keep track
  of modification identity, position, the enzyme for introducing it
  on the RNA, a specifier which determines the position on the RNA to
  be modified and the literature references each modification is
  associated with.

- [exomePeak2](/packages/exomePeak2) exomePeak2 provides bias awared
  quantification and peak detection on Methylated RNA
  immunoprecipitation sequencing data (MeRIP-Seq). MeRIP-Seq is a
  commonly applied sequencing technology to measure the
  transcriptome-wide location and abundance of RNA modification sites
  under a given cellular condition. However, the quantification and
  peak calling in MeRIP-Seq are sensitive to PCR amplification bias
  which is prevalent in next generation sequencing (NGS) techniques.
  In addition, the RNA-Seq based count data exhibits biological
  variation in small reads count. exomePeak2 collectively address
  these challanges by introducing a rich set of robust data science
  models tailored for MeRIP-Seq. With exomePeak2, users can perform
  peak calling, modification site quantification, and differential
  analysis with a straightforward one-step function. Alternatively,
  users could define personalized methods for their own analysis
  through multi-step functions and diagnostic plots.

- [ExploreModelMatrix](/packages/ExploreModelMatrix) Given a sample
  data table and a design formula, generate an interactive
  application to explore the resulting design matrix.

- [FRASER](/packages/FRASER) Detection of rare aberrant splicing
  events in transcriptome profiles. The workflow aims to assist the
  diagnostics in the field of rare diseases where RNA-seq is
  performed to identify aberrant splicing defects.

- [frenchFISH](/packages/frenchFISH) FrenchFISH comprises a nuclear
  volume correction method coupled with two types of Poisson models:
  either a Poisson model for improved manual spot counting without
  the need for control probes; or a homogenous Poisson Point Process
  model for automated spot counting.

- [GCSConnection](/packages/GCSConnection) Create R 'connection'
  objects to google cloud storage buckets using the Google REST
  interface. Both read and write connections are supported. The
  package also provide functions to view and manage files on Google
  Cloud.

- [GeneTonic](/packages/GeneTonic) This package provides a Shiny
  application that aims to combine at different levels the existing
  pieces of the transcriptome data and results, in a way that makes
  it easier to generate insightful observations and hypothesis -
  combining the benefits of interactivity and reproducibility, e.g.
  by capturing the features and gene sets of interest highlighted
  during the live session, and creating an HTML report as an artifact
  where text, code, and output coexist.

- [GGPA](/packages/GGPA) Genome-wide association studies (GWAS) is a
  widely used tool for identification of genetic variants associated
  with phenotypes and diseases, though complex diseases featuring
  many genetic variants with small effects present difficulties for
  traditional these studies. By leveraging pleiotropy, the
  statistical power of a single GWAS can be increased. This package
  provides functions for fitting graph-GPA, a statistical framework
  to prioritize GWAS results by integrating pleiotropy. 'GGPA'
  package provides user-friendly interface to fit graph-GPA models,
  implement association mapping, and generate a phenotype graph.

- [glmGamPoi](/packages/glmGamPoi) Fit linear models to overdispersed
  count data. The package can estimate the overdispersion and fit
  repeated models for matrix input. It is designed to handle large
  input datasets as they typically occur in single cell RNA-seq
  experiments.

- [gmoviz](/packages/gmoviz) Genetically modified organisms (GMOs)
  and cell lines are widely used models in all kinds of biological
  research. As part of characterising these models, DNA sequencing
  technology and bioinformatics analyses are used systematically to
  study their genomes. Therefore, large volumes of data are generated
  and various algorithms are applied to analyse this data, which
  introduces a challenge on representing all findings in an
  informative and concise manner. `gmoviz` provides users with an
  easy way to visualise and facilitate the explanation of complex
  genomic editing events on a larger, biologically-relevant scale.

- [GPA](/packages/GPA) This package provides functions for fitting
  GPA, a statistical framework to prioritize GWAS results by
  integrating pleiotropy information and annotation data. In
  addition, it also includes ShinyGPA, an interactive visualization
  toolkit to investigate pleiotropic architecture.

- [HIPPO](/packages/HIPPO) For scRNA-seq data, it selects features
  and clusters the cells simultaneously for single-cell UMI data. It
  has a novel feature selection method using the zero inflation
  instead of gene variance, and computationally faster than other
  existing methods since it only relies on PCA+Kmeans rather than
  graph-clustering or consensus clustering.

- [iSEEu](/packages/iSEEu) iSEEu (the iSEE universe) contains diverse
  functionality to extend the usage of the iSEE package, including
  additional classes for the panels, or modes allowing easy
  configuration of iSEE applications.

- [LACE](/packages/LACE) LACE is an algorithmic framework that
  processes single-cell somatic mutation profiles from cancer samples
  collected at different time points and in distinct experimental
  settings, to produce longitudinal models of cancer evolution. The
  approach solves a Boolean Matrix Factorization problem with
  phylogenetic constraints, by maximizing a weighed likelihood
  function computed on multiple time points.

- [MatrixGenerics](/packages/MatrixGenerics) S4 generic functions
  modeled after the 'matrixStats' API for alternative matrix
  implementations. Packages with alternative matrix implementation
  can depend on this package and implement the generic functions that
  are defined here for a useful set of row and column summary
  statistics. Other package developers can import this package and
  handle a different matrix implementations without worrying about
  incompatibilities.

- [MEAT](/packages/MEAT) This package estimates epigenetic age in
  skeletal muscle, using DNA methylation data generated with Illumina
  Infinium technology (HM27, HM450 and HMEPIC).

- [metaseqR2](/packages/metaseqR2) Provides an interface to several
  normalization and statistical testing packages for RNA-Seq gene
  expression data. Additionally, it creates several diagnostic plots,
  performs meta-analysis by combinining the results of several
  statistical tests and reports the results in an interactive way.

- [methylSig](/packages/methylSig) MethylSig is a package for testing
  for differentially methylated cytosines (DMCs) or regions (DMRs) in
  whole-genome bisulfite sequencing (WGBS) or reduced representation
  bisulfite sequencing (RRBS) experiments.  MethylSig uses a beta
  binomial model to test for significant differences between groups
  of samples. Several options exist for either site-specific or
  sliding window tests, and variance estimation.

- [MicrobiotaProcess](/packages/MicrobiotaProcess) MicrobiotaProcess
  is an R package for analysis, visualization and biomarker discovery
  of microbial datasets. It supports calculating alpha index and
  provides functions to visualize rarefaction curves. Moreover, it
  also supports visualizing the abundance of taxonomy of samples. And
  It also provides functions to perform the PCA, PCoA and
  hierarchical cluster analysis. In addition, MicrobiotaProcess also
  provides a method for the biomarker discovery of metagenome or
  other datasets.

- [mitch](/packages/mitch) mitch is an R package for multi-contrast
  enrichment analysis. At it’s heart, it uses a rank-MANOVA based
  statistical approach to detect sets of genes that exhibit
  enrichment in the multidimensional space as compared to the
  background. The rank-MANOVA concept dates to work by Cox and Mann
  (https://doi.org/10.1186/1471-2105-13-S16-S12). mitch is useful for
  pathway analysis of profiling studies with one, two or more
  contrasts, or in studies with multiple omics profiling, for example
  proteomic, transcriptomic, epigenomic analysis of the same samples.
  mitch is perfectly suited for pathway level differential analysis
  of scRNA-seq data. The main strengths of mitch are that it can
  import datasets easily from many upstream tools and has advanced
  plotting features to visualise these enrichments.

- [MOMA](/packages/MOMA) This package implements the inference of
  candidate master regulator proteins from multi-omics' data (MOMA)
  algorithm, as well as ancillary analysis and visualization
  functions.

- [MsCoreUtils](/packages/MsCoreUtils) MsCoreUtils defines low-level
  functions for mass spectrometry data and is independent of any
  high-level data structures. These functions include mass spectra
  processing functions (noise estimation, smoothing, binning),
  quantitative aggregation functions (median polish, robust
  summarisation, ...), missing data imputation, data normalisation
  (quantiles, vsn, ...) as well as misc helper functions, that are
  used across high-level data structure within the R for Mass
  Spectrometry packages.

- [netboxr](/packages/netboxr) NetBox is a software tool for
  performing network analysis on human interaction networks. It is
  pre-loaded with a Human Interaction Network (HIN) derived from four
  literature curated data sources, including the Human Protein
  Reference Database (HPRD), Reactome, NCI-Nature Pathway Interaction
  (PID) Database, and the MSKCC Cancer Cell Map.

- [netDx](/packages/netDx) netDx is a general-purpose algorithm to
  build a patient classifier from heterogenous patient data. The
  method converts data into patient similarity networks at the level
  of features. Feature selection identifies features of predictive
  value to each class. Methods are provided for versatile predictor
  design and performance evaluation using standard measures. netDx
  natively groups molecular data into pathway-level features and
  connects with Cytoscape for network visualization of pathway
  themes. For method details see: Pai et al. (2019). netDx:
  interpretable patient classification using integrated patient
  similarity networks. Molecular Systems Biology. 15, e8497

- [NoRCE](/packages/NoRCE) While some non-coding RNAs (ncRNAs) have
  been found to play critical regulatory roles in biological
  processes, most remain functionally uncharacterized. This presents
  a challenge whenever an interesting set of ncRNAs set needs to be
  analyzed in a functional context. Transcripts located close-by on
  the genome are often regulated together, and this spatial proximity
  hints at a functional association. Based on this idea, we present
  an R package, NoRCE, that performs cis enrichment analysis for a
  given set of ncRNAs. Enrichment is carried out by using the
  functional annotations of the coding genes located proximally to
  the input ncRNAs. NoRCE allows incorporating other biological
  information such as the topologically associating domain (TAD)
  regions, co-expression patterns, and miRNA target information.
  NoRCE repository includes several data files, such as cell line
  specific TAD regions, functional gene sets, and cancer expression
  data. Additionally, users can input custom data files. Results can
  be retrieved in a tabular format or viewed as graphs. NoRCE is
  currently available for the following species: human, mouse, rat,
  zebrafish, fruit fly, worm and yeast.

- [NPARC](/packages/NPARC) Perform non-parametric analysis of
  response curves as described by Childs, Bach, Franken et al.
  (2019): Non-parametric analysis of thermal proteome profiles
  reveals novel drug-binding proteins.

- [OpenStats](/packages/OpenStats) Package contains several methods
  for statistical analysis of genotype to phenotype association in
  high-throughput screening pipelines.

- [optimalFlow](/packages/optimalFlow) Optimal-transport techniques
  applied to supervised flow cytometry gating.

- [packFinder](/packages/packFinder) Algorithm and tools for in
  silico pack-TYPE transposon discovery. Filters a given genome for
  properties unique to DNA transposons and provides tools for the
  investigation of returned matches. Sequences are input in DNAString
  format, and ranges are returned as a dataframe (in the format
  returned by as.dataframe(GRanges)).

- [peco](/packages/peco) Our approach provides a way to assign
  continuous cell cycle phase using scRNA-seq data, and consequently,
  allows to identify cyclic trend of gene expression levels along the
  cell cycle. This package provides method and training data, which
  includes scRNA-seq data collected from 6 individual cell lines of
  induced pluripotent stem cells (iPSCs), and also continuous cell
  cycle phase derived from FUCCI fluorescence imaging data.

- [PloGO2](/packages/PloGO2) Functions for enrichment analysis and
  plotting gene ontology or KEGG pathway information for multiple
  data subsets at the same time. It also enables encorporating
  multiple conditions and abundance data.

- [pmp](/packages/pmp) Methods and tools for (pre-)processing of
  metabolomics datasets (i.e. peak matrices), including filtering,
  normalisation, missing value imputation, scaling, and signal drift
  and batch effect correction methods. Filtering methods are based
  on: the fraction of missing values (across samples or features);
  Relative Standard Deviation (RSD) calculated from the Quality
  Control (QC) samples; the blank samples. Normalisation methods
  include Probabilistic Quotient Normalisation (PQN) and
  normalisation to total signal intensity. A unified user interface
  for several commonly used missing value imputation algorithms is
  also provided. Supported methods are: k-nearest neighbours (knn),
  random forests (rf), Bayesian PCA missing value estimator (bpca),
  mean or median value of the given feature and a constant small
  value. The generalised logarithm (glog) transformation algorithm is
  available to stabilise the variance across low and high intensity
  mass spectral features. Finally, this package provides an
  implementation of the Quality Control-Robust Spline Correction
  (QCRSC) algorithm for signal drift and batch effect correction of
  mass spectrometry-based datasets.

- [PubScore](/packages/PubScore) Calculates the importance score for
  a given gene. The importance score is the total counts of articles
  in the pubmed database that are a result for that gene AND each
  term of a list.

- [randRotation](/packages/randRotation) A collection of methods for
  performing random orthogonal transformations (random rotations) on
  high-dimensional data (e.g. microarray or RNA-seq data) with batch
  structure. The random rotation approach allows exact testing of
  dependent test statistics with linear models following arbitrary,
  particularly non-linear, batch effect correction methods. The
  package further provides methods for estimating the local degrees
  of freedom of data mappings.

- [receptLoss](/packages/receptLoss) receptLoss identifies genes
  whose expression is lost in subsets of tumors relative to normal
  tissue. It is particularly well-suited in cases where the number of
  normal tissue samples is small, as the distribution of gene
  expression in normal tissue samples is approximated by a Gaussian.
  Originally designed for identifying nuclear hormone receptor
  expression loss but can be applied transcriptome wide as well.

- [reconsi](/packages/reconsi) Improves simultaneous inference under
  dependence of tests by estimating a collapsed null distribution
  through resampling. Accounting for the dependence between tests
  increases the power while reducing the variability of the false
  discovery proportion. This dependence is common in genomics
  applications, e.g. when combining flow cytometry measurements with
  microbiome sequence counts.

- [regutools](/packages/regutools) RegulonDB has collected,
  harmonized and centralized data from hundreds of experiments for
  nearly two decades and is considered a point of reference for
  transcriptional regulation in Escherichia coli K12. Here, we
  present the regutools R package to facilitate programmatic access
  to RegulonDB data in computational biology. regutools provides
  researchers with the possibility of writing reproducible workflows
  with automated queries to RegulonDB. The regutools package serves
  as a bridge between RegulonDB data and the Bioconductor ecosystem
  by reusing the data structures and statistical methods powered by
  other Bioconductor packages. We demonstrate the integration of
  regutools with Bioconductor by analyzing transcription factor DNA
  binding sites and transcriptional regulatory networks from
  RegulonDB. We anticipate that regutools will serve as a useful
  building block in our progress to further our understanding of gene
  regulatory networks.

- [rfaRm](/packages/rfaRm) rfaRm provides a client interface to the
  Rfam database of RNA families. Data that can be retrieved include
  RNA families, secondary structure images, covariance models,
  sequences within each family, alignments leading to the
  identification of a family and secondary structures in the
  dot-bracket format.

- [rhdf5filters](/packages/rhdf5filters) Provides a collection of
  compression filters for use with HDF5 datasets.

- [ribor](/packages/ribor) The ribor package provides an R Interface
  for .ribo files. It provides functionality to read the .ribo file,
  which is of HDF5 format, and performs common analyses on its
  contents.

- [ribosomeProfilingQC](/packages/ribosomeProfilingQC) Ribo-Seq (also
  named ribosome profiling or footprinting) measures translatome
  (unlike RNA-Seq, which sequences the transcriptome) by direct
  quantification of the ribosome-protected fragments (RPFs). This
  package provides the tools for quality assessment of ribosome
  profiling. In addition, it can preprocess Ribo-Seq data for
  subsequent differential analysis.

- [RNAAgeCalc](/packages/RNAAgeCalc) It has been shown that both DNA
  methylation and RNA transcription are linked to chronological age
  and age related diseases. Several estimators have been developed to
  predict human aging from DNA level and RNA level. Most of the human
  transcriptional age predictor are based on microarray data and
  limited to only a few tissues. To date, transcriptional studies on
  aging using RNASeq data from different human tissues is limited.
  The aim of this package is to provide a tool for across-tissue and
  tissue-specific transcriptional age calculation based on GTEx
  RNASeq data.

- [ROCpAI](/packages/ROCpAI) The package analyzes the Curve ROC,
  identificates it among different types of Curve ROC and calculates
  the area under de curve through the method that is most accuracy.
  This package is able to standarizate proper and improper pAUC.

- [ROSeq](/packages/ROSeq) A rank based approach to modeling gene
  expression with filtered and normalized read count matrix. Takes in
  the complete filtered and normalized read count matrix, the
  location of the two sub-populations and the number of cores to be
  used.

- [rrvgo](/packages/rrvgo) Reduce and visualize lists of Gene
  Ontology terms by identifying redudance based on semantic
  similarity.

- [rSWeeP](/packages/rSWeeP) The SWeeP method was developed to favor
  the analizes between amino acids sequences and to assist alignment
  free phylogenetic studies. This method is based on the concept of
  sparse words, which is applied in the scan of biological sequences
  and its the conversion in a matrix of ocurrences.  Aiming the
  generation of low dimensional matrices of Amino Acid Sequence
  occurrences.

- [sarks](/packages/sarks) Suffix Array Kernel Smoothing (see
  https://academic.oup.com/bioinformatics/article-abstract/35/20/3944/5418797),
  or SArKS, identifies sequence motifs whose presence correlates with
  numeric scores (such as differential expression statistics)
  assigned to the sequences (such as gene promoters). SArKS smooths
  over sequence similarity, quantified by location within a suffix
  array based on the full set of input sequences. A second round of
  smoothing over spatial proximity within sequences reveals
  multi-motif domains. Discovered motifs can then be merged or
  extended based on adjacency within MMDs. False positive rates are
  estimated and controlled by permutation testing.

- [scClassify](/packages/scClassify) scClassify is a multiscale
  classification framework for single-cell RNA-seq data based on
  ensemble learning and cell type hierarchies, enabling sample size
  estimation required for accurate cell type classification and joint
  classification of cells using multiple references.

- [scHOT](/packages/scHOT) Single cell Higher Order Testing (scHOT)
  is an R package that facilitates testing changes in higher order
  structure of gene expression along either a developmental
  trajectory or across space. scHOT is general and modular in nature,
  can be run in multiple data contexts such as along a continuous
  trajectory, between discrete groups, and over spatial orientations;
  as well as accommodate any higher order measurement such as
  variability or correlation. scHOT meaningfully adds to first order
  effect testing, such as differential expression, and provides a
  framework for interrogating higher order interactions from single
  cell data.

- [scMAGeCK](/packages/scMAGeCK) scMAGeCK is a computational model to
  identify genes associated with multiple expression phenotypes from
  CRISPR screening coupled with single-cell RNA sequencing data
  (CROP-seq)

- [SCOPE](/packages/SCOPE) Whole genome single-cell DNA sequencing
  (scDNA-seq) enables characterization of copy number profiles at the
  cellular level. This circumvents the averaging effects associated
  with bulk-tissue sequencing and has increased resolution yet
  decreased ambiguity in deconvolving cancer subclones and
  elucidating cancer evolutionary history. ScDNA-seq data is,
  however, sparse, noisy, and highly variable even within a
  homogeneous cell population, due to the biases and artifacts that
  are introduced during the library preparation and sequencing
  procedure. Here, we propose SCOPE, a normalization and copy number
  estimation method for scDNA-seq data. The distinguishing features
  of SCOPE include: (i) utilization of cell-specific Gini
  coefficients for quality controls and for identification of
  normal/diploid cells, which are further used as negative control
  samples in a Poisson latent factor model for normalization; (ii)
  modeling of GC content bias using an expectation-maximization
  algorithm embedded in the Poisson generalized linear models, which
  accounts for the different copy number states along the genome; (iii) a 
  cross-sample iterative segmentation procedure to identify
  breakpoints that are shared across cells from the same genetic
  background.

- [scry](/packages/scry) Many modern biological datasets consist of
  small counts that are not well fit by standard linear-Gaussian
  methods such as principal component analysis. This package provides
  implementations of count-based feature selection and dimension
  reduction algorithms. These methods can be used to facilitate
  unsupervised analysis of any high-dimensional data such as
  single-cell RNA-seq.

- [scTHI](/packages/scTHI) scTHI is an R package to identify active
  pairs of ligand-receptors from single cells in order to study,among
  others, tumor-host interactions. scTHI contains a set of signatures
  to classify cells from the tumor microenvironment.

- [selectKSigs](/packages/selectKSigs) A package to suggest the
  number of mutational signatures in a collection of somatic
  mutations using calculating the cross-validated perplexity score.

- [SimFFPE](/packages/SimFFPE) This package simulates artifact
  chimeric reads specifically generated in next-generation sequencing
  (NGS) process of formalin-fixed paraffin-embedded (FFPE) tissue.

- [SingleCellSignalR](/packages/SingleCellSignalR) Allows single cell
  RNA seq data analysis, clustering, creates internal network and
  infers cell-cell interactions.

- [snapcount](/packages/snapcount) snapcount is a client interface to
  the Snaptron webservices which support querying by gene name or
  genomic region. Results include raw expression counts derived from
  alignment of RNA-seq samples and/or various summarized measures of
  expression across one or more regions/genes per-sample (e.g.
  percent spliced in).

- [sparseMatrixStats](/packages/sparseMatrixStats) High performance
  functions for row and column operations on sparse matrices. For
  example: col / rowMeans2, col / rowMedians, col / rowVars etc.
  Currently, the optimizations are limited to data in the column
  sparse format. This package is inspired by the matrixStats package
  by Henrik Bengtsson.

- [spicyR](/packages/spicyR) spicyR provides a series of functions to
  aid in the analysis of both immunofluorescence and mass cytometry
  imaging data as well as other assays that can deeply phenotype
  individual cells and their spatial location.

- [spqn](/packages/spqn) The spqn package implements spatial quantile
  normalization (SpQN). This method was developed to remove a
  mean-correlation relationship in correlation matrices built from
  gene expression data. It can serve as pre-processing step prior to
  a co-expression analysis.

- [struct](/packages/struct) Defines and includes a set of
  class-based templates for developing and implementing data
  processing and analysis workflows, with a strong emphasis on
  statistics and machine learning. The templates can be used and
  where needed extended to 'wrap' tools and methods from other
  packages into a common standardised structure to allow for
  effective and fast integration. Model objects can be combined into
  sequences, and sequences nested in iterators using overloaded
  operators to simplify and improve readability of the code.
  STATistics Ontology (STATO) has been integrated and implemented to
  provide standardised definitions for methods, inputs and outputs
  wrapped using the class-based templates.

- [structToolbox](/packages/structToolbox) An extensive set of data
  (pre-)processing and analysis methods and tools for metabolomics
  and other omics, with a strong emphasis on statistics and machine
  learning. This toolbox allows the user to build extensive and
  standardised workflows for data analysis. The methods and tools
  have been implemented using class-based templates provided by the
  struct (Statistics in R Using Class-based Templates) package. The
  toolbox includes pre-processing methods (e.g. signal drift and
  batch correction, normalisation, missing value imputation and
  scaling), univariate (e.g. ttest, various forms of ANOVA,
  Kruskal–Wallis test and more) and multivariate statistical methods
  (e.g. PCA and PLS, including cross-validation and permutation
  testing) as well as machine learning methods (e.g. Support Vector
  Machines). The STATistics Ontology (STATO) has been integrated and
  implemented to provide standardised definitions for the different
  methods, inputs and outputs.

- [SynExtend](/packages/SynExtend) Shared order between genomic
  sequences provide a great deal of information. Synteny objects
  produced by the R package DECIPHER provides quantitative
  information about that shared order. SynExtend provides tools for
  extracting information from Synteny objects.

- [TAPseq](/packages/TAPseq) Design primers for targeted single-cell
  RNA-seq used by TAP-seq. Create sequence templates for target gene
  panels and design gene-specific primers using Primer3. Potential
  off-targets can be estimated with BLAST. Requires working
  installations of Primer3 and BLASTn.

- [TBSignatureProfiler](/packages/TBSignatureProfiler) Signatures of
  TB progression, TB disease, and other TB disease states have been
  created. This package makes it easy to profile RNA-Seq data using
  these signatures and common signature profiling tools including
  ASSIGN, GSVA, and ssGSEA.

- [tidybulk](/packages/tidybulk) This is a collection of utility
  functions that allow to perform exploration of and calculations to
  RNA sequencing data, in a modular, pipe-friendly and tidy fashion.

- [timeOmics](/packages/timeOmics) timeOmics is a generic data-driven
  framework to integrate multi-Omics longitudinal data measured on
  the same biological samples and select key temporal features with
  strong associations within the same sample group. The main steps of
  timeOmics are: 1. Plaform and time-specific normalization and
  filtering steps; 2. Modelling each biological into one time
  expression profile; 3. Clustering features with the same expression
  profile over time; 4. Post-hoc validation step.

- [TreeAndLeaf](/packages/TreeAndLeaf) TreeAndLeaf package comes as
  an alternative to solve problems regarding dendrogram plotting,
  such as the lack of space when the dendrogram is too large and the
  need for adding more layers of information. It treats a whole
  dendrogram as a tree, in which the observations are represented by
  the leaves.

- [tscR](/packages/tscR) Clustering for time series data using slope
  distance and/or shape distance.

- [vasp](/packages/vasp) Discovery of genome-wide variable
  alternative splicing events from short-read RNA-seq data and
  visualizations of gene splicing information for publication-quality
  multi-panel figures.

- [weitrix](/packages/weitrix) Data type and tools for working with
  matrices having precision weights and missing data. This package
  provides a common representation and tools that can be used with
  many types of high-throughput data. The meaning of the weights is
  compatible with usage in the base R function "lm" and the package
  "limma". Calibrate weights by scaling weights row-wise to account
  for known predictors of precision. Find PCA-like components of
  variation even with many missing values, rotated so that individual
  components may be meaningfully interpreted. DelayedArray matrices
  and BiocParallel are supported.


New Data Experiment Packages
=====================

There are 10 new data experiment packages in this release of Bioconductor.

- [curatedAdipoArray](/packages/curatedAdipoArray) A curated dataset
  of Microarrays samples. The samples are MDI- induced pre-adipocytes
  (3T3-L1) at different time points/stage of differentiation under
  different types of genetic (knockdown/overexpression) and
  pharmacological (drug treatment) perturbations. The package
  documents the data collection and processing. In addition to the
  documentation, the package contains the scripts that was used to
  generated the data.

- [dorothea](/packages/dorothea) This package contains human and
  mouse TF regulons. The human regulons were curated and collected
  from different types of evidence such as literature curated
  resources, ChIP-seq peaks, TF binding site motifs and interactions
  inferred directly from gene expression. The mouse regulons were
  constructed by mapping the human gene symbols to their orthologs in
  mice. Those regulons can be coupled with any statistical method
  that aims to analyse gene sets to infer TF activity from gene
  expression data. Preferably the statistical method viper is used.

- [gpaExample](/packages/gpaExample) Example data for the GPA
  package, consisting of the p-values of 1,219,805 SNPs for five
  psychiatric disorder GWAS from the psychiatric GWAS consortium
  (PGC), with the annotation data using genes preferentially
  expressed in the central nervous system (CNS).

- [HighlyReplicatedRNASeq](/packages/HighlyReplicatedRNASeq)
  Gene-level count matrix data for bulk RNA-seq dataset with many
  replicates. The data are provided as easy to use
  SummarizedExperiment objects. The source data that is made
  accessible through this package comes from
  https://github.com/bartongroup/profDGE48.

- [optimalFlowData](/packages/optimalFlowData) Data files used as
  examples and for testing of the software provided in the
  optimalFlow package.

- [scTHI.data](/packages/scTHI.data) Data for the vignette and
  tutorial of the package scTHI.

- [SingleCellMultiModal](/packages/SingleCellMultiModal)
  SingleCellMultiModal is an ExperimentHub package that serves
  multiple datasets obtained from GEO and other sources and
  represents them as MultiAssayExperiment objects. The current focus
  is on datasets that use new technologies such as scNMT and scM&T.

- [spatialLIBD](/packages/spatialLIBD) Inspect interactively the
  spatial transcriptomics 10x Genomics Visium data from Maynard,
  Collado-Torres et al, 2020 analyzed by Lieber Institute for Brain
  Development researchers and collaborators.

- [spqnData](/packages/spqnData) Bulk RNA-seq from GTEx on 4,000
  randomly selected, expressed genes. Data has been processed for
  co-expression analysis.

- [WGSmapp](/packages/WGSmapp) This package provides whole-genome
  mappability tracks on human hg19/hg38 assembly. We employed the
  100-mers mappability track from the ENCODE Project and computed
  weighted average of the mappability scores if multiple ENCODE
  regions overlap with the same bin. “Blacklist” bins, including
  segmental duplication regions and gaps in reference assembly from
  telomere, centromere, and/or heterochromatin regions are included.
  The dataset consists of three assembled .bam files of single-cell
  whole genome sequencing from 10X for illustration purposes.



New Annotation Packages
=====================

There are 5 new annotation packages in this release of Bioconductor.

- [EpiTxDb.Hs.hg38](/packages/EpiTxDb.Hs.hg38)
- [EpiTxDb.Sc.sacCer3](/packages/EpiTxDb.Sc.sacCer3)
- [EpiTxDb.Mm.mm10](/packages/EpiTxDb.Mm.mm10)
- [MafH5.gnomAD.r3.0.GRCh38_3.11.0](/packages/MafH5.gnomAD.r3.0.GRCh38_3.11.0)
- [phastCons30way.UCSC.hg38_3.11.1](/packages/phastCons30way.UCSC.hg38_3.11.1)


New Workflow Packages
=====================

There is 1 new workflow package in this release of Bioconductor. 

- [fluentGenomics](/packages/fluentGenomics) An extended workflow
  using the plyranges and tximeta packages for fluent genomic data
  analysis. Use tximeta to correctly import RNA-seq transcript
  quantifications and summarize them to gene counts for downstream
  analysis. Use plyranges for clearly expressing operations over
  genomic coordinates and to combine results from differential
  expression and differential accessibility analyses.


NEWS from new and existing Software Packages
===================================


[ACE](/packages/ACE)
---

                 Changes in version 1.5.2 (2020-04-14)                  

- updated maintainer e-mail address

                 Changes in version 1.5.1 (2020-01-14)                  

- fixed a bug in getadjustedsegments that joined adjacent chromosomes
  with same segment values

[affylmGUI](/packages/affylmGUI)
---------

                       Changes in version 1.62.0                        

- Update affylmGUI to use AnnotationDbi and the new ".db" style
  probe-set annotation packages.

- Reduce overlap with the limma package.  affylmGUI now uses the limma
  functions lmFit(), eBayes(), topTable() and limmaUsersGuide()
  directly.

- Remove functions ALGchangeLog(), ALGlimmaUsersGuide(),
  deleteItemFromList(), toptable2() and topTable2().

- Add affylmGUI-package help file. Split existing help file into two.

- Various internal code updates and simplifications.

[alevinQC](/packages/alevinQC)
--------

                        Changes in version 1.3.1                        

- Added the possibility of providing custom cell barcode sets

[AlphaBeta](/packages/AlphaBeta)
---------

                 Changes in version 1.2.1 (2020-04-20)                  

- Fixed bugs.

- Made the following significant changes + Using ape package to make
  tree. + Added some plots.

- Added SOMA functions.

- Plotting estimates

[alsace](/packages/alsace)
------

                       Changes in version 1.22.1                        

- replaced two calls to 'class' to 'inherits'

[AneuFinder](/packages/AneuFinder)
----------

                       Changes in version 1.15.2                        

BUG FIXES

- Compatibility update: Replaced deprecated
  fetchExtendedChromInfoFromUCSC() with getChromInfoFromUCSC().

[AnnotationHub](/packages/AnnotationHub)
-------------

                       Changes in version 2.19.0                        

BUG FIX

- (2.19.12) change has_internet to nslookup to avoid locations that do
  not have access to google dns.

NEW FEATURES

- (2.19.8) Fixed messages to be more informative

- (2.19.8) trycatch around bfcneedsupdate/bfcdownload for sqlite
  database to prevent error when internet is spotty and databse source
  cannot be checked for a newer version.

- (2.19.4) trycatch around bfcneedsupdate call will prevent error when
  internet is spotty and resource cannot be checked for a newer
  version.

- (2.19.3) New function removeResources for removing locally downloaded
  cached resources. Nicer interface than using cache. Improved
  documentation on use of setting a cache to NULL to also remove
  resources.

- (2.19.2) Adds option ask to constructor. Default: True. Ask will ask
  user to create default AnnotationHub location and if in an non
  interactive session utilize a temporary directory. If False, the
  default directory is created and utilized without prompting.

USER-VISIBLE MODIFICATIONS

- (2.19.1) Improve ERROR message when resource isn't found. Also
  involved updating RecordStatus function. If resource hasn't been
  completely removed from database, gives when the resource was
  removed. If the resource is available but after the snapshot date
  being utilized, specialized message.

[AnnotationHubData](/packages/AnnotationHubData)
-----------------

                       Changes in version 1.17.0                        

MODIFICATIONS

- 1.17.3 add check for valid Title and Description in metadata file. It
  should not be empty or NA

- 1.17.2 add XML as valid source type

- 1.17.1 add GSEMatrix as valid source type

[anota2seq](/packages/anota2seq)
---------

                        Changes in version 1.9.1                        

- Fixed coding that would results in errors in R version 4.0 related to
  used of class(...) == "x".

[APAlyzer](/packages/APAlyzer)
--------

                 Changes in version 1.1.2 (2020-03-18)                  

- Add the download_testbam to download the testing data bam files.

- Add extra section 'Complete Analysis Example' in vignettes

- Revise some typos in vignettes and mans.

                 Changes in version 1.1.1 (2020-03-15)                  

- Add the PAS2GEF to convert GTF to PAS reference.

- Add the REF4PAS to generate reference region for both 3'UTR and
  introic PAS at the same time.

- Add the APAVolcano and APABox to plot the APAdiff results.

- Add Adjusted p-value option in APAdiff

- Allow the user to set argument minMQS to filter minimum mapping
  quality score of counted reads.

- Add the build-in PAS reference for hg38 and mm10.

- Add the build-in PAS reference for all PASs including conserved and
  non-conserved.

- Update vignettes and mans.

- PAS reference regions are now moved to a separate repo.

                 Changes in version 1.1.0 (2020-02-27)                  

- Revise some typos in vignettes and mans.

                 Changes in version 1.0.1 (2020-01-28)                  

- Revise some typos in vignettes and mans.

[aroma.light](/packages/aroma.light)
-----------

                 Changes in version 3.17.1 (2019-12-09)                 

CODE REFACTORING

- Now importing throw() from R.oo instead of R.methodsS3.

                 Changes in version 3.17.0 (2019-10-30)                 

- The version number was bumped for the Bioconductor devel version,
  which is now BioC 3.11 for R (>= 3.6.1).

[artMS](/packages/artMS)
---------

                        Changes in version 1.6.0                        

- Major package updates. It addresses most of the warnings and messages from required packages

- Analysis of Protein Acetylation now supported.

- General to all Quality control plots: update default names for all pdf outputs

- QC artmsQualityControlEvidenceExtended():

    + New plots: peptide and protein overlap across bioreplicates and conditions (plotPEPTOVERLAP and plotPROTOVERLAP)
    + Updates and improves all the plots, affecting font size and other
    aesthetic options
    + Change parameter "plotIC" to "plotPCA" in artmsQualityControlEvidenceExtended()

- QC artmsQualityControlEvidenceBasic():

   + Depracated: Plot correlation distributions
   + Updated and improved all the plots, affecting font size and other aesthetic

- More data testing datasets available:

  + artms_data_ph_contrast: contrast object for available the PH example dataset
  + artms_data_ph_config: configuration yaml object for the available PH example dataset
  + artms_data_ph_msstats_results: results data object from running artmsQuantification
  + artms_data_ph_msstats_modelqc: modelqc data object from running artmsQuantification

- Vignette updated

- Other minor fixes / improvements

                 Changes in version 1.4.3 (2020-02-08)                  

- artmsAnalysisQuantifications(), several updates:

  + change default value for "l2fc_thres" to 1
  + change defult value for "output_dir" to "analysis_quant"
  + NEW option: "outliers". It allows to remove outliers from abundance data

- Fix bug affecting artmsQualityControlSummaryExtended()

                 Changes in version 1.4.2 (2019-12-06)                  

- Fixed a crash when impute fails due to missing outlier data (part 2)

                 Changes in version 1.4.1 (2019-12-06)                  

- Fixed a crash when impute fails due to missing outlier data



[ASpediaFI](/packages/ASpediaFI)
---------

                 Changes in version 1.1.4 (2020-04-01)                  

- Changed formats of outputs from ASpediaFI

                 Changes in version 1.1.3 (2020-03-27)                  

- Changed maintainer's email address

                 Changes in version 1.1.2 (2020-03-04)                  

- Modified to ensure that permutation is performed only on gene nodes.

                 Changes in version 1.1.1 (2019-12-12)                  

- Added permutation test to DRaWR

[ASSIGN](/packages/ASSIGN)
------

                 Changes in version 1.23.1 (2020-04-18)                 

- ASSIGN no longer changes the working directory

- ASSIGN requires that output files do not exist before functions run

[ATACseqQC](/packages/ATACseqQC)
---------

                       Changes in version 1.11.9                        

- Fix a issue for splitGAlignmentsList, splitGAlignments by supply
the header info to mergeBam.

                       Changes in version 1.11.8                        

- Fix a issue 'mergeBam' 'destination' exists, 'overwrite' is FALSE
for splitGAlignmentsByCut.

                       Changes in version 1.11.7                        

- fix a typo in doc for readBamFile.

                       Changes in version 1.11.6                        

- fix the sample for shiftGAlignments.

                       Changes in version 1.11.5                        

- add function shiftGAlignments for single end reads.

                       Changes in version 1.11.4                        

- Fix the issue of if there is no reads in bam file for
shiftGAlignmentsList.

                       Changes in version 1.11.2                        

- Fix the issue of "&#91;E::sam_parse1&#93; unrecognized type N".

                       Changes in version 1.11.1                        

- Add flag parameter for splitBam.

[Autotuner](/packages/Autotuner)
---------

                        Changes in version 1.1.2                        

- Updated ploting function of DKE to make it easier to evaluate the
  results. o Added a new xlim filter to DKE ploting function o Added
  some flags to check for file types while using AutoTuner

                 Changes in version 1.0.1 (2019-07-06)                  

- Added DOI to cite code through zenodo.

[BANDITS](/packages/BANDITS)
-------

                        Changes in version 1.3.4                        

- R.utils added to Imports (needed to load gzipped equivalence classes
  from Salmon v. 1.2.0 onwards)

                        Changes in version 1.3.3                        

- citation updated

                        Changes in version 1.3.2                        

- added a new constraint: min 5 counts per group are required to
  analyze a group

                        Changes in version 1.3.1                        

- bug fixed in 'test_DTU' function (giving error: "object 'n_genes' not
  found")

- Added 1 section of the vignette for inference with 3 groups

                        Changes in version 1.2.2                        

- citation updated

                        Changes in version 1.2.1                        

- fixed bugs and incorporated edits up to version 1.3.2

[BASiCS](/packages/BASiCS)
------

                 Changes in version 2.0.0 (2020-04-01)                  

- Improve UI for BASiCS_TestDE BASiCS_TestHVG/BASiCS_TestLVG and
  related functions by creating classes. Usage of objects documented in
  vignette.

- Reduce code duplication and refactor plots to use ggplot2.

- Add rudimentary unit tests for loading and some class methods. Fix
  several minor bugs.

- Change barplot in variance decomposition function to use ggplot2.

                 Changes in version 1.99.1 (2020-04-14)                 

- Minor change in documentation of `BASiCS_VarianceDecomp` to use
  `ggplot2` as reference linl (instead of `RColorBrewer`) link to

                 Changes in version 1.99.0 (2020-04-07)                 

- Minor changes in vignette to increase figure sizes

- Version number updated to follow Bioc guidelines

                 Changes in version 1.9.26 (2020-03-25)                 

- Bugfix in BASiCS_Sim

                 Changes in version 1.9.25 (2020-03-13)                 

- `ConstrainType` deprecated within `BASiCS_MCMC` (internal option
  only)

- In `tests_mcmc_arguments.R`, adds `set.seed` to avoid random failure
  of `makeExampleBASiCS_Data`

                 Changes in version 1.9.24 (2020-03-13)                 

- Removes R from Depends in DESCRIPTION.

- Minor fix in the documentation of `BASiCS_PriorParam` to pass
  BiocCheck.

                 Changes in version 1.9.23 (2020-03-10)                 

- Adds `BASiCS_PriorParam` function. This function makes it easier to
  alter parameters for the prior distributions used in BASiCS. See
  issue #154.

- Adds RBFMinMax argument to `BASiCS_PriorParam`, which controls
  whether RBF are set at the minimum and maximum starting mu values.

- Adds MinGenesPerRBF argument to `BASiCS_PriorParam`. This controls
  the minimum number of genes between around the centre of each RBF.

                 Changes in version 1.9.22 (2020-03-09)                 

- Bugfix in plot method when Param="sigma2"

                 Changes in version 1.9.21 (2020-03-05)                 

- Minor fix in `BASiCS_MCMC` to fix warning (definition of `PriorMu`)

                 Changes in version 1.9.20 (2020-02-16)                 

- Fix bug in handling of tail probability tests that meant
  probabilities were not calculated correctly when Epsilon* = 0.

                 Changes in version 1.9.19 (2020-02-16)                 

- Adds a while loop in BASiCS_Sim to ensure it doesn't fail if some
  generated cells have zero counts.

                 Changes in version 1.9.18 (2020-02-16)                 

- Merges empirical Bayes changes with those from divide-and-conquer

- Minor changes in unit tests to account for empirical Bayes prior (
  taking ParamPrior$mu.mu as a vector, rather than a scalar)

                 Changes in version 1.9.17 (2020-02-16)                 

- Implementation of empirical Bayes (EB) approach for `mu` in
  `BASiCS_MCMC`

- Additional parameter `PriorMu` incorporated to `BASiCS_MCMC`. This
  enables internal calculation of the EB prior.

- Creates `.EmpiricalBayesMu` to derive EB prior for mu.

- `HiddenBASiCS_MCMC_InputCheck`, `HiddenBASiCS_MCMC_GlobalParams` and
  `HiddenBASiCS_MCMC_ExtraArgs` renamed as `.BASiCS_MCMC_InputCheck`,
  `.BASiCS_MCMC_GlobalParams` and `.BASiCS_MCMC_ExtraArgs`.

- Fixes minor bug in `.EmpiricalBayesMu`

- Fixes EB approach for no-spikes case (regression and no regression)

- Updates documentation of `BASiCS_MCMC` to include `PriorMu` as a
  parameter

- Removes dependency to `scater` functions in `.EmpiricalBayesMu`.

                 Changes in version 1.9.16 (2020-02-16)                 

- Adds extra line in `test_sim.R` to avoid warning in BiocCheck

                 Changes in version 1.9.15 (2020-02-16)                 

- Use expect_equal in scran test (now all suitable tests)

- Minor change in `test_differential_test_reg.R` to use 3 instead of 2
  decimal points. This hopefully avoids issues associated to a bug in
  `round`. For details see:
  https://bugs.r-project.org/bugzilla/show_bug.cgi?id=17668

                 Changes in version 1.9.14 (2020-02-10)                 

- Changes coda::effectiveSize for a more efficient internal
  implementation.

- Adds `CheckESS` and `MinESS` arguments to `BASiCS_TestDE` to filter
  genes for differential expression testing based on effective sample
  size.

                 Changes in version 1.9.13 (2020-02-05)                 

- Adds option for prior scaling parameters (relevant for new divide and
  conquer approach only; does not affect UI)

                 Changes in version 1.9.12 (2020-01-17)                 

- Fixes bug in `BASiCS_CorrectOffset` for no-spikes case

                 Changes in version 1.9.11 (2020-01-11)                 

- Add unit tests for each MCMC update.

                 Changes in version 1.9.10 (2019-12-13)                 

- Further clean up in `BASiCS_TestDE`. Creates new internal functions
  to avoid duplicated code across different tests: `.TestResults`,
  `.ShowTestResults`

                 Changes in version 1.9.9 (2019-12-12)                  

- Version bump after pull request merge

- R version requirement set to 3.6.0

- `dimnames<-` added for `BASiCS_Chain` objects

- Additional input checks for `BASiCS_DenoisedCounts` and
  `BASiCS_DenoisedRates`

                 Changes in version 1.9.8 (2019-12-12)                  

- Clean-up for `HiddenTailProbTestDE` -> renamed as `.TailProb`.
  Similar for `.ThresholdSearch`, `.EFDR` and `.EFNR`

- Extra unit test for `GenesSelect` parameter in `BASiCS_TestDE`

- Removes `HiddenThresholdSearchDetectHVG_LVG` as no longer needed.

- Changes in `BASiCS_DetectHVG` and `BASiCS_DetectLVG` to reuse same
  hidden functions as in `BASiCS_TestDE` (extra potential changes
  exist)

                 Changes in version 1.9.7 (2019-12-05)                  

- Fixes bug in mu updates (log(mu) - mu.mu, rather than log(mu -
  mu.mu)). This does not affect default prior, only empirical Bayes
  approach.

                 Changes in version 1.9.6 (2019-12-04)                  

- Minor changes in `test_misc.R` to expose hidden functions

                 Changes in version 1.9.5 (2019-12-04)                  

- Optional parameter added to all mcmc samplers to enable user-defined
  prior mean for log(mu).

- Requirement of R (>= 4.0.0) added as suggested by BiocCheck

                 Changes in version 1.9.4 (2019-12-04)                  

- Example chain objects updated to satisfy identifiability constrain

- `test_vardecomp.R` updated accordingly

                 Changes in version 1.9.3 (2019-12-04)                  

- Introduces `BASiCS_Draw` to enable synthetic data generation based on
  a `BASiCS_MCMC` object (similar to splatter)

- Style changes to `newBASiCS_Data`

- Changes in `test_data_examples.R` to cover all cases.

                 Changes in version 1.9.2 (2019-12-04)                  

- Adds `LazyData: true` in DESCRIPTION to pass R CMD Check (R-devel
  requirement)

- Updates `BASiCS_LoadChain` to remove dependency to `data.table`

- Alan added to README.md. Funding acknowledgement updated.

                 Changes in version 1.9.1 (2019-11-28)                  

- In `HiddenBASiCS_MCMC_Start`, `scran::computeSumFactors` was replaced
  by `scran::calculateSumFactors` in order to match new version of
  `scran`.

[batchelor](/packages/batchelor)
---------

                        Changes in version 1.4.0                        

- Support the use of arbitrary design matrices in regressBatches().

- Allow lists of objects to be directly passed into the ... for many
  functions.

- Added the clusterMNN() function for performing MNN on cluster
  centroids.

- Added get.variance= option to fastMNN() to return variance explained
  from PCA. Support d=NA to skip the PCA step altogether.

- Modified correctExperiments() to preserve non-conflicting rowData()
  fields.

[bayNorm](/packages/bayNorm)
-------

                       Changes in version 1.5.14                        

- Modify Main_mean_NB_spBay.

                       Changes in version 1.5.13                        

- Activate "Allow sparse matrix as output (valid only for mean version
  procedure)". Hence you can work on dgCMatrix throughout the process.
  Input which is not of type dgCMatrix will be converted to dgCMatrix
  at first.

                       Changes in version 1.5.12                        

- Deprecated: Allow sparse matrix as output (valid only for mean
  version procedure). Since sparse matrix is useful only when the
  matrix it is sparse.

- Calibrate documentations.

                       Changes in version 1.5.11                        

- Allow sparse matrix as output (valid only for mean version
  procedure).

- Let version of RELEASE_3_10 to be 1.4.11.

                       Changes in version 1.5.10                        

- Update RELEASE_3_10 1.4.2 to 1.4.3.

                        Changes in version 1.5.9                        

- Change back to Rcpp EstPrior function (sp_mat version).

- Use dgCMatrix instead of matrix in R.

- define ARMA\_64BIT\_WORD 1 and so on allows for big sparse matrix
  (learn from the liger R package).

                        Changes in version 1.5.8                        

- Change back to rowSums instead of using Rcpp EstPrior function.

                        Changes in version 1.5.7                        

- Add useful function file.

- Use Rcpp EstPrior function.

                        Changes in version 1.5.6                        

- as_matrix for transforming sparse matrix to matrix in R efficiently
  than as.matrix.

                        Changes in version 1.5.5                        

- Make BB_fun faster: speed up the access of row of dgCMatrix object.

- Modify EstPrior, using rowSums from Matrix package to calculated
  variance. Hence no need for Rcpp EstPrior function.

                        Changes in version 1.5.4                        

- Support sparse matrix: dgTMatrix and dgCMatrix.

                        Changes in version 1.5.3                        

- Add Rcpp EstPrior function.

                        Changes in version 1.5.2                        

- Vectorize computation in EstPrior function.

                        Changes in version 1.5.0                        

- Set default setting of BETA_vec=NULL.

- Update citation.

[BEclear](/packages/BEclear)
-------

                 Changes in version 2.3.1 (2020-03-18)                  

- added helpful output in case the gradient descent diverges

[BgeeCall](/packages/BgeeCall)
--------

                        Changes in version 1.2.0                        

- Moved from workflow to software section

                        Changes in version 1.1.1                        

- Better manage transcript version

- Manage output directory when calls are generated for several
  libraries

- Manage transcript version when calls are generated for several
  libraries

                        Changes in version 1.1.0                        

- Change kmer size of kallisto index from 21 to 15 for libraries with
  readLength <= 50bp

- Can choose the output directory

- By default use a simpler arborescence for the output directory

- Can use reference intergenic sequences generated by the community

- Can use custom reference intergenic sequences from a local fastq file

                        Changes in version 1.0.0                        

- Public release of BgeeCall package. Enjoy!!!

[BgeeDB](/packages/BgeeDB)
------

                       Changes in version 2.13.1                        

- Fix issue in the getData() function when trying to uncompress huge
  tar.gz files (like all human data or GTeX experiment)

[bigPint](/packages/bigPint)
-------

                 Changes in version 1.3.2 (2020-03-04)                  

Users can now input SummarizedExperiment data type to create any
bigPint plot.

[bioCancer](/packages/bioCancer)
---------

                       Changes in version 1.15.02                       

- update help page

- rename www/imgs to www/logo: confict with www/imgs of rediant.data
  package.

                       Changes in version 1.15.01                       

- Change the address of reactomeFI from
  "http://reactomews.oicr.on.ca:8080/" to "http://cpws.reactome.org/"

- update ReactomeFI2018.RDS

[BiocCheck](/packages/BiocCheck)
---------

                        Changes in version 1.23                         

BUG FIX

- (1.23.4) Update locations for NEWS

- (1.23.1) Fix False Positive class == check

[BiocDockerManager](/packages/BiocDockerManager)
-----------------

                 Changes in version 0.99.0 (2020-03-18)                 

- Submit to Bioconductor

[BiocFileCache](/packages/BiocFileCache)
-------------

                        Changes in version 1.11                         

BUG FIX

- (1.11.5) Update dplyr functions for dplyr_1.0.0 release

- (1.11.2) Fixes run of examples. Different result manual vs
  interactive so have conditional that will allow complete example if
  run by user rather than run automatically through the checks.

NEW FEATURES

- (1.11.1) Add makeCachedActiveBinding(). Allows data to be cached
  during ActiveBinding call. Avoids callback function executing data
  retrieval everytime.

- (1.11.3) Only asks for / reports use of temporary cache once per
  session. https://github.com/Bioconductor/TENxBrainData/issues/7

[BioCor](/packages/BioCor)
------

                        Changes in version 1.12                         

- Move to NEWS.md.  - Corrected "value of length: 2 type: logical".
- Avoid cropping the images.  - Updated documentation.

                       Changes in version 1.10.1                        

- Added a pkgdown website

[BiocParallel](/packages/BiocParallel)
------------

                        Changes in version 1.22                         

USER VISIBLE CHANGES

- (v 1.20.2) don't advance random number stream when used 'internally'.
  This behavior alters reproducibility of existing scripts relying
  (probably implicitly) on the advancing stream.
  https://github.com/Bioconductor/BiocParallel/issues/110

BUG FIXES

- (v 1.20.1) bplapply(), bpmapply(), bpvec() propagate names on
  arguments more correctly,
  https://github.com/Bioconductor/BiocParallel/issues/108

[BiocSet](/packages/BiocSet)
-------

                         Changes in version 1.1                         

BUG FIX

- (1.1.4) Fix tests for release

- (1.1.3) Update dplyr functions in preparation for dplyr 1.0.0
  release.

- (1.1.1) Fix tests that test the length of org objects.

NEW FEATURES

- (1.1.2) Allow contruction of sets from lists,
  `BiocSet(list(a=letters, b=LETTERS))`.

[biocViews](/packages/biocViews)
---------

                       Changes in version 1.55.0                        

NEW FEATURES

- (1.55.2) printNEWS and getPackageDescriptions now have argument
  option 'relativeLink` that allows for relative links instead of hard
  coded urls for website. Important for release announcement.

BUG FIX

- (1.55.1) If new package fails to build still include in release
  announcement and don't fail out for lack of version number.

[biosigner](/packages/biosigner)
---------

                       Changes in version 1.15.4                        

MINOR MODIFICATION

- biosigner.Rmd vignette: minor update

                       Changes in version 1.15.2                        

NEW FEATURE

- biosigner.Rmd vignette: ropls::view method replaces ropls::strF
  (deprecated)

[biotmle](/packages/biotmle)
-------

                       Changes in version 1.11.0                        

- Change of estimation backend from the `tmle` package to the `drtmle`
  package.

- Removal of option to have repeated subjects since unsupported in new
  backend.

- Adds argument `bppar_debug` to facilitate debugging around
  parallelization.

[brainflowprobes](/packages/brainflowprobes)
---------------

                        Changes in version 1.1.1                        

SIGNIFICANT USER-VISIBLE CHANGES

- Documentation website is now available at
http://LieberInstitute.github.io/brainflowprobes/. It gets updated
with every commit on the master branch (bioc-devel) using GitHub
Actions and pkgdown.

[BRGenomics](/packages/BRGenomics)
----------

                       Changes in version 0.99.30                       

- Change all ncores options to default to getOption("mc.cores", 2L) -
Rename n-dimensional binning functions aggregateByNdimBins() and
densityInNdimBins() - Redone documentation website, splitting up the
vignette - Updated package vignettes/guide pages in several places -
Other minor internal changes

                       Changes in version 0.99.25                       

- Bug fixes with expand_ranges arguments affecting
getCountsByRegions(), subsampleGRanges(), and getSpikeInNFs() -
Expanded testing, particularly for expand_ranges arguments and
import_bam() - Added options in mergeReplicates() - Various small doc
updates and minor internal changes

                       Changes in version 0.99.23                       

- Added support for non-basepair-resolution GRanges throughout, via
the expand_ranges argument - Substantial performance benefits for
less-sparse datasets (e.g.  whole read coverage) - Supported
everywhere, including counting functions, subsampling, merging,
normalization, etc.  - Rewrite of mergeGRangesData(): - Substantial
performance improvements for most datasets - No longer requires
basepair-resolution GRanges objects - Added options and flexibility
for merging reads as well as coverage data - Add a mergeReplicates()
function - Rewrite of makeGRangesBRG() that significantly improves
performance for sparser datasets (the datasets for which using the
function makes the most sense) - subsampleGRanges() no longer returns
a basepair-resolution GRanges by default - When field=NULL,
applyNFsGRanges() no longer returns a basepair-resolution GRanges by
default - Add use_bin_numbers option to n-dimension binning
functions; setting to false allows returning of bin values (the bin
center) instead of the ordinal bin numbers (indexes) - Quietly adding
support for GRangesList objects (not fully tested)

                       Changes in version 0.99.15                       

- Add pre-filtering to counting functions for performance - Some
additional clarification of readcounts vs. coverage signal in
counting and import functions - Change tidyChromosomes test - Remove
indirect links in doc pages (use only exact names of man pages)

                       Changes in version 0.99.10                       

- Code modifications to pass Bioconductor test builds on Windows: -
Make all examples and tests single core - Internally (not exported)
redefine mcMap (current implementation in package parallel needs to
be modified) - In tests and examples, test if on Windows before
attempting any bigWig file import

                       Changes in version 0.99.0                        

- Changes for Bioconductor submission: - Move to package versioning
0.99.x - Update R requirement to version 4.0 - Add new branch R3 to
allow users to install under R version >=3.5 - Various minor
formatting changes to codebase

                        Changes in version 0.8.1                        

- Fixed bug in import_bam() that produced warnings when shift
argument used to shift both 5' and 3' ends of reads (i.e. when
length(shift) == 2) - Updated included external datasets to be much
smaller - Included bam file now <200 reads - Included bigWig and
bedGraph files derived from the bam file - Minor update to vignette
to reflect the change in the bam file - Updated the included data()
objects (PROseq and PROseq_paired) - Shifted PRO-seq 3' bases to
remove the run-on base, and updated associated package tests - Added
xz compression to the files - Streamlined some method dispatch in
genebodies() and getDESeqDataSet() functions

                        Changes in version 0.8.0                        

- Add support for lists in data import functions - Add the
convenience function applyNFsGRanges() - Significant internal changes
to import_bam() - New test for paired-end reads (deprecated use of
Rsamtools::testPairedEndBam()) - Avoids any internal use of
bpiterate() - Dropped dependency on GenomicFiles package

                       Changes in version 0.7.10                        

- Add intersectByGene() and reduceByGene() functions - Minor vignette
updates

                        Changes in version 0.7.8                        

- Substantially updated vignette - Fully load rtracklayer (so
completely exported to users) - Add isBRG() function - Fixed bug in
spikeInNormGRanges() that failed to remove spike-in reads (aside from
maintaining those reads, normalization was otherwise correct) - Fixed
minor bug in which metaSubsample() automatically added rownames with
list input

                        Changes in version 0.7.7                        

- Bug fix in aggregateByNdimensionalBins() affecting simultaneous
aggregation of multiple data - Minor updates to documentation,
including an error in getDESeqResults() - Slightly expanded vignette

                        Changes in version 0.7.5                        

- Update bootstrapping functions - Add blacklisting support for
metaSubsample() - Related to blacklisting, NA values now ignored in
bootstrapping - Add additional melt options for signal counting
functions - Further expanded support for list inputs (lists of
GRanges datasets), including in getStrandedCoverage() - Add explicit
support for blacklisting in getDESeqDataSet() - Rewrite n-dimensional
binning functions, and add function for aggregating data within
n-dimensional bins - Changed arguments in binNdimensions() to only
accept dataframe inputs - Add densityInNdimensionalBins() function to
count points in each bin - Add aggregateByNdimensionalBins() function
to aggregate data within bins using arbitrary functions - Added
arguments for setting sample names in spike-in/normalization
functions - Various improvements and streamlining for method dispatch
and flexibility

                        Changes in version 0.7.0                        

- Added functions for counting and filtering spike-in reads - Added
functions for generating spike-in normalization factors - Added
support for lists of GRanges datasets throughout, including all
signal counting functions - Updating signal counting functions with a
blacklist argument, for ignoring reads from user-supplied regions -
Added wrappers for import_bam() for several common use cases - Update
getCountsByPositions(): - Added a melt option for returning melted
dataframes - Now returns an error by default if multi-width regions
are given (must be explicit) - Changed argument order in
getMaxPositionsBySignal()

                        Changes in version 0.5.6                        

- Update import_bam() function - Added support for paired-end bam
files - Added the shift argument - Made metaSubsample() functions
robust to unevaluated inputs - Small performance improvement to
genebodies() function - Multicore usage is again the default for
getDESeqResults()

                        Changes in version 0.5.3                        

- Substantial performance improvement for mergeGRangesData() - Make
single-core the default for getDESeqResults() - Fixed errant warning
message in binNdimensions() for integer inputs - Update namespace to
fully import GenomicRanges and S4Vectors - Changed R dependency,
evidently required by the updated data objects - Various
documentation updates - Added package documentation page - Minor
changes to examples

                        Changes in version 0.5.0                        

- Increased support for multiplexed GRanges objects across all
functions - Increased performance for bootstrapping over multiple
fields/multiplexed GRanges objects - Removed requirement to set
'field' argument for gettings counts over multiplexed GRanges

                        Changes in version 0.4.7                        

- Rewrote mergeGRangesData() to support the creation of multiplexed
GRanges objects - Made getCountsByRegions() and
getCountsByPositions() to return integers if input signal is integer

                        Changes in version 0.4.5                        

- Added and modified numerous arguments - Increased support for
normalization factors across signal counting functions - Modified
behavior of bootstrapping over multiple fields by removing explicit
access to .Random.seed

                        Changes in version 0.4.1                        

- Various documentation updates

                        Changes in version 0.4.0                        

- Added a NEWS.md file to track changes - Added support for bam files

[BridgeDbR](/packages/BridgeDbR)
--------
                       Changes in version 1.21.6                        

NEW FEATURES

- adds a getProperties() method

- adds a maps() method that can map multiple identifiers

SIGNIFICANT USER-LEVEL CHANGES

- the map() method now returns a data frame

                       Changes in version 1.21.5                        

BUG FIXES

- fixed links in the Vignette and some further clarification

                       Changes in version 1.21.4                        

BUG FIXES

- updated the Apache Derby library to 10.5.3.0 (fixing compatibility with newer ID mapping databases)

- replaced RCurl with curl to fix downloading on Windows

                       Changes in version 1.21.3                        

BUG FIXES

- force RCurl to use TSL1.2

- added clarification of external data sources used

                       Changes in version 1.21.2                        

BUG FIXES

- updated URL of the /data/gene_database/ location to point directly to bridgedb.github.io

- now uses the smaller Bs mapping database as test data in the example

- updated the code for getDatabase() to allow downloading data from any location provided by /data/gene_database/

                       Changes in version 1.21.1                        

BUG FIXES

- no longer tries to get Derby file names online


[BUSpaRse](/packages/BUSpaRse)
--------

                 Changes in version 1.1.4 (2020-04-23)                  

- Now tr2g_* functions can also extract transcriptomes.  - By
default, tr2g_* functions write the tr2g.tsv to disk.  - Write
filtered GTF and GFF3 files.  - List of available gene and transcript
biotypes from Ensembl can be queried by data().  - RefSeq GFF3
annotations can now be used in tr2g_gff3.

                 Changes in version 1.1.3 (2020-04-06)                  

- Allow filtering gene and transcript biotypes in tr2g and
get_velocity_files - Allow removing scaffolds and haplotypes from
genome annotations for tr2g

                 Changes in version 1.1.2 (2020-02-06)                  

- Removed RcppParallel and data.table dependencies, and hence all C++
multithreading - Added knee_plot function - Changed default in
read_count_output to tcc = FALSE

                 Changes in version 1.1.1 (2020-01-29)                  

- Sort GRanges for RNA velocity to make sure that exons are in the
right order; split no longer sorts within each element of
GRangesList.  - Export subset_annot, which makes sure that all
chromosomes in the annotation are present in the genome prior to
transcriptome extraction.

[CAGEfightR](/packages/CAGEfightR)
----------

                        Changes in version 1.7.5                        

- MAJOR/BREAKING UPDATE: quantifyCTSSs is now a generic with new
  defaults. When importing BigWig files, tiles are no specifed as a
  single number of tiles across the genome, rather than size in
  basepairs. The new default is to import the entire genome as a single
  tile. This should be the fastest option for small datasets, for
  larger datasets is usually faster to use multiple tiles.

- quantifyCTSSs now accepts bedGraph files. Unlike BigWig import, this
  works on Windows.

- New convert*-family of functions which can convert CTSSs between
  BigWig/BED/bedGraph formats.

- New checkCTSSs for checking whether CTSSs-files are correctly
  formatted, and if not, will hopefully output some useful diagnostics.

- New quickGenes wrapper function for collapsing cluster-level data to
  gene-level data. Check the vignette for example of usage.

- New shapeMean function.

- Many internal functions are now exposed as utils*-function. This
  should make it easier for power users and/or developers to modify and
  extend CAGEfightR.

- Some internal changes to quantifyCTSSs and quantifyClusters for
  slightly better performance.

- Fixed a small bug where assignGeneID counted NA as a gene when
  outputting a message. Note, this never affected the actual output of
  the function, just the message produced.

[CAGEr](/packages/CAGEr)
-----

                       Changes in version 1.30.0                        

BUG FIXES

- Correct usage of `&&` for vector comparison in pairs.DataFrame().

- Adjust to latest SummarizedExperiment MultiAssayExperiment versions.

[CAMERA](/packages/CAMERA)
------

                       Changes in version 1.43.2                        

BUG FIXES

- Change several && causing failure with _R_CHECK_LENGTH_1_CONDITION_
  and _R_CHECK_LENGTH_1_CONDITION_

                       Changes in version 1.43.1                        

BUG FIXES

- Improve vignette layout and fix typos, thanks to Anton Nikolas Waniek

[Cardinal](/packages/Cardinal)
--------

                       Changes in version 2.5.12                        

BUG FIXES

- Drawing a region-of-interest using 'selectROI()' should now update
  the plot properly in RStudio devices

- Fixed issues with automatically guessing a reasonable mass tolerance
  for certain pre-processing methods

- In 'mzBin()', the previous value of 'centroided()' from the original
  dataset is now preserved after binning

                       Changes in version 2.5.11                        

BUG FIXES

- Fixed bug in 'spatialShrunkenCentroids()' that produced in NaNs when
  discriminant scores were very large

- Fixed bug in 'spatialFastmap()' where subsetting produced vectors
  instead of matrices due to omitting 'drop=FALSE'

                       Changes in version 2.5.10                        

NEW FEATURES

- Added 'topFeatures()' method for 'spatialKMeans()'

- Added coercion to 'DataFrame' from 'MSImagingExperiment'

SIGNIFICANT USER-VISIBLE CHANGES

- Deprecated legacy classes (MSImageSet, etc.): class definitions will
  remain for supporting datasets from CardinalWorkflows, but methods
  operating on them will be defunct in BioC 3.12

                        Changes in version 2.5.9                        

NEW FEATURES

- Added 'aggregate()' method for imaging experiments including
  'summarizePixels()' and 'summarizeFeatures()'

- Added 'subset()' method for imaging experiments including
  'subsetPixels()' and 'subsetFeatures()'

SIGNIFICANT USER-VISIBLE CHANGES

- Deprecated 'dplyr' verbs in favor of the above functions; this is to
  remove the (rather large) dependency on the tidyverse for a
  relatively small functionality

BUG FIXES

- Fix bug in 'spatialDGMM()' printing caused by a change in
  default.stringsAsFactors() == FALSE in R 4.0

                        Changes in version 2.5.8                        

NEW FEATURES

- Automatic estimation of mass resolution will now work for 'processed'
  imzML with centroid spectra

- New getter/setter options for Cardinal options such as
  'getCardinalBPPARAM()' and 'setCardinalBPPARAM()'

SIGNIFICANT USER-VISIBLE CHANGES

- Default BPPARAM backend is now set to 'SerialParam()'; use
  'setCardinalBPPARAM()' to change the backend

- Expose '.view' argument of 'matter::chunk_apply()' in 'pixelApply()',
  'featureApply()', and 'spatialApply()'

- Previously-deprecated functions 'generateSpectrum()' and
  'generateImage()' are now defunct

- Removed defunct functions 'Binmat()' and 'topLabels()'

BUG FIXES

- Fix large intensity text cutoffs in 'image()' colorkey

                        Changes in version 2.5.7                        

BUG FIXES

- Fix bug when using 'rbind()' or 'cbind()' on 'MassDataFrame' and
  'PositionDataFrame'

                        Changes in version 2.5.6                        

BUG FIXES

- Fix bug when assigning a dense matrix via iData()<- for an
  'MSProcessedImagingExperiment' object

                        Changes in version 2.5.5                        

BUG FIXES

- Fix bug in 'show' method for 'SimpleImageList' caused by
  class(matrix) -> c("matrix", "array")

                        Changes in version 2.5.4                        

BUG FIXES

- Fix bugs in legacy classes caused by a change in
  default.stringsAsFactors() == FALSE in R 4.0

                        Changes in version 2.5.3                        

SIGNIFICANT USER-VISIBLE CHANGES

- The 'resolution' argument in all methods has been redefined to always
  mean **full bin widt for both "mz" and "ppm" units

                        Changes in version 2.5.2                        

SIGNIFICANT USER-VISIBLE CHANGES

- The 'pixelApply()', 'featureApply()', and 'spatialApply()' methods
  now internally use 'matter::chunk_apply()'

                        Changes in version 2.5.1                        

BUG FIXES

- Fixed bug in 'mzBin()' binning spectra incorrectly

- Fixed bug in contrast enhancement with missing intensities

[cbaf](/packages/cbaf)
----

                 Changes in version 1.10.0 (2020-04-08)                 

New Features

- A unified method for increasing the consistency of cancer terms used
  by various functions. This enables easier maintanance and faster
  inclusion of new terms whenever available.

- Terms are updated. Package can recongize even more cancer studies!

- Incluion of additional checkpoint to control the name of requested
  cancer studies.

[CellBench](/packages/CellBench)
---------

                        Changes in version 1.3.2                        

Bug Fixes

- apply_methods and time_methods now correctly produces results for
custom data.frames whose columns are not sorted from left to right

[CellMixS](/packages/CellMixS)
--------

                        Changes in version 1.3.7                        

- Add input checks for colnames

- Include score and warning for identical cells

[ceRNAnetsim](/packages/ceRNAnetsim)
-----------

                Changes in version 0.99.16 (2019-11-20)                 

- Fixed documentation fails

- Re-arranged vignettes

                 Changes in version 0.99.0 (2019-10-07)                 

- Submitted to Bioconductor

[CeTF](/packages/CeTF)
----

                       Changes in version 0.99.19                       

- Fix netGOTFPlot, RIFPlot and densityPlot functions

- Fix warning messages when ploting with ggplot2 package (densityPlot,
  enrichPlot, histPlot, RIFPlot, SmearPlot)

                       Changes in version 0.99.18                       

- Fix SmearPlot bugs.

                       Changes in version 0.99.17                       

- Updating citation.

- Adding CITATION file.

                       Changes in version 0.99.16                       

- Development version.

- Fix netGOTFPlot dependencies and add the label size parameter.

- Update vignette.

                       Changes in version 0.99.15                       

- Development version.

- Adding 'level' parameter to perform groupGO analysis in getGroupGO
  function.

- Adding getEnrich function.

- Adding enrichPlot and heatPlot function.

- Adding RIFPlot function.

- Adding diffusion function.

- Adding CircosTargets function.

- Fix netGOTFPlot function bugs.

- Updated vignette.

[ChAMP](/packages/ChAMP)
-----

                       Changes in version 2.17.3                        

- Fixed bug in runCombat, as some factors are numeric format.

[ChemmineR](/packages/ChemmineR)
---------

                       Changes in version 3.40.0                        

NEW FEATURES

- Prior to this version, V3000 sdf files could be read, but would be
  written back out in V2000 format. Now these files will be written
  back out as V3000 files.

[ChIPpeakAnno](/packages/ChIPpeakAnno)
------------

                       Changes in version 3.21.7                        

- Fix a potentical bug reported by millerh1.

                       Changes in version 3.21.6                        

- add new function findMotifsInPromoterSeqs. HANGES IN VERSION 3.21.5

- replace all `class` by `is`.

                       Changes in version 3.21.4                        

- Fix a bug in getAllPeakSequence when there is peak out side of
  genome.

                       Changes in version 3.21.3                        

- Update function summarizePatternInPeaks.

                       Changes in version 3.21.2                        

- Add function getGO.

                       Changes in version 3.21.1                        

- remove RangedData from ChIPpeakAnno.

[ChIPseeker](/packages/ChIPseeker)
----------

                       Changes in version 1.23.1                        

- update GEO data (51079/762820 GSM) (2019-12-20, Fri)

[chromstaR](/packages/chromstaR)
---------

                       Changes in version 1.13.1                        

BUG FIXES

- Compatibility update: Replaced deprecated
  fetchExtendedChromInfoFromUCSC() with getChromInfoFromUCSC().

[CHRONOS](/packages/CHRONOS)
-------

                       Changes in version 1.15.1                        

- Updates for _R_CHECK_LENGTH_1_LOGIC2_, _R_CHECK_LENGTH_1_CONDITION_,
  classEq.

[CiteFuse](/packages/CiteFuse)
--------

                       Changes in version 0.99.7                        

- Improved Vignettes - Add examples - Fix for BiocCheck - Remove Rcpp
implementation

[cleaver](/packages/cleaver)
-------

                 Changes in version 1.25.1 (2019-12-30)                 

- Add missing space in vignette.

[CluMSID](/packages/CluMSID)
-------

                        Changes in version 1.3.1                        

- fixed bug in mergeMS2spectra when used with external peaktable

[clusterExperiment](/packages/clusterExperiment)
-----------------

                        Changes in version 2.7.2                        

Bugs: 

- Fix error from R 4.0 changing definition of treatment of NULL ("For
  consistency, N <- NULL; N&#91;&#91;1&#93;&#93; <- val now turns N into a list also
  when val) has length one. This enables dimnames(r1)&#91;&#91;1&#93;&#93; <- "R1" for
  a 1-row matrix r1, fixing PR#17719 reported by Serguei Sokol.").
  Replaced all `<- c()` with `<- vector(...)` and the appropriate
  class.

                        Changes in version 2.7.1                        

Bugs: 

- Fix error in making sample dendrogram

[clusterProfiler](/packages/clusterProfiler)
---------------

                       Changes in version 3.15.3                        

- incorporate clusterProfiler.dplyr (2020-03-12, Thu) - arrange,
filter, group_by, mutate, rename, select, slice and summarize

                       Changes in version 3.15.2                        

- remove Suggests of KEGG.db as it will be deprecated in Bioconductor
3.11 (2020-01-14, Tue) - optimize enrichGO to use less memory
(2019-12-13, Fri) - re-implement read.gmt without using GSEABase, and
my own version is much more fasta :)

                       Changes in version 3.15.1                        

- e.g. user can pass fun=enrichGO to compareCluster without quoting
enrichGO (2019-12-02, Mon) - add keytype and readable info in
compareCluster output - mv compareClusterResult class defintion to
DOSE (2019-11-02, Sat) - mv fortify, barplot and dotplot for
compareClusterResult to enrichplot.

[clustifyr](/packages/clustifyr)
---------

                 Changes in version 0.99.0 (2019-11-06)                 

- Update to tutorials

- bug fixes

- cleanup for bioc

                 Changes in version 0.0.3 (2019-09-06)                  

- support of SingleCellExperiment objects and more

- bug fixes

                 Changes in version 0.0.2 (2019-05-29)                  

- Updated discussions, benchmark, and visualization example pages

- reorganized `clustifyrdata` data package

                 Changes in version 0.0.1 (2019-02-01)                  

- Added direct support for `seurat` and other object classes

- Updated readme and tutorial pages

- Moved some larger reference data to `clustifyrdata`

[cmapR](/packages/cmapR)
-----

                 Changes in version 0.99.0 (2020-01-13)                 

- NOTE: updated function names replacing '.' with '_' (ex:
  'parse_gctx')

- Submitted to Bioconductor

[CNVfilteR](/packages/CNVfilteR)
---------

                        Changes in version 1.1.6                        

SIGNIFICANT USER-VISIBLE CHANGES

- plotVariantsForCNV() allows multiple visual adjustements

MINOR

- Added error control in loadVCFs(): heterozygous.range and
  homozygous.range cannot overlap

                        Changes in version 1.1.5                        

SIGNIFICANT USER-VISIBLE CHANGES

- Added ignore.unexpected.rows parameter to loadCNVcalls() function to
  ignore CNV calls which deletion/duplication value is not the expected
  one.

BUG FIXES

- Some SNV callers can produce calls with a genotype indicating that
  the variant actually was not found. Variants with GTs like "0/0 ./.
  1/0" are now discarded.

- Certain SNV callers can produce SNV calls with more than two alt
  values (multiple alt alleles). These variants are now discarded.

                        Changes in version 1.1.4                        

SIGNIFICANT USER-VISIBLE CHANGES

- Improved CNVfilteR accuracy. margin.pct parameter was added to
  filterCNVs() and ht.deletions.threshold default value was modified to
  30.

- Loading variants process optimized (multiple times faster)

- filterCNVs() function optimized

- min.total.depth default value modified to 10

- Added multiple modifications to the vignette

BUG FIXES

- Fixed bug when plotting a CNV with no SNV variants falling in it

                        Changes in version 1.1.3                        

BUG FIXES

- Fixed bug when processing variants in the limits of
  heterozygous.range and heterozygous.range

                        Changes in version 1.1.2                        

SIGNIFICANT USER-VISIBLE CHANGES

- Added sample.name parameter to loadCNVcalls() function to allow
  explicitly setting sample name

- deletion and duplication parameters can be vectors in loadCNVcalls()
  function, so multiple values are allowed

BUG FIXES

- Fixed bug happening in loadSNPsFromVCF() when ref and alt support was
  0

- Fixed bug when processing list.support.field in loadSNPsFromVCF()

- Added error control when allele frequency is not numeric

- Minor fixes

[CNVPanelizer](/packages/CNVPanelizer)
------------

                       Changes in version 1.19.1                        

- Fix due to change of Matrix class type in R version 4

[CNVRanger](/packages/CNVRanger)
---------

                        Changes in version 1.4.0                        

- New function `cnvOncoPrint` to visualize CNV calls on overlapping
  genomic features across a sample population

[cogena](/packages/cogena)
------

                 Changes in version 1.21.2 (2020-04-21)                 

- update coExp for R4.0 due to Matrix is now Array.

[cola](/packages/cola)
----

                        Changes in version 1.3.2                        

- add a new vignette introducing functional enrichment

[combi](/packages/combi)
-----

                       Changes in version 0.99.9                        

- Removed bplapply functionality because of Windows errors

                       Changes in version 0.99.7                        

- Created a print method for combi objects - Stricter argument
checking for exported functions - Consequent naming of results

                        Changes in version 0.1.2                        

- Reduce build time using

                        Changes in version 0.1.0                        

- First private GitHub document

[compcodeR](/packages/compcodeR)
---------

                       Changes in version 1.23.1                        

- Moved vignette to Rmd

[ComplexHeatmap](/packages/ComplexHeatmap)
--------------

                        Changes in version 2.3.4                        

- add `alter_graphic()` to automatically generate alteration graphic
  functions.

- add `label`/`annotation_label` argument in `SingleAnnotation()` and
  `HeatmapAnnotation()`.

- improved the subsetting methods for `comb_mat` class

- rewrite `smartAlign2()`

                        Changes in version 2.3.3                        

- support **gridtext** package

- fixed the bug of node stack overflow when there are many identical
  rows for drawing the dendrogram. see
  (http://www.thinkingincrowd.me/2016/06/06/How-to-avoid-Stack-overflow-error-on-recursive/)

- support to set legend gaps

- print important messages, e.g. when number of rows are more than
  2000, interanlly use_raster is turned on and the message should be
  printed to users.

- UpSet plots: optimize the processing of many sets (current solution
  consumes huge memory)

- UpSet plots: the "intersect" and "union" modes are currectly
  calculated.

- `anno_block()`: add `show_name` argument.

                        Changes in version 2.3.2                        

- `anno_simple()`: fixed a bug when pch are all NA in a slice

- adjust code according to grid 4.0.

- move scripts in test_not_run/ to tests/ folder

- `Heatmap()`: `cluster_row_slice`/`cluster_column_slice` set to TRUE
  by default for character matrix and when dendrogram is already
  provided.

- `smartAlign2()`: improved the code that positions are correctly
  calculated.

- row titles are in the correct order if they are set as "template".

                        Changes in version 2.3.1                        

- `anno_points()`: allows images as symbols.

- add `HEATMAP_LEGEND_PADDING` and `ANNOTATION_LEGEND_PADDING` options
  in `ht_opt`

- `oncoPrint()`: print messages if there are NA values in the input
  matrix

[ConsensusClusterPlus](/packages/ConsensusClusterPlus)
--------------------

Changes in Version 1.52.0

- Brief R code update for compatibilty with R 4.0.  
- Improved messages.  
- Deprecated kmdist clustering option


[consensusDE](/packages/consensusDE)
-----------

                 Changes in version 1.5.1 (2019-12-03)                  

- strand_mode argument replaces ignore_strand in buildSummarized.

- fragments default in buildSummarized changed to FALSE

- update vignette

[CoreGx](/packages/CoreGx)
------

                        Changes in version 1.0.0                        

- Implemented molecularProfiles as SummarizedExperiments instead of
ExpressionSets - Modifying generic implementation to add ...; allows
for additional arguments in setMethods() for CoreGx generics -
Removing package from CRAN] - Submitting package to Bioconductor

[csaw](/packages/csaw)
----

                       Changes in version 1.22.0                        

- Removed deprecated functionality in consolidateTests() and related
  functions.

- Modified the method used to count the number of up/down windows in
  combineTests(). Reported a representative test in the output along
  with its log-fold changes.

- Renamed mixedClusters() to mixedTests() with deprecation of the
  former.

- Added minimalTests() for cluster-level summarization with a minimum
  proportion of significant tests.

- Standardized the output of getBestTest(), empiricalFDR(),
  mixedTests() and minimalTests() to that of combineTests(). In
  particular, getBestTest() now returns the index of its best window in
  the rep.test field.

[ctgGEM](/packages/ctgGEM)
------

                 Changes in version 0.99.0 (2020-01-07)                 

- submitted to Bioconductor

[customProDB](/packages/customProDB)
-----------

                       Changes in version 1.26.1                        

BUG FIXES

- Fix a small bug in function PrepareAnnotationRefseq.R

[cydar](/packages/cydar)
-----

                       Changes in version 1.12.0                        

- prepareCellData() now returns a List rather than a half-complete
  CyData object.

- expandRadius() and neighborDistances() work explicitly on the List
  produced by prepareCellData().

- countCells() will compute weighted-median intensities for unused
  markers as well. Minor bugfix to avoid stochastic results due to
  numerical imprecision when computing medians.

- intensities() has a new mode= argument to return intensities for
  unused markers.

[cytomapper](/packages/cytomapper)
----------

                 Changes in version 0.99.5 (2020-04-10)                 

- Added script to generate the toy dataset

- Modified the toy dataset

- Added more explanation on datasets to vignette

- Bug fixes when colouring cells

- Unit tests on all hidden functions

- Account for over-saturation when merging colours

- Fixed labelling of colour legends

- Corrected typos in docs and vignette

- Avoid recalculation of min/max values in legend function

                 Changes in version 0.99.4 (2020-04-01)                 

- Fixed docs to pass checks on Windows

- Aesthetic changes to scale bar, image title and legend title

- Removed channelNames as Image generic

- removed exported plot method

                 Changes in version 0.99.0 (2020-03-30)                 

- Formatted the package for Bioconductor submission

                 Changes in version 0.1.13 (2020-03-28)                 

- Full vignette and correction to docs

                 Changes in version 0.1.12 (2020-03-25)                 

- Renamed package from SingleCellMapper to cytomapper

- Renamed ImageList class to CytoImageList class

                 Changes in version 0.1.11 (2020-03-24)                 

- Complete documentation of all package functions

                 Changes in version 0.1.10 (2020-03-21)                 

- interpolate parameter

- Bug fix to achieve correct aspect ratios

                 Changes in version 0.1.9 (2020-03-20)                  

- scale, return_plot, return_images, legend, margin, display parameters

- Finalised unit tests

                 Changes in version 0.1.8 (2020-03-18)                  

- Option to remove legend

- Cleaned up representation of legend on plot

                 Changes in version 0.1.7 (2020-03-17)                  

- Added scaleImages and normalize functions

- Expanded shared parameters for plotPixels and plotCells

                 Changes in version 0.1.6 (2020-03-10)                  

- Unit tests on plotCells and plotPixels

- Improved setting colours

- Rewrite to reduce validity checks

                 Changes in version 0.1.5 (2020-03-03)                  

- plotCells function

- plotPixels function

- Options for further customization of scale_bar

- Options for image_title customization

- Started vignette draft

                 Changes in version 0.1.4 (2020-02-16)                  

- Full docs and test on: - loadImages - ImageList-class

                 Changes in version 0.1.3 (2020-02-16)                  

- Full docs and tests on: - ImageList-naming - ImageList-subsetting

                 Changes in version 0.1.2 (2020-02-03)                  

- channelNames generic function for Image and ImageList object

- Reduced the size of example data

- Wrote docs on data

                 Changes in version 0.1.1 (2020-02-02)                  

- Simplified structure of loadImages function

- Created validity checks for path and file inputs

- Fixed bugs to pass R CMD check

- test

                 Changes in version 0.1.0 (2020-01-16)                  

- Initial commit

- Creation of the SingleCellMapper package

- Created inst/extdata folder with example images and data

- Build the ImageList class

- Created first unit tests for ImageList constructor

- test

[CytoML](/packages/CytoML)
------

                        Changes in version 3.11                         

API Changes

- Rename argument sampNLoc -> sample_names_from in open_flowjo_xml -
All parsers (flowjo/cytobank/diva_to_gatingset) now return GatingSet
based on cytoset rather than ncdfFlowSet - Add trans argument to
cytobank_to_gatingset to allow overriding of transformations from
gatingML file (#76) - gatingset_to_flowjo now uses a docker image
with a compiled converter: hub.docker.com/r/wjiang2/gs-to-flowjo -
Some updates to how flowjo_to_gatingset searches for FCS files (#77)
- Add include_empty_tree option to flowjo_to_gatingset to include
samples without gates - Allow gatingset_to_flowjo to take a path to a
GatingSet archive directory - Add gating_graphGML to replace
gating.graphGML method for openCyto::gating generic - Filter samples
by panel when parsing cytobank experiment and add ce_get_samples,
ce_get_panels

Fixes/internal changes

- Automatic time scaling of samples from FlowJo workspaces now
handled by flowjo_to_gatingset RGLab/cytolib#33 - Handle change to
default stringsAsFactors=FALSE in R 4.0 - Eliminated extra
intermediate files left in temp directory during workspace parsing -
Switch usage of GatingSetList to merge_gs_list - Solve some Windows
build issues - Switch from experimental::filesystem to
boost::filesystem in C++ FlowJo parser - Add CytoML XSD to
installation

                        Changes in version 3.10                         

API Changes

- Change handling of quad gates according to RGLab/cytolib#16

- Renaming of methods:

- openWorkspace -> open_diva_xml, open_flowjo_xml -
cytobankExperiment -> open_cytobank_experiment - cytobank2GatingSet
-> cytobank_to_gatingset - parseWorkspace -> flowjo_to_gatingset,
diva_to_gatingset - getSampleGroups -> fj_ws_get_sample_groups,
diva_get_sample_groups - getSamples -> fj_ws_get_samples,
diva_get_samples - getKeywords -> fj_ws_get_keywords -
getCompensationMatrices -> ce_get_compensations - getTransformation
-> ce_get_transformations - compare.counts ->
gs_compare_cytobank_counts

- Renaming of classes:

- divaWorkspace -> diva_workspace - flowJoWorkspace ->
flowjo_workspace

- Add CytoML.par.set, CytoML.par.get for setting parameters in CytoML
namespace

Fixes/internal changes

- Make gatingset_to_cytobank export cytobank ML with attribute
namespaces - Allow diva_to_gatingset to use compensation matrix from
xml - Pass ... args from cytobank_to_gatingset appropriately down to
FCS parser - Fix some issues with scaling of gates parsed from Diva
workspace (#64) - Guard against unsupported transformations being
added to GatingSet during Diva parsing - Switch diva_to_gatingset to
using flowjo_log_trans instead of logtGml2_trans - Fix ported
flowUtils::xmlTag to enable self-closing tags - Make gating.graphGML
lookup tailored gates by FCS name as well as file id - Add some
flexibility to getSpilloverMat used in gatingset_to_flowjo

[DAMEfinder](/packages/DAMEfinder)
----------

                       Changes in version 0.99.3                        

- Change some files to other folders

                       Changes in version 0.99.2                        

- Fix spacing (ggplot code looks funky now)

- Remove make_bigwigs func

- Fix examples for plotting functions

                       Changes in version 0.99.1                        

- Fix description file

                       Changes in version 0.99.0                        

- First version submitted to Bioconductor

                        Changes in version 0.2.0                        

- New SNP-based ASM score. Includes VCF and BAM files as input.

- New p-value assignment methods.

- New plotting functions for MDS and Methyl-circle-plots.

                       Changes in version 0.1.13                        

- Last version before S.Orjuela started working on it.

[DaMiRseq](/packages/DaMiRseq)
--------

                        Changes in version 2.0.0                        

- Since version 2.0.0 of the software, DaMiRseq offers a solution to
  solve two distinct problems, in supervised learning analysis: (i)
  finding a small set of robust features, and (ii) building the most
  reliable model to predict new samples;

- The functions DaMiR.EnsembleLearning2cl_Training,
  EnsembleLearning2cl_Test and EnsembleLearning2cl_Predict were
  deprecated and replaced by DaMiR.EnsL_Train, DaMiR.EnsL_Test and
  DaMiR.EnsL_Predict, respectively;

- We have created a new function DaMiR.ModelSelect to select the best
  model in a machine learning analysis;

- We have created two new functions DaMiR.iTSnorm and DaMiR.iTSadjust
  to normalize and adjust the gene espression of independent test sets;

- Two types of expression value distribution plot were added to the
  DaMiR.Allplot function.

[dearseq](/packages/dearseq)
-------

                 Changes in version 1.0.0 (2019-11-26)                  

- this is a reboot of the tcgsaseq package

[deepSNV](/packages/deepSNV)
-------

                 Changes in version 1.99.3 (2013-07-25)                 

Updates

- A few changes to shearwater vignette

- Renamed arguments pi.gene and pi.backgr in makePrior()

Bugfixes

- Fixed bug in bf2Vcf() when no variant is called

                 Changes in version 1.99.2 (2013-07-11)                 

Updates

- Updated CITATION

- Added verbose option to bam2R to suppress output

- Changed mode() to "integer" for value of loadAllData()

Bugfixes

- Fixed bug when only one variant is called in bf2Vcf()

                 Changes in version 1.99.1 (2013-06-25)                 

Updates

- Using knitr for prettier vignettes

- Including shearwater vignette

Bugfixes

- fixed issues with deletions in bf2Vcf()

- makePrior() adds background on all sites

                 Changes in version 1.99.0 (2013-04-30)                 

Updates

- New shearwater algorithm

- Including VCF output through summary(deepSNV, value="VCF")

[DEGreport](/packages/DEGreport)
---------

                       Changes in version 1.23.3                        

- Fix: pass min_counts parameter to degPlotCluster

                       Changes in version 1.23.2                        

- Fix: adopt class(counts)[1] to check classes

                       Changes in version 1.23.1                        

- Feature: add prefix to clusters plot in `degPlotCluster`

[DelayedArray](/packages/DelayedArray)
------------

                       Changes in version 0.14.0                        

NEW FEATURES

- Support 'type(x) <- new_type' to change the type of a DelayedArray
  object.

- 1D-style single bracket subsetting of DelayedArray objects now
  supports subsetting by a numeric matrix with one column per
  dimension.

SIGNIFICANT USER-VISIBLE CHANGES

- No more parallel evaluation by default, that is, getAutoBPPARAM() now
  returns NULL on a fresh session instead of one of the parallelization
  backends defined in BiocParallel. It is now the responsibility of the
  user to set the parallelization backend (with setAutoBPPARAM()) if
  they wish things like matrix multiplication, rowsum() or rowSums()
  use parallel evaluation again. Also BiocParallel has been moved from
  Depends to Suggests.

- Replace arrayInd2() and linearInd() with Lindex2Mindex() and
  Mindex2Lindex(). The new functions are implemented in C for better
  performances and they properly handle L-index values greater than
  INT_MAX (2^31 - 1) in the input and output.

- 2x speedup to coercion from DelayedArray to SparseArraySeed or
  dgCMatrix.

DEPRECATED AND DEFUNCT

- arrayInd2() and linearInd() are now deprecated in favor of
  Lindex2Mindex() and Mindex2Lindex().

BUG FIXES

- Fix handling of linear indices >= 2^31 in 1D-style single bracket
  subsetting of DelayedArray objects.

- rowsum() & colsum() methods for DelayedArray objects now respect
  factor level ordering (issue #59).

- Coercion from DelayedMatrix to dgCMatrix now propagates the dimnames.

- No more quotes around the NA values of a DelayedArray of type
  "character".

- Better error message when Ops methods for DelayedArray objects reject
  their operands.

[DeMixT](/packages/DeMixT)
------

                        Changes in version 1.2.5                        

- Added pi01 and pi02 as input values for users to initialize the
  proportion estimation.

- Added nspikein as an input value in the DeMixT, DeMixT_S1 and
  DeMixT_GS functions to specify how many spike-in normal reference
  samples need to be generated; Setting nspikein at null as a default
  value, the number of spike-in normal reference samples equal the
  min(200, 0.3 x My), where My is the number of mixed samples; By
  setting nspikein equals 0, no spike-in normal reference will be
  generated; If the input value of data.N2 is not null, nspikein will
  be forced to be 0.

- Added DeMixT_GS function, new proposed gene selection method which
  applies profile likelihood, for proportion estimation.

- Added simulate_2comp function for users to simulate test data for
  2-component de-convolution.

- Added simulate_3comp function for users to simulate test data for
  3-component de-convolution.

- Added row names and column names for all output values.

- Added gene.selection.method as an input value for DeMixT function.
  The default is 'GS'.

- Added ngene.Profile.selected as an input value for DeMixT function.
  The default is NA.

[DepecheR](/packages/DepecheR)
--------

                 Changes in version 1.3.3 (2020-03-18)                  

- Hiearchical clustering and ordering of the columns in the
  Cluster_centers heatmap from depeche.

- Note fixed regarding the use of ::: in GroupStatPlot

                 Changes in version 1.3.2 (2020-02-05)                  

- Bug fix in groupProbPlot solving issues when the cell number is low.

- Update of dWilcox.

                 Changes in version 1.3.1 (2019-12-05)                  

- Vinjette changes o Introduction of a vinjette for the groupProbPlot
  function.  o Fixing a minor bug in the general vinjette.

[derfinder](/packages/derfinder)
---------

                       Changes in version 1.21.7                        

BUG FIXES

- Noticed an issue with GenomicState::gencode_genomic_state() that
ultimately was due to makeGenomicState() and the transition in R to
have as default data.frame(stringsAsFactors = FALSE) instead of TRUE.

                       Changes in version 1.21.5                        

SIGNIFICANT USER-VISIBLE CHANGES

- Now makeGenomicState() now restores the
GenomicFeatures::isActiveSeq() on the txdb object before finishing to
avoid issues, like running regionReport::renderReport() on two
different sets of regions (different chrs). There's a new unit test
for this.

BUG FIXES

- Removed knitrBootstrap citation code.  - Made a test less
susceptible to numerical precision.

                       Changes in version 1.21.2                        

SIGNIFICANT USER-VISIBLE CHANGES

- Documentation website is now available at
http://lcolladotor.github.io/derfinder/. It gets updated with every
commit on the master branch (bioc-devel) using GitHub Actions and
pkgdown.

                       Changes in version 1.21.1                        

SIGNIFICANT USER-VISIBLE CHANGES

- Use GenomeInfoDb::getChromInfoFromUCSC() when possible instead of
data from biovizBase::hg19Ideogram for getting the hg19 chromosome
lengths.

[derfinderHelper](/packages/derfinderHelper)
---------------

                       Changes in version 1.21.1                        

SIGNIFICANT USER-VISIBLE CHANGES

- Documentation website is now available at
http://leekgroup.github.io/derfinderHelper/. It gets updated with
every commit on the master branch (bioc-devel) using GitHub Actions
and pkgdown.

[derfinderPlot](/packages/derfinderPlot)
-------------

                       Changes in version 1.21.4                        

SIGNIFICANT USER-VISIBLE CHANGES

- Documentation website is now available at
http://leekgroup.github.io/derfinderPlot/. It gets updated with every
commit on the master branch (bioc-devel) using GitHub Actions and
pkgdown.

                       Changes in version 1.21.3                        

SIGNIFICANT USER-VISIBLE CHANGES

- Code in all examples has been re-styled thanks to
styler::style_pkg(), see dev/02_update.R for details.

                       Changes in version 1.21.2                        

SIGNIFICANT USER-VISIBLE CHANGES

- Switched from Travis to GitHub actions through usethis.  - Build
the documentation website automatically with pkgdown.  - Added a
dev/01_setup.R script.  - Now using a README.Rmd file.

                       Changes in version 1.21.1                        

SIGNIFICANT USER-VISIBLE CHANGES

- Use GenomeInfoDb::getChromInfoFromUCSC() when possible instead of
data from biovizBase::hg19Ideogram for getting the hg19 chromosome
lengths.

[DESeq2](/packages/DESeq2)
------

                       Changes in version 1.28.0                        

- For lfcShrink(), changed order of `type` options: "normal" will no
  longer be first, as it under-performed "apeglm" and "ashr" in Zhu et
  al (2018). The new order is type=c("apeglm", "ashr", "normal").

- Added arguments to estimateDispersions: useCR (logical), and
  weightThreshold (numeric, default of 1e-2). The first argument
  optionally avoid the calculation of Cox-Reid adjustment term. The
  second argument sets the threshold for subsetting the design matrix
  and GLM weights when calculating the adjustment term. In addition,
  baseMean uses weights when calculating the mean of normalized counts,
  if weights matrix is provided.

                       Changes in version 1.27.12                       

- For lfcShrink(), changed order of `type` options: "normal" will no
  longer be first, as it under-performed "apeglm" and "ashr" in Zhu et
  al (2018). The new order is type=c("apeglm", "ashr", "normal").

                       Changes in version 1.27.9                        

- Added arguments to estimateDispersions: useCR (logical), and
  weightThreshold (numeric, default of 1e-2). The first argument
  optionally avoid the calculation of Cox-Reid adjustment term. The
  second argument sets the threshold for subsetting the design matrix
  and GLM weights when calculating the adjustment term. In addition,
  baseMean uses weights when calculating the mean of normalized counts,
  if weights matrix is provided.

[DEsubs](/packages/DEsubs)
------

                       Changes in version 1.13.1                        

- Updates for _R_CHECK_LENGTH_1_LOGIC2_, _R_CHECK_LENGTH_1_CONDITION_,
  classEq.

[DEWSeq](/packages/DEWSeq)
------

                 Changes in version 1.1.4 (2020-04-07)                  

- DEWSeq version bump for Bioconductor 3.11

- added missing dependency R.utils

                 Changes in version 1.0.2 (2020-02-03)                  

- Updated vignette to add htseq-clip pypi links

- LRT tests to results_DEWSeq function

[DIAlignR](/packages/DIAlignR)
--------

                       Changes in version 0.99.26                       

- Fixed visualization function.

- Calculating global alignment before iterating over multipeptides.

- Added summary after the alignment.

- Fixed area integrator for missing boundaries.

                       Changes in version 0.99.25                       

- Added filePath, Spectra file name, and RunID in one data frame. No
  need to pass datapath to each function.

- Separate functions to get precursor IDs, OpenSWATH features and
  chromatogram indices for each precursor across runs.

- Using multi-peptide and updating it in the parent frame.

- Added Alignment rank column.

- Copied C++ code for peak integration from OpenMS.

                       Changes in version 0.99.13                       

- Added as.list for S4 objects

                       Changes in version 0.99.12                       

- Added boxcar, Gaussian kernel and loess for chromatogram smoothing

- Started documentation for C++ code

- Added support for Microsoft Visual Studio compiler

- Using lists instead of matrix while updating RT and intensities

- Moved python code out of the src directory

                       Changes in version 0.99.11                       

- Made changes as suggested in initial review. Updated man/ pages
  associated with function.

- Link to the file for generating example data is updated.

                       Changes in version 0.99.10                       

- Made changes as suggested in initial review.
  https://github.com/Bioconductor/Contributions/issues/1328#issuecomment-576806498

                       Changes in version 0.99.9                        

- Added option to include mz pointer in getOswFiles and getXIC4AlignObj

                       Changes in version 0.99.8                        

- Removed R version dependency, have rda files instead of Rdata

                       Changes in version 0.99.7                        

- Lowered R version dependency

                       Changes in version 0.99.4                        

- Upadted version number

- Corrected tests

- Updated tests to reflect mzR updates, included travis-CI build

- Moved package to Bioconductor

[DiffBind](/packages/DiffBind)
--------

                       Changes in version 2.15.1                        

- Fix issue when extracing SummarizedExperiment from DBA

[diffcyt](/packages/diffcyt)
-------

                        Changes in version 1.7.6                        

- Add 'show_logFC' and 'show_all_cols' options to 'topTable'.

[DiscoRhythm](/packages/DiscoRhythm)
-----------

                 Changes in version 1.3.2 (2020-01-16)                  

- Fixed UI bug preventing job submission under some conditions

[distinct](/packages/distinct)
--------

                        Changes in version 1.0.0                        

- Submitted to Bioconductor

[dittoSeq](/packages/dittoSeq)
--------

                        Changes in version 1.0.0                        

- Submitted to Bioconductor.

[DOSE](/packages/DOSE)
----

                       Changes in version 3.13.2                        

- fixed issue caused by R v4.0.0 (2020-03-12, Thu) - length > 1 in
coercion to logical - https://github.com/YuLab-SMU/DOSE/pull/32

                       Changes in version 3.13.1                        

- remove S4Vectors dependencies (2019-12-19, Thu) - extend
setReadable to support compareClusterResult (2019-12-02, Mon) - add
gene2Symbol, keytype and readable slots for compareClusterResult -
move compareClusterResult class definition from clusterProfiler
(2019-11-01, Fri)

[dpeak](/packages/dpeak)
-----

                 Changes in version 1.0.0 (2020-04-14)                  

- Formatted package for Bioconductor. Changes include:

- Added runnable examples

- Reformatted source code to match Bioconductor practices

[DropletUtils](/packages/DropletUtils)
------------

                        Changes in version 1.8.0                        

- Added the downsampleBatches() function for convenient downsampling of
  batches.

- Preliminary support for using the output of write10xCounts() back in
  Cellranger.

- Support reading in 10X output files via prefixes in read10xCounts().
  Automatic detection of whether a file is Gzipped or not.

- Added chimericDrops() to remove chimeric molecules due to
  within-sample re-priming.

- Added hashedDrops() to demultiplex cell hashing experiments.

- Added maximumAmbience() to estimate the maximum contribution of the
  ambient profile.

[Dune](/packages/Dune)
----

                 Changes in version 0.9.0 (2019-03-15)                  

- Faster Dune version (from O(nlog(n)) to O(n))

[dupRadar](/packages/dupRadar)
--------

                       Changes in version 1.16.1                        

- Moved project site from Sourceforge to Github.

[easyRNASeq](/packages/easyRNASeq)
----------

                       Changes in version 2.23.4                        

- Ported changed from 2.22.5

                       Changes in version 2.23.3                        

- Ported changed from 2.22.3 & 4

                       Changes in version 2.23.2                        

- Ported changed from 2.22.2

                       Changes in version 2.23.1                        

- Ported changed from 2.22.1

                       Changes in version 2.22.5                        

- Moved the vignette data from the FTP to Git

                       Changes in version 2.22.4                        

- Moved the vignette data retrieval to the biocFileCache

                       Changes in version 2.22.3                        

- Ensured that misformatted gff3 prefix containing extra white space
  characters are gracefully handled.

                       Changes in version 2.22.2                        

- Fixed a namespace issue

- Extended the cleanup step of the vignette

                       Changes in version 2.22.1                        

- Added missing steps to the vignette

- Corrected the RobinsonDelhomme2014 metadata

- Fixed a bug introduced by a dropped functionality in asMates

- Updated package version dependencies

[edgeR](/packages/edgeR)
-----

                       Changes in version 3.30.0                        

- New function voomLmFit() that combines the limma voom-lmFit pipeline
  with loss of residual df due to zero counts as for glmQLFit(). The
  new function is more robust to zero counts than running voom() and
  lmFit() separately. The new function allows sample quality weights
  and intra-block correlations to be estimated it incorporates the
  functionality of duplicateCorrelation() and voomWithQualityWeights()
  as well.

- New function SE2DGEList() to convert a SummarizedExperiment object
  into a DGEList object.

- S3 methods for SummarizedExperiment objects are added to the
  following functions: aveLogCPM(), calcNormFactors(), cpm(),
  cpmByGroup(), estimateDisp(), filterByExpr(), glmFit(), glmQLFit(),
  plotMD(), plotMDS(), predFC(), rowsum(), rpkm(), rpkmByGroup() and
  sumTechReps().

- New cpm and rpkm methods for DGEGLM and DGELRT objects.

- New function effectiveLibSizes() to extract normalized library sizes
  from an edgeR data object or fitted model object.

- Add as.data.frame methods for DGEExact and DGELRT objects and remove
  the 'optional' argument from as.data.frame.TopTags().

- readBismark2DGE() now forces 'files' to be character vector.

- Add warning messages when filterByExpr() is used without specifying
  group or design.

[eisaR](/packages/eisaR)
-----

                         Changes in version 1.0                         

- Initial version - R implementation of Exon-Intron Split Analysis
(EISA, doi: 10.1038/nbt.3269) - functionality for extracting spliced
and unspliced transcript sequences, as well as intron sequences. e.g.
for use in RNA velocity analysis

[ELMER](/packages/ELMER)
-----

                       Changes in version 2.11.0                        

Changes:

- Minor bug fixes to support R 4.0

[EnhancedVolcano](/packages/EnhancedVolcano)
---------------

                         Changes in version 1.6                         

- removed legendVisible and pLabellingCutoff parameters

- deprecated transcriptPointSize, transcriptLabSize, transcriptLabCol,
  transcriptLabFace, transcriptLabhjust, transcriptLabvjust,
  boxedlabels, and drawconnectors parameters

- now pad xlim by +/- 1

- moved default position of labels to be central to its respective
  point

- added 'colGradient', which colours points continuously based on
  p-value

[EnMCB](/packages/EnMCB)
-----

                       Changes in version 0.99.12                       

NEW FEATURES

- Update vignettes.

SIGNIFICANT USER-VISIBLE CHANGES

- Delete unused data.

BUG FIXES

- Some fixes related to test functions.

[ENmix](/packages/ENmix)
-----

                       Changes in version 1.22.6                        

- add function: combp,ipdmr,mhtplot,p.qqplot

                       Changes in version 1.22.2                        

- removed function bmiq.mc and ComBat.mc

[EnrichmentBrowser](/packages/EnrichmentBrowser)
-----------------

                       Changes in version 2.18.0                        

- Gene sets: supporting additional gene set databases - getGenesets(db
  = "msigdb") to obtain gene sets from the MSigDB for 11 different
  species (via msigdbr) - getGenesets(db = "enrichr") to obtain gene
  set libraries from Enrichr for 5 different species

- Gene ID mapping for gene sets and gene regulatory networks (function
  `idMap`)

- Implementation of a variance-stabilizing transformation for RNA-seq
  data (function `normalize` with argument `norm.method = "vst"`) to
  simplify the application of legacy enrichment methods

[enrichplot](/packages/enrichplot)
----------

                        Changes in version 1.7.3                        

- update cnetplot color scale to tolerate with skewed foldchange
(2020-03-13, Fri) - https://github.com/YuLab-SMU/enrichplot/pull/40

                        Changes in version 1.7.1                        

- cnetplot for compareClusterResult (compareCluster output)
(2019-12-02, Mon) - move barplot, dotplot and fortify methods of
compareClusterResult from clusterProfiler (2019-11-2, Sat)

[ensembldb](/packages/ensembldb)
---------

                       Changes in version 2.11.1                        

- Fix mapIds issue.

- Add gc_count to the transcripts table.

- Ensure `mapIds` returns a result equal to the number of input keys -
  even if none of them could be mapped (issue #104).

[ensemblVEP](/packages/ensemblVEP)
----------

                       Changes in version 1.30.0                        

- add support for Ensembl release 99

[EpiTxDb](/packages/EpiTxDb)
-------

                 Changes in version 0.99.0 (2020-02-25)                 

- Submitted to Bioconductor

[epivizrServer](/packages/epivizrServer)
-------------

                       Changes in version 999.999                       

- This NEWS file is only a placeholder. The version 999.999 does not
  really exist. Please read the NEWS on Github: <URL:
  https://github.com/epiviz/epivizrServer>

[ERSSA](/packages/ERSSA)
-----

                        Changes in version 1.5.1                        

- Better implementation of jitter plot in ggplot2_dotplot

- plot_marginalPlot set to plot median as default, previously set to
  plot mean

[evaluomeR](/packages/evaluomeR)
---------

                 Changes in version 1.3.41 (2019-01-07)                 

- Minor bug fix for R 4.0 devel.

                 Changes in version 1.3.4 (2019-12-19)                  

- Adding 'plotMetricsClusterComparison' method.

                 Changes in version 1.3.3 (2019-12-16)                  

- Refactoring 'getOptimalKValue' method

- Adding stabilitySet and qualitySet

                 Changes in version 1.3.2 (2019-11-12)                  

- Added new methods: Dependency with FPC added.
  Stability/stabilityRange also outputs cluster information like
  centers.

                 Changes in version 1.3.0 (2019-11-12)                  

- Added new methods: plotMetricsMinMax, plotMetricsBoxplot,
  plotMetricsCluster, plotMetricsCluster and plotMetricsViolin

[exomePeak2](/packages/exomePeak2)
----------

                Changes in version 0.99.92 (2020-03-10)                 

- Fix several issues commented by the bioconductor reviewer

[ExperimentHub](/packages/ExperimentHub)
-------------

                       Changes in version 1.13.0                        

NEW FEATURES

- (1.13.5) New function removeResources for removing locally downloaded
  cached resources. Nicer interface than using cache. Improved
  documentation on use of setting a cache to NULL to also remove
  resources.

- (1.13.4) Adds option ask to constructor. Default: True. Ask will ask
  user to create default ExperimentHub location and if in an non
  interactive session utilize a temporary directory. If False, the
  default directory is created and utilized without prompting.

BUG FIXES

- (1.13.8) Updated has_internet to nslookup. Certain locations block
  has_internet from running (block google dns). switched so can query
  the hub directly

- (1.13.7) Fixed messages to be more informative

- (1.13.2) Aaron's corrections to setExperimentHubOptions.

USER-VISIBLE CHANGES

- (1.13.1) Match ERROR message when resource not found in database.
  Instead of a generic "unknown key" give more detailed information.
  This is consistent with AnnotationHub/Hub-classes. If resource hasn't
  been completely removed from database, gives when the resource was
  removed. If the resource is available but after the snapshot date
  being utilized, specialized message.

[ExperimentHubData](/packages/ExperimentHubData)
-----------------

                       Changes in version 1.13.0                        

MODIFICATIONS

- 1.13.1 add check for valid Title and Description in metadata file. It
  should not be empty or NA

[ExploreModelMatrix](/packages/ExploreModelMatrix)
------------------

                       Changes in version 0.99.0                        

- Prepared for Bioconductor submission

                        Changes in version 0.2.0                        

- Added pseudoinverse, variance inflation factor, coefficient
correlation and co-occurrence plots.  - Arguments flipCoord,
textSize, textSizeLabs, lineWidth, addColor and colorPalette in
VisualizeDesign have been renamed (adding Fitted as a suffix) to
allow corresponding arguments for other types of plots.

                        Changes in version 0.1.0                        

- Initial version

[fcoex](/packages/fcoex)
-----

                 Changes in version 1.1.1 (2020-05-08)                  

- Updated fcoex example object "fc".

[fgsea](/packages/fgsea)
-----

                       Changes in version 1.13.4                        

- mapIdsList for faster conversion of leadingEdges

- experimental support of hypergeometric tests with fora() and
  collapsePathwaysORA()

                       Changes in version 1.13.3                        

- experimental support for one-tailed tests

                       Changes in version 1.13.2                        

- fgsea() implementation changed from simple to multilevel algorithm

- proper handling of duplicate genes in gene sets for fgseaSimple() and
  fgseaMultilevel()

[fishpond](/packages/fishpond)
--------

                        Changes in version 1.4.0                        

- Added isoformProportions(), which can be run after scaleInfReps() and
  optionally after filtering out transcripts using labelKeep(). Running
  swish() after isoformProportions() will produce differential
  transcript usage (DTU) results, instead of differential transcript
  expression (DTE) results. Example in vignette. + Default number of
  permutations increased from 30 to 100. It was observed that there was
  too much fluctuation in the DE called set for nperms=30 across
  different seeds, and setting to 100 helped to stabilize results
  across seeds, without increasing running time too much. For further
  reduced dependence on the seed, even higher values of nperms (e.g.
  200, 300) can be used.

                        Changes in version 1.3.8                        

- Added isoformProportions(), which can be run after scaleInfReps() and
  optionally after filtering out transcripts using labelKeep(). Running
  swish() after isoformProportions() will produce differential
  transcript usage (DTU) results, instead of differential transcript
  expression (DTE) results. Example in vignette.

                        Changes in version 1.3.4                        

- Default number of permutations increased from 30 to 100. It was
  observed that there was too much fluctuation in the DE called set for
  nperms=30 across different seeds, and setting to 100 helped to
  stabilize results across seeds, without increasing running time too
  much. For further reduced dependence on the seed, even higher values
  of nperms (e.g. 200, 300) can be used.

[flowCore](/packages/flowCore)
--------

                        Changes in version 2.0.0                        

- keyword<- behaves now as normal replacement method, i.e.
keyword(fr) <- list(name = value) will replace the entire keyword
list instead of partial updating/inserting, to achieve latter, use
keyword(fr)&#91;"name"&#93; <- value - &#91; method (e.g. fr&#91;, 1:4&#93;) no longer
deletes $PnX keywords automatically so that it is compatible with
cytoframe

                        Changes in version 1.53                         

- 'workFlow' class and related classes and methods now defunct #148
-- superseded by flowWorkspace - Add support for multiple data
segments in 'read.FCSheader' - Fixed some edge cases with 'read.FCS
'with column.pattern argument #157 - Added 'transform_gate' methods
for geometric transformations of gates - Added 'collapse_desc' for
coercing keyword lists in to character - 'spillover' method for
calculating compensation matrix from a 'flowSet' of controls moved to
'flowStats' - Added 'filter_keywords' for filtering lists of keywords
- Deprecate description and description<- for keyword and keyword<-
to reduce redundancy and give consistent behavior for cytoframe
RGLab/flowWorkspace #311

[flowSpecs](/packages/flowSpecs)
---------

                 Changes in version 1.1.4 (2020-04-27)                  

- Chaning usage from flowCore::Description to flowCOre::keyword due to
  deprecation.

                 Changes in version 1.1.3 (2020-04-21)                  

- Adding an outlier removal step in the internal peakIdenti function

- Removing .internal calls in anticipation of the R 4.0 deprecation

- Minor bug fixes in internal fucntions.

                 Changes in version 1.1.2 (2020-01-27)                  

- bug fix enabling multiple FSC measurements when generating the unmix
  matrix

                 Changes in version 1.1.1 (2019-11-28)                  

- bug fix enabling multiple SSC measurements when generating the unmix
  matrix

[flowSpy](/packages/flowSpy)
-------

                 Changes in version 1.1.5 (2020-04-20)                  

- Fixed installation errors in Linux and Windows

- Fixed legend name of figure

- Fixed errors in building vignette

                 Changes in version 1.0.4 (2020-03-19)                  

- Fixed installation errors in Linux and Windows

                 Changes in version 1.0.3 (2020-03-17)                  

- Fixed bugs in `runDiff`

                 Changes in version 1.0.1 (2020-02-15)                  

- Add uniform downsampling in function `processingCluster`

- Fixed some warnings in the checking process of Bioc 3.10

[FoldGO](/packages/FoldGO)
------

                 Changes in version 1.5.2 (2020-04-23)                  

- Not fold-change-specific bars removed from plots

- Citation file added

                 Changes in version 1.5.1 (2020-04-21)                  

- issue with incompatible regex for inst/extdata directory fixed

- R version dependency switched to 4.0

[FRASER](/packages/FRASER)
------

                       Changes in version 0.99.1                        

- Introduction of the shinyApp

                       Changes in version 0.99.0                        

- This is the initial setup of the FRASER package.

- Use SummarizedExperiment as superclass

- Integrate package into the Bioconductor infrastructure

[GAPGOM](/packages/GAPGOM)
------

                 Changes in version 1.3.1 (2020-04-10)                  

Changed

- Author name in Readme, DESCRIPTION and LICENSE

- Copyright year 2018 -> 2020

Fixed

- `test_parsing.R` to work with latest Bioc (3.11)

- R depends version for Bioc 3.11

[GARS](/packages/GARS)
----

                        Changes in version 1.8.0                        

- bug fixed and publication added

[GateFinder](/packages/GateFinder)
----------

                        Changes in version 1.7.1                        

- changed conditional check on class to be consistent with R 4.0.0 for
  Bioconductor 3.11 release

[gCMAP](/packages/gCMAP)
-----

                       Changes in version 1.31.1                        

- CHANGE: The gCMAP and gCMAPWeb packages are deprecated and will be
  removed from Bioconductor version 3.12.

[gCMAPWeb](/packages/gCMAPWeb)
--------

                       Changes in version 1.27.1                        

- CHANGE: The gCMAP and gCMAPWeb packages are deprecated and will be
  removed from Bioconductor version 3.12.

[gCrisprTools](/packages/gCrisprTools)
------------

Version: 1.15.0
Category: Various code updates, new functions added for visualization,
        compatibility with modern R objects. Russell Bainer
        redesignated as official maintainer.
Text:

[gdsfmt](/packages/gdsfmt)
------

                       Changes in version 1.24.0                        

NEW FEATURES

- new option 'allow.error' in `openfn.gds()` for data recovery

- new option 'log.only' in `diagnosis.gds()`

- new C functions `GDS_Node_Unload()` and `GDS_Node_Load()` in R_GDS.h

- new sparse array data types in GDS (SparseReal32, SparseReal64,
  SparseInt8, SparseInt16, SparseInt32, SparseInt64, SparseUInt8,
  SparseUInt16, SparseUInt32, SparseUInt64)

- the opened gds file will be closed when the object is garbage
  collected

UTILITIES

- zlib updated to v1.2.11 from v1.2.8

- xz updated to v5.2.4 from v5.2.3

- LZ4 updated to v1.9.2 from v1.7.5

- `showfile.gds()` does not return the object of 'gds.class'

- new 'n' and 'depth' in `print.gdsn.class()` and `print.gds.class()`

[GeneNetworkBuilder](/packages/GeneNetworkBuilder)
------------------

                       Changes in version 1.29.1                        

- Add BiocStyle and magick into Suggests

[GeneSelectMMD](/packages/GeneSelectMMD)
-------------

                       Changes in version 2.31.2                        

- reformat NEWS file

                       Changes in version 2.31.1                        

- update DESCRIPTION: maintainer's email address; exclude 'Biobase' and
  'survival' from "Imports"

- update NAMESPACE: add "Import(Biobase)"

- fixed a bug in trans.R: the 'diff' in "tmp > 1 || diff < eps" is a
  vector

- revised visual.R: shorten strings in 'mytitle'

- change Weiliang's email address in all .Rd files

- revised plotHistDensity.Rd: shorten strings in 'mytitle'

[GENESIS](/packages/GENESIS)
-------

                       Changes in version 2.17.4                        

- Fix a bug that prevented use of multiple environment variables in
  GxE.

                       Changes in version 2.17.3                        

- Allow divergence to be NULL in pcair and pcairPartition.

                       Changes in version 2.17.1                        

- Fix a bug in null model when starting variance component is close to 0.

[geNetClassifier](/packages/geNetClassifier)
---------------

                       Changes in version 1.27.1                        

- Minor update for R 4.0.

[GeneTonic](/packages/GeneTonic)
---------

                       Changes in version 0.99.0                        

Other notes

- GeneTonic is now submitted to Bioconductor!

                       Changes in version 0.10.0                        

New features

- The functions for comparing different res_enrich, namely gs_radar,
gs_summary_overview_pair, and gs_horizon were internally rewritten to
accept correctly the comparison elements - The vignette now covers
completely the usage cases, with the full description of the user
interface of GeneTonic - The introductory tours are available for all
the main panels of GeneTonic. Feel free to try them out!

Other notes

- Some widgets have been added in the UI of GeneTonic to enable finer
control of the output aspect - Examples and unit tests have been
further expanded, with better messages for checking progress -
check_colors verify that color palettes are correctly provided - Info
on GeneTonic is now provided with modal dialog windows, rather than
in a separate tab - The tab names in the main app were slightly
edited to better describe their content - The info boxes are now
shown with a uniform style, based on the bs4Card UI element - The UI
elements have now a better spacing throughout the different tabs -
Soon the package will be submitted to Bioconductor!

                        Changes in version 0.9.0                        

New features

- GeneTonic sports a blazing new hex sticker - say bye to the
original draft!  - The overview DT datatables has some styling with
color bars - e.g.  for DE results - to enhance the visual perception
of numeric values (e.g. log2FoldChange) - gs_heatmap can now take a
custom list of gene identifiers (when no geneset is passed) - The
color palettes in enrichment maps now respect the values and the
range specified of the numeric values to be used for mapping to
colors - gs_mds is now optionally returning a data.frame, to be
further used for custom plotting or downstream processing -
gs_summary_overview now has coloring enabled by the variable of
choice

Other notes

- The UI has received some restyling (e.g. in the choice of the icons
for the dropdown menus, or the name of some buttons) - Added tour
contents for most of the functionality - Added link to the demo
instance - Added examples for overlap functions, gene info buttons,
map2color, and deseqresult2df - Extended documentation of some
parameters - Some functions have gained an alias for calling them:
gs_spider is equivalent to gs_radar, and gs_sankey is equivalent to
gs_alluvial

                        Changes in version 0.8.0                        

New features

- GeneTonic now delivers bundled example objects to make examples and
tests slim - gs_volcano can now plot points by different colors
according to the columns of interest - GeneTonic has a fully fledged
manual describing its functionality and user interface

Other notes

- Now using ids for genes and genesets for exchanging information in
the app - Added examples for all functions - Most tabs have working
tours - anchor and text elements

                        Changes in version 0.7.0                        

New features

- Introduced a uniform interface for calculating different
similarity/distance matrices. This enables the usage in the different
functions that might need such matrices for further downstream
processing (e.g. enrichment_map(), gs_mds()) - First appearance of
gs_dendro() to display distance matrices with some visualization
sugar, as an alternative to other methods - The n_gs and gs_ids are
exposed to more functions to enable custom subsets of the enrichment
results to be inspected

                        Changes in version 0.6.0                        

New features

- gs_heatmap now relies on ComplexHeatmap, to avoid the issues with
Shiny of not displaying the outputs in the app, and enabling a
comfortable heatmap annotation - Many functions gain the possibility
to pass a set of custom geneset identifiers to be added to the top N
sets (default): among these, gs_mds, gs_volcano (parameter:
gs_labels), gs_alluvial, ggs_network, enrichment_map, and
enhance_table (using gs_ids)

Other notes

- gs_ggheatmap got renamed to gs_scoresheat - The report generated
from the bookmarked content is expanded in its default content

                        Changes in version 0.5.0                        

New features

- GeneTonic now enforces a format for res_enrich, and provides some
conversion functions, shake_*(). Requirements are specified in the
documentation, if an appropriate converter does not (yet) exist.  -
The reporting feature is active to some extent on the bookmarked
content.

                        Changes in version 0.4.0                        

New features

- Added functionality for bookmarking - Bookmarking can work (PoP) by
pressing a key (left control)!  - gene_plot can enforce a plot type
overriding the default based on the number of samples per condition -
GeneTonic uses now bs4Dash and many of its nice features, replacing
the previous implementation based on shinydashboard

                        Changes in version 0.3.0                        

Other notes

- Rearranging the order of parameters to harmonize it across
functions, and uniforming similarly called parameters

                        Changes in version 0.2.0                        

New features

- added geneset scoring calculation and corresponding heatmap

                        Changes in version 0.1.0                        

New features

- much of the functionality available, in a proof of concept format

                        Changes in version 0.0.1                        

New features

- backbone of the project started!

[GenomeInfoDb](/packages/GenomeInfoDb)
------------

                       Changes in version 1.24.0                        

NEW FEATURES

- Add getChromInfoFromUCSC(), getChromInfoFromNCBI(), and
  getChromInfoFromEnsembl().

DEPRECATED AND DEFUNCT

- Deprecate fetchExtendedChromInfoFromUCSC() in favor of
  getChromInfoFromUCSC().

[GenomicDataCommons](/packages/GenomicDataCommons)
------------------

                       Changes in version 1.11.1                        

- add support for fetching field descriptions

[GenomicFeatures](/packages/GenomicFeatures)
---------------

                       Changes in version 1.40.0                        

NEW FEATURES

- Small improvements to makeTxDbFromEnsembl(): - check validity of user
  supplied transcript attribute code - report nb of fetched transcripts
  and genes during progress - produced TxDb will "remember" which
  transcript attribute code was used

SIGNIFICANT USER-VISIBLE CHANGES

- genes() now prints a message when genes are dropped.

- getChromInfoFromUCSC() is now defined in the GenomeInfoDb package.
  Note that the new function is a much improved version of the old one
  but also the returned data frame has its columns named differently!
  See ?getChromInfoFromUCSC in the GenomeInfoDb package for the
  details.

- Argument 'goldenPath_url' was renamed 'goldenPath.url' in functions
  makeTxDbFromUCSC(), makeFeatureDbFromUCSC(),
  makeTxDbPackageFromUCSC(), and makeFDbPackageFromUCSC().

- Default value for argument 'circ_seqs' was changed from
  DEFAULT_CIRC_SEQS to NULL in all functions that have this argument.

- Global constant DEFAULT_CIRC_SEQS (character vector) was removed. It
  is now defined in GenomeInfoDb but no longer exported, at least for
  now.

BUG FIXES

- Make asGFF() method for TxDb objects robust to absence of CDS ranges.

[GenomicRanges](/packages/GenomicRanges)
-------------

                       Changes in version 1.40.0                        

NEW FEATURES

- Add trim() method for GRangesList objects.

[GenomicScores](/packages/GenomicScores)
-------------

                        Changes in version 2.0.0                        

USER VISIBLE CHANGES

- Added a shiny app available through the function 'igscores()'.

- Added support for genomic score packages (e.g., MafH5.*) that use
  HFD5 files as a backend.

- Added a new function 'rgscores()' for randomly sampling genomic
  scores.

[GEOquery](/packages/GEOquery)
--------

                       Changes in version 2.55.1                        

Bug fixes:

* Fix test failure due to stricter boolean check lengths
&#91;#568ae94fb95, from @russHyde&#93;

* Fix failure on empty file &#91;#88741521a0e, from @vlakam&#93;

[GGPA](/packages/GGPA)
----

                Changes in version 0.99.10 (2020-04-14)                 

- Formatted package for Bioconductor. Changes include:

- Added runnable examples

- Reformatted source code to match Bioconductor practices

                        Changes in version 0.9.6                        

- Made the following significant changes

- The vignette was updated by incorporating the illustration of the
  workflow to download the prior disease graph from the companion
  website (http://www.chunglab.io/ddnet).

- plot(): Improve visualization, e.g., using a circle mode.

                        Changes in version 0.9.5                        

- Made the following significant changes

- GGPA(): Allow to incorporate a prior phenotype graph.

- plot(): Use GGally (ggnet2) instead of Rgraphviz for the phenotype
  graph visualization.

                        Changes in version 0.9.1                        

- The first version of GGPA R package.

[ggtree](/packages/ggtree)
------

                        Changes in version 2.1.6                        

- Now geom_tiplab() works with unrooted layouts (ape, equal_angle and
daylight) (2020-04-23, Thu) -
https://github.com/YuLab-SMU/ggtree/issues/292 - bug fixed of
layoutEqualAngle (2020-04-09, Thu) - in tibble v=3.0.0, df$x = NA
will store df$x as lgl variable and assign numeric value to df$x will
throw error. Now change to df$x = 0 in layoutEqualAngle.

                        Changes in version 2.1.5                        

- bug fixed of calculating inset width and height -
https://github.com/YuLab-SMU/ggtree/issues/289

                        Changes in version 2.1.4                        

- import aplot::xrange

                        Changes in version 2.1.3                        

- move xlim2 and ylim2 to aplot package (2020-03-30, Mon) -
https://cran.r-project.org/package=aplot - remove mutate_ as it was
deprecated in dplyr (2020-03-25, Wed) - fixed mutate bug caused by
new version of dplyr and tidytree in daylight layout (2020-03-16,
Mon) - https://github.com/YuLab-SMU/ggtree/issues/282

                        Changes in version 2.1.2                        

- expand_scale was deprecated in ggplot2 v=3.3.0, import expansion
instead (2020-03-12, Thu) - bug fixed of determined layout for
ggplot(tree) + geom_tiplab() (2020-01-26, Sun) -
https://github.com/YuLab-SMU/ggtree/issues/272 - set unrooted layout
to use coord_fixed by default (2020-01-25, Sat) - if layout = "ape",
use ape unrooted layout -
https://github.com/YuLab-SMU/ggtree/pull/273

                        Changes in version 2.1.1                        

- export geom_highlight as an alias of geom_hilight (2020-01-08, Wed)
- set clip="off" for all layouts (2019-12-06, Fri) - not passing
fontface if geom is image or phylopic in geom_tiplab and geom_nodelab
(2019-11-29, Fri) - import and re-export guide_legend,
scale_colour_manual, scale_color_manual, scale_fill_manual and margin
from ggplot2 - offspring method for ggtree object (2019-11-21, Thu) -
fixed revts to work with collapse (2019-11-18, Mon) - convert roxygen
documents using markdown (2019-11-01, Fri) - extend xlim2 and ylim2
to support discrete scale.  - xlim2 and ylim2 to uniformize axis
limits of ggplot objects (2019-10-31, Thu) -
https://yulab-smu.github.io/treedata-book/chapter7.html#composite_plot
-
https://yulab-smu.github.io/treedata-book/docs/chapter10.html#axis_align
- fixed fontface warning message when align=TRUE in geom_tiplab
(2019-10-30, Wed) - https://github.com/YuLab-SMU/ggtree/issues/260

[gmoviz](/packages/gmoviz)
------

                       Changes in version 0.99.8                        

- Submitted to Bioconductor

[GNET2](/packages/GNET2)
-----

                        Changes in version 1.3.7                        

- The retuun of extract_edges() now contains the group score.

                        Changes in version 1.3.3                        

- Add function to extract the network from the gnet result

                        Changes in version 1.3.2                        

- Fix plot order issue.

- Add group split procedure for module groups with too many genes.

                        Changes in version 1.3.1                        

- Fix the initial group number limit.

[GOfuncR](/packages/GOfuncR)
-------

                        Changes in version 1.7.1                        

USER-LEVEL CHANGES

- update GO-graph (version 23-Mar-2020)

[GOSemSim](/packages/GOSemSim)
--------

                       Changes in version 2.13.1                        

- add new citation (2020-03-19, Thu) - fixed compiling error due to
the change of Rcpp - https://github.com/YuLab-SMU/GOSemSim/issues/27

[GPA](/packages/GPA)
---

                        Changes in version 1.1.0                        

- Made the following significant changes

- Incorporate ShinyGPA developed by Emma Kortemeier as a part of the R
  package.

- Add fitAll() to fit GPA models for possible pairs of phenotypes.

- Add shinyGPA() to open the shiny app for the interactive pleiotropy
  visualization.

                 Changes in version 0.99.9 (2020-04-14)                 

- Formatted package for Bioconductor. Changes include:

- Added runnable examples

- Reformatted source code to match Bioconductor practices

                        Changes in version 0.9.3                        

- Made the following significant changes

- Incorporate R package vignette.

- assoc(), fdr(): allows association mapping & local FDR for specific
  association status.

- aTest(), pTest(): message() was replaced with cat().

                        Changes in version 0.9.2                        

- Made the following significant changes

- Add se() to provide parameter estimates and their standard errors.

                        Changes in version 0.9.1                        

- Made the following significant changes

- Add fdr() to provide local FDR that a SNP is not associated with each
  phenotype.

- Add assoc() for association mapping.

- The function aTest() and methods show() and cov() for class 'GPA' now
  provide estimates and standard error for the ratio of q in each
  association class over the baseline.

- In GPA(), 'initBetaMean' argument is replaced with 'initBetaAlpha'.

- Generally improve robustness of functions by checking input for
  arguments.

                        Changes in version 0.9.0                        

- The first version of GPA R package.

[graphite](/packages/graphite)
--------

                 Changes in version 1.33.1 (2020-10-24)                 

- Updated all pathway data.

[gscreend](/packages/gscreend)
--------

                        Changes in version 1.0.1                        

- The data input function raises errors, when column naming concerning
  sgRNAs or genes in not clear.

- Functions that can only be run after analysing data raise errors when
  you try to run them before the model has been fitted and statistics
  computed.

[GSEABenchmarkeR](/packages/GSEABenchmarkeR)
---------------

                        Changes in version 1.8.0                        

- Extended support for benchmarking user-defined inputs: - mixing of
  pre-defined and user-defined enrichment methods (functions `runEA`
  and `evalTypeIError`) - simplified passing on of additional arguments
  to user-defined enrichment methods for functions `runEA` and
  `evalTypeIError`

- The TCGA RNA-seq compendium can now also be obtained via
  curatedTCGAData using `loadEData("tcga", mode = "cTD")`

[GSVA](/packages/GSVA)
----

                        Changes in version 1.36                         

USER VISIBLE CHANGES

- Improved management of parallel calculations by using the
  BiocParallel package. This facilitates calculating GSVA scores from
  gene expression data matrices with thousands of samples. See
  arguments 'parallel.sz' and 'BPPARAM' in the manual page of the
  'gsva()' function.

- Improved implementation of the SSGSEA method ('method="ssgsea" in the
  call to 'gsva()') that makes calculations with this option about one
  order of magnitude faster. Improvement thanks to Alexey Sergushichev
  (https://github.com/rcastelo/GSVA/pull/15).

- The function 'gsva()' now issues a warning when one or more gene sets
  are singletons, i.e., they are formed by just one feature/gene.

[Gviz](/packages/Gviz)
----

                       Changes in version 1.32.0                        

NEW FEATURES

- Exposed the possibility to specify centromere shape

- `AlignmentsTracks` can visualize "indels"

- SequenceTrack has new subclass: SequenceRNAStringSetTrack

- added RNASequenceTrack

BUG FIXES

- Fixed issue in .buildRange (internal) to correctly build package on
  windows

[Harman](/packages/Harman)
------

                       Changes in version 1.15.0                        

- Previously, if no random seed is specified to the harman function, a
  seed of 0 was used. Now a random seed between 0 and 1e9 is picked at
  random.

[HDF5Array](/packages/HDF5Array)
---------

                       Changes in version 1.16.0                        

NEW FEATURES

- New h5writeDimnames()/h5readDimnames() functions for writing/reading
  the dimnames of an HDF5 dataset to/from the HDF5 file. See
  ?h5writeDimnames for more information.

- Add full support for HDF5Array objects of type "raw": -
  writeHDF5Array() now works on a DelayedArray object of type "raw" (it
  creates an H5 dataset of type H5T_STD_U8LE). - The HDF5Array()
  constructor now should return an HDF5Array object of type "raw" when
  pointed to an H5 dataset with an 8-bit width type (e.g.
  H5T_STD_U8LE, H5T_STD_U8BE, H5T_STD_I8LE, H5T_STD_I8BE, H5T_STD_B8LE,
  H5T_STD_B8BE, etc...)

- Add 'H5type' argument to writeHDF5Array().

- h5mread() now supports contiguous (i.e. unchunked) string data.

SIGNIFICANT USER-VISIBLE CHANGES

- HDF5Array objects now find their dimnames in the HDF5 file.
  writeHDF5Array() and as(x, "HDF5Array") know how to write the
  dimnames to the HDF5 file, and the HDF5Array() constructor knows how
  to find them. See ?writeHDF5Array for more information.

BUG FIXES

- Fix bug causing character data to be truncated when written to HDF5
  file.

- Fix h5mread() inefficiency when the user selection covers full
  chunks.

- h5mread() now handles character NAs consistently with
  rhdf5::h5read().

- Fix writeHDF5Array() error on character array filled with NAs.

[HIPPO](/packages/HIPPO)
-----

                 Changes in version 0.99.3 (2020-03-25)                 

- Submitted to Bioconductor

[HPAanalyze](/packages/HPAanalyze)
----------

                         Changes in version 1.5                         

- Changes in version 1.5.3 

    + Update built-in data to HPA version 19.


- Changes in version 1.5.2 

    + Minor bug fixes.


- Changes in version 1.5.1 

    + Add documentation with pkgdown. Minor updates to documentation. 
    + tidyr and readr dependencies have been replaced with base R solutions. 
    + Importing data from hpar is no longer supported. 
    + hpaDownload function was updated to correctly download data in HPA version 19.


- Changes in version 1.5.0 

    + Starting devel for Bioconductor 3.10

[hpar](/packages/hpar)
----

                        Changes in version 1.29                         

Changes in version 1.29.1

- Update email address <2020-03-29 Sun>

[HTSeqGenie](/packages/HTSeqGenie)
----------

                       Changes in version 4.17.1                        

- fix bug in targetLenght plotting in reportSummaryQC

- make countGenomicFeatuers work with subclasses of GRangesList

[hypeR](/packages/hypeR)
-----

                       Changes in version 1.3.00                        

- Force version bump for bioconductor release 3.10

[ideal](/packages/ideal)
-----

                       Changes in version 1.12.0                        

New features

- Implemented a series of gentle fail mechanisms to avoid abrupt
crashing of the app if the expected objects or formats are not
provided

Other notes

- Alignments of UI elements have been adjusted - Updated citation
info - the preprint for ideal, "ideal: an R/Bioconductor package for
Interactive Differential Expression Analysis", is now published at
bioRxiv, https://doi.org/10.1101/2020.01.10.901652

[idr2d](/packages/idr2d)
-----

                        Changes in version 1.0.4                        

- changed max_gap default value from 0 to -1 to conform with convention

- updated reference to IDR2D publication

                        Changes in version 1.0.3                        

- added option to disable jitter

- added min_value and max_value parameters to estimate_idr2d_hic

- fix block size display for draw_hic_contact_map

                        Changes in version 1.0.2                        

- added parser for HiC-Pro contact maps (.matrix and .bed files)

- added function to visualize contact maps, showing only reproducible
  blocks

                        Changes in version 1.0.1                        

- fixed alpha level in rank and value scatterplots

- bold font face for axis and legend titles

[IgGeneUsage](/packages/IgGeneUsage)
-----------

                 Changes in version 1.1.6 (2020-04-01)                  

- Corresponding methodology published in Bioinformatics

- Speedup in posterior prediction

[immunoClust](/packages/immunoClust)
-----------

                       Changes in version 1.19.4                        

- CHANGES * code cleaning

                       Changes in version 1.19.3                        

- CHANGES * fixes a compiler error under windows

                       Changes in version 1.19.1                        

- CHANGES * bugfix in transformParam.immunoMeta

[infercnv](/packages/infercnv)
--------

                 Changes in version 1.3.6 (2020-03-20)                  

- Fix a read.table() call who's behavior was changed by the latest R
  base devel changes.

                 Changes in version 1.3.5 (2020-03-02)                  

- Fix saving file name for preliminary object.

- Added an option to plot_cnv(), plot_chr_scale, that allows to plot
  chromosomes proportionally to their full size. Either a list of
  chromosomes sizes is provided, or the last gene stop position + 10k
  bp is assumed. Gene width is proportional to the area in the
  chromosomes they "cover" (midpoints between genes). This option is
  slower than the default as rasterization can not be applied when
  drawing the heatmap.

                 Changes in version 1.3.4 (2020-02-18)                  

- Added options to save rds results after each step (save_rds) or final
  results (save_final_rds), with default TRUE.

- Fix size of pdf output settings to use the given width and height
  values instead of the "USr" format.

- Fixes in plot_cnv() when using observation groups with 1 or 2 cells.

- Fix checks in plot_cnv() of references for when subclusters are not
  defined in the object.

- Fix (sub)cluster definition when an annotation only has 1 cell
  (although this should probably be avoided).

                 Changes in version 1.3.3 (2020-02-07)                  

- Large speed improvements in plot_cnv() that are more significant the
  bigger the matrix size.

- plot_cnv() now uses clustering information stored in the infercnv
  object for the references instead of rerunning the clustering every
  time.

- Fix assumption that length(class(obj)) == 1, in this case for matrix
  objects for R 2020-01-28 update.

- Made color.palette an exported function so that it can be used for
  the "custom_color_pal" option.

- Updated some logging to be more accurate.

                 Changes in version 1.3.2 (2019-12-06)                  

- Added an new option "custom_color_pal" to plot_cnv() so a user can
  provide their choice of colors for the heatmap.

                 Changes in version 1.2.2 (2019-12-06)                  

- Fixed check of how HMM results are reported by so that add_to_seurat
  works when HMM_report_by is "cell" but HMM analysis_mode was
  "subclusters".

- Fix check of HMM_report_by setting in add_to_seurat when the option
  was not specified.

                 Changes in version 1.2.1 (2019-11-14)                  

- Fix _R_CHECK_LENGTH_1_LOGIC2_ related error because an argument was
  checked before matching to it's list of potential arguments.

- Fix text outputs for CNV reports when running with report_by="cell".

- Fix add_to_seurat method to handle CNVs reported by cells in HMM
  predictions.

- Fix how options used for a run are stored in the object and compared
  when trying to reload previous results so that it handles arguments
  given as variables in the call to run(). This also fixes the
  command-line script to run infercnv.

[InPAS](/packages/InPAS)
-----

                       Changes in version 1.19.1                        

- replace all `class` by `is`.

[INSPEcT](/packages/INSPEcT)
-------

                       Changes in version 1.16.1                        

- added the slot 'version' that store the version used in each object
  created (IMPORTANT! objects without this slot, i.e. created before
  this version cannot be used anymore and must be regenerated)

- added 2 wrap-up function to run the entire INSPEcT pipeline with a
  single command line

- ratePvals modified (calculated at the time of modeling via
  calculateRatePvals method and stored into slot ratePvals of the class
  INSPEcT)

- changes in model params accession (modeling parameters are set via
  modelRates and only accessed via modelingParams; model selection
  parameters are set via calculatePvals and only accessed via
  modeingParams)

- modifications to INSPEcT-GUI (Added Fix-Y-Axis & pvals in Y-labels,
  Confidence intervals controlled by a button rather than the checkbox)

- converted gene class names from a,c,b for synthesis, processing and
  degradation to s,p,d

[intansv](/packages/intansv)
-------

                       Changes in version 1.29.0                        

Notes

- Fix build error on Windows and Mac.

[IntEREst](/packages/IntEREst)
--------

                       Changes in version 1.12.0                        

BUG FIXES

- Replacing class( x ) == "foo" with is( x , "foo") .

[IRanges](/packages/IRanges)
-------

                       Changes in version 2.22.0                        

SIGNIFICANT USER-VISIBLE CHANGES

- Resync with change to smoothEnds() in R 4.0.  In R 4.0,
  stats::smoothEnds() always returns an integer vector when the input
  is an integer vector. smoothEnds() on an IntegerList now reflects
  this: it returns an IntegerList object instead of a NumericList
  object.

DEPRECATED AND DEFUNCT

- RangedData objects are now defunct.  RangedData objects are defunct
  in BioC 3.11. They were deprecated in BioC 3.9 and, before that,
  their used has been discouraged in favor of GRanges or GRangesList
  objects since BioC 2.12, that is, since 2014.

BUG FIXES

- Fix restrict() method for RangesList objects for when ranges are
  dropped.

[iSEE](/packages/iSEE)
----

                       Changes in version 1.99.9                        

- Fixed slot names in the ReducedDimensionPlot() man page.

                       Changes in version 1.99.8                        

- Protect against transient invalid selected index; fixes #400.  -
Forced renderDT to rerun expression upon panel reorg.

                       Changes in version 1.99.7                        

- Fixed out-of-date vignette content.

                       Changes in version 1.99.6                        

- Added static screenshots to vignettes.  - Added GitHub Actions for
continuous integration and deployment.  - Updated Docker base image
to bioconductor/bioconductor_docker:devel.

                       Changes in version 1.99.5                        

- Export utilities relevant to downstream panel development.

                       Changes in version 1.99.4                        

- Added options for dynamic choice of single/multiple selection
sources.  - Fixed bug around NA groupings in subsetPointsByGrid().  -
Explicitly notify the user when removing invalid panels supplied by a
landing page.

                       Changes in version 1.99.3                        

- Fixed occurences of rowData in RowDotPlot panels.  - Refactored
.create_visual_box().  - Refactored visual parameter sections into
generics.  - Apply global option selected.color to single selections.
- Added panel.width, panel.height, and assay to global options.  -
Added .checkboxInputHidden().

                       Changes in version 1.99.2                        

- Added extension points to the API.  - Added global settings using
iSEEOptions.  - Added .allowableYAxisChoices() and
.allowableXAxisChoices() methods to intercept choices of x/y-axis
variables.

                       Changes in version 1.99.1                        

- Allowed customization of landing pages (from calling iSEE() in
no-SE mode) for enterprise deployments.  - Allowed export of plot and
table panel outputs as PDF and CSV files, respectively.  - Fixed
handling of se objects missing dimnames.  - Added createCustomPlot()
and createCustomTable() to provide on-ramp for making customized
panels.  - Allowed global setting using iSEEOptions.  - Expanded
class and slot names from "Feat" to "Feature", "Samp" to "Sample",
"RedDimPlot" to "ReducedDimensionPlot".  - Renamed "StatTable" to
"DataTable".

                       Changes in version 1.99.0                        

- Refactored panel implementation as S4 classes.  - Display a spinner
while panels are rerendering.  - Refactored heatmap panel to use
ComplexHeatmap.

                        Changes in version 1.7.2                        

- Added notification on birthday.  - Enabled hiding of the *DataPlot
UI elements.

                        Changes in version 1.7.1                        

- Fix rbind() of data.frame and DataFrame objects.  - Fix error
related to using && with variable length greater than 1.  - Replace
deprecated scater argument.  - Replace deprecated functions:
SingleCellExperiment::clearSpikes(),
SingleCellExperiment::clearSizeFactors().

                        Changes in version 1.7.0                        

- Bioconductor release.

[iSEEu](/packages/iSEEu)
-----

                       Changes in version 0.99.8                        

- Trigger new build on the single package builder.

                       Changes in version 0.99.7                        

- Replace !="" by nzchar().  - Use 4-space indent.  - Add info on how
to contribute to iSEEu, including coding style.

                       Changes in version 0.99.6                        

- Trigger new build on the single package builder.

                       Changes in version 0.99.5                        

- Add more suggested packages.  - Add screenshot images to the
vignette.  - More realistic examples in the vignette.  - Fix
GeneSetTable to acknowledge initial value of "Selected".

                       Changes in version 0.99.4                        

- Fix to GitHub action R-CMD-check.

                       Changes in version 0.99.3                        

- Set up GitHub action R-CMD-check.  - Remove Travis CI covr code
coverage and pkgdown site deployment.  - Fix man page warning.

                       Changes in version 0.99.2                        

- Create observer for "Assay" in DiffStatTable.  - Update NEWS.

                       Changes in version 0.99.1                        

- Trigger new build on the single package builder.

                       Changes in version 0.99.0                        

- Added panel DifferentialStatisticsTable.  - Added panel
DynamicReducedDimensionPlot.  - Added panel GeneSetTable.  - Added
panel MAPlot.  - Added panel ReducedDimensionHexPlot.  - Added panel
VolcanoPlot.  - Added mode modeEmpty.  - Added mode modeGating.  -
Added mode modeReducedDim.

                        Changes in version 0.1.0                        

- iSEEu is officially born!

[IsoCorrectoR](/packages/IsoCorrectoR)
------------

                        Changes in version 1.4.1                        

- Modified molecule regex such that now more than 99 atoms per element
  can be used in the molecular formulas. Thanks to user "Farheen" by
  bringing up this issue on the Bioconductor support site
  (https://support.bioconductor.org/p/127001)

[IsoformSwitchAnalyzeR](/packages/IsoformSwitchAnalyzeR)
---------------------

                 Changes in version 1.9.5 (2020-04-22)                  

- Update type: Minor.

- Update of createSwitchAnalyzeRlist() documentation.

- Clean up of code for removal of fusion transcripts in importRdata()

                 Changes in version 1.9.4 (2020-04-21)                  

- Update type: Minor.

- importRdata() was also updated to handle edge-case fusion
  transcripts.

                 Changes in version 1.9.3 (2020-04-20)                  

- Update type: Minor.

- The creation of the switchAnalyzeRlist was updated to better handle
  cases where multiple isoforms have the same isoform_id (which could
  potentially be fusion transcripts). This was done by introducing the
  "removeFusionTranscripts" argument in both importGTF() and
  createSwitchAnalyzeRlist().

- Update date for version 1.9.3 was updated.

                 Changes in version 1.9.2 (2020-04-20)                  

- Update type: Minor

- analyzeSignalP and its documentation was further updated to handle
  edge case senarios from SignalP-5 predictions.

- In extractSequence() the 'filterAALength' argument which removed to
  short and long sequences were split into to arguments:
  'removeShortAAseq' (default is TRUE) and 'removeLongAAseq' (default
  is FALSE) to allow more nuanced control.

- Fixed a problem in extractSequence() where the onlySwitchingGenes did
  not work if the nuclotide/amino acid sequences was already stored in
  the switchAnalyzeRlist.

- extractConsequenceSummary() was modified to now also plot
  consequences analyzed but where no differences was found. This
  behaviour can be controled with the 'removeEmptyConsequences'
  argument.

- Small documentation improvements.

                 Changes in version 1.9.1 (2019-10-18)                  

- (Version bump due to Bioconductor release)

- Update type: Minor.

- importCufflinksFiles() was updated to have the isoformNtFasta option
  making it easier to work with non-model organisms.

- importGTF() was updated to allow the fasta file pointed to by the
  isoformNtFasta argument to contain extra sequences (which are then
  just ignored). This is what importRdata() already did.

- Corrected a copy/paste mistake where analyzeCPC2 was suggested to be
  run with a codingCutoff of 0.725. This have now been corrected to 0.5
  - which was the default value all the time.

- A bug in isoformToGeneExp() which could cause problems when using
  Gencode data was fixed. It was also updated to give better warning
  message in case of different ids in the quantification and
  annotation.

- Due to the repport of pfam results whith missing data (which should
  not affect its usage within IsoformSwitchAnalyzeR) analyzePFAM() now
  uses readr::read_fwf() in combination with readr::fwf_empty() instead
  of read.table() to import the fixed width file (fwf) with the pfam
  results into R. analyzePFAM() will give warnings if it finds missing
  data.

- analyzeSignalP() was updated to better handle stand-alone versions of
  SignalP-5 as well as edge case senarios from SignalP-5 predictions.

- Various documentation updates.

- switchPlotTopSwitches() was updated to give better warnings for edge
  case senarios.

- the createSwitchAnalyzeRlist() was updated add the version number of
  IsoformSwitchAnalyzeR when the switchAnalyzeRlist was created.

[isomiRs](/packages/isomiRs)
-------

                       Changes in version 1.15.2                        

FIX

- Fix error for class type when using `counts` method.

                       Changes in version 1.15.1                        

FIX

- Fix bug when dataframe is empty at isoSelect: Thanks to @DrHogart.
  https://github.com/miRTop/mirtop/issues/65#issuecomment-593845084

- Fix class error: using is now.

[KnowSeq](/packages/KnowSeq)
-------

                 Changes in version 1.1.15 (2019-11-11)                 

- Novel Biological Based Feature Selection method named as Disease
  Association Feature Selection (DA-FS) added Further versions

- MAC_OS alignment tools support

- Incorporation of RUV to batch effect methods

[limma](/packages/limma)
-----

                       Changes in version 3.44.0                        

New functionality

- Add new argument `package` to changeLog(). changeLog() now works for
  any package rather than only for limma.

Code improvements

- Improved treatment of NA coefficients by contrasts.fit().
  contrasts.fit() has always removed coefficients that were NA because
  of singularities in the design matrix, but any extra NA coefficients
  caused by NA expression values would cause all the stdev.unscaled
  values returned by contrasts.fit() to be NA for those genes.
  contrasts.fit() now returns non-NA coefficients and non-NA
  stdev.unscaled values if all the NA coefficients are multiplied by
  zero contrast multipliers.

- Speed improvement for contrasts.fit() if some input coefficients are
  not involved (have zero multipliers) in any of the contrasts.

- contrasts.fit() now preserves the `pivot` component of the fit object
  instead of removing it.

- voom now checks explicitly for NA or negative counts and gives an
  informative error message.

- fitFDist() now allows a covariate trend even with fewer than 4
  useable observations. (Useable means positive residual df and finite
  variance and covariate values.)

- Internal code cleaned up to avoid partial matching of function
  arguments, attributes or list component names. The automatic package
  tests are now run with the warnPartialMatchArgs, warnPartialMatchAttr
  and warnPartialMatchDollar options all set to TRUE.

Documentation

- More explanation of `covariates` argument to removeBatchEffects().

Bug fixes

- Fix bug in contrasts.fit() when some coefficients are NA due to NA
  expression values and the corresponding contrast multipliers are
  zero. The new code fixes the improvement originally introduced in
  limma 3.35.9. The new code will return non-NA contrasts if the NA
  coefficients all get 0 weight in the contrast. Previous code was
  correct if all genes had the same NA coefficients but could give an
  avoidable NA result if some genes had NA coefficients but the first
  gene did not.

- Improvements to subsetting MArrayLM objects by column. Previously the
  presence of non-estimable coefficients sometimes caused the
  cov.coefficient matrix to be subsetted incorrectly; that is now
  fixed. Subsetting zero columns is now allowed even when F-statistics
  are present.

- Fix to classifyTestsF() when one or more of the coefficients have
  zero variance. Previously the zero variance will introduce NAs into
  the correlation matrix and hence cause an error. This is now avoided.
  This also fixes eBayes() when one or more of the coefficients are
  identically zero (usually caused by an all-zero contrast).

[lionessR](/packages/lionessR)
--------

                 Changes in version 1.1.1 (2020-04-06)                  

- Fixed bug: custom function in aggregate network

[Maaslin2](/packages/Maaslin2)
--------

                        Changes in version 1.1.2                        

- Small changes to pass build tests for latest bioconductor release.

                        Changes in version 1.1.1                        

- Updated reference to sync up with manuscript.

[maftools](/packages/maftools)
--------

                       Changes in version 2.4.00                        

BUG FIXES

- lollipopPlot2 error with zero mutation Issue: #480 -
inferHeterogeneity PR #473 - oncoplot cnv crosses border #472 - Avoid
wrongly naming columns in annovarToMaf Issue: #457 - Duplicated color
code Issue: #452 - oncoplot with exprsTbl leads to wrong alignment
Issue: #451 - plotMafSummary with single sample Issue: #449 - Avoid
empty spaces and quoates while reading funcotator MAF Issue: #403
#397

ENHANCEMENTS

- OncogenicPathways improvements Issue: #505 - Added showSum, colNC,
nShiftSymbols, sigSymbolsSize, sigSymbolsFontSize, pvSymbols
arguments to somaticInteractions PR: #505 Thanks zmiimz - Added
anno_height, drawBox, drawExtraBar arguments to oncoplot PR: #501
Thanks Kai Gu - Added decreasing argument to tcgaComapre Issue: #497
- Added startgain and startlost to annovarToMaf Issue: #487 - y_lims
argument in gisticChromPlot Issue: #485 - Added support for
highlighting multiple additionalFeatures Issue: #476 - Pairwise
t-test and mutational load in tcgaCompare Issue: #453 - Added
titleText argument to oncoplot Issue: #448 - Highlight mutated genes
on gisticChromplot Issue: #443 - Added legend_height argument to
oncoplot Issue: #346 - Added bgBorderCol domainBorderCol showLegend
argument to lollipopPlot

NEW FUNCTIONS AND FEATURES

- tmb Simple tumor mutation burden estimation function.  - setdiffMAF
and intersectMAF. Set operations for MAF files. PR: #508 Thanks
Shixiang Wang - tcgaDriverBP Compare genes to known TCGA drivers and
their biological pathways - vafCompare plots vaf distribution of
target genes from two cohorts Issue: #495 - coBarplot side-by-side
barplot for maf comparison Issue: #486 - Group genes in oncoplot by
specific pathways. This can be done by setting pathways = 'auto' or
by providing a two column data.frame/tsv-file with Gene names and
their corresponding pathway belongings.  - Oncoplot gains
rightBarData and leftBarData arguments for user specific side bar
plots.  - Default flatUI colors for lollipop plots

Deprecated

- geneCloud and pancanComparison

[MAGeCKFlute](/packages/MAGeCKFlute)
-----------

                        Changes in version 1.7.2                        

- Allow user customized data input.

- Prioritize the documentation.

                        Changes in version 1.6.4                        

- Config Travis CI test.

- Download Depmap data automatically and save as rdata.

                        Changes in version 1.6.3                        

- Download GO terms from source database.

- Use human genes to do enrichment analaysis in pipelines.

- Revise BatchRemove to have the same parameters as paper mentioned.

- Revise FluteRRA to surport only gene summary as input.

                        Changes in version 1.6.2                        

- Debug Depmap data loading process

                        Changes in version 1.6.1                        

- Incorporate gene id conversion between organisms into TransGeneID.

- Incorporate visualization functions from enrichplot.

- Add new functions for visualization, such as ScatterView and BarView.

- Debug the read file commands.

- Add additional document for enrichment analysis.

- Integrate Depmap data into the pipeline.

[MatrixGenerics](/packages/MatrixGenerics)
--------------

                       Changes in version 0.0.99                        

- Added a NEWS.md file to track changes to the package.

[matter](/packages/matter)
------

                       Changes in version 1.13.10                       

NEW FEATURES

- Added 's_nnzero()' stream-stat for non-zero entries

- Added 'push()' and 'pull()' generics for future use

                       Changes in version 1.13.9                        

BUG FIXES

- Minor updates to 'chunk_apply()' internals

                       Changes in version 1.13.8                        

BUG FIXES

- Fixed bugs due to logical condition length > 1

                       Changes in version 1.13.7                        

BUG FIXES

- Fixed to internal function 'linearInd()'

                       Changes in version 1.13.6                        

BUG FIXES

- Fixed bug caused by change in R 4.0 so that class of matrix is now
  c("matrix", "array")

                       Changes in version 1.13.5                        

NEW FEATURES

- Support for 'view="chunk"' when specifying 'pattern'

                       Changes in version 1.13.4                        

NEW FEATURES

- Added 'pattern' argument to 'chunk_mapply()'

                       Changes in version 1.13.3                        

NEW FEATURES

- Implemented 'outfile' argument for writing results to a file for
  'chunk_apply()' and 'chunk_mapply()'

                       Changes in version 1.13.2                        

NEW FEATURES

- Added 'pattern' argument to 'chunk_apply()'

                 Changes in version 1.13.1 (2019-10-30)                 

NEW FEATURES

- Added 'chunk_mapply()'

[mbkmeans](/packages/mbkmeans)
--------

                 Changes in version 1.3.4 (2020-03-25)                  

- Change default batch size to 500 in `mbkmeans`.

- Change default init_fraction to be the same as the batch size in
  `mbkmeans`.

                 Changes in version 1.3.3 (2020-03-12)                  

- Change to `blockApply` to compute labels.

                 Changes in version 1.3.2 (2020-02-04)                  

- Added option `compute_labels=TRUE` to optionally avoid computing
  labels and return only the centroids.

[mCSEA](/packages/mCSEA)
-----

                        Changes in version 1.7.0                        

- GSEA algorithm updated to fGSEAMultiLevel method.

[MEAT](/packages/MEAT)
----

                 Changes in version 0.99.0 (2020-01-09)                 

- Submitted to Bioconductor

[MEB](/packages/MEB)
---

                        Changes in version 1.2.0                        

- The latest one.

                        Changes in version 1.1.0                        

- failed update.

[metagene2](/packages/metagene2)
---------

                 Changes in version 1.3.3 (2019-11-13)                  

- Fixed a bug which caused overlapping contiguous regions to be
  assigned the wrong coverage values.

                 Changes in version 1.3.1 (2019-11-05)                  

- Added the metagene2_heatmap function, which generates heatmap
  representation of the binned coverages underlying the metagene2
  object.

[metaseqR2](/packages/metaseqR2)
---------

                 Changes in version 0.99.2 (2020-04-22)                 

NEW FEATURES

- First submission of metaseqR2

[methylGSA](/packages/methylGSA)
---------

                        Changes in version 1.5.9                        

- Bug fixes on Shiny app - Updated the definition of "promoter" on
Shiny app

                        Changes in version 1.5.5                        

- In methylRRA(ORA), the overlapping between significant genes and
genes in the gene set can be printed out

                        Changes in version 1.5.2                        

- Updated the definition of "promoter"

                        Changes in version 1.5.1                        

- Minor updates on package vignette

[methylKit](/packages/methylKit)
---------

                       Changes in version 1.13.4                        

IMPROVEMENTS AND BUG FIXES

- set stringAsFactors to FALSE - changed when reading data and tabix
  files - this solves class missmatches downstream - update methylKit
  data with stringsAsFactors off

- methRead: - filter for given context when pipeline is cytosinereport
  - order data when processing with methread, this ensures that
  methylRaw and methylRawDB contain exact same data - add additional
  test for methread, check whether methread with or without tabix
  produces same data when processing bismark coverage and cytosine
  report

- unite: - make sure .CpG.dinuc.unifyOld equals .CpG.dinuc.unify - add
  test to check that destranding result is same for methylBase and
  methylBaseDB

                       Changes in version 1.13.3                        

IMPROVEMENTS AND BUG FIXES

- getMethylDiff: - fixes bugs in method for methylDiffDB - suffix will
  be actual type now, was "type" before - removed whitespaces in
  filename, fixes https://github.com/al2na/methylKit/issues/171 - add
  extensive test for getmethylDiff function

- applyTbxByChunk: - Allow optional writing of colnames and rownames if
  returntype is text. Find details in
  https://github.com/al2na/methylKit/issues/169

- percMethylation: - return rownames for methylDB when rowids is TRUE,
  fixes https://github.com/al2na/methylKit/issues/170 - add extensive
  tests for percMethylation

- reorganize: - fix bug in method for methylBaseDB introduced with new
  tabix header - add extensive testing for reorganize

- assocComp: - add further checks for proper annotation

- removeComp: - set default comp to NULL, to trigger earlier fail - fix
  bug method for methylBaseDB introduced with new tabix header

- reconstruct: - update checks for percent methylation matrix - fix bug
  with fread

- add extensive tests for batch control functions

- diffMethPerChr: - prevent from plotting when no chrom left - add
  extensive tests for diffMethPerChr

                       Changes in version 1.13.2                        

IMPROVEMENTS AND BUG FIXES

- fread.gzipped: - add pre-checks wheter file exists and is not empty -
  switch to R.utils::gunzip instead of building shell command manually

- tileMethylCounts: remove orphan warning triggered by incomplete
  temporary methylRaw object, fixes
  https://github.com/al2na/methylKit/issues/175

- update vignette: - add mincov argument to methread - add paragraph to
  tiling section about reducing coverage threshold + update example
  code

- update docs: - rebuild docs with more recent version of roxygen -
  exclude examples from internal function

- increase version requirement for tabix files with header

                       Changes in version 1.13.1                        

IMPROVEMENTS AND BUG FIXES

- fix issues related to build checking: - logical vectors of size
  longer than one - docs for methseg contain line longer than 100 chars

- fix methSeg2bed error when seg.mean=0 and scores=NA by replacing
  score with zero if average seg.mean is zero
  https://github.com/al2na/methylKit/issues/123

- update vignette: - explain how to get wig/bigwig file from methylKit
  https://github.com/al2na/methylKit/issues/157 - explain how to deal
  with soft clipping in processBismarkAln
  https://github.com/al2na/methylKit/issues/159 - add section about
  exporting and loading tabix objects starting from version 1.13.1

[methylSig](/packages/methylSig)
---------

                 Changes in version 0.99.0 (2020-03-23)                 

- Refactor functions and workflow from pre-0.99.0 releases o
  methylSigReadData() replaced with the functions: o
  bsseq::read.bismark() o filter_loci_by_coverage() o
  filter_loci_by_location() o methylSigTile() replaced with the
  functions: o tile_by_regions() o tile_by_windows() o Differential
  testing should be preceded with: o filter_loci_by_group_coverage() o
  binomialDiffCalc() is replaced by diff_binomial() o methylSigCalc()
  is replaced by diff_methylSig() o methylSigDSS() is replaced by
  diff_dss_fit() and diff_dss_test()

- See "Using methylSig" vignette for full example.

- See "Updating methylSig Code" vignette for how to retrofit pre-0.99.0
  code.

[MetNet](/packages/MetNet)
------

                 Changes in version 1.5.3 (2020-03-09)                  

- change tests for rtCorrection, do not use function levels to test for
  if `+`, `.` or `?` is present in the transformation list

- update vignette that formulas are displayed properly

                 Changes in version 1.5.2 (2020-01-12)                  

- change unit tests for clr and aracne that it doesn't fail in Windows

- change ppm calculation in structural, use m/z of precursors instead
  of m/z differences

- change unit tests for clr and aracne that it doesn't fail in Windows

                 Changes in version 1.5.1 (2019-12-05)                  

- rename combineStructuralStatistical to combine

- combine accepts directly the output of createStructuralAdjacency,
  before the function combineStructuralStatistical accepted a numeric
  matrix (first entry of the output of createStatisticalAdjacency)

- remove the functions consensusAdjacency, createStatisticalAdjacency,
  createStatisticalAdjacencyList

- add functions statistical, getLinks, topKnet, threshold with improved
  functionality for thresholding the statistical adjacency matrices
  (based on hard thresholds, or top1, top2 or mean consensus matrix
  calculation according to Hase et al.) to replace removed functions

- rename createStructuralAdjacency to structural

- change vignette to markdown

- add parameter directed in structural to allow for directed networks

- add parameter values to specify if respectively min, max or all
  values from the corresponding feature pairs (in the upper and lower
  triangle) should be taken in statistical

[microbiome](/packages/microbiome)
----------

                 Changes in version 1.9.9 (2019-12-25)                  

- noncore_members removed

- Added aggregate_rare function

- Bioc polishing

                 Changes in version 1.9.2 (2019-12-25)                  

- Divergence function rewritten for clarity * Switched from sapply to
  vapply

- Removed noncore_abundance

- Fixing plot_core axis labeling

- is.compositional function added

- spreadplot function added

- removed ready made themes from functions

- Renamed is.compositional to is_compositional

- Fixed a bug in core_members (also non-compositional detection now
  allowed)

- removed rm.na option from aggregate_taxa

- Deprecating noncore_* functions (replacing with rare_* functions
  everywhere)

- Removed variable_members function

- Support removed from R-3.3.3 and lower

[microbiomeDASim](/packages/microbiomeDASim)
---------------

                 Changes in version 1.1.6 (2020-03-10)                  

- F1000 manuscript with package details accepted and available at
  <https://doi.org/10.12688/f1000research.20660.2>

- Citation updated within package

- Require study duration be specified when simulating observations

- Implemented functionality to allow random asynchronous timepoints
  samples along the study duration

- Additional functions to approximate observed longitudinal microbiome
  data

- Convert simulated data into common data objects in either the
  metagenomeSeq or phyloseq packages

- Existing function changes:

- gen_norm_microbiome (must specify t_interval for study duration and
  additional option to specify asynchronous time points using
  asynch_time=TRUE)

- mvrnorm_sim (must specify t_interval for study duration and
  additional option to specify asynchronous time points using
  asynch_time=TRUE)

- New functions:

- conversion_functions (convert simulated data to metagenomeSeq or
  phyloseq object)

- sim_obs_functions (host of functions that allow users to specify
  discrete sampling timepoints and approximate observed longitudinal
  patterns)

[miRSM](/packages/miRSM)
-----

                        Changes in version 1.5.1                        

- Update miRSM function <2020-02-01, Sat>

[miRspongeR](/packages/miRspongeR)
----------

                    Changes in version 1.13.1-1.13.2                    

- Update vignettes <2020-03-25, Wed>.

[missMethyl](/packages/missMethyl)
----------

                       Changes in version 1.21.4                        

- Bug fix: Fixed bug that resulted in negative value for odds. Bug only
  occurred in rare cases when CpG universe was smaller than all CpGs on
  array AND if some large collections had more genes than the universe.

                       Changes in version 1.20.3                        

- Added fract.counts logical parameter to enable fractional counting of
  CpGs to be turned off in gometh, gsameth, goregion and gsaregion.
  This switches off the adjustment for CpGs that map to multiple genes.
  This is primarily intended for comparison with earlier versions and
  we strongly encourage it to be set to TRUE.

[mitch](/packages/mitch)
-----

                       Changes in version 0.99.15                       

- Making changes in response to bioconductor review

                        Changes in version 0.2.2                        

- Test

[mixOmics](/packages/mixOmics)
--------

                       Changes in version 6.12.0                        

new features / enhancements

- circosPlot and plotLoadings bug caused by features with NAs fixed -
circosPlot's inconsistentcy of blocks with identical X names fixed -
consensus and weighted consensus plots now supported for plotIndiv
with relevant block analyses - plotLoadings's feature name trimming
can be customised - block.splsda bug which could drop some Y factors
with near.zero.variance=TRUE fixed - perf.block.splsda now supports
calculation of combined and per-block AUC - model improvement
significance can be custmoised in all perf and tune functions -
perf.block.splsda is now much faster and supports FORK clusters -
tune.(s)pls(da), perf.(s)plsda now support FORK clusters

bug fixes

- plotIndiv(..., ind.names = FALSE) warning/bug fixed -
tune.block.splsda bug on Windows parallelisation fixed - perf and
tune functions' issue when choosing the optimum component resolved -
added option to suppress auroc from printing all the AUCs

[MLInterfaces](/packages/MLInterfaces)
------------

                       Changes in version 1.67.10                       

- Fix es2df (for R 4.0.0) setting stringsAsFactors = TRUE

- Add unit test

                       Changes in version 1.67.1                        

- rda has been moved to 'Enhances' and its use is conditional its
  presence <2019-01-05>

[Modstrings](/packages/Modstrings)
----------

                 Changes in version 1.3.9 (2019-03-02)                  

- added incompatibleModifications(), which returns a logical for manual
  subsetting

- added combineModifications() to allow modifications to be combined
  manually

- added separate() for GRanges to allow modification to be separated,
  if possible

                 Changes in version 1.3.4 (2019-02-05)                  

- added removeIncompatibleModifications()

                 Changes in version 1.3.3 (2019-02-05)                  

- combineIntoModstrings() is now 'vectorized' for multiple sequences.
  Sequences with the same name are treated independently and are
  modified equally

- add stop.on.error option for combineIntoModstrings()

                 Changes in version 1.3.2 (2019-12-01)                  

- Display ModDNA and ModRNA sequences in color

[MOMA](/packages/MOMA)
----

                 Changes in version 1.0.0 (2020-04-22)                  

- Bioconductor Initial Release

[MotifDb](/packages/MotifDb)
-------

                        Changes in version 1.30                         

NEW FEATURES

- now has HOCOMOCO version 11, with all motifs classified, and
  queryable, by category ("core", "secondary") and reliability (A, B,
  C, D).

[motifStack](/packages/motifStack)
----------

                       Changes in version 1.31.2                        

- reduce the compiling time for vignettes.

                       Changes in version 1.31.1                        

- fix the issue invloved by replace class(x)=="something" to is(x,
  "something").

[MsCoreUtils](/packages/MsCoreUtils)
-----------

                        Changes in version 0.99                         

MsCoreUtils 0.99.3

- Trigger build.

MsCoreUtils 0.99.2

- Provide more comprehensive description.  - Add vignette.

MsCoreUtils 0.99.1

- Additional functions, and using Author@R to specify (unique)
RforMassSpectrometry Package Maintainer.

MsCoreUtils 0.99.0

- First release of MsCoreUtils with core function to get extreme
values, grouping/matching, noise/smoothing, similarity measurements,
various helper function, and function to process (impute and
normalise) quantitative features.

[MSnbase](/packages/MSnbase)
-------

                        Changes in version 2.13                         

Changes in 2.13.9

- Typo in OnDiskMSnExp man page <2020-04-26 Sun> - Fix
plot,&#91;On&#93;MSnExp for centroided data <2020-04-26 Sun> - Use free x
scaling in plot,MSnExp's facet <2020-04-26 Sun>

Changes in 2.13.8

- Remove duplicated COM= when writing mgf <2020-04-24 Fri>

Changes in 2.13.7

- Make writeMgfData example runnable <2020-04-20 Mon> - Fix issue
when writing mgf with a connection (PR #505, by Rico Derks)
<2020-04-20 Mon> - Add writeMgfData file/connection unit test
<2020-04-20 Mon>

Changes in 2.13.6

- When reading MzTab files, the column names aren't converted to
legal colnames (my make.names) - check.names in read.delim is now
(internally) set to FALSE. This will make it easier to write MzTab
objects back to files. See issue #501 for a discussion.  -
Re-introduce MzTab write support (contributed by Steffen Neumann).
See #502.

Changes in 2.13.5

- Support reading of MzTab-M 2.0 format (contributed by Steffen
Neumann).  - Use UTF-8 encoding.

Changes in 2.13.4

- Fix failing unit test: in R 4.0, data.frames not don't cast strings
to factors anymore <2020-03-24 Tue>

Changes in 2.13.3

- exprsToRatios is now defunct.

Changes in 2.13.2

- Need ProtGenerics 1.19.3 for impute generic

Changes in 2.13.1

- Need ProtGenerics 1.19.2 - Deprecate filterZero and exprsToRatios

Changes in 2.13.0

- Bioconductor 3.11 (devel)

[msPurity](/packages/msPurity)
--------

                       Changes in version 1.13.2                        

- Update dev to match bug fixes in master

                       Changes in version 1.13.1                        

- Update dev to match bug fixes in master

                       Changes in version 1.13.0                        

- Bioconductor dev (automatic version bump)

                       Changes in version 1.12.2                        

- Fix to allow large files to be processed with combineAnnotation (in
  some cases the summary report should be excluded)

                       Changes in version 1.12.1                        

- Bug fix for combineAnnotations - MS1 lookup not handling CAMERA
  adducts correctly

- Fillpeaks bug fix see github #68

- Documentation fixes

[MSstatsTMT](/packages/MSstatsTMT)
----------

                 Changes in version 1.4.7 (2020-04-24)                  

- Update the NEWS file

                 Changes in version 1.4.6 (2020-04-14)                  

- Fix bug in groupComparison() for unbalanced design

- Use df approximation from lmerTest to perform group comparison

                 Changes in version 1.4.5 (2020-03-01)                  

- Add new function OpenMStoMSstatsTMTFormat()

                 Changes in version 1.4.4 (2020-02-01)                  

- Fix bug in PDtoMSstatsTMTFormat(): remove redundant rows when
  combining multiple fractions

                 Changes in version 1.4.3 (2019-12-28)                  

- Fix bug in groupComparisonTMT(): very few measurements case
  previously doesn't work

                 Changes in version 1.4.2 (2019-12-20)                  

- Add the column 'issue' to the output of groupComparisonTMT()

                 Changes in version 1.4.1 (2019-10-31)                  

- Fix the bug in the PDtoMSstatsTMTFormat() due to different PD version

[MultiAssayExperiment](/packages/MultiAssayExperiment)
--------------------

                       Changes in version 1.14.0                        

New features

- exportClass creates a number of .csv data files for exporting data
- Allow vector input i for selecting assays in longFormat (@lgatto,
#266) - Updates to 'Using MultiAssayExperiment with DelayedMatrix'
vignette

Bug fixes and minor improvements

- Warn when colData rownames and ExperimentList colnames are empty (@LTLA #262) 
- Add informative error message for ExperimentList (@lgatto, #265) 
- Informative warning when dropping ExperimentList element columns (@lwaldron) 
- Fixes to constructor functions, MultiAssayExperiment and MatchedAssayExperiment (@lgatto, #267 #268, @lwaldron) 
- Add warning when j in mae&#91;i, j, k&#93; is longer than colData rows 
- Strict argument matching between generic and methods 
- Updates due to class(matrix()) 
- UpsetSamples more robust to differences in names between split sampleMap and names(ExperimentList) (@jonocarroll, #269) 
- Refactored and improved UpsetSamples 
- ExperimentList propagation of mcols and metadata (@vobencha, #270) 
- Enforcement of validObject with replacement methods colData and sampleMap (@vobencha, #271)

[mzR](/packages/mzR)
---

                       Changes in version 2.21.1                        

- inject a new PSI-MS.obo controlled vocabulary data-version: 4.1.30
  date: 30:08:2019 16:10 saved-by: Gerhard Mayer

[NADfinder](/packages/NADfinder)
---------

                       Changes in version 1.11.1                        

- replace all class by is.

[NBAMSeq](/packages/NBAMSeq)
-------

                 Changes in version 1.3.1 (2020-02-18)                  

- The default gamma parameter in NBAMSeq changed from 2 to 2.5.

- The NBAMSeq function now supports fitting linear model after fitting
  GAM model.

[netDx](/packages/netDx)
-----

                 Changes in version 0.99.0 (2019-11-11)                 

- Submitted to Bioconductor

[ngsReports](/packages/ngsReports)
----------

                        Changes in version 1.3.3                        

- Added line plot to plotDupLevels()

                        Changes in version 1.3.2                        

- Changed default template to use DT instead of kable for tables and
  removed kableExtra dependency

- Added outputDir as an argument to writeHtmlReport()

[normr](/packages/normr)
-----

                       Changes in version 1.13.1                        

- Changed deprecated 'GenomeInfoDb::fetchExtendedChromInfoFromUCSC' to
  'GenomeInfoDb::getChromInfoFromUCSC'

[NPARC](/packages/NPARC)
-----

                 Changes in version 0.99.0 (2019-10-29)                 

- Submitted to Bioconductor

[omicplotR](/packages/omicplotR)
---------

                        Changes in version 1.7.1                        

- bug fix to cause by Shiny 1.4.0 (renderDataTable causes error in
  jquery). solution is to import DT package and specify
  DT::renderDataTable and DT::dataTableOutput.

- ensured that ALDEx2 dependency for aldex.plot feature is correctly
  implemented

- added Brandon Lieng as author

                        Changes in version 1.7.0                        

- bioconductor dev version 3.11, no changes since 1.5.4

[OncoSimulR](/packages/OncoSimulR)
----------

                Changes in version 2.17.92 (2020-04-27)                 

- Fixed test.Z-magellan failure in Mac.

                Changes in version 2.17.91 (2020-04-24)                 

- Fixed Wrestrict warnings in Windows from MAGELLAN's sources.

                 Changes in version 2.17.9 (2020-03-16)                 

- With R-devel, stringsAsFactors = FALSE by default. Fix tests

- POM documentation.

                 Changes in version 2.17.8 (2020-02-02)                 

- With newest R-devel (01-28) getting "the condition has length > 1":
  fixed.

                 Changes in version 2.17.7 (2020-01-30)                 

- Removed dependency on nem, to be deprecated. Adding nem's code (file
  nem_transitive_reduction.R).

                 Changes in version 2.17.6 (2020-01-22)                 

- Further additions of MAGELLAN's functionality and Additive model to
  rfitness

                 Changes in version 2.17.5 (2019-12-19)                 

- Oooops: forgot Makevars.win

                 Changes in version 2.17.4 (2019-12-19)                 

- fno-common: better way of dealing with verbose in magellan.

                 Changes in version 2.17.3 (2019-12-17)                 

- Compiles with fno-common (for gcc 10).

                 Changes in version 2.17.2 (2019-12-17)                 

- Fixed error "length(x) = 5 > 1' in coercion to 'logical(1)'"

                 Changes in version 2.17.1 (2019-11-24)                 

- rfitness: clarified log=TRUE and truncate_at_0 after log.

- Magellan_stats: really return a vector.

                 Changes in version 2.17.0 (2019-10-25)                 

- Bumped version for BioC-3.11

[openCyto](/packages/openCyto)
--------

                        Changes in version 3.11                         

Enhancements

- Allow control of mininum number of events for each step of
GatingTemplate #298

Bug Fixes

- Handle a few more edge cases in .improvedMinDensity - Added a fix
to the density estimate used by gate_tautstring

                        Changes in version 3.10                         

API Changes

Simple renaming

- gate_flowClust_1d -> gate_flowclust_1d - gate_flowClust_2d ->
gate_flowclust_2d - tautStringGate -> gate_tautstring - templateGen
-> gh_generate_template - add_pop_init -> gs_add_gating_method_init -
add_pop -> gs_add_gating_method - remove_pop ->
gs_remove_gating_method - get.helperGates -> gs_get_helpergates -
toggle.helperGates -> gt_toggle_helpergates - delete.helperGates ->
gs_delete_helpergates - getNodes -> gt_get_nodes - getParent ->
gt_get_parent - getChildren -> gt_get_children - getGate ->
gt_get_gate - gating -> gt_gating - registerPlugins ->
register_plugins - Classes and methods no longer exported

- registerGatingFunction - gtMethod - gtPopulation - polyFunctions -
ppMethod

Bug Fixes

- Some minor fixes to gt_toggle_helpergates - Fix "positive" argument
to gate_mindensity to match doc

[oposSOM](/packages/oposSOM)
-------

                        Changes in version 2.5.2                        

- Bugfix for class(...) error occuring in R > 4.0.0

                        Changes in version 2.5.1                        

- Minor bugfixes

                        Changes in version 2.5.0                        

- Version bump of Bioconductor

[ORFik](/packages/ORFik)
-----

                       Changes in version 1.7.17                        

SIGNIFICANT USER-VISIBLE CHANGES

- Added pmapToTranscriptsF, a much faster pmapToTranscripts.

- All of ORFik now supports weights for representing duplicated reads.
  This will speed up all function related to NGS data a lot.

                        Changes in version 1.7.0                        

SIGNIFICANT USER-VISIBLE CHANGES

- The ORFik experiment syntax is ready, to simplifiy ORFik usage with
  big data.

[packFinder](/packages/packFinder)
----------

                 Changes in version 1.0.0 (2020-04-25)                  

- Minor bugfixes

- Generate package graphics

- Allow for partial TSD matching

- Add methods for BLAST classification

- Add method to collapse overlapping sequences

                 Changes in version 0.99.0 (2019-10-28)                 

- Package Submitted to Bioconductor

- Core package functionality implemented

[PAST](/packages/PAST)
----

                 Changes in version 1.2.8 (2020-03-16)                  

- fixed warnings from missing documentation

                 Changes in version 1.2.7 (2020-03-11)                  

- brought documentation in line with code

                 Changes in version 1.2.6 (2020-03-11)                  

- fixed a warning in plot_pathways

                 Changes in version 1.2.5 (2020-03-10)                  

- fixed an error in loading large annotations

                 Changes in version 1.2.4 (2020-03-04)                  

- brought code in line with documentation

                 Changes in version 1.2.3 (2020-02-27)                  

- added correct citation after paper publication

                 Changes in version 1.2.1 (2019-12-17)                  

- minor corrections to package based on original method

[PCAtools](/packages/PCAtools)
--------

                        Changes in version 2.0.0                        

- added parallelPCA function to perform Horn's parallel analysis, which
  chooses an ideal number of principal components to retain (courtesy
  Aaron Lun)

- added findElbowPoint function, which finds the elbow point in the
  curve of variance explained and which can also be used to determine
  the number of principal components to retain (courtesy Aaron Lun)

- user can now specify custom labels for points

- fixed bug with singlecol parameter for biplot colouring everything
  black

- added p-value adjustment for eigencorplot (aleighbrown)

- eigencorplot now automatically converts non-numeric columns to
  numeric (aleighbrown)

[peakPantheR](/packages/peakPantheR)
-----------

                 Changes in version 1.1.2 (2020-04-05)                  

- Correct test tolerance for Win on R 4.0.0

                 Changes in version 1.1.1 (2020-03-04)                  

- Compatibility with R 4.0.0

[peco](/packages/peco)
----

                Changes in version 0.99.16 (2020-04-10)                 

Bug fixes

- Fix AppVeyor build (use R release) - Fix Travis build (use R
release) - Require R >= 2.10 - Ignore Rmd-related files when building
package

                Changes in version 0.99.15 (2020-04-08)                 

Minor changes

- Relaxed minimum versions for dependencies

                Changes in version 0.99.14 (2020-03-23)                 

Bug fixes

- Tavis-CI and AppVeyor

                Changes in version 0.99.13 (2020-03-23)                 

Bug fixes

- Debug cycle_npreg_outsample example

Bug fixes

- Debug vignettes.Rmd

                Changes in version 0.99.11 (2020-03-23)                 

Major changes

- Rename fit_trendfilter_generic to fit_trendfilter

                 Changes in version 0.99.7 (2020-01-01)                 

Major changes

- Backward compatibiltiy with R3.5.0

                 Changes in version 0.99.6 (2020-01-01)                 

Major changes

- Inclue new vignette example

Bug fixes

- Debug cycle_npreg_outsample and data_transform_quantile

                 Changes in version 0.99.3 (2019-09-19)                 

Major changes

- cycle_npreg_outsample outputs SingleCellExperiment object -
data_transform_quantile inputs and outputs SingleCellExperiment
object

Bug fixes

- @import and @importFrom fields - BiocCheck notes

Major changes

- Store data in SingleCellExperiment objects

                 Changes in version 0.99.1 (2019-09-16)                 

Bug fixes

- R CMD Check warnings and errors - R CMD BiocCheck warnings and
errors

                 Changes in version 0.99.0 (2019-09-11)                 

Submitted to Bioconductor

[phantasus](/packages/phantasus)
---------

                        Changes in version 1.7.6                        

- Fix bug after change to stringAsFactors=FALSE

                        Changes in version 1.7.4                        

- Removed Matrix.utils dependency

                        Changes in version 1.7.3                        

- Volcano plot tool

                        Changes in version 1.7.2                        

- Better GEO loader

- DESeq2 experimental support

- Initial support for hierarchical preloaded folder

[PhyloProfile](/packages/PhyloProfile)
------------

                       Changes in version 1.1.12                        

- Added pseudo IDs for Holozoa and Holomycota clades and their non-NCBI
  sub-clades

                       Changes in version 1.1.11                        

- Increased the resolution of the phylostratigraphy function

                       Changes in version 1.1.10                        

- Improved the rank indexing function

                        Changes in version 1.1.6                        

- Removed grDevices, grid, stats, utils from Imports list in
  DESCRIPTION

                        Changes in version 1.1.5                        

- Added progress indicators for data loading and processing steps

- Removed shinycssloaders

                        Changes in version 1.1.3                        

- Fixed bug listing taxa by selecting a supertaxon from a higher level

- Added message for downloading preprocessed data

                        Changes in version 1.1.2                        

- Fixed bug sorting taxa when taxonomy rank is set as strain

                        Changes in version 1.1.1                        

- Commented examples of OMA parser functions

- Turned off tests for OMA parser functions

[Pigengene](/packages/Pigengene)
---------

                Changes in version 1.13.14 (2020-03-03)                 

Changes in existing functions

- inherits(a, b) is now used instead of class(a)==b.

                 Changes in version 1.13.6 (2020-01-03)                 

Changes in existing functions

- doSave was added to the combine.networks() function.

                 Changes in version 1.13.4 (2019-11-19)                 

Bug Fixes

- In the combine.networks() function, the netwok typo was fixed.

[plgem](/packages/plgem)
-----

                       Changes in version 1.59.2                        

- bug fixes: --fixed syntax error in `plgem.obsStn' --imported function
  `is' into NAMESPACE

                       Changes in version 1.59.1                        

- bug fixes: --updated class-checks to avoid package failures in R 4.0,
  as reported in
  https://stat.ethz.ch/pipermail/bioc-devel/2020-January/016081.html
  and
  http://bioconductor.org/developers/how-to/troubleshoot-build-report/#classEq
  --imported several functions into NAMESPACE from `grDevices',
  `graphics' and `stats'

[plyranges](/packages/plyranges)
---------

                       Changes in version 1.7.16                        

- refactoring of select internals, improved speed when casting a
GRanges -> DataFrame

                       Changes in version 1.7.15                        

- further fixes to reduce/disjoin internals

                       Changes in version 1.7.14                        

- fixes reduce/disjoin internals cleans up disjoin cases when an
expansion occurs

                       Changes in version 1.7.13                        

- set tidyselect version to be v 1.0 
- set coverage method for delegating ranges 
- fix docs for bam reading

                       Changes in version 1.7.11                        

- move from tidyselect::vars_select() to tidyselect::eval_select()

                        Changes in version 1.7.7                        

- update handling of list columns, expand_ranges() no longer takes
cartesian product if lists are parallel. summarize() properly handles
list column output without blowing out number of columns.

                        Changes in version 1.7.6                        

- adds method for dplyr::sample_n()

                        Changes in version 1.7.5                        

- fixed issue #62 for Ranges construction, the as_granges() and
as_iranges() functions now handle List columns correctly 
- added in helper functions for dealing with names in Ranges. See ?ranges-names
for details.

                        Changes in version 1.7.4                        

- added slice() for Ranges, and GroupedRanges 
- internals of grouping have been overhauled, but there shouldn't be any user facing changes. It 
is now much faster to generate groupings.  
- group information can be interrogated with dplyr::group_keys() 
- a GRangesList can be obtained automatically from a GroupedGenomicRanges with dplyr::group_split() 
- group indices can be generated with dplyr::group_indices()

                        Changes in version 1.7.3                        

- shift_downstream() and shift_upstream() now properly handle vector
amounts of shift. Fixes issue #73

                        Changes in version 1.7.2                        

- Left outer join overlap operations now work if either x or y have no metadata columns see #70 
- Left outer join overlap operations will also correctly behave in situations when there are no non-overlapping
ranges.  
- Left outer join overlaps no longer modify seqinfo [see here](https://support.bioconductor.org/p/125623/)
- patch left outer join when x or y are IRanges, flesh out overlaps documentation.

                        Changes in version 1.7.1                        

- Reformatting NEWS.md so no longer softlinks to inst/NEWS

[pmp](/packages/pmp)
---

                       Changes in version 0.99.6                        

- Updated vignettes.

                       Changes in version 0.99.2                        

- Support for SummarizedExperiment class.

- Updated documentation and vignette.

- Added functions for QCRSC algorithm of signal-batch correction for MS
  data.

                        Changes in version 0.2.6                        

- glog transformation can use fixed lambda value input.

- Plot to visualise output of glog lambda value optimisation.

                        Changes in version 0.2.5                        

- Pre-calculated mean reference values can be used with PQN
  normalisation function.

                 Changes in version 0.2.4 (2019-08-07)                  

- Updated documetation and usage examples.

- Refactored glog scaling function.

- PQN normalisation supports using all samples to calculate correction
  factor.

[podkat](/packages/podkat)
------

                       Changes in version 1.19.1                        

- fix in nullModel() method to ensure compatibility with R 4.0

                       Changes in version 1.19.0                        

- new branch for Bioconductor 3.11 devel

[polyester](/packages/polyester)
---------

                       Changes in version 1.99.3                        

- NB function now exported

- note that version 1.99.3 on GitHub was version 1.1.0 on Bioconductor.

                       Changes in version 1.99.2                        

- bug fix in fragment generation (last 2 bases of transcript were never
  sequenced)

[proDA](/packages/proDA)
-----

                         Changes in version 1.1                         

- Only convert 0's to NA's if there are no NA's in the dataset (commit
  06df44a, fix issue #3)

[pRoloc](/packages/pRoloc)
------

                        Changes in version 1.27                         

Changes in version 1.27.6

- Depend on MLInterfaces 1.67.10

Changes in version 1.27.5

- import stats4::plot <2020-03-26 Thu>

Changes in version 1.27.4

- Import missing mclust::mclustBIC

Changes in version 1.27.3

- Remove exprsToRatio man page (function in MSnbase and is
deprecated)

Changes in version 1.27.2

- Fix errors related to R-devel

Changes in version 1.27.1

- Merged plotting payes PR

Changes in version 1.27.0

- Version bump for Bioc 3.11 (devel)

[pRolocGUI](/packages/pRolocGUI)
---------

                        Changes in version 1.21                         

CHANGES IN VERSION 1.21.3

- Depend on latest pRoloc

CHANGES IN VERSION 1.21.2

- Keep only one maintainer

CHANGES IN VERSION 1.21.1

- Update email

[ProtGenerics](/packages/ProtGenerics)
------------

                       Changes in version 1.19.3                        

- new impute generic

                       Changes in version 1.19.2                        

- new filterNA generic

                       Changes in version 1.19.1                        

- new aggregateFeatures generic

[psichomics](/packages/psichomics)
----------

                       Changes in version 1.12.1                        

- Alternative splicing events can now be represented via diagrams: -
Redesign of alternative splicing event selection (graphical
interface) - plotSplicingEvent() plots diagram representation of
alternative splicing events - In the visual interface, alternative
splicing event diagrams were added below distribution plots (to
quickly illustrate higher and lower values of alternative splicing
quantification) and in annotation page - User-provided junction
quantification loading: - Support junction coordinates from
mitochondrial, Z and W chromosomes - Fix issues with files containing
splice junctions within random, alt and unknown chromosomes by
discarding those rows (a warning is raised) - Alternative splicing
annotation: - listSplicingAnnotations() can now be filtered by
species, assembly and data of available annotations - Improve
import/export of data groups from/to a file, including colour support
- Copy-edit and improve all tutorials, welcome message and help tab -
Include link to article in Methods in Molecular Biology:
https://doi.org/10.1007/978-1-0716-0301-7_10

More bug fixes and minor changes

- TCGA and SRA data loading: - loadFirebrowseData() now returns
expected data when asking for multiple datasets (such as in the case
of performing a pan-cancer analysis) in both visual and command-line
interface - SRA projects containing only one column of extra
information in sample metadata are now correctly loaded instead of
raising an error (loadSRAproject()) - Data loading and manipulation:
- Copy-edit information on the format of user-provided files - Warn
when discarding rows with duplicated rownames after loading
user-provided files - getGtexDataTypes() is now exported, as expected
- parseSplicingEvent() now returns the coordinates as numeric if
coords = TRUE and char = FALSE - Improve dialog when trying to load a
local folder without any supported files available - Alternative
splicing annotation: - Confirmation dialog is not displayed any more
when creating a folder (specially useful while running the visual
interface) - Allow to select cache directory of AnnotationHub
(command-line interface) - Fix prepareGeneQuant() discarding the
argument strandedness if either stranded or stranded (reverse) - Data
grouping: - Samples not associated with any subject are now kept when
exporting groups to a file - Fix issues related with importing groups
based on a file with groups of splicing events and/or genes - Inform
user when groups are successfully loaded from a file and whether any
group elements are discarded in the process - Show number of genes
contained in pre-made gene lists - Discard unavailable genes when
creating group of genes based on pre-made gene lists (unless these
are automatically created at startup) - Density plot
(plotDistribution()): - Fix visual bug when plotting a group with
only one sample (if one data point is available for a group, only the
rug plot is drawn) - After hiding all plot series, rug plots of the
different groups can be distinguished based on the Y axis (different
arbitrary Y values are given to each rug plot series) - Rug plot
labels now show data values if sample names are not provided - Rug
plot labels can now be rotated (rotation is not enabled by default
given Highcharts issues that may occur at different zoom levels and
depending on proximity between different sample values) - Fix
misguiding example in function documentation - When hovering the
values in the rug plot, the colour of the tooltip is now the same
used for the rug points as expected - Survival analysis (p-value
plot): - Fix alternative splicing quantification cutoff being
selected based on the one whose difference has the highest (instead
of the lowest) p-value - Fix plot line label presenting "p < 0.05"
independently of the threshold used for significance - Gene,
transcript and protein annotation: - Show available annotation
information and query PubMed even if Ensembl is down - Avoid app
crash when searching for PubMed articles too many times - Improve
warning/error alerts: - Fix alerts crashing the visual interface -
Improve message formatting regarding HTML-containing alerts - Improve
unit tests and function documentation

[PureCN](/packages/PureCN)
------

                       Changes in version 1.18.0                        

SIGNIFICANT USER-VISIBLE CHANGES

- callAlterations: columns C and seg.mean now provide the values of the
  segment listed in seg.id. This changes the behaviour in cases where
  the gene contains breakpoints and thus multiple segments overlap
  (#112)

BUGFIXES

- Fix for bug that can result in crash when candidates were provided in
  runAbsoluteCN and test.purity, max.ploidy and/or min.ploidy were set
  to non-default values

[qcmetrics](/packages/qcmetrics)
---------

                        Changes in version 1.25                         

CHANGES IN VERSION 1.25.1

- update email

[qpgraph](/packages/qpgraph)
-------

                        Changes in version 2.22                         

USER VISIBLE CHANGES

- Added new function filterCollinearities() to aid in identifying and
  discarding collinear pairs of variables/genes.

[qsmooth](/packages/qsmooth)
-------

                 Changes in version 1.3.1 (2020-03-01)                  

- Fixed bug on conditionals longer than 1

[RaggedExperiment](/packages/RaggedExperiment)
----------------

                       Changes in version 1.12.0                        

Bug fixes and minor improvements

- Adjust to changes from SummarizedExperiment::assay (withDimnames
argument) - Reference simplifyTCGA helper function from TCGAutils in
examples - Restore original real-world qreduceAssay example and
include alternative qreduceTCGA example

[randRotation](/packages/randRotation)
------------

                Changes in version 0.99.12 (2020-04-25)                 

- Submitted to Bioconductor

[Rbowtie](/packages/Rbowtie)
-------

                       Changes in version 1.27.1                        

NEW FEATURES

- updated bowtie to version 1.2.3 (now also supports bowtie2 indices
  and compilation on intel, gcc 8 and gcc 9)

[RcisTarget](/packages/RcisTarget)
----------

                         Changes in version 1.7                         

- New function: reRank

- ImportRankings: Added arguments "indexCol" and "warnMissingColumns"

- Class rankingRcisTarget: Added methods rownames and colnames

- Support for .parquet databases

[RCy3](/packages/RCy3)
----

                        Changes in version 2.8.0                        

- New functions: - mergeNetworks - analyzeNetwork

- Better messaging for... - App-related functions, like installApp -
  When style changes are applied to "default" style

[ReactomeGSA](/packages/ReactomeGSA)
-----------

                 Changes in version 1.1.4 (2020-04-20)                  

- bioRxiv paper is out!

                 Changes in version 1.1.3 (2020-04-11)                  

- Added custom function to calculate average cluster expression for
  Seurat objects

                 Changes in version 1.1.2 (2020-03-13)                  

- Add support manually track the status of submitted analyses

- Added additional signature to add_dataset to support expression
  values as matrix

                 Changes in version 1.1.1 (2020-02-19)                  

- Compress JSON requests using gzip to improve performance for large
  datasets

[receptLoss](/packages/receptLoss)
----------

                 Changes in version 0.99.8 (2020-03-01)                 

- Updated bug in 'toMatrix' function in scripts.R, allowing it to
accept the objects with multiple classes

                 Changes in version 0.99.6 (2020-03-01)                 

- Updated the 'toMatrix' function in scripts.R, allowing it to accept
the RangedSummarizedExperiment class, and added option for specifying
rownames

                 Changes in version 0.99.5 (2020-02-14)                 

- update .Rbuildignore

                 Changes in version 0.99.4 (2020-02-12)                 

- removed .Rproj files from git repo

- Updated gitignore file to include .Rproj files

                 Changes in version 0.99.3 (2020-02-12)                 

- removed .DS_Store and .Rproj files from git repo

                 Changes in version 0.99.2 (2020-02-12)                 

- exported 'toMatrix' function to fix R CMD check error.

                 Changes in version 0.99.1 (2020-02-11)                 

- As per Martin Morgan's feedback from BioConductor review, added a
function, 'toMatrix', to accept SummarizedExperiment and other
matrix-like objects

                 Changes in version 0.99.0 (2020-02-05)                 

Changes:

- Passed R CMD Check and devtools::build() - Submitted to
Bioconductor

[reconsi](/packages/reconsi)
-------

                       Changes in version 0.99.2                        

- Reducing runtimes examples

                       Changes in version 0.99.1                        

- Semantic changes by Joris Meys

                       Changes in version 0.99.0                        

- Submission to BioConductor

                        Changes in version 0.1.2                        

- Prior weights for bootstrap samples modified

                        Changes in version 0.1.1                        

- Enforce normality assumption on resample instances and collapsed null

[recount](/packages/recount)
-------

                       Changes in version 1.13.2                        

SIGNIFICANT USER-VISIBLE CHANGES

- Documentation website is now available at
http://leekgroup.github.io/recount/. It gets updated with every
commit on the master branch (bioc-devel) using GitHub Actions and
pkgdown.

                       Changes in version 1.13.1                        

- Mention in reproduce_ranges() the link to
https://support.bioconductor.org/p/126148/#126173 which shows how to
update the gene symbols in the RSE objects in recount.

[regionReport](/packages/regionReport)
------------

                       Changes in version 1.21.11                       

BUG FIXES

- Cropping images through magick::image_trim() as done by default by
BiocStyle::html_document() can fail on Linux. This could be an
ImageMagick issue or an issue about lack of resources. The full
investigative report is at
https://stat.ethz.ch/pipermail/bioc-devel/2020-April/016650.html.
This is related to:
https://github.com/yihui/knitr/issues/1785#issuecomment-574723631
https://github.com/yihui/knitr/issues/1796
https://github.com/Bioconductor/BiocStyle/issues/65#issuecomment-552832630
https://github.com/ropensci/magick/issues/171
https://github.com/ropensci/magick/issues/194 In regionReport version
1.21.10 I have opted by using crop = NULL to disable cropping of
images by BiocStyle::html_document() and thus avoid the issues with
ImageMagick either coming from magick, from the version of
ImageMagick installed on the Linux Bioconductor build machine and
devel docker, or from resources in these two Linux environments as
described in the investigative report. About a month ago I also saw
failures on Windows on Bioc 3.10. Whether they were caused by ggbio
1.35.1 or this issue will remain a mystery. But it's likely that this
magick::image_trim() issue also affected the Bioconductor windows
builder. The related bioc-devel threads are:
https://stat.ethz.ch/pipermail/bioc-devel/2020-April/016538.html
https://stat.ethz.ch/pipermail/bioc-devel/2020-March/016365.html

                       Changes in version 1.21.10                       

BUG FIXES

- Add crop = NULL to all template Rmarkdown files. Drop magick as a
depencency since cropping is not used on the reports (as done through
BiocStyle by default). This is again related to responding to
https://stat.ethz.ch/pipermail/bioc-devel/2020-April/016645.html.

                       Changes in version 1.21.9                        

BUG FIXES

- Ran another test in preparation for
https://stat.ethz.ch/pipermail/bioc-devel/2020-April/016645.html
after the test in version 1.21.8 failed. The code will now be run in
the example of derfinderReport() instead of the vignette.

                       Changes in version 1.21.8                        

BUG FIXES

- In an attempt to fix the bug I reported at
https://stat.ethz.ch/pipermail/bioc-devel/2020-April/016645.html and
in preparation to reply to this thread, I'm experimenting with
suggesting that magick is installed, at which point BiocStyle will
try to crop. Another option might be to use crop = NULL on all Rmd
files.

                       Changes in version 1.21.5                        

BUG FIXES

- The example in renderReport() was failing because the first time
the function was called, it ran derfinder::makeGenomicState() which
in turn uses GenomicFeatures::isActiveSeq(). derfinder version 1.21.5
fixed this bug and is thus required by regionReport now.

                       Changes in version 1.21.4                        

SIGNIFICANT USER-VISIBLE CHANGES

- Add links to the example reports created when deploying the
documentation website with pkgdown::deploy_to_branch(). This
eliminates the need for the vignetttes/Makefile and the fake
vignettes. It should also avoid confusing pkgdown.

                       Changes in version 1.21.3                        

SIGNIFICANT USER-VISIBLE CHANGES

- Documentation website is now available at
http://leekgroup.github.io/derfinderPlot/. It gets updated with every
commit on the master branch (bioc-devel) using GitHub Actions and
pkgdown.

                       Changes in version 1.21.2                        

SIGNIFICANT USER-VISIBLE CHANGES

- Now use GenomeIndoDb::getChromInfoFromUCSC() instead of
data(hg19Ideogram, package = 'biovizBase') to get the hg19 chromosome
lengths.

[regutools](/packages/regutools)
---------

                       Changes in version 0.99.14                       

SIGNIFICANT USER-VISIBLE CHANGES

- Now connect_database() as a bfc parameter instead of path.

                       Changes in version 0.99.13                       

SIGNIFICANT USER-VISIBLE CHANGES

- Now connect_database() uses BiocFileCache::BiocFileCache() and
BiocFileCache::bfcrpath() to download the database from Dropbox when
it's not available from AnnotationHub::query() and caches the file so
you don't have to re-download it multiple times.

                       Changes in version 0.99.12                       

SIGNIFICANT USER-VISIBLE CHANGES

- Documentation website is now available at
http://comunidadbioinfo.github.io/regutools/. It gets updated with
every commit on the master branch (bioc-devel) using GitHub Actions
and pkgdown.

                       Changes in version 0.99.11                       

BUG FIXES

- Manual adjustment for lines > 80 chr in parameters description.

                       Changes in version 0.99.10                       

BUG FIXES

- Bump Version to run R CMD check

                       Changes in version 0.99.9                        

BUG FIXES

- Add a suggested biocView Transcription.

                       Changes in version 0.99.8                        

BUG FIXES

- Change biocViews from AnnotationData to Software

                       Changes in version 0.99.7                        

BUG FIXES

- Fix a bug in pkgdown that is currently resolved by Rscript -e
'remotes::install_github("r-lib/pkgdown#1276")'. Details at
https://github.com/r-lib/pkgdown/pull/1276 and related issues.

                       Changes in version 0.99.6                        

BUG FIXES

- Manual adjust for some lines > 80 characters

                       Changes in version 0.99.5                        

BUG FIXES

- Reindent lines and try line-wrapping for lines > 80 characters

                       Changes in version 0.99.4                        

NEW FEATURES

- Added a NEWS.md file to track changes to the package.

                       Changes in version 0.99.3                        

SIGNIFICANT USER-VISIBLE CHANGES

- DESCRIPTION file now contains an article-like summary in the field
regutools package Description.  - DESCRIPTION was updated to show
only one maintainer as required by
http://bioconductor.org/developers/package-guidelines/#description

BUG FIXES

- Fix a bug in the function get_regulatory_network() related with
Cytoscape conection.

                       Changes in version 0.99.2                        

SIGNIFICANT CHANGES

- Removed the regutools.Rproj file (you can still have it, but it's
not version controlled).  - Bump R depends to 4.0 as required by
BiocCheck. We'll see if it breaks.

                       Changes in version 0.99.1                        

BUG FIXES

- Address as many errors, warnings and notes as possible from
http://bioconductor.org/spb_reports/regutools_buildreport_20200302121040.html#malbec2_check_anchor.
For the Windows error, it's likely related to mode = "w" versus mode
= "wb" for utils::download.file() as in
https://github.com/LieberInstitute/spatialLIBD/commit/7ac24dc72cc75e42c9af8382c661517f4a64f08d.

                       Changes in version 0.99.0                        

NEW FEATURES

- Submitted to Bioconductor

[rfaRm](/packages/rfaRm)
-----

                 Changes in version 0.99.0 (2020-04-08)                 

- Submitted to Bioconductor

[rGREAT](/packages/rGREAT)
------

                       Changes in version 1.19.1                        

- add 'tsv' option in `getEnrichmentTables()`

[rhdf5](/packages/rhdf5)
-----

                       Changes in version 2.32.0                        

NEW FEATURES

- Added support for writing and reading datasets using the R 'raw'
  datatype.

- HDF5 functions H5Tset_strbuf and H5Tget_strbuf are now exposed.

CHANGES

- R 'logical' values are now stored as 8-bit integers rather than
  32-bit integers.  (Response to suggestions in
  https://github.com/grimbough/rhdf5/pull/55)

- Default compression level is 6 for all functions, rather than a mix
  of 6 or 7 in different places.

- Character vectors writen as HDF5 strings are now null padded by
  default, rather than null terminated.  (Response to
  https://github.com/grimbough/rhdf5/pull/50)

BUG FIXES

- Writing data.frames with more than one factor column no longer leads
  to memory explosion.

- Bug in h5ls() which could lead to failure in printing the dataset
  dimensions has been fixed.

- Patched bug in h5write which left an unclosed HDF5 datatype object
  when writing strings.

[rhdf5filters](/packages/rhdf5filters)
------------

                       Changes in version 0.99.0                        

- Initial submission to Bioconductor

[ribor](/packages/ribor)
-----

                       Changes in version 0.99.7                        

- All parameter names are in singular version. The parameters `region`
  and `experiment` are the parameter names for all applicable
  functions. The `experiment` parameter can still take in a list of
  multiple experiments and the `region` parameter can as well except
  for the case of getting the length distribution.

- Validator and accessor methods have been added. Use ?Ribo for more
  details on the available methods.

- The return types for the reader functions are currently either
  S4Vectors DataFrame or the R native data.frame, depending on the
  presence of the added `compact` parameter.

- Documentation has been added to be more thorough.

                       Changes in version 0.99.0                        

- Submitted to Bioconductor

[ribosomeProfilingQC](/packages/ribosomeProfilingQC)
-------------------

                       Changes in version 0.99.91                       

- change Y-axis lable from Frequence to Frequency for plot
Distance2Codon.  - update vignette.

                       Changes in version 0.99.9                        

- reset the test_estimatePsite.

                       Changes in version 0.99.8                        

- decrease the testing time.

                       Changes in version 0.99.6                        

- fix the warning missing link of plot in plotTE.

                       Changes in version 0.99.5                        

- update ggBar function to accept more colors.  - update readsEndPlot
function with shift parameter.  - update plotDistance2Codon function
with a range of plot window - add codonUsage function.

                       Changes in version 0.99.4                        

- update the description in DESCRIPTION.  - update help files.

                       Changes in version 0.99.3                        

- change the link from AtomicList to AtomicList-class and
RUVs-methods to RUVs.  - shorter lines < 80 characters.  - spead up
coverageDepth

                       Changes in version 0.99.2                        

- remove .DS_Store

                       Changes in version 0.99.1                        

- decrease the size of the package - simplify and add parameter
anchor for the estimatePsite function.  - add parameter normByLibSize
for function translationalEfficiency - add more test files.  - update
documentations.

                       Changes in version 0.99.0                        

- Submit to Bioconductor.

                        Changes in version 0.0.6                        

- Remove sapply.

                        Changes in version 0.0.1                        

- Create the package.

[RMassBank](/packages/RMassBank)
---------

                       Changes in version 2.15.3                        

- Fix an issue if saved InfoLists have missing columns

                       Changes in version 2.15.2                        

- Merge PR by Adelene Lai to add CompTox information

- Merge PR by Hendrik Treutler to 1) add more CH$NAME fields and 2)
  more Adducts and 3) Selenium

[RNAAgeCalc](/packages/RNAAgeCalc)
----------

                 Changes in version 0.99.1 (2019-08-30)                 

- Package created

[RNAdecay](/packages/RNAdecay)
--------

                        Changes in version 1.6.1                        

- Updates for Bioconductor 3.11: new function: hl_plot, fixed
  decay_plot bugs, rewrote cols function.

[RNAmodR](/packages/RNAmodR)
-------

                 Changes in version 1.1.1 (2019-11-10)                  

- added missing implementation for compare and subset functions

[ropls](/packages/ropls)
-----

                       Changes in version 1.19.16                       

INTERNAL MODIFICATION

- '.header' function: 'class(x) == ' removed from code to take into
  account the R 4.0 changes

                       Changes in version 1.19.14                       

INTERNAL MODIFICATION

- 'strF' function: use of is.array instead of is.matrix to take into
  account the R 4.0 changes

                       Changes in version 1.19.12                       

INTERNAL MODIFICATION

- 'strF' function: replacement by 'is(x, )': cont'd

                       Changes in version 1.19.10                       

INTERNAL MODIFICATION

- 'strF' function: 'class(x) ==' replaced by 'is(x, )' as required by
  R/Bioconductor

                       Changes in version 1.19.8                        

NEW FEATURE

- 'view' method: 'standardizeL' argument for the standardization of
  column values for display

                       Changes in version 1.19.6                        

BUG FIXED

- 'view' method: ExpressionSet with empty metadata

                       Changes in version 1.19.4                        

NEW FEATURE

- 'view' method: plot extended to data.frames and ExpressionSet

                       Changes in version 1.19.2                        

BUG FIXED

- 'view' method: in case the matrix has no row nor column names

[ROSeq](/packages/ROSeq)
-----

                       Changes in version 1.99.01                       

--- no bug, but removed extra files from man folder

                       Changes in version 0.99.23                       

--- updated authors names

                       Changes in version 0.99.22                       

--- no bug, but removed some part of code that was not necessary

                       Changes in version 0.99.21                       

--- no bug, but corrected package version format

                       Changes in version 0.99.9                        

--- no bug

                       Changes in version 0.99.8                        

--- no bug, but version dependancy changed 3.6 to 4.0

                       Changes in version 0.99.7                        

--- error was removed

                       Changes in version 0.99.6                        

--- got one error, now is removed

                       Changes in version 0.99.5                        

--- no bug, but removed notes

                       Changes in version 0.99.4                        

--- no bug, but version dependancy changed 3.6 to 4.0

                       Changes in version 0.99.3                        

Bugfixes

- no bug, but i corrected all previous issues of 11 Dec as shown
below - corrected i have listed all program in documentation -
changed according to <- or = - used importFrom methods of the package
instead of the whole package.  - print removed from code - used
seq_len() this time instead of : - no need for message required -
used vapply function this time -
getd2logAdbda<-getd2logAdb2<-getd2logAdadb<-getd2logAda2 issue solved
- last issue of BiocManager::install('ROSeq') solved

                       Changes in version 0.99.2                        

Bugfixes

- no bug

                       Changes in version 0.99.1                        

                        Changes in version 0.0.1                        

Bugfixes

- removing src folder

                     Changes in version 0.0.0.9000                      

setup

- added NEWS.md creation

[RProtoBufLib](/packages/RProtoBufLib)
------------

                        Changes in version 2.0.0                        

- update protobuf to 3.10

                       Changes in version 1.99.0                        

- bundle the protobuf c++ library and expose its headers and static
libraries

[rpx](/packages/rpx)
---

                        Changes in version 1.23                         

Changes in version 1.23.1

- Define modifiable rpx_fix_issue_5 <2019-11-26 Tue>

Changes in version 1.23.0

- New devel version

[rrvgo](/packages/rrvgo)
-----

                       Changes in version 0.99.1                        

- First release to Bioconductor

[Rsamtools](/packages/Rsamtools)
---------

                         Changes in version 2.4                         

BUG FIXES

- (v 2.3.2; from v 2.2.2) Correctly handle '*' ('unknown') RNAME during
  paired-end processing. See
  https://github.com/Bioconductor/Rsamtools/issues/16.

- (v 2.3.5) Fix regression introduced by v 2.3.2

NEW FEATURES

- (v 2.3.1) Don't require BAM files to have @SQ lines; allows parsing
  PacBio 'unaligned' BAM files.
  (https://github.com/Bioconductor/Rsamtools/issues/15 ; jayoung)

[rScudo](/packages/rScudo)
------

                        Changes in version 1.3.1                        

USER VISIBLE CHANGES

- Updated vignette and converted it to html

[Rsubread](/packages/Rsubread)
--------

                        Changes in version 2.2.0                        

- Improve cellCounts() on the identification of cell barcodes arising
  from ambient RNAs.

[rSWeeP](/packages/rSWeeP)
------

                Changes in version 0.99.12 (2020-02-08)                 

- 2000+ sequences multifasta error removed

                 Changes in version 0.99.7 (2020-01-21)                 

- Warning removed

                 Changes in version 0.99.6 (2020-01-21)                 

- Error removed

                 Changes in version 0.99.5 (2020-01-21)                 

- An even bigger FASTA has been used in the "Quick Start" section

                 Changes in version 0.99.4 (2020-01-21)                 

- The package no longer relies on seqinr, now operates only with
  Biostrings functions and variables

- The Title and the Description have been updated to be more precise in
  describing the package.

- Despcritional comments have been added to all codes.

- The sWeeP functions now accept AAStringset as well file names as a
  valid entry

- The sWeeP function now uses an S4 Generic and S4 methods

- The user input variable baseMatrix has now a validating step on the
  code

- The sWeeP function now uses Biostrings::readAAStringSet for FASTA
  input.

- All functions now only use message(...) to communicate with users.

- All 1:length(...) instances have been replaced with seq_len

- On all codes, the function tidy_source() from the formatR package has
  been used to standardizing code formating.

- All operations that do not affect have to the package working been
  removed.

- Some functions and variables naming has been changed aiming to follow
  the lowerCamelCase convention (the sWeeP function still as it's, due
  it is the method initials)

- aa2int() has been replaced by a vector inside the aa2num2()

- The data set datastring has been removed.

- fas2mat receives now an AAStringSet as entry

- Instead of reading the R object datastring and writing it as a FASTA
  file, a FASTA file has been included directly in inst/extdata/

- A bigger FASTA has been used in the "Quick Start" section

- All non-R package source files have been removed

                 Changes in version 0.99.0 (2019-05-15)                 

- Submitted to Bioconductor

[RTCA](/packages/RTCA)
----

                     Changes in version 2009-07-13                      

- combineRTCA(list): Additional column is renamed into Plate. The vlues
  is evaluated from list item names. When the list has no name, an
  integer index beginning from 1 is used. Special attentions to list
  partially with names is noted in the documentation.

- parseRTCA(file, dec=".",phenoData, skipWell,...): Example is added in
  the documentation how to import pre-configured phenoData. Details
  section in the documentation is re-written to describe the process of
  parsing.

- RTCA-class: Experiment ID added to RTCA class

- Makefile: add Makefile to simplify common tasks like check and
  install

- plotGridEffect: takes 'column' instead of 'col' as mode parameter,
  and renders the mode as the title of the legend. Documentation
  updated.

- plotRTCA: is removed from the package and is substituted by the plot
  function.

[RTCGAToolbox](/packages/RTCGAToolbox)
------------

                       Changes in version 2.18.0                        

New features

- Warning for Windows users added when file paths are too long -
getGISTICPeaks now requires a FirehoseGISTIC data object obtained
from getFirehoseData

Bug fixes and minor improvements

- Consolidate GISTIC data download methods in getFirehoseData and
getGISTICPeaks - Increase robustness of internal helper functions
that work with strands - 'TCGA' sample column identification is less
strict

[RUVcorr](/packages/RUVcorr)
-------

                       Changes in version 1.19.0                        

- Changed class calling.

- New vignette.

[rWikiPathways](/packages/rWikiPathways)
-------------

                        Changes in version 1.8.0                        

- Replaced RJSONIO with rjson

- Replaced all but one sapply() with vapply()

- Deprecated getColoredPathway

- New features - Upgraded findPathwaysByXXX, listPathways and
  getPathwayInfo functions to return dataframes - Upgraded
  findPathwaysByLiterature to present literature fields

- Doc fixes - Updated tests and vignettes using dataframes

[S4Vectors](/packages/S4Vectors)
---------

                       Changes in version 0.26.0                        

NEW FEATURES

- Add TransposedDataFrame objects.

- Add make_zero_col_DFrame() for constructing a zero-column DFrame
  object. Intended for developers to use in other packages and
  typically not needed by the end user.

- Add internal generic makeNakedCharacterMatrixForDisplay() to
  facilitate implementation of show() methods. Also add
  cbind_mcols_for_display() helper for use within
  makeNakedCharacterMatrixForDisplay() methods.

SIGNIFICANT USER-VISIBLE CHANGES

- Rename parallelSlotNames() internal generic -> vertical_slot_names().
  Also add new horizontal_slot_names() internal generic (no methods
  yet).

BUG FIXES

- Fix bug causing segfault in C function 'select_hits()' when 'nodup'
  is TRUE.

[SAIGEgds](/packages/SAIGEgds)
--------

                        Changes in version 1.2.0                        

- `seqAssocGLMM_SPA()` can save to a RDS file

                        Changes in version 1.0.2                        

- fix an issue when there is no covariate in the formula (e.g., y ~ 1)

                        Changes in version 1.0.1                        

- add a new option for the threshold of missing variant to
  `seqAssocGLMM_SPA()`

- change the output column 'AC.alt' to 'mac' in `seqAssocGLMM_SPA()`

- retry model fitting at least 10 times if matrix is singular or not
  positive definite, or large variance is observed

[sarks](/packages/sarks)
-----

                 Changes in version 0.99.0 (2020-02-14)                 

- Submitted to Bioconductor

[scater](/packages/scater)
------

                       Changes in version 1.16.0                        

- Added coldata_merge= argument to aggregateAcrossCells() for custom
  column metadata aggregation.  Also include averaged reduced dimension
  results for each group.

- Added label_format= argument to plotReducedDim() for greater control
  over the axis labels.

- Added geometric= argument to librarySizeFactors() to compute size
  factors with a geometric mean.

- Added the runMultiUMAP() function to perform a combined UMAP on
  multiple feature sets.

- Added the medianSizeFactors() function to perform a DESeq2-like size
  factor calculation.

- Modified isOutlier() so that min_diff= now acts on the log2 scale
  when log=TRUE.  Added share_medians=, share_mads= and share_missing=
  options for sharing information across batches.

- Exposed various developer utilities for downstream packages.

- Modified sumCountsAcrossCells() to always return a
  SummarizedExperiment object, regardless of the class of ids. Return
  the number of cells used for summation in the colData.

- Restored capability to create a data.frame from a SCE with
  makePerCellDF() and makePerFeatureDF().

- Added ggcells() and ggfeatures() for convenient creation of ggplot
  objects from a SCE.

[scClassify](/packages/scClassify)
----------

                       Changes in version 0.99.3                        

- Improve coding format - sapply changed to vapply - Add examples -
Improve vignettes

[schex](/packages/schex)
-----

                        Changes in version 1.1.4                        

- Option to remove NAs for plot_hexbin_meta*.

                        Changes in version 1.1.3                        

- Updated the vignettes and example code to incorporate changes.  -
Added bivariate plotting feature.

                        Changes in version 1.1.2                        

- Changes to vignettes to comply with changes in Seurat

                        Changes in version 1.1.1                        

- Deprecated plot_hexbin_gene.  - Updated plot_hexbin_feature to use
all features, including genes.  - Introduced plot_hexbin_fc for fold
change visualization.  - Introduced plot_hexbin_meta_plus and
plot_hexbin_feature_plus to plot cluster outlines.  - Introduced
plot_hexbin_meta_shiny, plot_hexbin_feature_shiny and
plot_hexbin_density_shiny for interactive visualization.  - Changed
internal structure.  - Updated vignettes.  - Included two new
vignettes.  - New README.

[SCOPE](/packages/SCOPE)
-----

                       Changes in version 0.99.13                       

- Integrate mouse genome and update plot_iCN()

                       Changes in version 0.99.12                       

- add algorithms with shared clonal memberships

                       Changes in version 0.99.11                       

- get_gc() dontrun

                       Changes in version 0.99.8                        

- minor edit

                       Changes in version 0.99.7                        

- second-round revision

                       Changes in version 0.99.6                        

- first-round revision

                       Changes in version 0.99.5                        

- Enable user-define bin length and offer SoSplot

                       Changes in version 0.99.4                        

- Add .bed file for hg38

                       Changes in version 0.99.3                        

- Test if SCOPE is linked to WGSmapp

                       Changes in version 0.99.2                        

- Bump new version to trigger a build

                       Changes in version 0.99.1                        

- Add session_info() to the vignette

                       Changes in version 0.99.0                        

- SCOPE getting ready for submission to Bioc

[scPCA](/packages/scPCA)
-----

                 Changes in version 1.1.14 (2020-04-26)                 

- Fixing broken link in an internal function documentation page.

                 Changes in version 1.1.12 (2020-04-21)                 

- Updated citations - Fixed typos in documentation

                 Changes in version 1.1.11 (2020-02-02)                 

- Added more SPCA algorithm options - SPCA via variable projection -
Randomized SPCA via variable projection - New vignette section
comparing performance of SPCA algorithms - Improvements to code
coverage

                 Changes in version 1.1.5 (2020-01-18)                  

- Fixed issue with matrix normalization - Misc. big fixes -
Improvements to code coverage

                 Changes in version 1.1.2 (2020-01-08)                  

- Added hierarchical clustering options for clustering based
cross-validation

[scPipe](/packages/scPipe)
------

                 Changes in version 1.9.4 (2020-04-06)                  

- Changed hashmap package calls to hash package as hashmap is no longer
  on CRAN.

                 Changes in version 1.9.1 (2019-12-23)                  

- Changed instances of deprecated `plotQC()` to `plotHighestExprs()`

[scran](/packages/scran)
-----

                       Changes in version 1.16.0                        

- Added the quickSubCluster() function for convenient subclustering.

- Added the bootstrapCluster() function to convenient bootstrapping of
  cluster stability.

- Added the coassignProb() function to compute coassignment
  probabilities of alternative groupings.

- combineMarkers() and findMarkers() report a summary effect size for
  each cluster.

- Added the multiMarkerStats() function to combine statistics from
  multiple findMarkers() calls.

- Added the clusterPurity() function to evaluate cluster purity as a
  quality measure.

- Added the pseudoBulkDGE() function to easily and safely perform
  pseudo-bulk DE analyses. Also added the decideTestsPerLabel() and
  summarizeTestsPerLabel() utilities.

- Added the clusterSNNGraph() and clusterKNNGraph() wrapper functions
  for easier graph-based clustering. Provided a k-means pre-clustering
  option to handle large datasets.

[scry](/packages/scry)
----

                         Changes in version 1.0                         

- Initial release of the scry package for analysis of
high-dimensional data consisting of small counts (such as single-cell
RNA-seq).  Features included in this release:

- Feature selection with deviance - Dimension reduction with GLM-PCA
- Approximate GLM-PCA using Pearson and deviance residuals from a
null model.  - Optional adjustment for categorical batch effects for
all methods.  - Support for SingleCellExperiment and
SummarizedExperiment objects - Support for matrix and sparse Matrix
objects.

[scTensor](/packages/scTensor)
--------

                        Changes in version 1.4.0                        

- verbose parameter was added in cellCellRank() and cellCellDecomp()

- set.seed(1234) was added in example and vignette

- .cellCellDecomp.CabelloAguilar() and .cellCellDecomp.Halpern() were
  added

- The num.iter 300　of NTD was changed to 30

- schex package was imported for visualing all the two dimensional gene
  plots

- The rule of cellCellRanks() based on the singular value of SVD was
  changed to a reconstruction error based rule, using NMF with
  matricised tensor in each mode

- The CCI-tensor is normalized in each frontral slice so that the total
  value is 1 (.frontal.normalization)

                        Changes in version 1.2.1                        

- Some bugs were fixed

[scTHI](/packages/scTHI)
-----

                 Changes in version 0.99.6 (2020-03-02)                 

- Submitted to Bioconductor

[SDAMS](/packages/SDAMS)
-----

                        Changes in version 1.7.1                        

- add reference.

[selectKSigs](/packages/selectKSigs)
-----------

                       Changes in version 0.99.5                        

- Remove git track src-i386 and src-x64.

                       Changes in version 0.99.4                        

- Address all error messages.

                       Changes in version 0.99.3                        

- Address all comments from the Bioconductor reviewer.

                       Changes in version 0.99.2                        

- Remove selectKSigs.Rproj.

                       Changes in version 0.99.1                        

- Submitted to Bioconductor.

[SeqArray](/packages/SeqArray)
--------

                       Changes in version 1.28.0                        

NEW FEATURES

- new function `seqUnitSlidingWindows()`

- new function `seqUnitApply()`

- new variable "$variant_index", "$sample_index" in `SeqGetData()`,
  `seqBlockApply()` and `seqUnitApply()` to get the indices of selected
  variants

- new arguments '.padNA' and '.envir' in `seqGetData()`

- new functions `seqSetFilterAnnotID()` and `seqGDS2BED()`

- multicore function in `seqBED2GDS(, parallel=)`

- new package-wide option `options(seqarray.nofork=TRUE)` to disable
  forking

- new option 'minor' in `seqAlleleFreq()` and `seqAlleleCount()`

- new option 'verbose' in `seqMissing()`, `seqAlleleFreq()` and
  `seqAlleleCount()`; '.progress' is deprecated, but still can be used
  for compatiblity

- `seqAlleleFreq()` and `seqAlleleCount()` work on
  'annotation/format/DS', if 'genotype/data' is not available

UTILITIES

- `seqAddValue()` adds vectors, matrices and data frame to
  "annotation/info"

- `seqBED2GDS()` allows a single file name without the extended file
  names (.bed, .fam, .bim)

- allele flip in `seqBED2GDS()` to allow major allele to be reference

- rewrite `seqGetData()` for faster loading

- significantly improve `seqBlockApply()` on 'annotation/info/VARIABLE'
  (https://github.com/zhengxwen/SeqArray/issues/59)

- add a S3 method `print.SeqVCFHeaderClass()` for `seqVCF_Header()`

- new option '.tolist' in `seqGetData()`, `seqBlockApply()` and
  `seqUnitApply()`

- String "." in a VCF file are converted to a blank string (missing
  value) in `seqVCF2GDS()`

- add a class name 'SeqVarDataList' to the returned 'list(length,
  data)' from `seqGetData()`

- new option `seqMissing(, per.variant=NA)`

- add `comment.char=""` to `seqBED2GDS()`

                       Changes in version 1.26.2                        

NEW FEATURES

- multiple variable names are allowed in `seqGetData(, var.name=)`

BUG FIXES

- fix `seqGetData(, "genotype", .useraw=NA)`
  (https://github.com/zhengxwen/SeqArray/issues/58)

                       Changes in version 1.26.1                        

BUG FIXES

- fails to correctly select duplicate indices in `seqSetFilter(f,
  variant.sel=)`

[SeqVarTools](/packages/SeqVarTools)
-----------

                       Changes in version 1.25.1                        

- Bug fix in minorAlleleCount when all samples have the same sex.

[sevenbridges](/packages/sevenbridges)
------------

                       Changes in version 1.17.1                        

Bug Fixes

- Fix build issues in R 4.0.0, in which the the default value for
stringsAsFactors is changed to FALSE. Updated the affected
data.frame() calls which rely on the default automatic vector
expansion behavior with explicit stringsAsFactors = TRUE.

[SharedObject](/packages/SharedObject)
------------

                        Changes in version 1.1.6                        

- Add memory check on Linix system

                        Changes in version 1.1.1                        

- Refactoring code

- Add supports for package developer at C++, R levels

[signatureSearch](/packages/signatureSearch)
---------------

                 Changes in version 1.0.6 (2020-04-19)                  

- Supported searching refdb parallelly by using multiple cores on a
single machine

                 Changes in version 1.0.5 (2020-04-10)                  

- Fix bug: fixed null issue and throw warning messages when up or
down gene sets share zero identifiers with refdb for gess_lincs
method.

                 Changes in version 1.0.4 (2020-04-02)                  

- Added instructions for GESS batch queries in vignette - Added runWF
function to run entire GESS/FEA workflow

                 Changes in version 1.0.3 (2020-02-07)                  

- Supported converting feaResult object to enrichResult object in the
clusterProfiler package so that the plotting functionalities in the
latter package such as dotplots and gene-concept networks could be
applied to the FEA enrichment results - Supported searching against
subset of refdb (subsetted specific columns (treatments) in the refdb
(e.g. lincs)) for GESS methods - Updated comp_fea_res function to
reduce number of characters in description - Added functions to draw
different types of query GESs from refdb - Added deprof2subexpr
function to get a subset of gene expression values from a
differential expression profile

                 Changes in version 1.0.2 (2020-01-21)                  

- Fix bug: the enrichment results from DSEA methods and some of TSEA
methods were added an aditional 'ont' column where the GO itmes were
subsetted to the selected ontology

                 Changes in version 1.0.1 (2019-11-10)                  

- Support windows by not depending gCMAP package - Deal with HDF5
files with functions in HDF5Array package

[signeR](/packages/signeR)
------

                       Changes in version 1.13.1                        

- fix issue with pseudo-variants on vcf

[SimFFPE](/packages/SimFFPE)
-------

                 Changes in version 0.99.0 (2019-12-12)                 

- Submitted to Bioconductor

[SingleCellExperiment](/packages/SingleCellExperiment)
--------------------

                       Changes in version 1.10.0                        

- Removed deprecated modes for getting and setting reducedDims.

[SingleR](/packages/SingleR)
-------

                        Changes in version 1.2.0                        

- Added support for consolidating labels from multiple references via
  combineResults().

- Added mappings to standardized Cell Ontology terms in all *Data()
  functions.

- Changed the name of the labels input of plotScoreDistribution() to
  labels.use for consistency across functions.

- Fixed a label from adipocytes to astrocytes in BlueprintEncodeData().

- Removed umlauts from labels (e.g., naive) in
  NovershternHematopoieticData() to avoid problems with Windows.

- Perform PCA before clustering in aggregateReference() for speed and
  memory efficiency.

- Modified genes="all" behavior in trainSingleR() to report DE-based
  markers for fine-tuning only.

[singscore](/packages/singscore)
---------

                        Changes in version 1.8.0                        

- implemented a novel approach to scoring that uses measurements from
panel-based tests such as RT-qPCR and nanostring - implemented a
function that returns a set of stable genes we have identified in
carcinomas and in blood - fixed bugs in the multiScore function that
produced errors when scoring a single sample - fixed bugs in the
generateNull function that produced errors when assessing a single
geneset - fixed bugs in the vignette

[sitePath](/packages/sitePath)
--------

                        Changes in version 1.2.2                        

- Bug fix: infinity loop might occur when using 'fixationSites' or
  'multiFixatoinSites' caused by internal function. There's no solution
  for version 1.2.x, so choices are given to possibly avoid it.

- Bug fix: malfunctional 'setSiteNumbering'.

                        Changes in version 1.2.1                        

- Bug fix: plot warning caused by 'tab' character

- Bug fix: bad design causing 'multiFixationSites' extremely slow

- Bug fix: 'fixationSites' gives replicated tips when combined

[snapcount](/packages/snapcount)
---------

                 Changes in version 0.99.0 (2019-09-17)                 

- Submitted to Bioconductor

[SNPRelate](/packages/SNPRelate)
---------

                       Changes in version 1.22.0                        

- 'allow.fork=TRUE' is the default in `snpgdsOpen()` since v1.21.2

- add `print.snpgdsPCAClass`, `print.snpgdsEigMixClass`,
  `print.snpgdsPCASNPLoadingClass`, `print,
  snpgdsEigMixSNPLoadingClass`, `print.snpgdsIBDClass`,
  `print.snpgdsDissClass`, `print.snpgdsIBSClass`

- fix a Win32 compiler issue with gcc-4.9

- update `snpgdsBED2GDS()` with comment.char=""

                       Changes in version 1.20.1                        

- Bug fix: eigenvalues and TraceXTX are correctly calculated in
  `snpgdsPCA(, algorithm="randomized")`

[sojourner](/packages/sojourner)
---------

                        Changes in version 1.1.4                        

- Modified readme file for new user.

                        Changes in version 1.1.3                        

- bug fixes for Bioc3.11

                        Changes in version 1.1.2                        

- SDMTools dependency removed

                        Changes in version 1.1.1                        

- rowr dependency removed.

                        Changes in version 1.1.0                        

- BioC3.10 release.

[sparseMatrixStats](/packages/sparseMatrixStats)
-----------------

                 Changes in version 0.0.99 (2020-04-03)                 

- Submitted to Bioconductor

[SpatialCPie](/packages/SpatialCPie)
-----------

                         Changes in version 1.3                         

- (1.3.4) Fixed type error when resolutions only have one cluster

- (1.3.3) Fixed vignette field name

- (1.3.2) Renamed cluster tree to cluster graph

[splatter](/packages/splatter)
--------

                 Changes in version 1.12.0 (2020-04-20)                 

- Add checks for cycles in the Splat path.from parameter.

- Use alternative algorithm if fitting dropout fails in splatEstimate.

- Adjust paths example in vignette.

- Replace defunct functions in vignettes.

- Minor fixes for compatibility with updates to other packages.

[spqn](/packages/spqn)
----

                        Changes in version 0.99                         

- Submission to Bioconductor. This package implements the SpQN method
  for normalizing correlation matrices, as described in Wang, Hicks and
  Hansen (2020) bioRxiv.

[SSPA](/packages/SSPA)
----

                       Changes in version 2.27.1                        

- convert Rnw vignette to Rmd

- fixed _R_CHECK_LENGTH_1_LOGIC2_

[statTarget](/packages/statTarget)
----------

                         Changes in version 2.0                         

NEW FEATURES

- New GUI o Mouse Hover for help information o .log file

- New Signal correction o Combat for QC-free Signal correction o
  QC-RFSC methods for metabolomics and proteomics data

- New feature slection o Random Forest and the Permutation based
  variable importance measures o new MDSplot for Random Forest o
  P-value based importance plot

- New data preprocessing o PQN/SUM/none normalization o center/none
  Scaling method

                       Changes in version 1.17.3                        

- shiftCor-coCV was added as the cutoff value (0-100) of CV for
  controlling the number of features.

- To fixed bugs for skiping the roc analysis once the number of samples
  in any groups was less than 5.

                       Changes in version 1.17.2                        

- Check and remove the missing values in pvalues for volcanoplot
  functions

[struct](/packages/struct)
------

                       Changes in version 0.99.10                       

- update vignettes

- update documentation

- add 'signature' input to struct_obj_method

                       Changes in version 0.99.9                        

- subsetting of iterators with a single model now converts to a model
  sequence with 1 step

                       Changes in version 0.99.8                        

- documentation changes for Bioconductor checks

                       Changes in version 0.99.7                        

- fixed broken is_valid for .outputs

- minor documentation updates

                       Changes in version 0.99.6                        

- .params and.outputs inherited correctly

- fix some documentation issues

                       Changes in version 0.99.5                        

- added as.DatasetExperiment for converting SummarizedExperiment
  objects

                 Changes in version 0.99.4 (2020-02-04)                 

- Set initial value for entity on creation if not provided

- Rename enum 'list' to 'allowed'

                 Changes in version 0.99.3 (2020-02-03)                 

- Improved use of class constructors

                       Changes in version 0.99.2                        

- incremental changes in response to bioconductor feedback

                       Changes in version 0.99.1                        

- incremental changes in response to bioconductor feedback

                       Changes in version 0.99.0                        

- Bioconductor submission

                        Changes in version 0.4.1                        

- Final working release before bioconductor submission

                 Changes in version 0.2.0 (2019-07-25)                  

- Introduction of seq_out slot for more flexible workflows (currently
  method objects only)

[Structstrings](/packages/Structstrings)
-------------

                 Changes in version 1.3.1 (2019-12-01)                  

- Display Dot Bracket sequences in color

[structToolbox](/packages/structToolbox)
-------------

                       Changes in version 0.99.10                       

- add Metabolmics BiocView

- fix missing sections at end of vignette

                       Changes in version 0.99.9                        

- add proteomics vignette

- add mean_of_medians

- merge all vignettes

- fix bug in kfold_xval_grid

                       Changes in version 0.99.8                        

- added stratified splitting (stratified_split)

- added Gastric Cancer vignette

- add AUC metric

- add predicted and ROC plots for PLSDA

                       Changes in version 0.99.7                        

- minor documentation updates

                       Changes in version 0.99.6                        

- fix issues using class() due to changes in R 4.0.0

                       Changes in version 0.99.5                        

- fix dimnames mismatch for knn-impute

- add additional vignettes

                       Changes in version 0.99.4                        

- Add SVM, nroot transform, constant sum norm

- Add funtionality to autoscaling

- Fix some issues with permutation testing of model sequences

- Add option to impute by samples to knn

- Add custom feature labels option to pca loadings

                       Changes in version 0.99.3                        

- Fix dimnames mismatch errors

- Updated struct depency to 0.99.5

                       Changes in version 0.99.2                        

- renamed some charts containing .'s to use underscores

- renamed sbcms dataset to MTBLS79

                       Changes in version 0.99.1                        

- minor updates to pass Bioconductor checks

                       Changes in version 0.99.0                        

- use new struct v0.99.4 class constructor functions

                        Changes in version 0.8.4                        

- Final working version befor bioconductor updates

                 Changes in version 0.4.0 (2019-07-25)                  

- Minor bug fixes

- Added tsne method

- Added sbcms dataset

- Added methods related to signal/batch correction

[SubCellBarCode](/packages/SubCellBarCode)
--------------

                 Changes in version 1.2.6 (2020-04-08)                  

- Changed the NEWS.md file to track changes to the package.

- Fixed bug.

[SummarizedExperiment](/packages/SummarizedExperiment)
--------------------

                       Changes in version 1.18.0                        

NEW FEATURES

- SummarizedExperiment objects with assays of > 4 dimensions are now
  fully supported.

SIGNIFICANT USER-VISIBLE CHANGES

- By default the assays() and assay() setters now reject inconsistent
  dimnames. By default the dimnames on the supplied assay(s) must be
  identical to the dimnames on the SummarizedExperiment object. The
  user now must use 'withDimnames=FALSE' if it's not the case or they
  get an error. This is for symmetry with the behavior of the assays()
  and assay() getters (see issue #35). Unfortunately this change is
  likely to break existing code but at least the fix is easy.

- dimnames() now returns NULL instead of list(NULL, NULL) on a
  SummarizedExperiment object with no dimnames. This is consistent with
  matrix objects.

- Swap positions of arguments '...' and 'withDimnames' in assays()
  setter and getter. So now it's: assays(x, withDimnames=TRUE, ...)
  assays(x, withDimnames=TRUE, ...) <- value

- Add 'withDimnames' argument to the assay() getter/setter. So now
  it's: assay(x, i, withDimnames=TRUE, ...)  assay(x, i,
  withDimnames=TRUE, ...) <- value Note that before this change, the
  user was able to explicitly set 'withDimnames' when calling assay()
  but since this was not a formal argument it was forwarded to assays()
  via the ellipsis. Having it as a formal argument makes it easier to
  discover and allows tab completion.

[SWATH2stats](/packages/SWATH2stats)
-----------

                       Changes in version 1.17.5                        

update

- .Rmd vignettes: remove tinytex option

                       Changes in version 1.17.4                        

update

- convert_protein_ids: Also count empty strings as NA and count unique
  non-mapped identifiers.

                       Changes in version 1.17.3                        

update

- plot_correlation_between_samples.R, plot_variation.R,
  plot_variation_vs_total.R: improve plots and do theme_bw

                       Changes in version 1.17.2                        

NEW FEATURES

- filter_on_min_peptides.R: add ability to deal with PeptideSequence
  column

- filter_on_max_peptides.R: add ability to deal with PeptideSequence
  column

                       Changes in version 1.17.1                        

UPDATE

- convert_protein_ids: count unique non-mapped IDs.

                       Changes in version 1.17.0                        

NEW FEATURES

- SWATH2stats in BioC 3.11 development release

[SynExtend](/packages/SynExtend)
---------

                       Changes in version 0.99.30                       

- Vignette and help files edited for clarity

                       Changes in version 0.99.1                        

- SynExtend submitted to Bioconductor - Added function gffToDataFrame
- Added function NucleotideOverlap - Added function PairSummaries

[systemPipeR](/packages/systemPipeR)
-----------

- With the upgrades provided in this release, systemPipeR has become a much more
  generic data analysis workflow environment that is no longer limited to
  analyzing just NGS data. Now it can be efficiently used for data analysis
  tasks in many omics areas, including genomics, proteomics, metabolomics and
  drug discovery.
- A workflow control class (SYSargsList) has been added allowing users to manage
  multiple-step workflows from a single container. This way one can select and
  execute multiple workflow steps with standard R subsetting syntax,
  e.g. runWF&#91;1:3&#93;.
- Various improvements have been added to systemPipeR’s new command-line
  interface including the recently introduced SYSargs2 class that supports the
  Common Workflow Language (CWL).
- Utilities have been added to visualize workflow designs and topologies with different graphical layouts.
- Improvements have been added to monitor the run status of workflows as well as
  better tracking of warning and error messages. This includes the generation of
  both scientific and technical status reports.

[TAPseq](/packages/TAPseq)
------

                 Changes in version 0.99.0 (2020-02-21)                 

- Initial submission to Bioconductor

[TargetSearch](/packages/TargetSearch)
------------

                       Changes in version 1.44.0                        

NEW FEATURES

- The class tsRim allows subsetting by the `[` operator. This makes
  possible to choose specific marker for plotting, eg, in
  `checkRimLim`.

- New function `updateRI` to update/correct/force the time of the RI
  markers. Just like the old function `fixRI`, but it also corrects CDF
  files.

BUG FIXES

- Make sure matrix dimensions are not dropped in NCDF extraction.

- Fix check warnings due to plot() being moved to `base`.

- Clean-up NAMESPACE

[TBSignatureProfiler](/packages/TBSignatureProfiler)
-------------------

                     Changes in version 0.0.0.9000                      

- Added a NEWS.md file to track changes to the package.

[TCGAutils](/packages/TCGAutils)
---------

                        Changes in version 1.8.0                        

New features

- README.md now includes a cheat sheet for reference - mergeColData
and oncoPrintTCGA sections updated/included in the vignette

Minor changes and bug fixes

- translateBuild more robust to consistent inputs - translateBuild
returns vector output instead of single string as before -
makeSummarizedExperimentFromGISTIC now has a more open interface with
... input to RTCGAToolbox::getGISTICPeaks - oncoPrintTCGA now uses
seqlevels from input throughout

[TOAST](/packages/TOAST)
-----

                        Changes in version 1.1.1                        

- Add options for partial reference-free deconvolution.

[tofsims](/packages/tofsims)
-------

                        Changes in version 099.1                        

SIGNIFICANT USER-VISIBLE CHANGES

- changed function behvaiour in the whole package from call-by-ref to
  call-by value. Adjusted accordingly all examples and the vignette.

INTERNALS

- depends now on ProtGenerics from which it uses 'mz'

- exchanged various print() with message()

[topconfects](/packages/topconfects)
-----------

                        Changes in version 1.3.1                        

- Fix y axis labels on confects_plot

[topdownr](/packages/topdownr)
--------

                         Changes in version 1.9                         

- New version for Bioc 3.11 (devel)

Changes in version 1.9.4

- Fix unit tests that check for "matrix" class. (class(m) now returns
c("matrix", "array") in r-devel) &#91;2019-12-17&#93;.

Changes in version 1.9.3

- Adapt analysis vignette to changed condition argument &#91;2019-11-22&#93;.

Changes in version 1.9.2

- Set conditions="ScanDescription" as new default for
readTopDownFiles. The creation of FilterString IDs in the method
files was deprecated since over a year. conditions="FilterString" is
still possible for backward-compatibility &#91;2019-11-22&#93;.  - Fix error
message handling in .validFilename and .translateThermoIdToScanId
&#91;2019-11-22&#93;.

Changes in version 1.9.1

- Remove defaultMs1Settings and defaultMs2Settings &#91;2019-11-18&#93;.  -
Fix .rbind for lists with mixed data.frame and DataFrame
&#91;2019-11-18&#93;.

[TPP](/packages/TPP)
---

                       Changes in version 3.15.1                        

- remove dependency to the sme package because it is not available any
  more on CRAN.

[TPP2D](/packages/TPP2D)
-----

                 Changes in version 1.3.0 (2020-04-17)                  

- Carry-over detection is not used anymore

- Bootstrapping is done based on resampling of alternative model
  residuals onto null model fits

- `getFDR` replaces `computeFdr` (now deprecated)

- option to moderate F is now available in function `getFDR`

[trackViewer](/packages/trackViewer)
-----------

                       Changes in version 1.23.5                        

- replace class function by is function.

                       Changes in version 1.23.4                        

- fix a issue of "EXPR must be a length 1 vector".

                       Changes in version 1.23.3                        

- fix a issue of vignette when visit ebi proteins api.

                       Changes in version 1.23.2                        

- plot multiple genes in one track.

                       Changes in version 1.23.1                        

- add interactionData track.

[tradeSeq](/packages/tradeSeq)
--------

                 Changes in version 1.1.07 (2020-02-27)                 

- added testing against fold change cut-off for all DE tests

- default cut-off for deciding the rank of the variance-covariance
  matrix changed in patternTest and earlyDETest. This can impact the
  results of these tests. To return to original behaviour, set
  `eigenThresh=1e-8` argument.

[transcriptR](/packages/transcriptR)
-----------

                       Changes in version 1.15.6                        

- Updated NEWS file to track changes in the package.

                       Changes in version 1.15.5                        

- Modified vignette to replace Rclass and Rfunction sweave syntax with
  markdown syntax.

                       Changes in version 1.15.4                        

- Modified vignette to add images via knitr::include_graphics.

                       Changes in version 1.15.3                        

- Changed vignette output format to html_document.

                       Changes in version 1.15.2                        

- Reformatted author section in the vignette yaml header.

                       Changes in version 1.15.1                        

- Fixed bug in the predictStrand function caused by the new behaviour
  of the data.frame introduced in R 4.0. Have to manually set
  stringsAsFactors=TRUE.

[TreeAndLeaf](/packages/TreeAndLeaf)
-----------

                       Changes in version 0.99.0                        

- Submitted to Bioconductor &#91;2019-12-16&#93;.

[treeio](/packages/treeio)
------

                       Changes in version 1.11.3                        

- change according to dplyr (v=1.0.0) (2020-04-09, Thu) - remove
mutate_, rename_, select_ and group_by_ - remove data_frame for it
was deprecated in tibble (v=3.0.0)

                       Changes in version 1.11.2                        

- update citation (2020-02-18, Tue) - phyloxml parser read.phyloxml
(2019-12-05, Thu)

                       Changes in version 1.11.1                        

- support jplace version 1 (2019-11-25, Mon) -
https://github.com/YuLab-SMU/treeio/pull/25 - offspring return
integer(0) instead of throw error if input .node is a tip
(2019-11-21, Thu)

[tRNAdbImport](/packages/tRNAdbImport)
------------

                  Changes in version 1.5 (2020-02-05)                   

- import of RNA sequences now disregards the insertion character "_"

- added tRNAdb_reference and tRNAdb_pmid columns to output

[tRNAscanImport](/packages/tRNAscanImport)
--------------

                 Changes in version 1.7.3 (2020-01-21)                  

- fixed get.tRNAprecursor() function if length of flanking sequences
  was not equal

- fixed istRNAscanGRanges() not returing a boolean value on fail

- removed remove.LowerCase argument from import.tRNAscanAsGRanges()
  since it was redundant

[tscR](/packages/tscR)
----

                 Changes in version 0.99.2 (2020-05-06)                 

- Reviewer's Suggestions Fixed

[TSRchitect](/packages/TSRchitect)
----------

                       Changes in version 1.13.5                        

- Corrected line in loadTSSobj() that caused a build error in the Bioc
  devel branch (3.11).

                       Changes in version 1.13.4                        

- Added feature to writeTSS() that generates two separate bedgraph
  files (plus and minus) instead of one.

                       Changes in version 1.13.3                        

- Updated documentation for loadTSSobj() for the sake of clarity.

                       Changes in version 1.13.2                        

- Corrected minor bugs in non-user-level functions detTSR() and
  detTSS()

- Various updates to Show-methods.R

                       Changes in version 1.13.1                        

- Version bump due to release of Bioconductor v. 3.10.

[TVTB](/packages/TVTB)
----

                 Changes in version 1.13.2 (2020-04-11)                 

Bug fix

- tSVE vignette output format set to HTML.

                 Changes in version 1.13.1 (2020-03-26)                 

Bug fix

- parallelSlotNames() was renamed vertical_slot_names() in S4Vectors
  0.25.14.

[tximeta](/packages/tximeta)
-------

                        Changes in version 1.6.0                        

- Added PLOS Computational Biology citation! :-)

- Added function `splitSE` to split one assay of a SummarizedExperiment
  into multiple assays, each containing features of a given type.

- Added a wrapper function makeDGEList() to simplify making a DGEList
  for use with edgeR. See vignette for example. + tximeta will now make
  use of EnsDb created and distributed on AnnotationHub, unless
  useHub=FALSE. Also, a new function retrieveDb() can be called on a
  SummarizedExperiment to retrieve the underlying TxDb or EnsDb. +
  tximeta can now use `customMetaInfo` argument to locate a custom
  metadata information file such as `meta_info.json`, which should
  contain a tage, `index_seq_hash`, with the SHA-256 hash value of the
  reference transcripts.

- Added `markDuplicateTxps` argument to add `hasDuplicate` and
  `duplicates` columns to rowData of SummarizedExperiment. One note is
  that, for efficiency, this argument and `cleanDuplicateTxps` will now
  share a duplicates CharacterList that is stored in the BiocFileCache,
  with the name `dups-...`. Therefore, if you have previously used
  `cleanDuplicateTxps`, you may need to bfcremove() any `dups-...`
  entries. Summarization to gene level will keep track of `numDupSets`
  per gene which informs about the number of transcripts sets
  (equivalence classes by transcript sequence content). + If during the
  indexing step, user didn't use --gencode for a Gencode transcriptome
  file, tximeta will deal with this internally now by stripping all
  characters after the vertical bar `|`, in order to match long
  transcript names in the `quant.sf` files to the correct transcript
  names in the GTF.

                       Changes in version 1.5.28                        

- Added function `splitSE` to split one assay of a SummarizedExperiment
  into multiple assays, each containing features of a given type.

                       Changes in version 1.5.19                        

- Added `markDuplicateTxps` argument to add `hasDuplicate` and
  `duplicates` columns to rowData of SummarizedExperiment. One note is
  that, for efficiency, this argument and `cleanDuplicateTxps` will now
  share a duplicates CharacterList that is stored in the BiocFileCache,
  with the name `dups-...`. Therefore, if you have previously used
  `cleanDuplicateTxps`, you may need to bfcremove() any `dups-...`
  entries. Summarization to gene level will keep track of `numDupSets`
  per gene which informs about the number of transcripts sets
  (equivalence classes by transcript sequence content).

                       Changes in version 1.5.16                        

- Added a wrapper function makeDGEList() to simplify making a DGEList
  for use with edgeR. See vignette for example.

                       Changes in version 1.5.11                        

- tximeta can now use `customMetaInfo` argument to locate a custom
  metadata information file such as `meta_info.json`, which should
  contain a tage, `index_seq_hash`, with the SHA-256 hash value of the
  reference transcripts.

                        Changes in version 1.5.8                        

- tximeta will now make use of EnsDb created and distributed on
  AnnotationHub, unless useHub=FALSE. Also, a new function retrieveDb()
  can be called on a SummarizedExperiment to retrieve the underlying
  TxDb or EnsDb.

                        Changes in version 1.5.3                        

- If during the indexing step, user didn't use --gencode for a Gencode
  transcriptome file, tximeta will deal with this internally now by
  stripping all characters after the vertical bar `|`, in order to
  match long transcript names in the `quant.sf` files to the correct
  transcript names in the GTF.

[tximport](/packages/tximport)
--------

                       Changes in version 1.16.0                        

- Moved alevin arguments into a new 'list' argument, alevinArgs. As of
  this version the possible values for alevinArgs are: filterBarcodes,
  tierImport, forceSlow (all logical).

- Added alevinArgs argument `tierImport`, which will import the "tier"
  information from alevin on the quantification assessment.

- Add an alevinArgs argument `filterBarcodes`, which will only import
  cells with barcodes in the `whitelist.txt` file.

- Fixed bug where the bootstrap matrices from alevin were not aligned
  by cell with the counts matrix. This affected the variance and the
  infReps list. The fix will also be propogated to tximport v1.14.1
  (Oct 2019 release).

                       Changes in version 1.15.12                       

- Moved alevin arguments into a new 'list' argument, alevinArgs. As of
  this version the possible values for alevinArgs are: filterBarcodes,
  tierImport, forceSlow (all logical).

- Added alevinArgs argument `tierImport`, which will import the "tier"
  information from alevin on the quantification assessment.

                       Changes in version 1.15.10                       

- Add an alevinArgs argument `filterBarcodes`, which will only import
  cells with barcodes in the `whitelist.txt` file.

                       Changes in version 1.15.9                        

- Fixed bug where the bootstrap matrices from alevin were not aligned
  by cell with the counts matrix. This affected the variance and the
  infReps list. The fix will also be propogated to tximport v1.14.1
  (Oct 2019 release).

[UniProt.ws](/packages/UniProt.ws)
----------

                       Changes in version 2.27.0                        

BUG FIX

o (2.27.1) Fix bug when selecting column GENEID. The mapping mapped
both GENEID and ENTREZ_GENE to P_ENTREZGENE. When returning columsn
used match to identify but would only pick up first match which was
ENTREZ_GENE entry.

[universalmotif](/packages/universalmotif)
--------------

                        Changes in version 1.6.0                        

NEW FEATURES

- log_string_pval(): small utility function to obtain the log of
  string-formatted p-values (such as those often carried in
  MEME-formatted motifs which are smaller than R's double.xmin limit).
  Likely only a temporary solution.

- view_motifs(..., return.raw) option: instead of returning a plot type
  object, return the aligned motif matrices.

- view_motifs(..., dedup.names) option: allows plotting of motifs with
  duplicated names by appending a unique string.

- merge_motifs(..., new.name) option: assign a name to the new merged
  motif instead of collapsing the names of the merged motifs together.

- round_motif() utility: round down very low letter-position scores to
  zero.

MINOR CHANGES

- Removed most previously deprecated function arguments.

- Make sure view_motifs(..., use.type = "ICM") properly sets ylim.

- create_motif(): single motif positions can now be created.

- create_motif(), character input: nsites slot is left empty is input
  is a single string. It is still filled if the input consists of
  multiple strings.

- merge_motifs(): ALLR/ALLR_LL/KL/IS methods no longer add pseudocounts
  to the motifs. Instead, pseudocounts are added to temporary internal
  copies which are used for comparison and alignment. The original
  un-modified matrices are then combined.

- read_meme(): now supports DNA-LIKE, RNA-LIKE, and AA-LIKE alphabets,
  though these will be treated as regular DNA, RNA, and AA alphabets,
  respectively. Contribution from Spencer Nystrom
  (https://github.com/bjmt/universalmotif/pull/7).

- Some cleanup to documentation and vignettes.

- General code cleanup.

BUG FIXES

- write_meme() now includes altname slot if filled. Contribution from
  Spencer Nystrom (https://github.com/bjmt/universalmotif/pull/5).

- write_meme() checks for and removes any spaces/equal signs in motif
  names/altnames.

- view_motifs(..., use.type = "ICM"): check for zero IC motifs, as
  these cannot be plotted by ggseqlogo.

[variancePartition](/packages/variancePartition)
-----------------

                       Changes in version 1.17.10                       

- fix issue returning residuals from limma

- resolve issue where dream gives error: r&#91;cbind(1L:p, 1L:p)&#93; <- 1 :
  subscript out of bounds - only occured when no fixed effects were
  used

                       Changes in version 1.17.9                        

- Fix issues with compatability with R/4.0.0

                       Changes in version 1.17.8                        

- Better error message when response contains missing data

                       Changes in version 1.17.7                        

- Better error message when variable in formula does not exist

                       Changes in version 1.17.6                        

- comply with new Bioconductor check: _R_CHECK_LENGTH_1_LOGIC2_

                       Changes in version 1.17.5                        

- if contrasts for dream() is data.frame, convert to matrix

                       Changes in version 1.17.4                        

- Don't print warnings for residuals() when only one argument passed.

- fix bug with residuals evaluated with only fixed effects

                       Changes in version 1.17.3                        

- Allow sparseMatrix for gene expression.  Now saves memory by avoiding
  conversion to matrix.  Processing sparseMatrix will be slower, but
  memory usage will be low.

- dream(..., computeResiduals=TRUE) now computes residuals and allows
  use of residuals() function

                       Changes in version 1.17.2                        

- fix error in voomWithDreamWeights() when design matrix is null

                       Changes in version 1.17.1                        

- topTable(...,sort.by=) now is correct when and F-test is used

- fixed issue in classifyTestsF.MArrayLM2, now is much faster

[vasp](/packages/vasp)
----

                Changes in version 0.99.23 (2020-03-25)                 

- Released to the 'Devel' version of Bioconductor (accepted and
  builded)

                 Changes in version 0.99.0 (2020-03-03)                 

- Submitted to Bioconductor

[ViSEAGO](/packages/ViSEAGO)
-------

                         Changes in version 1.1                         

- add gene_symbol field for Custom2GO

- patch for Custom gene_symbols on merge_enrich_terms

- set esummary to 400 terms (instead of 500) by querying on
  merge_enrich_terms

- add gene frequency hover text on GOterms_heatmap

- add custom heatmap colors on GOterms_heatmap

- showIC print patch on GOterms_heatmap

- GO count update

- Uniprot2GO connection method update

- prevent bioconductor errors when check examples

- annotate upgrade

- merge_enrich_terms upgrade for custom

[waddR](/packages/waddR)
-----

                 Changes in version 1.1.4 (2020-03-16)                  

- Fixes a build fail due to erronous merge in the previous commit
  267f84b

                 Changes in version 1.1.3 (2020-03-12)                  

- Updates to Documentation

                 Changes in version 1.1.2 (2020-03-11)                  

- Fixes critical bug: o Removed the usage of the unavailable package
  'eva', which caused builds to fail

                 Changes in version 1.1.1 (2019-10-28)                  

- P-value reproducibility feature: o Now using nextRNGstream to
  generate independent seeds for each gene o Setting the seed (with
  set.seed()) only in .testWass o Removed use of seeds and the seed
  argument everywhere in WassersteinTest.R

[weitrix](/packages/weitrix)
-------

                       Changes in version 0.99.2                        

- Automatic package builder wants package to depend on R version 4.0.

                       Changes in version 0.99.1                        

- weitrix now uses the BiocParallel default parallel processing,
rather than the DelayedArray default. (DelayedArray does not use
parallel processing by default as of version 0.14.0.)

                       Changes in version 0.99.0                        

- Bioconductor submission.

                        Changes in version 0.1.0                        

- Initial version.

[xcms](/packages/xcms)
----

                        Changes in version 3.9.4                        

- Fix issue in centWave which skips peak detection depending on minimum
  peakwidth (issue #445): add parameter `extendLengthMSW` in
  `CentWaveParam`. Thanks to William Kumler for contributing the fix.

- Tentatively reduce memory requirements in `fillChromPeaks`.

- Fix issue #467 for fillPeaks() of an xcmsSet converted from an
  XCMSnSet

                        Changes in version 3.9.3                        

- Move multtest from Imports to Suggests to avoid duplicated method
  definition for plot (issue #459).

- Add support for peak filling from MS level > 1 to fillChromPeaks.

- featureValues gains parameter msLevel to extract feature values for
  features of all, or from a specific MS level.

- refineChromPeaks supports different MS levels.

- Added support to perform correspondence analysis on MS level > 1 and
  add the respective results to already present feature definitions.

- hasChromPeaks and hasFeatures gain parameter msLevel to check for
  presence of chromatographic peaks or features from a specific MS
  level.

                        Changes in version 3.9.2                        

- Fix featureChromatograms and chromatograms on a XCMSnExp object with
  features: features can be duplicated across rows (EICs).

- findChromPeaks: add parameter `add` to allow several rounds of peak
  detections on the same object.

- Small performance enhancement in fillChromPeaks.

- Better support for MS > 1 data in fillChromPeaks: skip MS level 2
  spectra for filling in.

- Add refineChromPeaks for XChromatogram and XChromatograms objects.

- Add groupOverlaps function to group arbitrary ranges.

- Add quantify,XCMSnExp object to quantify an XCMSnExp into a
  SummarizedExperiment.

- Fine-tune MergeNeighboringPeaks peak refinement method: the average
  of the 3 data points between candidate peaks is used to evaluate
  whether the peaks should be merged making the approach more robust
  against outliers. In addition, an ion chromatogram for candidate
  peaks is extracted with an m/z range expanded depending on the
  expandMz and ppm setting ensuring that low intensity data points
  between candidate peaks are not missed out (because their m/z might
  be slightly shifted on ToF instruments). The mzmin and mzmax of the
  merged peak represents also the minimum and maximum m/z of all data
  points in that extracted ion chromatogram.

                        Changes in version 3.9.1                        

- Fix problem of not shown/plotted peak positions in
  plotChromPeakSpectra for experiments in which peaks were not detected
  in the first sample(s).

- Add method *from_to* to missing value imputation method
  `imputeRowMinRand`.

- Show warning in findChromPeaks if empty spectra are detected.

- Add refineChromPeaks method and CleanPeaksParam class to allow
  removal of chromatographic peaks exceeding a user-definable maximal
  peak width.

- Add MergeNeighboringPeaksParam for refineChromPeaks to allow merging
  of chromatographic peaks close in m/z and retention time with a
  signal between them higher than a certain threshold (issue #414).

- Fix misspelled parameter `mzd` in LC-MS/MS vignette.

[YAPSA](/packages/YAPSA)
-----

                       Changes in version 1.13.3                        

This is a stable version after several important changes

- 95% Confidence intervals for exposures to mutational signatures can
  be computed

- Analysis of the PCAWG mutational signatures is supported, both for
  PCAWG SNV and PCAWG Indel signatures

The vignettes have been substantially changed

- new vignettes have been written for the new features confidence
  intervals and Indel signatures

- new vignettes were added for features already present in the package
  before, especially the optimal signature-specific cutoffs

- the topic of stratifiec analyses of mutational signatures was taken
  out of the main vignette and now is described in a vignette on its
  own.



NEWS from new and existing Data Experiment Packages
===================================



[curatedAdipoArray](/packages/curatedAdipoArray)
-----------------

                 Changes in version 0.99.0 (2019-10-10)                 

- Submitted to Bioconductor

[depmap](/packages/depmap)
------

Changes in Version 1.1.2

- 20Q1 data added

Changes in Version 1.1.1

- 19Q4 data added
- Updated loading functions (accessor functions)
- 19Q4 versions of rnai, RPPA and drug_sensitivity were not released for this
  quarter, and therefore the 19Q3 releases for these datasets are the most
  current versions.


[derfinderData](/packages/derfinderData)
-------------

                        Changes in version 2.5.3                        

SIGNIFICANT USER-VISIBLE CHANGES

- Now using BiocStyle for the vignette.

                        Changes in version 2.5.1                        

SIGNIFICANT USER-VISIBLE CHANGES

- Documentation website is now available at
  http://leekgroup.github.io/derfinderData/. It gets updated with every
  commit on the master branch (bioc-devel) using GitHub Actions and
  pkgdown.

[dorothea](/packages/dorothea)
--------

                 Changes in version 0.99.9 (2020-04-22)                 

- Integration of Travis CI

- Integrated unit test coverage with codecov

- Build github page via pkgdown

                 Changes in version 0.99.1 (2020-04-06)                 

- If the input to the viper wrapper is a Bioconductor class, this
  Bioconductor class is retured with added TF activities at appropiate
  slots

- Expanded the viper wrapper to the Bioconductor class
  SingleCellExperiment

                 Changes in version 0.99.0 (2020-04-03)                 

- Initial submission to Bioconductor

[dsQTL](/packages/dsQTL)
-----

                        Changes in version 2.17                         

USER VISIBLE CHANGES

- ch2locs (retrievable via dsQTL::getSNPlocs) has been changed at about
  1850 locations where rs numbers had been associated with hg19
  addresses; the dsQTL regions are hg18 as are all the chr2... SNP
  addresses.  Previously the discoverable rs numbers used in the
  Chicago distribution from
  http://eqtl.uchicago.edu/dsQTL_data/GENOTYPES/ had be mapped via
  SNPlocs...20111119, but now they come directly from the Chicago text
  file.

[FlowSorted.Blood.EPIC](/packages/FlowSorted.Blood.EPIC)
---------------------

                 Changes in version 1.5.2 (2019-11-05)                  

- Made the following significant changes o Added the projectCellType_CP
  function for projecting the cell counts on beta values

[HDCytoData](/packages/HDCytoData)
----------

                        Changes in version 1.6.3                        

- Add vignette "Examples and use cases"

- Add vignette "Contribution guidelines"

- Update documentation

[HighlyReplicatedRNASeq](/packages/HighlyReplicatedRNASeq)
----------------------

                       Changes in version 0.99.0                        

- Initial release of the package

[pRolocdata](/packages/pRolocdata)
----------

                       Changes in version 1.25.3                        

- update email

                       Changes in version 1.25.2                        

- add data from Shin et al. (2020)

- add data from Baryluk et al. (2020)

                       Changes in version 1.25.1                        

- add dynamic Itzhak data

[PtH2O2lipids](/packages/PtH2O2lipids)
------------

                     Changes in version 2016-04-21                      

- Initial release for Bioconductor

[RegParallel](/packages/RegParallel)
-----------

                        Changes in version 1.6.0                        

- vignette updates

- removed glm.nb

[RNAmodR.Data](/packages/RNAmodR.Data)
------------

                  Changes in version 1.1 (2020-02-13)                   

- Fixed snoRNAdb information regarding positions of RNU12 modifications
  and RNU2 modification type at position 11 and 61

[scRNAseq](/packages/scRNAseq)
--------

                        Changes in version 2.2.0                        

- Removed deprecated inbuilt datasets.

[scTHI.data](/packages/scTHI.data)
----------

                 Changes in version 0.99.6 (2020-03-02)                 

- Submitted to Bioconductor

[signatureSearchData](/packages/signatureSearchData)
-------------------

                 Changes in version 1.1.0 (2019-10-23)                  

- Initial version

[WGSmapp](/packages/WGSmapp)
-------

                       Changes in version 0.99                        

- Add mm10 blacklist regions
- Use lazy-loading of data
- Update R version Dependency to 3.6
- Fix NAs in seqlengths(mapp_hg38)
- Incorporate hg38gaps.txt
- Add seg region file for hg38
- Bump version number to trigger a new build
- Remove WGSmapp.Rproj and add GenomicRanges dependency
- WGSmapp getting ready for submission to Bioc



NEWS from new and existing Workflows
===================================

[fluentGenomics](/packages/fluentGenomics)
--------------

                       Changes in version 0.99.0                        

- Added a NEWS.md file to track changes to the package.

[recountWorkflow](/packages/recountWorkflow)
---------------

                       Changes in version 1.11.2                        

SIGNIFICANT USER-VISIBLE CHANGES

- Documentation website is now available at
http://LieberInstitute.github.io/recountWorkflow/. It gets updated
with every commit on the master branch (bioc-devel) using GitHub
Actions and pkgdown.  - Added a NEWS.md file to track changes to the
package.

BUG FIXES

- Fix the spelling of RIN to rin now that all SRA columns are in
lower case.  - Fix the DESCRIPTION file (had a missing comma).

                       Changes in version 1.11.1                        

NEW FEATURES

- Piggyback on GenomicState now that this package produces all the
pieces needed for making the ER plots.  - Potentially everything
works on Windows now that rtracklayer::import.bw() seems to work
there. But I haven't tested it.

BUG FIXES

- Update SRARunTable.txt reading code since they changed the format
on delivered by SRA. Adjust variables accordingly.



Deprecated and Defunct Packages
===============================

Eighteen software packages were removed from this release (after being deprecated
in Bioc 3.10): birte, brainImageR, charm, CNPBayes, condcomp, dSimer, exomePeak,
flipflop, GenomeGraphs, HTSanalyzeR, mlm4omics, Pbase, plateCore, Rchemcpp,
rHVDM, RnaSeqSampleSize, SEPA, SNPchip

Fifty Six software are deprecated in this release and will be removed in Bioc 3.12:
affypdnn, AnalysisPageServer, anamiR, BayesPeak, bgafun, biosvd, birta, CALIB,
CAMTHC, cellGrowth, chroGPS, cobindR, CTDquerier, CVE, DChIPRep, DEDS,
DupChecker, FEM, gCMAP, gCMAPWeb, geecc, Genominator, IdMappingAnalysis,
IdMappingRetrieval, IPPD, kimod, LMGene, lol, LVSmiRNA, M3D, manta,
MaxContrastProjection, MCRestimate, MergeMaid, mitoODE, MoPS, motifRG, MTseeker,
nem, PAPi, pcaGoPromoter, pint, plw, PowerExplorer, proteoQC, QUALIFIER, readat,
RefNet, RIPSeeker, SANTA, scfind, splicegear, sRAP, triform, Vega, waveTiling

DESeq has been deprecated in favor of DESeq2. Due to its high number of reverse
dependencies it will remain deprecated in 3.12 to be removed in 3.13.

Three experimental data packages were removed in this release (after being
deprecated in BioC 3.10): allenpvc, charmData, facopy.annot

Two experimental data packages are deprecated in this release and will be
removed in Bioc 3.12: MTseekerData, RIPSeekerData

Two annotation packages were removed this release: 
MafDb.ESP6500SI.V2.SSA137.hs37d5, MafDb.ESP6500SI.V2.SSA137.GRCh38

Nine annotation packages are deprecated in this release and will be removed in
Bioc 3.12: hom.At.inp.db, hom.Ce.inp.db, hom.Dm.inp.db, hom.Dr.inp.db,
hom.Hs.inp.db, hom.Mm.inp.db, hom.Rn.inp.db, hom.Sc.inp.db, KEGG.db. 

One workflow package is removed in this release (after being deprecated in BioC
3.10): BgeeCall (This package has moved to Software)

No workflow packages were deprecated in this release.