May 3, 2019

Bioconductors:

We are pleased to announce Bioconductor 3.9, consisting of 1741
software packages, 371 experiment data packages, 948 annotation
packages, and 27 workflows.

There are 105 new software packages, 13 new data experiment packages, 
4 new workflows, and many updates and improvements
to existing packages; Bioconductor 3.9 is compatible with R 3.6.0,
and is supported on Linux, 32- and 64-bit Windows, and Mac OS X.  This
release will include an updated Bioconductor [Amazon Machine Image][1]
and [Docker containers][2].

Visit [https://bioconductor.org][3]
for details and downloads.

[1]: https://bioconductor.org/help/bioconductor-cloud-ami/
[2]: https://bioconductor.org/help/docker/
[3]: https://bioconductor.org

Contents
--------

* [Getting Started with Bioconductor 3.9](#getting-started-with-bioconductor-35)
* [New Software Packages](#new-software-packages)
* [New Data Experiment Packages](#new-data-experiment-packages)
* [New Workflows](#new-workflows)
* [NEWS from new and existing software packages](#news-from-new-and-existing-software-packages)
* [NEWS from new and existing data experiment packages](#news-from-new-and-existing-data-experiment-packages)
* [NEWS from new and existing workflows](#news-from-new-and-existing-workflows)
* [Deprecated and Defunct Packages](#deprecated-and-defunct-packages)

Getting Started with Bioconductor 3.9
======================================

To update to or install Bioconductor 3.9:

1. Install R >=3.6.0.  Bioconductor 3.9 has been designed expressly for
   this version of R.

2. Follow the instructions at
   [http://bioconductor.org/install/](http://bioconductor.org/install/).

New Software Packages
=====================

There are 105 new software packages in this release of Bioconductor.

- [ADAM](https://bioconductor.org/packages/ADAM) ADAM is a GSEA R
  package created to group a set of genes from comparative samples
  (control versus experiment) belonging to different species
  according to their respective functions (Gene Ontology and KEGG
  pathways as default) and show their significance by calculating
  p-values referring togene diversity and activity. Each group of
  genes is called GFAG (Group of Functionally Associated Genes).

- [ADAMgui](https://bioconductor.org/packages/ADAMgui) ADAMgui is a
  Graphical User Interface for the ADAM package. The ADAMgui package
  provides 2 shiny-based applications that allows the user to study
  the output of the ADAM package files through different plots. It's
  possible, for example, to choose a specific GFAG and observe the
  gene expression behavior with the plots created with the
  GFAGtargetUi function. Features such as differential expression and
  foldchange can be easily seen with aid of the plots made with
  GFAGpathUi function.

- [adductomicsR](https://bioconductor.org/packages/adductomicsR)
  Processes MS2 data to identify potentially adducted peptides from
  spectra that has been corrected for mass drift and retention time
  drift and quantifies MS1 level mass spectral peaks.

- [alevinQC](https://bioconductor.org/packages/alevinQC) Generate QC
  reports summarizing the output from an alevin run. Reports can be
  generated as html or pdf files, or as shiny applications.

- [AMARETTO](https://bioconductor.org/packages/AMARETTO) Integrating
  an increasing number of available multi-omics cancer data remains
  one of the main challenges to improve our understanding of cancer.
  One of the main challenges is using multi-omics data for
  identifying novel cancer driver genes. We have developed an
  algorithm, called AMARETTO, that integrates copy number, DNA
  methylation and gene expression data to identify a set of driver
  genes by analyzing cancer samples and connects them to clusters of
  co-expressed genes, which we define as modules. We applied AMARETTO
  in a pancancer setting to identify cancer driver genes and their
  modules on multiple cancer sites. AMARETTO captures modules
  enriched in angiogenesis, cell cycle and EMT, and modules that
  accurately predict survival and molecular subtypes. This allows
  AMARETTO to identify novel cancer driver genes directing canonical
  cancer pathways.

- [animalcules](https://bioconductor.org/packages/animalcules)
  animalcules is an R package for utilizing up-to-date data
  analytics, visualization methods, and machine learning models to
  provide users an easy-to-use interactive microbiome analysis
  framework. It can be used as a standalone software package or users
  can explore their data with the accompanying interactive R Shiny
  application. Traditional microbiome analysis such as alpha/beta
  diversity and differential abundance analysis are enhanced, while
  new methods like biomarker identification are introduced by
  animalcules. Powerful interactive and dynamic figures generated by
  animalcules enable users to understand their data better and
  discover new insights.

- [atSNP](https://bioconductor.org/packages/atSNP) atSNP performs
  affinity tests of motif matches with the SNP or the reference
  genomes and SNP-led changes in motif matches.

- [BANDITS](https://bioconductor.org/packages/BANDITS) BANDITS is a
  Bayesian hierarchical model for detecting differential splicing of
  genes and transcripts, via differential transcript usage (DTU),
  between two or more conditions. The method uses a Bayesian
  hierarchical framework, which allows for sample specific
  proportions in a Dirichlet-Multinomial model, and samples the
  allocation of fragments to the transcripts. Parameters are inferred
  via Markov chain Monte Carlo (MCMC) techniques and a DTU test is
  performed via a multivariate Wald test on the posterior densities
  for the average relative abundance of transcripts.

- [batchelor](https://bioconductor.org/packages/batchelor) Implements
  a variety of methods for batch correction of single-cell (RNA
  sequencing) data. This includes methods based on detecting mutually
  nearest neighbors as well as a simple sparsity-preserving
  translation of the population means. Functions are also provided
  for global rescaling to remove differences in depth between
  batches, and to perform a principal components analysis that is
  robust to differences in the numbers of cells across different
  batches.

- [bigPint](https://bioconductor.org/packages/bigPint) Methods for
  visualizing large multivariate datasets using static and
  interactive scatterplot matrices, parallel coordinate plots,
  volcano plots, and litre plots. Includes examples for visualizing
  RNA-sequencing datasets and differentially expressed genes.

- [BiocSingular](https://bioconductor.org/packages/BiocSingular)
  Implements exact and approximate methods for singular value
  decomposition and principal components analysis, in a framework
  that allows them to be easily switched within Bioconductor packages
  or workflows. Where possible, parallelization is achieved using the
  BiocParallel framework.

- [BioMM](https://bioconductor.org/packages/BioMM) The identification
  of reproducible biological patterns from high-dimensional omics
  data is a key factor in understanding the biology of complex
  disease or traits. Incorporating prior biological knowledge into
  machine learning is an important step in advancing such research.
  We have proposed a biologically informed multi-stage machine
  learing framework termed BioMM specifically for phenotype
  prediction based on omics-scale data where we can evaluate
  different machine learning models with various prior biological
  meta information.

- [celda](https://bioconductor.org/packages/celda) celda leverages
  Bayesian hierarchical modeling to cluster genes, cells, or both
  simultaneously from single cell sequencing data.

- [CellBench](https://bioconductor.org/packages/CellBench) This
  package contains infrastructure for benchmarking analysis methods
  and access to single cell mixture benchmarking data. It provides a
  framework for organising analysis methods and testing combinations
  of methods in a pipeline without explicitly laying out each
  combination. It also provides utilities for sampling and filtering
  SingleCellExperiment objects, constructing lists of functions with
  varying parameters, and multithreaded evaluation of analysis
  methods.

- [CellMixS](https://bioconductor.org/packages/CellMixS) Evaluate
  Cellspecific Mixing Scores (CMS) for different batches/groups in
  scRNA-seq data.

- [CHETAH](https://bioconductor.org/packages/CHETAH) CHETAH is a
  accurate, selective and fast scRNA-seq classifier. Classification
  is guided by a reference dataset, preferentially also a scRNA-seq
  dataset. By hierarchical clustering of the reference data, CHETAH
  creates a classification tree that enables a step-wise,
  top-to-bottom classification. Using a novel stopping rule, CHETAH
  classifies the input cells to the cell types of the references and
  to "intermediate types": more general classifications that ended in
  an intermediate node of the tree.

- [CluMSID](https://bioconductor.org/packages/CluMSID) CluMSID is a
  tool that aids the identification of features in untargeted
  LC-MS/MS analysis by the use of MS2 spectra similarity and
  unsupervised statistical methods. It offers functions for a
  complete and customisable workflow from raw data to visualisations
  and is interfaceable with the xmcs family of preprocessing
  packages.

- [CNVRanger](https://bioconductor.org/packages/CNVRanger) The
  CNVRanger package implements a comprehensive tool suite for CNV
  analysis. This includes functionality for summarizing individual
  CNV calls across a population, assessing overlap with functional
  genomic regions, and association analysis with gene expression and
  quantitative phenotypes.

- [cola](https://bioconductor.org/packages/cola) Subgroup
  classification is a basic task in genomic data analysis, especially
  for gene expression data and methylation data. It can predict novel
  subgroups when there is nothing known about the data or it can test
  consistency between predicted subgroups with known annotations. The
  cola package provides a general framework for subgroup
  classification by consensus clustering. It has following features: 1. It modularizes the consensus clustering processes that various
  methods can be easily integrated. 2. It provides rich
  visualizations for interpreting the results. 3. It allows running
  multiple methods at the same time and provides functionalities to
  compare results in a straightforward way. 4. It provides a new
  method to extract features which are more efficient to separate
  subgroups. 5. It allows doing partitioning in a hierarchical way to
  detect subgroups with relatively smaller difference. 6. It
  generates detailed reports for the complete analysis.

- [CopyNumberPlots](https://bioconductor.org/packages/CopyNumberPlots)
  CopyNumberPlots have a set of functions extending karyoploteRs
  functionality to create beautiful, customizable and flexible plots
  of copy-number related data.

- [CoRegFlux](https://bioconductor.org/packages/CoRegFlux) CoRegFlux
  aims at providing tools to integrate reverse engineered gene
  regulatory networks and gene-expression into metabolic models to
  improve prediction of phenotypes, both for metabolic engineering,
  through transcription factor or gene (TF) knock-out or
  overexpression in various conditions as well as to improve our
  understanding of the interactions and cell inner-working.

- [cytofast](https://bioconductor.org/packages/cytofast)
  Multi-parametric flow and mass cytometry allows exceptional
  high-resolution exploration of the cellular composition of the
  immune system. Together with tools like FlowSOM and Cytosplore it
  is possible to identify novel cell types. By introducing cytofast
  we hope to offer a workflow for visualization and quantification of
  cell clusters for an efficient discovery of cell populations
  associated with diseases or other clinical outcomes.

- [dcanr](https://bioconductor.org/packages/dcanr) Methods and an
  evaluation framework for the inference of differential
  co-expression/association networks.

- [deco](https://bioconductor.org/packages/deco) This package
  discovers differential features in hetero- and homogeneous omic
  data by a two-step method including subsampling LIMMA and NSCA.
  DECO reveals feature associations to hidden subclasses not
  exclusively related to higher deregulation levels.

- [decompTumor2Sig](https://bioconductor.org/packages/decompTumor2Sig)
  Uses quadratic programming for signature refitting, i.e., to
  decompose the mutation catalog from an individual tumor sample into
  a set of given mutational signatures (either Alexandrov-model
  signatures or Shiraishi-model signatures), computing weights that
  reflect the contributions of the signatures to the mutation load of
  the tumor.

- [DelayedDataFrame](https://bioconductor.org/packages/DelayedDataFrame)
  Based on the standard DataFrame metaphor, we are trying to
  implement the feature of delayed operation on the DelayedDataFrame,
  with a slot of lazyIndex, which saves the mapping indexes for each
  column of DelayedDataFrame. Methods like show, validity check, [/[[
  subsetting, rbind/cbind are implemented for DelayedDataFrame to be
  operated around lazyIndex. The listData slot stays untouched until
  a realization call e.g., DataFrame constructor OR as.list() is
  invoked.

- [DeMixT](https://bioconductor.org/packages/DeMixT) DeMixT is a
  software package that performs deconvolution on transcriptome data
  from a mixture of two or three components.

- [DepecheR](https://bioconductor.org/packages/DepecheR) The purpose
  of this package is to identify traits in a dataset that can
  separate groups. This is done on two levels. First, clustering is
  performed, using an implementation of sparse K-means. Secondly, the
  generated clusters are used to predict outcomes of groups of
  individuals based on their distribution of observations in the
  different clusters. As certain clusters with separating information
  will be identified, and these clusters are defined by a sparse
  number of variables, this method can reduce the complexity of data,
  to only emphasize the data that actually matters.

- [DiscoRhythm](https://bioconductor.org/packages/DiscoRhythm) Set of
  functions for estimation of cyclical characteristics, such as
  period, phase, amplitude, and statistical significance in large
  temporal datasets. Supporting functions are available for quality
  control, dimensionality reduction, spectral analysis, and analysis
  of experimental replicates. Contains a R Shiny web interface to
  execute all workflow steps.

- [divergence](https://bioconductor.org/packages/divergence) This
  package provides functionality for performing divergence analysis
  as presented in Dinalankara et al, "Digitizing omics profiles by
  divergence from a baseline", PANS 2018. This allows the user to
  simplify high dimensional omics data into a binary or ternary
  format which encapsulates how the data is divergent from a
  specified baseline group with the same univariate or multivariate
  features.

- [DNABarcodeCompatibility](https://bioconductor.org/packages/DNABarcodeCompatibility)
  The package allows one to obtain optimized combinations of DNA
  barcodes to be used for multiplex sequencing. In each barcode
  combination, barcodes are pooled with respect to Illumina chemistry
  constraints. Combinations can be filtered to keep those that are
  robust against substitution and insertion/deletion errors thereby
  facilitating the demultiplexing step. In addition, the package
  provides an optimizer function to further favor the selection of
  barcode combinations with least redundancy of DNA barcodes.

- [doseR](https://bioconductor.org/packages/doseR) doseR package is a
  next generation sequencing package for sex chromosome dosage
  compensation which can be applied broadly to detect shifts in gene
  expression among an arbitrary number of pre-defined groups of loci.
  doseR is a differential gene expression package for count data,
  that detects directional shifts in expression for multiple,
  specific subsets of genes, broad utility in systems biology
  research. doseR has been prepared to manage the nature of the data
  and the desired set of inferences. doseR uses S4 classes to store
  count data from sequencing experiment. It contains functions to
  normalize and filter count data, as well as to plot and calculate
  statistics of count data. It contains a framework for linear
  modeling of count data. The package has been tested using real and
  simulated data.

- [enrichTF](https://bioconductor.org/packages/enrichTF) As
  transcription factors (TFs) play a crucial role in regulating the
  transcription process through binding on the genome alone or in a
  combinatorial manner, TF enrichment analysis is an efficient and
  important procedure to locate the candidate functional TFs from a
  set of experimentally defined regulatory regions. While it is
  commonly accepted that structurally related TFs may have similar
  binding preference to sequences (i.e. motifs) and one TF may have
  multiple motifs, TF enrichment analysis is much more challenging
  than motif enrichment analysis. Here we present a R package for TF
  enrichment analysis which combine motif enrichment with the PECA
  model.

- [epihet](https://bioconductor.org/packages/epihet) epihet is an
  R-package that calculates the epigenetic heterogeneity between
  cancer cells and/or normal cells. The functions establish a
  pipeline that take in bisulfite sequencing data from multiple
  samples and use the data to track similarities and differences in
  epipolymorphism,proportion of discordantly methylated sequencing
  reads (PDR),and Shannon entropy values at epialleles that are
  shared between the samples.epihet can be used to perform analysis
  on the data by creating pheatmaps, box plots, PCA plots, and t-SNE
  plots. MA plots can also be created by calculating the differential
  heterogeneity of the samples. And we construct co-epihet network
  and perform network analysis.

- [evaluomeR](https://bioconductor.org/packages/evaluomeR) Evaluating
  the reliability of your own metrics and the measurements done on
  your own datasets by analysing the stability and goodness of the
  classifications of such metrics.

- [fishpond](https://bioconductor.org/packages/fishpond) Fishpond
  contains methods for differential transcript and gene expression
  analysis of RNA-seq data using inferential replicates for
  uncertainty of abundance quantification, as generated by Gibbs
  sampling or bootstrap sampling.

- [GAPGOM](https://bioconductor.org/packages/GAPGOM) Collection of
  various measures and tools for lncRNA annotation prediction put
  inside a redistributable R package. The package contains two main
  algorithms; lncRNA2GOA and TopoICSim. lncRNA2GOA tries to annotate
  novel genes (in this specific case lncRNAs) by using various
  correlation/geometric scoring methods on correlated expression
  data. After correlating/scoring, the results are annotated and
  enriched. TopoICSim is a topologically based method, that compares
  gene similarity based on the topology of the GO DAG by information
  content (IC) between GO terms.

- [GladiaTOX](https://bioconductor.org/packages/GladiaTOX) GladiaTOX
  R package is an open-source, flexible solution to high-content
  screening data processing and reporting in biomedical research.
  GladiaTOX takes advantage of the tcpl core functionalities and
  provides a number of extensions: it provides a web-service solution
  to fetch raw data; it computes severity scores and exports ToxPi
  formatted files; furthermore it contains a suite of functionalities
  to generate pdf reports for quality control and data processing.

- [GNET2](https://bioconductor.org/packages/GNET2) Cluster genes to
  functional groups with E-M process. Iteratively perform TF
  assigning and Gene assigning, until the assignment of genes did not
  change, or max number of iterations is reached.

- [gpuMagic](https://bioconductor.org/packages/gpuMagic) The package
  aims to help users write openCL code with little or no effort. It
  is able to compile an user-defined R function and run it on a
  device such as a CPU or a GPU. The user can also write and run
  their openCL code directly by calling .kernel function.

- [graper](https://bioconductor.org/packages/graper) This package
  enables regression and classification on high-dimensional data with
  different relative strengths of penalization for different feature
  groups, such as different assays or omic types. The optimal
  relative strengths are chosen adaptively. Optimisation is performed
  using a variational Bayes approach.

- [HCABrowser](https://bioconductor.org/packages/HCABrowser) Search,
  browse, reference, and download resources from the Human Cell Atlas
  data portal. Development of this package is supported through funds
  from the Chan / Zuckerberg initiative.

- [HumanTranscriptomeCompendium](https://bioconductor.org/packages/HumanTranscriptomeCompendium)
  Provide tools for working with a compendium of human transcriptome
  sequences (originally htxcomp).

- [hypeR](https://bioconductor.org/packages/hypeR) Geneset enrichment
  analysis based on hyper-geometric test.

- [infercnv](https://bioconductor.org/packages/infercnv) Using
  single-cell RNA-Seq expression to visualize CNV in cells.

- [IsoCorrectoRGUI](https://bioconductor.org/packages/IsoCorrectoRGUI)
  IsoCorrectoRGUI is a Graphical User Interface for the IsoCorrectoR
  package. IsoCorrectoR performs the correction of mass spectrometry
  data from stable isotope labeling/tracing metabolomics experiments
  with regard to natural isotope abundance and tracer impurity. Data
  from both MS and MS/MS measurements can be corrected (with any
  tracer isotope: 13C, 15N, 18O...), as well as high resolution MS
  data from multiple-tracer experiments (e.g. 13C and 15N used
  simultaneously).

- [lipidr](https://bioconductor.org/packages/lipidr) lipidr an
  easy-to-use R package implementing a complete workflow for
  downstream analysis of lipidomics data. lipidr parses results
  exported from Skyline directly into R, allowing integration into
  current analysis frameworks. lipidr allows data inspection,
  normalization, univariate and multivariate analysis, displaying
  informative visualizations. lipidr also implements a novel Lipid
  Set Enrichment Analysis (LSEA), harnessing molecular information
  such as lipid class, chain length and unsaturation.

- [mbkmeans](https://bioconductor.org/packages/mbkmeans) Implements
  the mini-batch k-means algorithm for large datasets, including
  support for on-disk data representation.

- [Melissa](https://bioconductor.org/packages/Melissa) Melissa is a
  Baysian probabilistic model for jointly clustering and imputing
  single cell methylomes. This is done by taking into account local
  correlations via a Generalised Linear Model approach and global
  similarities using a mixture modelling approach.

- [metagene2](https://bioconductor.org/packages/metagene2) This
  package produces metagene plots to compare coverages of sequencing
  experiments at selected groups of genomic regions. It can be used
  for such analyses as assessing the binding of DNA-interacting
  proteins at promoter regions or surveying antisense transcription
  over the length of a gene. The metagene2 package can manage all
  aspects of the analysis, from normalization of coverages to plot
  facetting according to experimental metadata. Bootstraping analysis
  is used to provide confidence intervals of per-sample mean
  coverages.

- [miRspongeR](https://bioconductor.org/packages/miRspongeR) This
  package provides several functions to study miRNA sponge (also
  called ceRNA or miRNA decoy), including popular methods for
  identifying miRNA sponge interactions, and the integrative method
  to integrate miRNA sponge interactions from different methods, as
  well as the functions to validate miRNA sponge interactions, and
  infer miRNA sponge modules, conduct enrichment analysis of modules,
  and conduct survival analysis of modules.

- [mnem](https://bioconductor.org/packages/mnem) Mixture Nested
  Effects Models (mnem) is an extension of Nested Effects Models and
  allows for the analysis of single cell perturbation data provided
  by methods like Perturb-Seq (Dixit et al., 2016) or Crop-Seq
  (Datlinger et al., 2017). In those experiments each of many cells
  is perturbed by a knock-down of a specific gene, i.e. several cells
  are perturbed by a knock-down of gene A, several by a knock-down of
  gene B, ... and so forth. The observed read-out has to be
  multi-trait and in the case of the Perturb-/Crop-Seq gene are
  expression profiles for each cell. mnem uses a mixture model to
  simultaneously cluster the cell population into k clusters and and
  infer k networks causally linking the perturbed genes for each
  cluster. The mixture components are inferred via an expectation
  maximization algorithm.

- [Modstrings](https://bioconductor.org/packages/Modstrings)
  Representing nucleotide modifications in a nucleotide sequence is
  usually done via special characters from a number of sources. This
  represents a challenge to work with in R and the Biostrings
  package. The Modstrings package implements this functionallity for
  RNA and DNA sequences containing modified nucleotides by
  translating the character internally in order to work with the
  infrastructure of the Biostrings package. For this the ModRNAString
  and ModDNAString classes and derivates and functions to construct
  and modify these objects despite the encoding issues are
  implemenented. In addition the conversion from sequences to list
  like location information (and the reverse operation) is
  implemented as well.

- [MOFA](https://bioconductor.org/packages/MOFA) Multi-Omics Factor
  Analysis: an unsupervised framework for the integration of
  multi-omics data sets.

- [nanotatoR](https://bioconductor.org/packages/nanotatoR) Whole
  genome sequencing (WGS) has successfully been used to identify
  single-nucleotide variants (SNV), small insertions and deletions
  and, more recently, small copy number variants. However, due to
  utilization of short reads, it is not well suited for
  identification of structural variants (SV) and the majority of SV
  calling tools having high false positive and negative rates.Optical
  next-generation mapping (NGM) utilizes long fluorescently labeled
  native-state DNA molecules for de novo genome assembly to overcome
  the limitations of WGS. NGM allows for a significant increase in SV
  detection capability. However, accuracy of SV annotation is highly
  important for variant classification and filtration to determine
  pathogenicity.Here we create a new tool in R, for SV annotation,
  including population frequency and gene function and expression,
  using publicly available datasets. We use DGV (Database of Genomic
  Variants), to calculate the population frequency of the SVs
  identified by the Bionano SVCaller in the NGM dataset of a cohort
  of 50 undiagnosed patients with a variety of phenotypes. The new
  annotation tool, nanotatoR, also calculates the internal frequency
  of SVs, which could be beneficial in identification of potential
  false positive or common calls. The software creates a primary gene
  list (PG) from NCBI databases based on patient phenotype and
  compares the list to the set of genes overlapping the patient’s SVs
  generated by SVCaller, providing analysts with an easy way to
  identify variants affecting genes in the PG. The output is given in
  an Excel file format, which is subdivided into multiple sheets
  based on SV type. Users then have a choice to filter SVs using the
  provided annotation for identification of inherited, de novo or
  rare variants. nanotatoR provides integrated annotation and the
  expression patterns to enable users to identify potential
  pathogenic SVs with greater precision and faster turnaround times.

- [NBAMSeq](https://bioconductor.org/packages/NBAMSeq)
  High-throughput sequencing experiments followed by differential
  expression analysis is a widely used approach to detect genomic
  biomarkers. A fundamental step in differential expression analysis
  is to model the association between gene counts and covariates of
  interest. NBAMSeq a flexible statistical model based on the
  generalized additive model and allows for information sharing
  across genes in variance estimation.

- [netboost](https://bioconductor.org/packages/netboost) Boosting
  supported network analysis for high-dimensional omics applications.
  This package comes bundled with the MC-UPGMA clustering package by
  Yaniv Loewenstein.

- [ngsReports](https://bioconductor.org/packages/ngsReports) This
  package provides methods and object classes for parsing FastQC
  reports and output summaries from other NGS tools into R, as well
  as visualising the data loaded from these files.

- [OmicsLonDA](https://bioconductor.org/packages/OmicsLonDA)
  Statistical method that provides robust identification of time
  intervals where omics features (such as proteomics, lipidomics,
  metabolomics, transcriptomics, microbiome, as well as physiological
  parameters captured by wearable sensors such as heart rhythm, body
  temperature, and activity level) are significantly different
  between groups.

- [OVESEG](https://bioconductor.org/packages/OVESEG) An R package for
  multiple-group comparison to detect tissue/cell-specific marker
  genes among subtypes. It provides functions to compute OVESEG-test
  statistics, derive component weights in the mixture null
  distribution model and estimate p-values from weightedly aggregated
  permutations. Obtained posterior probabilities of component null
  hypotheses can also portrait all kinds of upregulation patterns
  among subtypes.

- [PAIRADISE](https://bioconductor.org/packages/PAIRADISE) This
  package implements the PAIRADISE procedure for detecting
  differential isoform expression between matched replicates in
  paired RNA-Seq data.

- [PAST](https://bioconductor.org/packages/PAST) PAST takes GWAS
  output and assigns SNPs to genes, uses those genes to find pathways
  associated with the genes, and plots pathways based on
  significance. Implements methods for reading GWAS input data,
  finding genes associated with SNPs, calculating enrichment score
  and significance of pathways, and plotting pathways.

- [pathwayPCA](https://bioconductor.org/packages/pathwayPCA) Apply
  the Supervised PCA and Adaptive, Elastic-Net, Sparse PCA methods to
  extract principal components from each pathway. Use these pathway-
  specific principal components as the design matrix relating the
  response to each pathway. Return the model fit statistic p-values,
  and adjust these values for False Discovery Rate. Return a data
  frame of the pathways sorted by their adjusted p-values. This
  package has corresponding vignettes hosted in the ''User Guides''
  page of <https://gabrielodom.github.io/pathwayPCA/index.html>, and
  the website for the development information is hosted at
  <https://github.com/gabrielodom/pathwayPCA>.

- [PCAtools](https://bioconductor.org/packages/PCAtools) Principal
  Components Analysis (PCA) is a very powerful technique that has
  wide applicability in data science, bioinformatics, and further
  afield. It was initially developed to analyse large volumes of data
  in order to tease out the differences/relationships between the
  logical entities being analysed. It extracts the fundamental
  structure of the data without the need to build any model to
  represent it. This 'summary' of the data is arrived at through a
  process of reduction that can transform the large number of
  variables into a lesser number that are uncorrelated, i.e., the
  principal components, whilst at the same time being capable of easy
  interpretation on the original data.

- [phemd](https://bioconductor.org/packages/phemd) Package for
  comparing and generating a low-dimensional embedding of multiple
  single-cell samples.

- [pipeFrame](https://bioconductor.org/packages/pipeFrame) pipeFrame
  is an R package for building a componentized bioinformatics
  pipeline. Each step in this pipeline is wrapped in the framework,
  so the connection among steps is created seamlessly and
  automatically. Users could focus more on fine-tuning arguments
  rather than spending a lot of time on transforming file format,
  passing task outputs to task inputs or installing the dependencies.
  Componentized step elements can be customized into other new
  pipelines flexibly as well. This pipeline can be split into several
  important functional steps, so it is much easier for users to
  understand the complex arguments from each step rather than
  parameter combination from the whole pipeline. At the same time,
  componentized pipeline can restart at the breakpoint and avoid
  rerunning the whole pipeline, which may save a lot of time for
  users on pipeline tuning or such issues as power off or process
  other interrupts.

- [PoTRA](https://bioconductor.org/packages/PoTRA) The PoTRA analysis
  is based on topological ranks of genes in biological pathways.
  PoTRA can be used to detect pathways involved in disease (Li, Liu &
  Dinu, 2018). We use PageRank to measure the relative topological
  ranks of genes in each biological pathway, then select hub genes
  for each pathway, and use Fishers Exact test to determine if the
  number of hub genes in each pathway is altered from normal to
  cancer (Li, Liu & Dinu, 2018). Alternatively, if the distribution
  of topological ranks of gene in a pathway is altered between normal
  and cancer, this pathway might also be involved in cancer (Li, Liu
  & Dinu, 2018). Hence, we use the Kolmogorov–Smirnov test to detect
  pathways that have an altered distribution of topological ranks of
  genes between two phenotypes (Li, Liu & Dinu, 2018). PoTRA can be
  used with the KEGG, Biocarta, Reactome, NCI, SMPDB and PharmGKB
  databases from the devel graphite library.

- [pram](https://bioconductor.org/packages/pram) Publicly available
  RNA-seq data is routinely used for retrospective analysis to
  elucidate new biology.  Novel transcript discovery enabled by large
  collections of RNA-seq datasets has emerged as one of such
  analysis.  To increase the power of transcript discovery from large
  collections of RNA-seq datasets, we developed a new R package named
  Pooling RNA-seq and Assembling Models (PRAM), which builds
  transcript models in intergenic regions from pooled RNA-seq
  datasets.  This package includes functions for defining intergenic
  regions, extracting and pooling related RNA-seq alignments,
  predicting, selected, and evaluating transcript models.

- [PrecisionTrialDrawer](https://bioconductor.org/packages/PrecisionTrialDrawer)
  A suite of methods to design umbrella and basket trials for
  preision oncology.

- [PrInCE](https://bioconductor.org/packages/PrInCE) PrInCE
  (Predicting Interactomes from Co-Elution) uses a naive Bayes
  classifier trained on dataset-derived features to recover
  protein-protein interactions from co-elution chromatogram profiles.
  This package contains the R implementation of PrInCE.

- [proBatch](https://bioconductor.org/packages/proBatch) The proBatch
  package facilitates batch effects analysis and correction in
  high-thoughput experiments. It was developed primarily for
  mass-spectrometry proteomics (DIA/SWATH), but could also be
  applicable to most omic data with minor adaptations. The package
  contains functions for diagnostics (proteome/genome-wide and
  feature-level), correction (normalization and batch effects
  correction) and quality control. Non-linear fitting based
  approaches were also included to deal with complex, mass
  spectrometry-specific signal drifts.

- [profileplyr](https://bioconductor.org/packages/profileplyr) Quick
  and straighforward visualization of read signal over genomic
  intervals is key for generating hypotheses from sequencing data
  sets (e.g. ChIP-seq, ATAC-seq, bisulfite/methyl-seq). Many tools
  both inside and outside of R and Bioconductor are available to
  explore these types of data, and they typically start with a bigWig
  or BAM file and end with some representation of the signal (e.g.
  heatmap). profileplyr leverages many Bioconductor tools to allow
  for both flexibility and additional functionality in workflows that
  end with visualization of the read signal.

- [projectR](https://bioconductor.org/packages/projectR) Functions
  for the projection of data into the spaces defined by PCA, CoGAPS,
  NMF, correlation, and clustering.

- [qckitfastq](https://bioconductor.org/packages/qckitfastq)
  Assessment of FASTQ file format with multiple metrics including
  quality score, sequence content, overrepresented sequence and
  Kmers.

- [qsmooth](https://bioconductor.org/packages/qsmooth) Smooth
  quantile normalization is a generalization of quantile
  normalization, which is average of the two types of assumptions
  about the data generation process: quantile normalization and
  quantile normalization between groups.

- [RCM](https://bioconductor.org/packages/RCM) Combine ideas of
  log-linear analysis of contingency table, flexible response
  function estimation and empirical Bayes dispersion estimation for
  explorative visualization of microbiome datasets. The package
  includes unconstrained as well as constrained analysis.

- [Rcwl](https://bioconductor.org/packages/Rcwl) The package can be a
  simple and user-friendly way to manage command line tools and build
  data analysis pipelines in R using Common Workflow Language (CWL).

- [RcwlPipelines](https://bioconductor.org/packages/RcwlPipelines) A
  collection of Bioinformatics pipeline recipes based on Rcwl.

- [RepViz](https://bioconductor.org/packages/RepViz) RepViz enables
  the view of a genomic region in a simple and efficient way. RepViz
  allows simultaneous viewing of both intra- and intergroup variation
  in sequencing counts of the studied conditions, as well as their
  comparison to the output features (e.g. identified peaks) from user
  selected data analysis methods.The RepViz tool is primarily
  designed for chromatin data such as ChIP-seq and ATAC-seq, but can
  also be used with other sequencing data such as RNA-seq, or
  combinations of different types of genomic data.

- [Rhisat2](https://bioconductor.org/packages/Rhisat2) An R interface
  to the HISAT2 spliced short-read aligner by Kim et al. (2015). The
  package contains wrapper functions to create a genome index and to
  perform the read alignment to the generated index.

- [rmelting](https://bioconductor.org/packages/rmelting) R interface
  to the MELTING 5 program
  (<https://www.ebi.ac.uk/biomodels/tools/melting/>) to compute
  melting temperatures of nucleic acid duplexes along with other
  thermodynamic parameters.

- [rScudo](https://bioconductor.org/packages/rScudo) SCUDO
  (Signature-based Clustering for Diagnostic Purposes) is a
  rank-based method for the analysis of gene expression profiles for
  diagnostic and classification purposes. It is based on the
  identification of sample-specific gene signatures composed of the
  most up- and down-regulated genes for that sample. Starting from
  gene expression data, functions in this package identify
  sample-specific gene signatures and use them to build a graph of
  samples. In this graph samples are joined by edges if they have a
  similar expression profile, according to a pre-computed similarity
  matrix. The similarity between the expression profiles of two
  samples is computed using a method similar to GSEA. The graph of
  samples can then be used to perform community clustering or to
  perform supervised classification of samples in a testing set.

- [scAlign](https://bioconductor.org/packages/scAlign) An
  unsupervised deep learning method for data alignment, integration
  and estimation of per-cell differences in -omic data (e.g. gene
  expression) across datasets (conditions, tissues, species). See
  Johansen and Quon (2019) <doi:10.1101/504944> for more details.

- [scds](https://bioconductor.org/packages/scds) In single cell RNA
  sequencing (scRNA-seq) data combinations of cells are sometimes
  considered a single cell (doublets). The scds package provides
  methods to annotate doublets in scRNA-seq data computationally.

- [scMerge](https://bioconductor.org/packages/scMerge) Like all gene
  expression data, single-cell RNA-seq (scRNA-Seq) data suffers from
  batch effects and other unwanted variations that makes accurate
  biological interpretations difficult. The scMerge method leverages
  factor analysis, stably expressed genes (SEGs) and (pseudo-)
  replicates to remove unwanted variations and merge multiple
  scRNA-Seq data. This package contains all the necessary functions
  in the scMerge pipeline, including the identification of SEGs,
  replication-identification methods, and merging of scRNA-Seq data.

- [scRecover](https://bioconductor.org/packages/scRecover) scRecover
  is an R package for imputation of single-cell RNA-seq (scRNA-seq)
  data. It will detect and impute dropout values in a scRNA-seq raw
  read counts matrix while keeping the real zeros unchanged, since
  there are both dropout zeros and real zeros in scRNA-seq data. By
  combination with scImpute, SAVER and MAGIC, scRecover not only
  detects dropout and real zeros at higher accuracy, but also improve
  the downstream clustering and visualization results.

- [scTensor](https://bioconductor.org/packages/scTensor) The
  algorithm is based on the non-negative tucker decomposition (NTD)
  of nnTensor.

- [sitePath](https://bioconductor.org/packages/sitePath) The package
  does hierarchical search for fixation events given multiple
  sequence alignment and phylogenetic tree. These fixation events can
  be specific to a phylogenetic lineages or shared by multiple
  lineages.

- [SMAD](https://bioconductor.org/packages/SMAD) Assigning
  probability scores to prey proteins captured in affinity
  purification mass spectrometry (AP-MS) expriments to infer protein-
  protein interactions. The output would facilitate non-specific
  background removal as contaminants are commonly found in AP-MS
  data.

- [SpatialCPie](https://bioconductor.org/packages/SpatialCPie)
  SpatialCPie is an R package designed to facilitate cluster
  evaluation for spatial transcriptomics data by providing intuitive
  visualizations that display the relationships between clusters in
  order to guide the user during cluster identification and other
  downstream applications. The package is built around a shiny
  "gadget" to allow the exploration of the data with multiple plots
  in parallel and an interactive UI. The user can easily toggle
  between different cluster resolutions in order to choose the most
  appropriate visual cues.

- [SpectralTAD](https://bioconductor.org/packages/SpectralTAD)
  SpectralTAD is an R package designed to identify Topologically
  Associated Domains (TADs) from Hi-C contact matrices. It uses a
  modified version of spectral clustering that uses a sliding window
  to quickly detect TADs. The function works on a range of different
  formats of contact matrices and returns a bed file of TAD
  coordinates. The method does not require users to adjust any
  parameters to work and gives them control over the number of
  hierarchical levels to be returned.

- [sRACIPE](https://bioconductor.org/packages/sRACIPE) sRACIPE
  implements a randomization-based method for gene circuit modeling.
  It allows us to study the effect of both the gene expression noise
  and the parametric variation on any gene regulatory circuit (GRC)
  using only its topology, and simulates an ensemble of models with
  random kinetic parameters at multiple noise levels. Statistical
  analysis of the generated gene expressions reveals the basin of
  attraction and stability of various phenotypic states and their
  changes associated with intrinsic and extrinsic noises. sRACIPE
  provides a holistic picture to evaluate the effects of both the
  stochastic nature of cellular processes and the parametric
  variation.

- [ssrch](https://bioconductor.org/packages/ssrch) Demonstrate
  tokenization and a search gadget for collections of CSV files.

- [Structstrings](https://bioconductor.org/packages/Structstrings)
  The Structstrings package implements the widely used dot bracket
  annotation for storing base pairing information in structured RNA.
  Structstrings uses the infrastructure provided by the Biostrings
  package and derives the DotBracketString and related classes from
  the BString class. From these, base pair tables can be produced for
  in depth analysis. In addition, the loop indices of the base pairs
  can be retrieved as well. For better efficiency, information
  conversion is implemented in C, inspired to a large extend by the
  ViennaRNA package.

- [StructuralVariantAnnotation](https://bioconductor.org/packages/StructuralVariantAnnotation)
  StructuralVariantAnnotation contains useful helper functions for
  dealing with structural variants in VCF format. The packages
  contains functions for parsing VCFs from a number of popular
  callers as well as functions for dealing with breakpoints involving
  two separate genomic loci encoded as GRanges objects.

- [SubCellBarCode](https://bioconductor.org/packages/SubCellBarCode)
  Mass-Spectrometry based spatial proteomics have enabled the
  proteome-wide mapping of protein subcellular localization (Orre et
  al. 2019, Molecular Cell). SubCellBarCode R package robustly
  classifies proteins into corresponding subcellular localization.

- [survtype](https://bioconductor.org/packages/survtype) Subtypes are
  defined as groups of samples that have distinct molecular and
  clinical features. Genomic data can be analyzed for discovering
  patient subtypes, associated with clinical data, especially for
  survival information. This package is aimed to identify subtypes
  that are both clinically relevant and biologically meaningful.

- [SynMut](https://bioconductor.org/packages/SynMut) There are
  increasing demands on designing virus mutants with specific
  dinucleotide or codon composition. This tool can take both
  dinucleotide preference and/or codon usage bias into account while
  designing mutants. It is a powerful tool for in silico designs of
  DNA sequence mutants.

- [TNBC.CMS](https://bioconductor.org/packages/TNBC.CMS) This package
  implements a machine learning-based classifier for the assignment
  of consensus molecular subtypes to TNBC samples. It also provides
  functions to summarize genomic and clinical characteristics.

- [topconfects](https://bioconductor.org/packages/topconfects) Find
  the largest confident effect sizes from a large collection, while
  maintaining False Discovery Rate and False Coverage-statement Rate
  control. The main application is differential gene expression
  analysis, providing genes ranked in order of confident log2 fold
  change.

- [TPP2D](https://bioconductor.org/packages/TPP2D) FDR-controlled
  analysis of 2D-TPP experiments by functional analysis of
  dose-response curves across temperatures.

- [TreeSummarizedExperiment](https://bioconductor.org/packages/TreeSummarizedExperiment)
  TreeSummarizedExperiment has extended SingleCellExperiment to
  include hierarchical information on the rows or columns of the
  rectangular data.

- [VCFArray](https://bioconductor.org/packages/VCFArray) VCFArray
  extends the DelayedArray to represent VCF data entries as
  array-like objects with on-disk / remote VCF file as backend. Data
  entries from VCF files, including info fields, FORMAT fields, and
  the fixed columns (REF, ALT, QUAL, FILTER) could be converted into
  VCFArray instances with different dimensions.

- [VennDetail](https://bioconductor.org/packages/VennDetail) A set of
  functions to generate high-resolution Venn,Vennpie plot,extract and
  combine details of these subsets with user datasets in data frame
  is available.

- [Xeva](https://bioconductor.org/packages/Xeva) Contains set of
  functions to perform analysis of patient-derived xenograft (PDX)
  data.


New Data Experiment Packages
=====================

There are 13 new data experiment packages in this release of Bioconductor.

- [adductData](https://bioconductor.org/packages/adductData) mzXML
  files from Grigoryan et al 2016 (Anal Chem).

- [bodymapRat](https://bioconductor.org/packages/bodymapRat) This
  package contains a SummarizedExperiment from the Yu et al. (2013)
  paper that performed the rat BodyMap across 11 organs and 4
  developmental stages. Raw FASTQ files were downloaded and mapped
  using STAR. Data is available on ExperimentHub as a data package.

- [chipseqDBData](https://bioconductor.org/packages/chipseqDBData)
  Sorted and indexed BAM files for ChIP-seq libraries, for use in the
  chipseqDB workflow. BAM indices are also included.

- [CluMSIDdata](https://bioconductor.org/packages/CluMSIDdata) This
  package contains various LC-MS/MS and GC-MS data that is used in
  vignettes and examples in the CluMSID package.

- [curatedAdipoChIP](https://bioconductor.org/packages/curatedAdipoChIP)
  A curated dataset of publicly available ChIP-sequencing of
  transcription factors, chromatin remodelers and histone
  modifications in the 3T3-L1 pre-adipocyte cell line. The package
  document the data collection, pre-processing and processing of the
  data. In addition to the documentation, the package contains the
  scripts that was used to generated the data.

- [curatedAdipoRNA](https://bioconductor.org/packages/curatedAdipoRNA)
  A curated dataset of RNA-Seq samples. The samples are MDI-induced
  pre-phagocytes (3T3-L1) at different time points/stage of
  differentiation. The package document the data collection,
  pre-processing and processing. In addition to the documentation,
  the package contains the scripts that was used to generated the
  data.

- [FlowSorted.CordBloodCombined.450k](https://bioconductor.org/packages/FlowSorted.CordBloodCombined.450k)
  Raw data objects to be used for umbilical cord blood cell
  proportion estimation in minfi and similar packages. The
  FlowSorted.CordBloodCombined.450k object is based in samples
  assayed by Bakulski et al, Gervin et al., de Goede et al., and Lin
  et al.

- [HCAData](https://bioconductor.org/packages/HCAData) This package
  allows a direct access to the dataset generated by the Human Cell
  Atlas project for further processing in R and Bioconductor, in the
  comfortable format of SingleCellExperiment objects (available in
  other formats here: http://preview.data.humancellatlas.org/).

- [macrophage](https://bioconductor.org/packages/macrophage) This
  package provides the output of running Salmon on a set of 24
  RNA-seq samples from Alasoo, et al. "Shared genetic effects on
  chromatin and gene expression indicate a role for enhancer priming
  in immune response", published in Nature Genetics, January 2018.
  For details on version numbers and how the samples were processed
  see the package vignette.

- [MOFAdata](https://bioconductor.org/packages/MOFAdata) A collection
  of datasets to accompany the R package MOFA and illustrate running
  and analysing MOFA models.

- [nanotubes](https://bioconductor.org/packages/nanotubes) Cap
  Analysis of Gene Expression (CAGE) data from "Identification of
  Gene Transcription Start Sites and Enhancers Responding to
  Pulmonary Carbon Nanotube Exposure in Vivo" by Bornholdt et al.
  supplied as CAGE Transcription Start Sites (CTSSs).

- [NestLink](https://bioconductor.org/packages/NestLink) Provides
  next-generation sequencing (NGS) and mass spectrometry (MS) sample
  data, code snippets and replication material used for developing
  NestLink. The NestLink approach is a protein binder selection and
  identification technology able to biophysically characterize
  thousands of library members at once without handling individual
  clones at any stage of the process. Data were acquired on NGS and
  MS platforms at the Functional Genomics Center Zurich.

- [oct4](https://bioconductor.org/packages/oct4) This package
  provides the output of running Salmon on a set of 12 RNA-seq
  samples from King & Klose, "The pioneer factor OCT4 requires the
  chromatin remodeller BRG1 to support gene regulatory element
  function in mouse embryonic stem cells", published in eLIFE, March 2017. For details on version numbers and how the samples were
  processed see the package vignette.


New Workflows
=====================

There are 4 new workflow packages in this release of Bioconductor.

- [BgeeCall](https://bioconductor.org/packages/BgeeCall) Reference
  intergenic regions are generated by the Bgee RNA-Seq pipeliene.
  These intergenic regions are used to generate all Bgee RNA-Seq
  present/absent expression calls. This package now allows to
  generate your own present/absent calls using both these intergenic
  regions and the expertise of Bgee, as long as your species is
  present in Bgee. The threshold of present/absent expression is no
  longer arbitrary defined but is calculated based on expression of
  all RNA-Seq libraries integrated in Bgee.

- [CAGEWorkflow](https://bioconductor.org/packages/CAGEWorkflow)
  Workflow for analyzing Cap Analysis of Gene Expression (CAGE) data
  using R/Bioconductor.

- [csawUsersGuide](https://bioconductor.org/packages/csawUsersGuide)
  A user's guide for the csaw package for detecting differentially
  bound regions in ChIP-seq data. Describes how to read in BAM files
  to obtain a per-window count matrix, filtering to obtain
  high-abundance windows of interest, normalization of
  sample-specific biases, testing for differential binding,
  consolidation of per-window results to obtain per-region
  statistics, and annotation and visualization of the DB results.

- [SingscoreAMLMutations](https://bioconductor.org/packages/SingscoreAMLMutations)
  This workflow package shows how transcriptomic signatures can be
  used to infer phenotypes. The workflow begins by showing how the
  TCGA AML transcriptomic data can be downloaded and processed using
  the TCGAbiolinks packages. It then shows how samples can be scored
  using the singscore package and signatures from the MSigDB.
  Finally, the predictive capacity of scores in the context of
  predicting a specific mutation in AML is shown.The workflow
  exhibits the interplay of Bioconductor packages to achieve a
  gene-set level analysis.


NEWS from new and existing Software Packages
===================================


[ACE](https://bioconductor.org/packages/ACE)
---

                 Changes in version 1.1.2 (2019-04-23)                  

- added citation information!

- fixed bug in getadjustedsegments and twosamplecompare with
  non-matching NAs

                 Changes in version 1.1.1 (2018-11-13)                  

- added forcesegmentsontemplate function

- speed boost by small coding change in model fitting

- added argument in loopsquaremodel to make returning the models
  optional and flexible (see documentation)

[ADAM](https://bioconductor.org/packages/ADAM)
----

                Changes in version 0.99.66 (2018-11-15)                 

- Fixed possible problem to access kegg id

                Changes in version 0.99.65 (2018-11-05)                 

- Fixed possible bugs

                 Changes in version 0.99.0 (2018-10-05)                 

- Submitted to Bioconductor

[AffiXcan](https://bioconductor.org/packages/AffiXcan)
--------

                        Changes in version 1.1.3                        

BUG FIXES

- Fix a bug in generating the vignette for 'devel' version of
  Bioconductor

[alevinQC](https://bioconductor.org/packages/alevinQC)
--------

                       Changes in version 0.99.0                        

- Preparation for Bioconductor submission

                        Changes in version 0.1.0                        

- Initial version

[amplican](https://bioconductor.org/packages/amplican)
--------

                        Changes in version 1.5.8                        

- amplican has been published now, doi: 10.1101/gr.244293.118

- clearly outperforms other tools on many benchmarks

- please cite the paper if you ar eusing amplican

[AneuFinder](https://bioconductor.org/packages/AneuFinder)
----------

                       Changes in version 1.11.1                        

BUG FIXES

- Fixed "object 'dfplot.seg' not found" bug when building vignette.

[AnnotationHub](https://bioconductor.org/packages/AnnotationHub)
-------------

                       Changes in version 2.15.0                        

SIGNIFICANT USER-VISIBLE CHANGES

- (2.15.9) The Hubs have been updated to use BiocFileCache as a backend
  utility for managing and downloading resources. This allows for file
  specific caching mechanism. This change is a significant user change
  as it changed the default directories for the Hubs
  (AnnotationHub/ExperimentHub) as well as the base directory no longer
  being hidden.

NEW FEATURES

- (2.15.15) Added BamFile Resource method

- (2.15.9) Add helper function to get additional information on
  resources. getInfoOnIds

- (2.15.2) Add function to list currently available DispatchClass

USER-VISIBLE MODIFICATIONS

- (2.15.3) Bioconductor will no longer accept random, Individual
  resources without an accompanied package.

- (2.15.4) We also encourage the use of AnnotationHub rather than very
  large annotation packages. We added sections for users wishing to
  convert existing packages to use AnnotationHub as well.

- (2.15.7) There were substantial updates to Rsamtools. Rsamtools no
  longer supports razip files. These files have been removed from the
  hub and vignette updated to use the twobit and gff files

BUG FIXES

- (2.15.5) Fixed bug. If resources are removed from the hub, this
  should only be reflected in the devel branch but still be available
  in release branch. The filter for exposed resources was refined to
  respect datadateremoved

[AnnotationHubData](https://bioconductor.org/packages/AnnotationHubData)
-----------------

                       Changes in version 1.13.0                        

NEW FEATURES

- Added ability to have multiple RDataPaths associated with single hub
  id for strongly associated files (like bam and its bai index file)

- DispatchClass are now validated against
  AnnotationHub::DispatchClassList() which contains currently available
  DispatchClass and brief description of loading process.

[anota2seq](https://bioconductor.org/packages/anota2seq)
---------

                        Changes in version 1.5.0                        

- We added a new option to the output parameter of the
  anota2seqGetOutput function, more details can be found in the
  function help.

- The quality control section of the vignette has been expanded with
  more details, furthermore we added an example of quality control of
  poorly normalized data as reference.

- We removed the apvSlope and apvSlopeP columns from the output of
  analysis for total mRNA and translated mRNA as these values are not
  generated for RNA source differential expression analysis.

[apeglm](https://bioconductor.org/packages/apeglm)
------

                        Changes in version 1.5.4                        

- Added possibility for weights and standard errors to bbEstDisp(). The
  standard errors are for the log of dispersion (theta).

[ASICS](https://bioconductor.org/packages/ASICS)
-----

                         Changes in version 2.0                         

New features

- preprocessing functions from PepsNMR package

Improvements

- changes in alignment function

- new step to improve library alignment and thus quantifications

[AssessORF](https://bioconductor.org/packages/AssessORF)
---------

                 Changes in version 1.1.7 (2019-04-29)                  

- Added a function, CompareAssessmentResults, for comparing two results
  objects against each other

- Added a processors parameter to MapAssessmentData for compatibility
  with DECIPHER functions

                 Changes in version 1.1.6 (2019-04-07)                  

- Added parameters to AssessGenes to bound the range in which to look
  for conserved starts that can serve as alternative options to a
  predicted start

- Added a parameter to AssessGenes that requires ORFs with protein
  evidence but no predicted start to have a certain minimum number of
  peptide hits in order to be included in the final output

                 Changes in version 1.1.5 (2019-01-11)                  

- Update to how conserved starts and conserved stops are utilized in
  AssessGenes

- Removed weighted score

- Made clarifications to the man pages

- Minor edits to the vignette

                 Changes in version 1.1.4 (2018-12-15)                  

- Updated AssessGenes to make sure predicted stops are valid for their
  corresponding genes, and users will be warned appropriately if a
  predicted stop is invalid

- Minor updates to plots

- Bug fixes on previous version's MapAssessmentData updates

                 Changes in version 1.1.3 (2018-11-12)                  

- Fixed a bug affecting coverage related to gaps in alignments in
  MapAssessmentData

- Used exponential moving averages to filter out bad alignment regions
  in MapAssessmentData

                        Changes in version 1.1.2                        

- Implemented a fix to the vignette so it builds without an internet
  connection

                        Changes in version 1.1.1                        

- Reduced time it takes to run MapAssessmentData examples # AssessORF
  0.99

[ATACseqQC](https://bioconductor.org/packages/ATACseqQC)
---------

                        Changes in version 1.7.8                        

- change the function of footprintsScanner.

                        Changes in version 1.7.7                        

- add parameter outPath for splitGAlignmentsByCut.

                        Changes in version 1.7.6                        

- fix a bug in PTscore.

                        Changes in version 1.7.5                        

- use file.copy instead of file.rename.

                        Changes in version 1.7.4                        

- add parameter outbam for shiftGAlignmentsList.

                        Changes in version 1.7.3                        

- Update documentation for Transcription start site (TSS) enrichment
values

                        Changes in version 1.7.2                        

- add the new biocViews tag "ImmunoOncology"

                        Changes in version 1.7.1                        

- add more documentation for bigFile parameter in readBamFile.R

[atSNP](https://bioconductor.org/packages/atSNP)
-----

                 Changes in version 0.99.0 (2018-12-29)                 

- Submitted to Bioconductor

[AUCell](https://bioconductor.org/packages/AUCell)
------

                         Changes in version 1.5                         

- Added function orderAUC()

- Error fixes in AUCell_plotTSNE()

[BANDITS](https://bioconductor.org/packages/BANDITS)
-------

                        Changes in version 1.0.0                        

- Submitted to Bioconductor

[BASiCS](https://bioconductor.org/packages/BASiCS)
------

                 Changes in version 1.5.34 (2019-04-20)                 

- Fixes typo in vignette

                 Changes in version 1.5.33 (2019-04-19)                 

- Updated ERCC formula in the vignette.

                 Changes in version 1.5.32 (2019-04-17)                 

- Minor change in `BASiCS_MCMC` to avoid storing adaptive variances in
  the `BASiCS_Chain` object when `StoreAdapt = TRUE`.

- New unit test to ensure the correct output in
  `BASiCS_Chain@parameters`

- Minor change in `Summary` method for `BASiCS_Chain` class to avoid
  calculating posterior summaries when `designMatrix` is part of the
  `parameters` slot

- Added `return` to the docummentation of `BASiCS_effectiveSize`.

                 Changes in version 1.5.31 (2019-04-15)                 

- Updated vignette to include ERCC calculation

                 Changes in version 1.5.30 (2019-04-11)                 

- `BASiCS_effectiveSize` has been created as a wrapper of
  `coda::effectiveSize`. This is to resolve issues when `Param =
  'epsilon'` as it contains `NA` for very lowly expressed genes that
  are excluded from the mean/over-dispersion trend.

                 Changes in version 1.5.29 (2019-04-10)                 

- Unit test associated to `BASiCS_Sim` updated to use a fixed seed and
  to check exact values being reproducible

                 Changes in version 1.5.28 (2019-04-09)                 

- Edited version of `BASiCS_Sim` to allow no spikes and multiple
  batches

- Documentation updated accordingly

                 Changes in version 1.5.27 (2019-04-08)                 

- Edits `BASiCS_TestDE` to avoid low posterior probability thresholds
  derived from the EFDR calibration. This alters some of the related
  'Hidden' functions.

- In `BASiCS_TestDE`, default values for `EFDR_M`, `EFDR_D` and
  `EFDR_R` were changed to 0.05.

- Unit tests updated accordingly

                 Changes in version 1.5.26 (2019-04-08)                 

- Similar changes applied to `utils_MCMCcppReg.cpp` and
  `utils_MCMCcppRegNoSpikes.cpp`.

                 Changes in version 1.5.25 (2019-04-07)                 

- Minor change in `muUpdateNoSpikes` (C++ function in
  `utils_MCMCNoSpikes.cpp`) to avoid WARNINGS in bioc-devel related to
  integers vs unsigned integers

                 Changes in version 1.5.24 (2019-04-05)                 

- Minor edit to the documentation of `makeExampleBASiCS_Data` (fixed
  link to `set.seed` documentation as it led to WARNINGS in Bioc-devel)

                 Changes in version 1.5.23 (2019-04-02)                 

- Minor edit to the documentation of `makeExampleBASiCS_Data` (explain
  why a fixed seed is no longer used)

                 Changes in version 1.5.22 (2019-04-02)                 

- `HiddenCheckValidCombination`, `HiddenGeneParams` and
  `HiddenGeneParams` moved from `Methods.R` to `HiddenUtils.R`

- `sapply` replaced by `vapply` in `HiddenCheckValidCombination`,
  `HiddenThresholdSearchDetectHVL_LVG.R` and
  `HiddenThresholdSearchTestDE.R`

- Use of `set.seed` removed from `makeExampleBASiCS_Data`

- Unit tests updated accordingly

- Adds missing running examples to Rds in order to pass R CMD BioCheck

                 Changes in version 1.5.21 (2019-04-01)                 

- Uses built-in chain in examples for `BASiCS_diagHist` and
  `BASiCS_diagPlot`

                 Changes in version 1.5.20 (2019-04-01)                 

- Updated documentation for `BASiCS_MCMC`, `BASiCS_DetectHVG`,
  `BASiCS_DetectLVG` and `BASiCS_VarThresholdSearchHVG_LVG`

                 Changes in version 1.5.19 (2019-03-31)                 

- `stats4` added to imports in order to import `plot` S4 generic.

- `KernSmooth` removed from DESCRIPTION.

- Maintainer field removed from DESCRIPTION (use Author@R only)

- Updates to travis integration to include BiocCheck

                 Changes in version 1.5.17 (2019-03-30)                 

- Minor changes in dependencies to pass R CMD check

- Minor typo fix in documentation of `newBASiCS_Data`

- `@export` call added to all methods

                 Changes in version 1.5.16 (2019-03-30)                 

- `SingleCellExperiment` moved to depends (DESCRIPTION)

- Explicit calls to external packages (e.g. `SingleCellExperiment::`)

- Unit tests for `BASiCS_diagHist` and `BASiCS_diagPlot`

- `BASiCS_diagHist`, `BASiCS_diagPlot` and `BASiCS_showFit` moved from
  methods to functions

- Internal functions marked as 'Hidden'

- `BASiCS_diagHist`, etc moved into separate .R files

                 Changes in version 1.5.15 (2019-03-27)                 

- Extended message to suggest additional QC when MCMC sampler fails

                 Changes in version 1.5.14 (2019-03-27)                 

- Brackets removed from `setGeneric` call (as per Hadley Wickham
  recommendation)

- Removes full import of `ggplot2` in `BASiCS_Package.R`

- `BASiCS_Package.R` is now in alphabetical order

                 Changes in version 1.5.13 (2019-03-27)                 

- `BASiCS_D_TestDE` from deprecated to defunct (follows BioC
  guidelines)

                 Changes in version 1.5.12 (2019-03-26)                 

- R (>= 3.6) is now a requirement

                 Changes in version 1.5.11 (2019-03-26)                 

- Edits in BASiCS_Package.R so that `roxygen2` creates `NAMESPACE`

- `testthat` moved from Imports to Suggests

                 Changes in version 1.5.10 (2019-03-26)                 

- Corrected ordering in TestDE outpout so that genenames always match

- Small fixes for R CMD check

- Ordering in `BASiCS_DetectHVG` and `BASiCS_DetectLVG` can only be
  done on GeneNames, GeneIndex and Prob

                 Changes in version 1.5.9 (2019-03-24)                  

- Alan O'Callaghan is now author

- `diagPlot` and `diagHist` methods added to facilitate diagnosis of
  convergence and mixing for MCMC chains

- Added explicit calls to `ggplot2` in plot methods.

- `BASiCS_DetectHVG` and `BASiCS_DetectLVG` edited based on the
  "regression" approach (i.e. using epsilon to detect HVGs/LVGs).

- In `BASiCS_DetectHVG` and `BASiCS_DetectLVG`, options for
  `OrderVariable` were simplified (only original order and based on
  probability)

- Temporary changes in no-spikes version have been restored

                 Changes in version 1.5.8 (2019-03-22)                  

- Changes DESCRIPTION to have a single maintainer

- Density plot added to plot Method for BASiCS_Chain objects

- Output plots for plot Method for BASiCS_Chain objects

                 Changes in version 1.5.7 (2019-03-22)                  

- Added default value for 'BatchInfo' within 'BASiCS_MCMC'

                        Changes in version 1.5.6                        

- Bug fix when not providing BatchInfo as factor

- Extra check if batch names contain spaces

- Added line breaks after individual error messages

- Bug fix that corrects for the number of factors in BatchInfo vector

- Small bug fix in 'test_mcmc_arguments.R'

                 Changes in version 1.5.5 (2019-03-14)                  

- Updated vignette to include recommended settings earlier.

                 Changes in version 1.5.4 (2019-02-12)                  

- Replaces `HPDinterval` by `coda::HPDinterval` in `BASiCS_showFit`
  method

- Same as above when calling `mcmc`

- Changes `BASiCS_showFit` such that fit line is on top of the dots

- Additional optional arguments (`markExcludedGenes`, `colour`,
  `GenesSel` and `colourGenesSel`) added to `BASiCS_showFit` for more
  flexibility

                 Changes in version 1.5.3 (2018-12-06)                  

- Removes `...` argument from `BASiCS_VarianceDecomp`

- Adds missing argument (`beside`) to documentation of
  `BASiCS_VarianceDecomp`

                 Changes in version 1.5.2 (2018-12-01)                  

- Fixes use of `BatchInfo` when it cannot be parsed to numeric

- Refactors the use of `...` argument (several functions)

                 Changes in version 1.5.1 (2018-11-29)                  

- `ImmunoOncology` added in biocViews as requested by BioC core team

[bayNorm](https://bioconductor.org/packages/bayNorm)
-------

                        Changes in version 1.1.6                        

- Suppress warning message came from BiocParallel.

                        Changes in version 1.1.4                        

- For a smoother progress bar, use foreach instead of BiocParallel.

[bcSeq](https://bioconductor.org/packages/bcSeq)
-----

                       Changes in version 1.5.11                        

- using new data structure for the library to speed up the mapping
  process.

                        Changes in version 1.5.9                        

- fix bugs in the examples

                        Changes in version 1.5.7                        

- fix bugs in the examples

                        Changes in version 1.5.3                        

- fix bugs in vignette for bioconductor new platform

[beachmat](https://bioconductor.org/packages/beachmat)
--------

                        Changes in version 2.0.0                        

- Converted beachmat into a header-only library to simplify linking,
  avoid ABI incompatibilities.

- Minor API change for external native support to avoid using C++
  classes with C-style linkage.

[BEclear](https://bioconductor.org/packages/BEclear)
-------

                 Changes in version 2.0.0 (2019-04-30)                  

- Code refactoring o Simplification of the source code o Changes to the
  API o Extraction of the preprocessing function o loss function is now
  exported o renaming of functions o Combination of calcPvals and
  calcMedians methods into one

- Documentation o Update of the README o New Vignette

- Serial execution of the methods is now the default

- Performance improvments o Usage of RCPP for further performance
  improvements

- Change from GPL-2 to GPL-3 license

- Removing of deprecated functions

- Tests o Implementation of further tests

[bigPint](https://bioconductor.org/packages/bigPint)
-------

                 Changes in version 0.99.7 (2019-01-28)                 

BigPint accepted to Bioconductor (Open source software for
bioinformatics)

                 Changes in version 0.99.0 (2018-09-30)                 

Vignette for bigPint was released on github.io

[bioCancer](https://bioconductor.org/packages/bioCancer)
---------

                       Changes in version 1.11.00                       

- Change the address of cgdsr: http://www.cbioportal.org/ instead
  http://www.cbioportal.org/public-portal/

[BiocCheck](https://bioconductor.org/packages/BiocCheck)
---------

                        Changes in version 1.19                         

NEW FEATURES

- (1.19.33) Add Authors@R vs Author/Maintainer check

- (1.19.29) Add non evaluated code chunks that have not been executed
  because of invalid syntax (```, ```R, ```r). Valid syntax for an
  evaluated code chunk takes the form ```{r} or ```{R}.

- (1.19.28) Check that vignette metadata set correctly

- (1.19.27) Check for Author@R or Author/Maintainer but not both

- (1.19.25) Check for use of remotes: in Description

- (1.19.18) Check for use of dontrun/donttest

- (1.19.15) Check vignetteEngine/vignetteBuilder minimially declared in
  Suggests

- (1.19.9) Check usage of donttest and dontrun in man page
  documentation

- (1.19.9) Update deprecated check to check for Bioconductor release
  and devel deprecated packages as specified in biocViews

- (1.19.7) More helpful ERROR when using a non valid command line
  option.

- (1.19.4) All checks module. Ability to turn on/off with flag options.
  See 'R CMD BiocCheck --help'

- (1.19.1) New Check options to turn off if in CRAN (--no-check-CRAN)
  and Bioconductor mailing list and support site (--no-check-bioc-help)

USER SIGNIFICANT CHANGES

- (1.19.8) Updated Documentation in vignette for flag/option controls
  and reorganization of code/checks. Grouped similar checks together
  and changed order of checks.

- (1.19.3) Remove Native Routine Registrations (use flag in R CMD check
  instead _R_CHECK_NATIVE_ROUTINE_REGISTRATION_)

- (1.19.2) Match CRAN standards of package size <= 5MB (updated from
  4MB)

BUG FIXES

- (1.19.26) Fix NEWS check to recognize NEWS.md

- (1.19.26) Check all repos for existing package not just software

- (1.19.23) Informative message if no biocViews term found

- (1.19.22) Fix output of system2 usage check

- (1.19.19) Test only closures for T/F

- (1.19.14) Fix ERROR when more than one VignetteEngine found

- (1.19.10) Add suggestion of FormatR package to formatting notes

- (1.19.10) Fix function lengths to be a NOTE and only display if
  functions are greater than 50 lines long.

- (1.19.6) Replace use of devtools::create with usethis::create_package
  as function was deprecated.

- (1.19.5) Fix length > 1 logical comparison in if statement

[BiocFileCache](https://bioconductor.org/packages/BiocFileCache)
-------------

                         Changes in version 1.7                         

BUG FIX

- (1.7.10) Fix date comparison to use as.POSIXlt instead of as.Date.
  This catches the rare edge case if a remote resource is updated twice
  in a day and a download happened in between the two.

- (1.7.9) Make creation of database atomic commits

- (1.7.8) Make adding resource to database atomic commits

- (1.7.4) Limit where possible opening database in RW mode. Use
  read-only mode whereever possible to avoid database locking

- (1.7.2) Fix bug in multi path addition through bfcadd/bfcrpaths. The
  function was assuming all files had same rtype and action. Now can
  vectorize

- (1.7.1) Fix man documentation example. Interactive vs non interactive
  session

[BiocNeighbors](https://bioconductor.org/packages/BiocNeighbors)
-------------

                        Changes in version 1.2.0                        

- findNeighbors() and queryNeighbors() now accept a vector of
  point-specific thresholds.

- Added an VP tree implemention in findVptree(), queryVptree(),
  buildVptree(). Supported dispatch to these methods in the generics.

- Added a HNSW implementation in findHnsw(), queryHnsw(), buildHnsw().
  Supported dispatch to these methods in the generics.

- Renamed buildNNIndex() to buildIndex().

- Converted findNeighbors() and queryNeighbors() into S4 methods.
  Created specific rangeFind*() and rangeQuery*() functions for KMKNN
  and VP tree algorithms.

- Modified AnnoyIndex class to hold the original data matrix. Created
  bnorder(), bndata() generics to obtain (possibly reordered) input
  matrices for all indexes.

- Supported Manhattan distance searching in all algorithms.

[BiocParallel](https://bioconductor.org/packages/BiocParallel)
------------

                        Changes in version 1.18                         

USER VISIBLE CHANGES

- (v 1.17.6) Initial use of registered BPPARAM does not advance random
  number seed, see
  https://stat.ethz.ch/pipermail/bioc-devel/2019-January/014526.html

- (v 1.17.7) Loading package does not advance random number seed, see
  https://stat.ethz.ch/pipermail/bioc-devel/2019-January/014535.html

- (v. 1.17.7) removed deprecated functions bplasterror(), bpresume(),
  bpcatchError() and field catch.error.

- (v. 1.17.7) Make logdir, resultdir fields of BiocParallelParam.

- (v. 1.17.7) replaced internal use of BatchJobs:::checkDir() (testing
  existence and read / write ability of log and other directories) with
  BiocParallelParam validity check.

- (v. 1.17.7) expose 'developer' interface, `?DeveloperInterface`

- (v. 1.17.11) on Windows, coerce `MulticoreParam(n)` to
  `MulticoreParam(1)` == SerialParam()`

BUG FIXES

- (v 1.17.4) port 1.16.3 (no '>' on SnowParam() worker end) and 1.16.4
  (bpRNGseed<-() accepts NULL)

- (v 1.17.5) port 1.16.4 (bpRNGseed() can reset seed to NULL), 1.16.5
  (number of available cores defaults to 1 if cannot be determined).

[biocViews](https://bioconductor.org/packages/biocViews)
---------

                       Changes in version 1.51.11                       

NEW FEATURES

- (1.51.7) Cross check views with manifest file so all expected
  bioconductor package get listed in biocViews. New argument to
  write_VIEWS manifestFile

- (1.51.9) New argument to write_VIEWS meatPath. For packages that fail
  to build on any platform, use cloned repository DESCRIPTION file to
  fill in as much information to the biocVIEWS entry.

- (1.51.10) Add new biocVIEW terms ImmunoOncolocy and
  ImmunoOncologyWorkflow.

- (1.51.12) Allow NEWS file to be in .md format. R 3.6 started allowing
  NEWS, NEWS.Rd and newly NEWS.md formats. Adjust code accordingly.

[BioMM](https://bioconductor.org/packages/BioMM)
-----

                       Changes in version 0.99.13                       

- updated the format of NEWS

- updated BioMMtutorial.Rmd

- updated data4demo.R

- replaced the generation of stage-2 data by BioMMreconData() with
  preprocessed 'stage2dataA' to reduce runtime

- increased repeatA1 to 50 for BioMM() example to avoid any NAs.

- created stage2dataA.rds

                       Changes in version 0.99.12                       

- updated the format of NEWS

                       Changes in version 0.99.11                       

- updated DESCRIPTION file

                       Changes in version 0.99.10                       

- minor bug fixed in getDataAfterFS()

- updated installation method

                       Changes in version 0.99.9                        

- removed uncessary print statements

- removed 'return' calls for some functions

- removed unnecessary 'which' statements

- replaced the usage of MulticoreParam()

- updated tutorial with smaller examples and models with less
  computation

                       Changes in version 0.99.8                        

- updated author, maintainer section in DESCRIPTION

- updated parallel computing cores used in the vignettes

                       Changes in version 0.99.7                        

- updated parallel evaluation environments

- updated installation method in README.md

- more cores (non-windows) for parallel computing used in the vignettes
  to reduce runtime

                       Changes in version 0.99.6                        

- updated plotRankedFeature() and getMetrics()

- increased core number for parallel computing

- removed Duplicated descriptions in *.Rd files

                       Changes in version 0.99.5                        

- reformatted code by using formatR

- bplapply() instead of mclapply

- message() instead of print() for end-user messages

- improved usage of which()

- use seq_along() instead of seq_len(length())

- updated installation method

- renamed R source files

                       Changes in version 0.99.4                        

- updated example in BioMMreconData()

                       Changes in version 0.99.3                        

- updated example in BioMM()

                       Changes in version 0.99.2                        

- updated R version dependency (from 3.5.2 to 3.6)

- Removed the file LICENSE

                       Changes in version 0.99.1                        

- updated R version dependency (from 3.5.0 to 3.5.2)

                       Changes in version 0.99.0                        

- Removed the file "BioMM.Rproj"

[biosigner](https://bioconductor.org/packages/biosigner)
---------

                       Changes in version 1.11.26                       

NEW FEATURE

- biosign and plot methods: 'fig.pdfC' and 'info.txtC' arguments have
  been added for compatibility reasons with other packages; they have
  the same role as 'file.pdfC' and '.sinkC' which now generate a
  deprecated warning.

                       Changes in version 1.11.24                       

MINOR MODIFICATION

- minor internal modification

                       Changes in version 1.11.22                       

MINOR MODIFICATION

- minor internal modification

                       Changes in version 1.11.20                       

MINOR MODIFICATION

- minor internal modification

                       Changes in version 1.11.18                       

MINOR MODIFICATION

- minor internal modification

                       Changes in version 1.11.16                       

MINOR MODIFICATION

- minor internal modification for the handling of ExpressionSet
  instances

                       Changes in version 1.11.14                       

MINOR MODIFICATION

- minor internal modification

                       Changes in version 1.11.12                       

MINOR MODIFICATION

- minor modification in the example of application to ExpressionSet
  instances

                       Changes in version 1.11.10                       

MINOR MODIFICATION

- minor update of example

                       Changes in version 1.11.8                        

MINOR MODIFICATION

- vignette minor modification

                       Changes in version 1.11.6                        

MINOR MODIFICATION

- dev.new() is no more called by the plot method

                       Changes in version 1.11.4                        

BUG FIXED

- error message when y is a character but not a factor is now fixed

                       Changes in version 1.11.2                        

NEW FEATURES

- Application to ExpressionSet: tiers now returned in the fData

[breakpointR](https://bioconductor.org/packages/breakpointR)
-----------

                        Changes in version 1.1.2                        

- Fixed bug in hotspotter function

- Improved handling of small chromosomes/contigs

- Fixed bug in masking certain genomic regions (maskRegions param)

- Improved handling of double SCE chromosomes when synchronizing read
  directionality across cells

[BUMHMM](https://bioconductor.org/packages/BUMHMM)
------

                        Changes in version 1.6.2                        

- Updated links to external functions in man pages.

[CAGEfightR](https://bioconductor.org/packages/CAGEfightR)
----------

                         Changes in version 1.5                         

- Added new functions for spatial analysis of clusters: findLinks finds
  nearby pairs of clusters (for example TSSs and enhancers) and
  calculates the correlation of expression between them. findStretches
  find stretches along the genome where clusters are within a certain
  distance of eachother (for example groups of enhancer forming a super
  enhancer) and calculates the average pairwise correlation between
  members.

- Changed the way clustering works: CAGEfightR uses coverage() to
  calculate genome-wide signals and now rounds the resulting signal to
  a certain number of digits (this can be modified via the
  CAGEfightR.round option), to prevent small positive or negative
  values due to floating point errors. This makes clustering more
  stable meaning the tuneTagClustering function is now deprecated. This
  should also increase the speed of most functions.

- CAGEfightR now uses GPos instead of GRanges for storing CTSSs, this
  should result in improved memory performance.

- Several changes to clusterBidirectionality: Balance is now calculated
  using the midpoint as well (preventing some rare cases where the
  midpoint could mask a single highly expressed CTSS), the pooled CTSS
  signal is now prefiltered for bidirectionality to increase speed and
  custom balance function can be provided (Bhattacharyya coefficient
  and Andersson's D are included).

- Added new check-functions to make it easier to check if objects are
  formatted correctly

[CAGEr](https://bioconductor.org/packages/CAGEr)
-----

                       Changes in version 1.26.0                        

DOCUMENTATION

- Split the "CAGE resources" documentation into a separate vignette.

[CAMERA](https://bioconductor.org/packages/CAMERA)
------

                       Changes in version 1.39.2                        

BUG FIXES

- Fix duplicate vignette name

[canceR](https://bioconductor.org/packages/canceR)
------

                       Changes in version 1.17.6                        

- Remove dependency of RSvgDevice. This package is not available for
  Windows os.

- Remove GSEA.1.0.R from /data folder

                       Changes in version 1.17.5                        

- import from grDevice new.dev savePlot

- remove /data/datalist

- import RSvgDevice (not available for windows OS?)

                       Changes in version 1.17.4                        

- modify doc for cbind.na and rbind.na functiONs

                       Changes in version 1.17.3                        

- modify doc for cbind.na and rbind.na functiONs

[Cardinal](https://bioconductor.org/packages/Cardinal)
--------

                       Changes in version 2.1.30                        

BUG FIXES

- Fixed bug where 'spatialShrunkenCentroids' classification would
  change the user 'options(Cardinal.progress)'

                       Changes in version 2.1.29                        

BUG FIXES

- Coercing to 'SpatialShrunkenCentroids2' now drops empty classes for
  segmentations (as expected)

                       Changes in version 2.1.28                        

BUG FIXES

- Fixed 'topFeatures' method for 'SpatialShrunkenCentroids2' where
  'statistic' was actually printing the 'centers'

- The 'collect' method for 'MSProcessedImagingExperiment' now preserves
  sparseness when pulling into memory

                       Changes in version 2.1.27                        

SIGNIFICANT USER-VISIBLE CHANGES

- Added 'mzFilter' as an alias for 'peakFilter' with more suitable
  defaults for non-peak-picked spectra

BUG FIXES

- Minor fixes to 'print' method for plots and images

- Minor fixes to margin padding for 'colorkey' in images

                       Changes in version 2.1.26                        

BUG FIXES

- Fixed 'print' method for plots and images to respect updating
  plotting parameters via '...' arguments

                       Changes in version 2.1.25                        

SIGNIFICANT USER-VISIBLE CHANGES

- Add 'run' argument to 'pixels' for 'MSImagingExperiment'

- Add 'run' argument to 'plot' for 'MSImagingExperiment'

BUG FIXES

- Fixed bug in 'ylab' with one-sided formulas in 'plot'

- Relaxed errors for out-of-range m/z values in 'features'

- Value range of 'colorkey' now obeys 'zlim' argument

- NULL values for plot limits no longer give errors

                       Changes in version 2.1.24                        

SIGNIFICANT USER-VISIBLE CHANGES

- Documented new 'options(Cardinal.dark=FALSE)' default

BUG FIXES

- Fixed linear subsetting of models using '['' for
  'SparseResultImagingExperiment' objects

- Check length of 'classControl' in 'segmentationTest'

                       Changes in version 2.1.23                        

NEW FEATURES

- Added 'slice' for slicing imaging datasets (as a data cube)

- Added 'alpha.power' argument for 'image' methods

SIGNIFICANT USER-VISIBLE CHANGES

- Added documentation for options() under '?Cardinal'

                       Changes in version 2.1.22                        

NEW FEATURES

- Added new 'Cardinal 2: Statistical methods' vignette

                       Changes in version 2.1.21                        

NEW FEATURES

- Added 'image3D' methods for new classes

                       Changes in version 2.1.20                        

SIGNIFICANT USER-VISIBLE CHANGES

- Updated documentation and simulation examples

BUG FIXES

- Fixed bug in 'summary' for 'SpatialShrunkenCentroids2'

                       Changes in version 2.1.19                        

NEW FEATURES

- Added 'colocalized' method for colocalization

- Added 'colocalized' method for colocalization

                       Changes in version 2.1.18                        

SIGNIFICANT USER-VISIBLE CHANGES

- Updated 'Cardinal 2: User guide' vignette

BUG FIXES

- Fixed bug in 'simulateSpectrum' when n > 1

                       Changes in version 2.1.17                        

NEW FEATURES

- Added 'topFeatures' method for extracting top-ranked features from
  statistical analyses

- Added 'summary' methods for new results objects

- Added 'SummaryDataFrame' for printing result summaries

SIGNIFICANT USER-VISIBLE CHANGES

- Deprecated 'topLabels' method -> use 'topFeatures'

                       Changes in version 2.1.16                        

NEW FEATURES

- Added dplyr verbs for 'DataFrame' and "XDataFrame'

                       Changes in version 2.1.15                        

NEW FEATURES

- Added new 'PCA' method for 'SparseImagingExperiment'

- Added new 'PLS' method for 'SparseImagingExperiment'

- Added new 'OPLS' method for 'SparseImagingExperiment'

SIGNIFICANT USER-VISIBLE CHANGES

- The 'selectROI' method will now use the last plot if no additional
  plotting arguments are given

- Setting 'resolution' on 'MSProcessedImagingExperiment' will now
  update the m/z values binning scheme

- The 'select' and 'filter' methods now using integers as row/col IDs
  without the '.id' argument

BUG FIXES

- Fixed bug in 'OPLS' methods causing cross-validation to produce
  slightly optimistic results

                       Changes in version 2.1.14                        

NEW FEATURES

- New 'mzAlign' processing method for spectral alignment

- New 'mzBin' processing method for spectral binning

- New 'crossValidate' method that cleans up 'cvApply' output

- New normalization methods: 'rms' and 'reference'

- New baseline reduction method: 'locmin'

- New peak picking method: 'mad'

- New 'darkmode' and 'lightmode' plotting options

- Added 'cvApply' methods for new classes

SIGNIFICANT USER-VISIBLE CHANGES

- Image plotting now uses different 'colorkey' legend placed beside
  plot that no longer obscures the image

- All plotting now uses different 'strip' labels placed above plot that
  no longer obscures the plot area

- All plotting now accept hidden 'dark=TRUE' argument to switch to
  plotting in new "dark mode"

- Changed/updated presets for 'presetImageDef' which provides presets
  for 'simulateImage' function

                       Changes in version 2.1.13                        

SIGNIFICANT USER-VISIBLE CHANGES

- Added message when printing a 'SparseImagingExperiment' object
  without un-applied pre-processing steps

BUG FIXES

- Added C routine registration with "C_" prefix for .Call

- Cleaned up '.o' objects in /src created by accident

                       Changes in version 2.1.12                        

NEW FEATURES

- Added 'meansTest' method for linear model-based hypothesis tests of
  mean-summarized images

- Added 'segmentationTest' method for linear model-based hypothesis
  tests of spatially-segmented images

                       Changes in version 2.1.11                        

SIGNIFICANT USER-VISIBLE CHANGES

- Added option to set the probability regularization parameter (p0) to
  'spatialDGMM' method

- Added option to initialize 'spatialDGMM' algorithm with either
  k-means or Gaussian mixture model

                       Changes in version 2.1.10                        

SIGNIFICANT USER-VISIBLE CHANGES

- Changed default colorscale for images to 'viridis'

- Updated 'simulateImage' presets to handle multiple runs

                        Changes in version 2.1.9                        

NEW FEATURES

- Added 'spatialDGMM' method for fitting feature-wise spatially-aware
  Dirichlet Gaussian mixture models

- Added 'predict' method for 'SpatialShrunkenCentroids2'

BUG FIXES

- Fixed 'spatialShrunkenCentroids' classification methods for new
  ('SparseImagingExperiment'-based) classes

- Cleaned up unit tests for statistical methods

                        Changes in version 2.1.8                        

BUG FIXES

- Updated 'PositionDataFrame' initialization due to change in S4Vectors
  [<-,DataFrame behavior

- Updated 'MassDataFrame' initialization due to change in S4Vectors
  [<-,DataFrame behavior

                        Changes in version 2.1.7                        

BUG FIXES

- Updated 'filter' signature due to dplyr changes

                        Changes in version 2.1.6                        

SIGNIFICANT USER-VISIBLE CHANGES

- Added ImmunoOncology biocViews term

                 Changes in version 2.1.5 (2018-12-14)                  

BUG FIXES

- Updated read/write methods for 'matter' filemode changes

                 Changes in version 2.1.4 (2018-12-12)                  

SIGNIFICANT USER-VISIBLE CHANGES

- In 'readImzML' and 'readAnalyze', changed defaults from
  'attach.only=FALSE' to 'attach.only=TRUE'

- In 'readImzML' and 'readAnalyze', changed defaults from
  'as=MSImageSet' to 'as=MSImagingExperiment'

BUG FIXES

- Subsetting large 'SparseImagingExperiment' objects with 'sparse_mat'
  imageData should no longer hang

                 Changes in version 2.1.3 (2018-12-12)                  

NEW FEATURES

- Added 'ResultImagingExperiment' class for results of statistical
  analyses of imaging experiments

- Added 'spatialFastmap' method for 'SparseImagingExperiment'

- Updated 'summarize' to accept a '.group_by' argument

- Added 'simulateSpectrum' and 'simulateImage' functions

SIGNIFICANT USER-VISIBLE CHANGES

- Automatically detect and setup 'layout' for new classes

- Using 'layout' now assumes (row, column) order when creating facet
  plots for new classes (only)

- Updated examples in documentation to use new classes

BUG FIXES

- Fixed bug in 'spatialKMeans' that caused 'spatialFastmap' to fail for
  datasets with fewer features than components

- Fixed bug where 'plot' did not facet over runs of a
  'SparseImagingExperiment' when using 'plusminus'

                 Changes in version 2.1.2 (2018-11-30)                  

SIGNIFICANT USER-VISIBLE CHANGES

- Updated 'spatialKMeans' to use new 'spatialFastmap'

- Updated 'spatialShrunkenCentroids' to use new spatially-aware
  discriminant scores calculation

                 Changes in version 2.1.1 (2018-11-30)                  

NEW FEATURES

- Added 'spatialFastmap' method for performing spatially-aware FastMap
  projection more easily

                 Changes in version 2.0.2 (2018-11-30)                  

BUG FIXES

- Fixed bug in 'reduceDimension.peaks' that caused peak intensities to
  be binned incorrectly

                 Changes in version 2.0.1 (2018-11-14)                  

SIGNIFICANT USER-VISIBLE CHANGES

- Updated 'peakFilter' to check for freq.min >= 1 to accomodate old
  behavior (counts)

[cbaf](https://bioconductor.org/packages/cbaf)
----

                 Changes in version 1.6.0 (2019-04-28)                  

New Features

- heatmapOutput now uses a more accurate algorithm for determining the
  margins.

- Terms updates. Package can recongize more cancers!

[celaref](https://bioconductor.org/packages/celaref)
-------

                       Changes in version 1.1.10                        

UPDATES

- Support for passing hdf5-backed SummarizedExperiment objects
  (internal conversion to sparse)

- Explicitly handle multiple assay() in the summarizedExperiment. Must
  have a named 'counts' assay, or, just the one unnamed assay.

BUG FIXES

- Subsampling per group should actually be used now.

                        Changes in version 1.1.8                        

UPDATES

- Updated vignette with large data handling.

- Doco updates

- make_ranking_violin_plot can pass parameters (e.g. rankmetrics) with
  ...

                        Changes in version 1.1.4                        

UPDATES

- Methods/options to subset large datatsets.

                        Changes in version 1.1.3                        

UPDATES

- None. Version bump for build only.

                        Changes in version 1.1.2                        

UPDATES

- Testing tests

                        Changes in version 1.1.1                        

BUG FIXES

- Internal SingleCellAssay coercion bugfix. Should no longer require
  library(MAST) call to work.

UPDATES

- Partying hard with unit tests woo

                        Changes in version 1.1.0                        

UPDATES

- Option to change the gene ranking metric. Maybe useful for similar
  cell types like PBMCs.

- Use sparse matricies to support larger datasets in less RAM.

                        Changes in version 1.0.1                        

UPDATES

- First bioconductor version

[celda](https://bioconductor.org/packages/celda)
-----

                Changes in version 0.99.34 (2019-04-23)                 

- Minor changes to the vignettes

                Changes in version 0.99.23 (2019-04-10)                 

- Remove pheatmap import

                Changes in version 0.99.22 (2019-04-09)                 

- Package celda, for bi-clustering of single-cell 'omics data.

                 Changes in version 0.99.8 (2019-03-11)                 

- Second submission to Bioconductor

                 Changes in version 0.99.0 (2018-05-15)                 

- First submission to Bioconductor

[CellBench](https://bioconductor.org/packages/CellBench)
---------

                       Changes in version 0.99.10                       

- Accepted into Bioconductor

New Features

- Added new vignettes for Tidyverse Patters and Method Wrappers

Modifications

- fn_arg_seq() now has a .strict argument to check if arguments
supplied are actually used in the function. Default is FALSE,
previously this check is always done, but it failed for functions
that use methods dispatch.  - pipeline_collapse() now has a data.name
argument for if the name of the dataset should be kept in the
pipeline string. Useful if only one dataset is used.  - arrow_sep()
now uses ascii glyphs (only left and right available) instead of
unicode. Unicode arrows fail when ggplots in rmarkdown is compiled
into PDF, a common enough use-case for this to be concerning.

                        Changes in version 0.0.7                        

Modifications

- purrr version number requirement set to (>= 0.3.0) because of
argument name change in partial() - Documentation reorganised to
clean up package documentation index.  - Added landing page for
?CellBench

                        Changes in version 0.0.6                        

New Features

- Added propagation for errors - Added task_error class for errors -
Added print method for task_error objects

                        Changes in version 0.0.5                        

New Features

- Added "Timing" vignette to explain time_methods - Changed
apply_methods() to continue on errors and return error object in
result column

                        Changes in version 0.0.4                        

Breaking Changes

- Changed .name arguments in time_methods() and apply_methods() to
name

New Features

- Added time_methods function - Added set_cellbench_bpparam for more
advanced parallelism options

                        Changes in version 0.0.3                        

New Features

- Implemented parallel application of methods to benchmark_tbl,
previously only worked for dataset lists - Added fn_list constructor
- Added data_list constructor

                        Changes in version 0.0.2                        

Bug Fixes

- Fixed bug in apply_methods() causing it to fail when more than 1
thread is used

Modifications

- Updated introduction vignette to describe multithreading and
function caching

                        Changes in version 0.0.1                        

- Minimal functioning package created - Compliant with
BiocCheck::BiocCheck() and goodpractice::goodpractice()

[CellMixS](https://bioconductor.org/packages/CellMixS)
--------

                       Changes in version 0.99.9                        

- Prepared Bioconductor submission

[ChAMP](https://bioconductor.org/packages/ChAMP)
-----

                       Changes in version 2.13.2                        

- Changed ebayesGSEA function with a new name as ebGSEA, and added KPMT
  function in it.

[CHETAH](https://bioconductor.org/packages/CHETAH)
------

                 Changes in version 0.99.0 (2019-02-28)                 

- Submitted to Bioconductor

[ChIPanalyser](https://bioconductor.org/packages/ChIPanalyser)
------------

                        Changes in version 1.4.1                        

Package Updates

- Updated Plotting functions

- Model quality assessment has been improved

- Noise Filter method for ChIP-seq Pre-processing

[ChIPpeakAnno](https://bioconductor.org/packages/ChIPpeakAnno)
------------

                       Changes in version 3.17.2                        

- fix a bug in toGRanges to accept GTF files.

                       Changes in version 3.17.1                        

- fix the bug in annotatePeakInBatch which can not accept bindingType
  parameter.

[ChIPseeker](https://bioconductor.org/packages/ChIPseeker)
----------

                       Changes in version 1.19.1                        

- add origin_label = "TSS" parameter to plotAvgProf (2018-12-12, Wed)
- https://github.com/GuangchuangYu/ChIPseeker/issues/91

[chromstaR](https://bioconductor.org/packages/chromstaR)
---------

                        Changes in version 1.9.2                        

BUGFIXES

- Bugfix for error when exporting empty peak lists.

[CHRONOS](https://bioconductor.org/packages/CHRONOS)
-------

                       Changes in version 1.11.1                        

- Replaced RBGL::brandes.betweenness.centrality() in pathwayMeasures()
  with igraph::betweenness() since the former is no longer available.

[cleaver](https://bioconductor.org/packages/cleaver)
-------

                 Changes in version 1.21.4 (2018-12-26)                 

- Use "https" in hyperlinks (where possible).

                 Changes in version 1.21.3 (2018-12-26)                 

- Correct cleavage rules for "pepsin" and "pepsin1.3". According to the
  peptidecutter documentation "pepsin1.3" is more specific (cleaves at
  [FL]) than "pepsin" (cleaves [FLWY]); see
  https://github.com/sgibb/cleaver/issues/6 for details. Thanks to Jean
  Manguy <jean@manguy.eu> for reporting this error.

                 Changes in version 1.21.2 (2018-11-03)                 

- Convert vignette from Sweave to rmarkdown.

- Another try to fix links in manual pages to avoid WARNING on Windows.

                 Changes in version 1.21.1 (2018-11-01)                 

- Add my ORCID to the Author field in DESCRIPTION.

- Fix links in manual pages to avoid WARNING on Windows.

[CluMSID](https://bioconductor.org/packages/CluMSID)
-------

                       Changes in version 0.99.13                       

- changed download instructions to BiocManager::install()

                       Changes in version 0.99.12                       

- Bioconductor review II: R, part three

- added MS2spectrum methods for MSnbase/ProtGenerics generics

- minor code changes with no visible effect for users

                       Changes in version 0.99.11                       

- fixed bug in new distanceMatrix(), now coerces numeric IDs to
  character

                       Changes in version 0.99.10                       

- Bioconductor review II: R, part two

- replaced nested for-loop inside distanceMatrix()

                       Changes in version 0.99.9                        

- Bioconductor review II: R, part one

- provided all 'type' argument options as vector

- added '...' argument to specplot()

- changed function to convert MSnbase objects from 'convertSpectrum(x)'
  to 'as.MS2spectrum(x)' / 'as(x, "MS2spectrum")'

- various minor code changes with no visible effect for users

                       Changes in version 0.99.8                        

- Bioconductor review I: DESCRIPTION and NAMESPACE

- added URL and BugReports to DESCRIPTION

- started importing all packages into NAMESPACE

- abandoned "CluMSID_" prefix and renamed respective functions

- changed all function names to camel case starting with lower case
  letters

                       Changes in version 0.99.7                        

- further reduced size of vignettes

                       Changes in version 0.99.6                        

- bugfix in MTBLS vignette

                       Changes in version 0.99.5                        

- modified data import for CluMSID MTBLS vignette

                       Changes in version 0.99.4                        

- reduced size of vignettes

                       Changes in version 0.99.3                        

- added missing dependency for is()

                       Changes in version 0.99.2                        

- started avoiding class() function to check for class

- minor code style change

                       Changes in version 0.99.1                        

- removed .RHistory

- changed import for "show"

- removed invalid character from GC_post.csv in CluMSIDdata which
  caused errors on Linux and OS X

                       Changes in version 0.99.0                        

- Bioconductor submission

                       Changes in version 0.1.14                        

- deleted inst/extdata (now in CluMSIDdata)

                       Changes in version 0.1.13                        

- updated R version dependency to 3.6

- code style changes

                       Changes in version 0.1.12                        

- updated vignette dependencies

                       Changes in version 0.1.11                        

- added missing Suggests

- updated documentation and GC-EI-MS vignette

                       Changes in version 0.1.10                        

- added splitPolarities() function

- added specplot() function

                        Changes in version 0.1.9                        

- Featurelist() and writeFeaturelist() now also work with lists of
  'pseudospectrum' objects

- bugfix for intensity extraction in extractPseudospectra()

- tutorial update

                        Changes in version 0.1.8                        

- added "..." to CluMSID_MDSplot(), CluMSID_HCplot(),
  CluMSID_OPTICSplot() to allow for customisation of plot appearance

                        Changes in version 0.1.7                        

- added polarity slot to 'MS2spectrum' class and respective show()
  generic

- added polarity accessor

- added extraction of polarity information from raw data to
  extractMS2spectra()

                        Changes in version 0.1.6                        

- added a NEWS file

- added "tolerance" argument for m/z tolerance to cossim() and
  distanceMatrix()

- added accessor functions for "MS2spectrum" and "pseudospectrum"
  objects

- added runnable examples for all exported functions

- some formatting and style changes

[clusterProfiler](https://bioconductor.org/packages/clusterProfiler)
---------------

                       Changes in version 3.11.1                        

- asis parameter in [.compareClusterResult (2018-12-24, Mon) -
https://github.com/GuangchuangYu/enrichplot/issues/17

[CNVPanelizer](https://bioconductor.org/packages/CNVPanelizer)
------------

                       Changes in version 1.15.1                        

- Fix test due to change of sample function

[CNVRanger](https://bioconductor.org/packages/CNVRanger)
---------

                        Changes in version 1.0.0                        

- Initial release of the 'CNVRanger' package

[cola](https://bioconductor.org/packages/cola)
----

                       Changes in version 0.99.15                       

- support t-SNE and UMAP.

                       Changes in version 0.99.14                       

- cola_report() supports `mc.cores`

                       Changes in version 0.99.13                       

- partition functions all support `mc.cores`

                       Changes in version 0.99.12                       

- implement get_consensus_matrix() by C++

- add regiester_NMF() and register_SOM()

                       Changes in version 0.99.9                        

- move TEMPLATE_DIR to `.onLoad()`.

                       Changes in version 0.99.5                        

- reduce the number of packages loaded

- improve the package according to comments in
  https://github.com/Bioconductor/Contributions/issues/941#issuecomment-452728439

                       Changes in version 0.99.4                        

- fixed warnings in Rd files

                       Changes in version 0.99.3                        

- remove venneuler from DESCRIPTION

                       Changes in version 0.99.0                        

- Submitted to Bioconductor

[coMET](https://bioconductor.org/packages/coMET)
-----

                 Changes in version 1.15.2 (2018-12-24)                 

- Update vignette (add warning=False)

                 Changes in version 1.15.1 (2018-12-24)                 

- Update script to snp_ensembL

[ComplexHeatmap](https://bioconductor.org/packages/ComplexHeatmap)
--------------

                       Changes in version 1.99.8                        

- add `title_gap` in `Legend()

- fixed a bug of wrong row title spaces when multiple heatmaps are
  vertically concatenated.

- fixed a bug of *_sub_title_side when the heatmap annotation is the
  first/last element in the heatmap list.

- zero-column/row heatmap is supported.

- improved calculation of axis breaks

                       Changes in version 1.99.7                        

- `UpSet()` supports adding complement sets.

- `make_comb_set()`: add `universal_set` and `complement_size`
  arguments.

- axes can be reversed in anno_* functions.

                       Changes in version 1.99.6                        

- adjust the size of heatmap annotations and add testing scripts.

- run multiple times k-means to get a consensus partition.

- `show_heatmap_legend` is set to FALSE if `rect_gp = gpar(type =
  "none")`.

- add `restore_matrix()`.

- add `row_names_centered`/`column_names_centered` arguments to
  `Heatmap()`.

- `gp` in `anno_text()` supports `fill` and `border`.

- `Legend` adds boxplot-style legend.

- adjustment according to annotation extension is improved.

                       Changes in version 1.99.5                        

- add `UpSet()` and some related functions to make Upset plots

- fixed bugs of drawing legends

- add `test_alter_fun()`

- `HeatmapAnnotation()`: fixed a bug for setting `height` when all
  annotations are simple annotations.

- `default_col()`: if the fraction of positive values in the matrix is
  in (0.3, 0.7), the color mapping is symmetric to zero.

- check `NA` values in `anno_boxplot()` and `anno_density()`.

- add `mc.cores` in `densityHeatmap()`.

                       Changes in version 1.99.4                        

- anno_mark() is now calculated in multiple slices.

- oncoPrint(): automatically split the alteration type if the separator
  is one of ";:,|".

- add anno_zoom()

                       Changes in version 1.99.1                        

- add `cluster_row_slices` and `cluster_column_slices` arguments in
  `Heatmap()`.

- fixed a bug when annotation_height with only one annotation

- order of k-means slices are reordered by slice mean of
  `row_reorder`/`column_reorder` if they are provided as vectors.

- remove **rsvg** from Suggests.

                       Changes in version 1.99.0                        

This a major update of the package. The main changes are

- support column split

- support align heatmaps vertically

- add a naive `AnnotationFunction` class to handle annotation functions

- add more annotation functions

[consensusDE](https://bioconductor.org/packages/consensusDE)
-----------

                 Changes in version 1.0.7 (2019-03-01)                  

- export RUV corrected summarized file before model fit

- option to summarize read by transcript, exon or cds added to
  buildSummarized

- added chromosome coordinates to annotation of merged results

- include gtf or txdb as meta-data in summarized object

- added option, htseq_dir, to import HTSEQ count files to
  buildSummarized

- fixed sample_table format check (refactorise)

                 Changes in version 1.0.6 (2019-12-31)                  

- add legend and label option to multi_de_plots

- fixing bug for paired mode model matrix in multi_de_pairs without RUV

- update vignette

                 Changes in version 1.0.5 (2019-12-16)                  

- fixing first column name when writing tables for edgeR, Voom and
  DEseq

- fixed RUV model coefficients carrying through

                 Changes in version 1.0.4 (2019-12-12)                  

- fixed ranking bug for combined data

- adding min_p to merged tables

                 Changes in version 1.0.3 (2019-12-10)                  

- fixed ensembl_annotate option for multi_de_pairs

- added BiocGenerics as depends

                 Changes in version 1.0.2 (2019-12-09)                  

- filter error update to buildSummarized

- add error reporting with incorrect pairs or group numbers

- add option to force_build to buildSummarized

- fix color bug in diag_plots

- fix annotation documentation for multi_de_pairs

- update filter option to permit from summarized experiment

- update vignette

                 Changes in version 1.0.1 (2019-12-07)                  

- Fixed bugs - build gtf from path

[coRdon](https://bioconductor.org/packages/coRdon)
------

                        Changes in version 1.1.3                        

- Implemented codonTable subsetting with "[" and "[[".

- Bug fix: ensured validity of codonTable when KEGG/COG annotations for
  some of the DNA sequences are missing.

[CoRegFlux](https://bioconductor.org/packages/CoRegFlux)
---------

                 Changes in version 0.99.0 (2019-01-15)                 

- Submitted to Bioconductor

[csaw](https://bioconductor.org/packages/csaw)
----

                       Changes in version 1.18.0                        

- Deprecated the type= argument in normOffsets().

- Removed the deprecated consolidateSizes() function.

- Deprecated the use of BPPARAM= in readParam(). Added the BPPARAM=
  argument to all relevant functions.

- Modified csawUsersGuide() to point to the workflow package.

[cTRAP](https://bioconductor.org/packages/cTRAP)
-----

                        Changes in version 1.0.3                        

- Add tag ImmunoOncology to BiocViews

                        Changes in version 1.0.2                        

- Fix comparison against CMap perturbations using gene set enrichment
analysis (the resulting score was the additive inverse of the real
scores)

                        Changes in version 1.0.1                        

- Update title, author names, version and README - Remove biomaRt
dependency - By default, getL1000conditions now shows CMap
perturbation types except for controls - Compare against CMap
perturbations (compareAgainstL1000 function): - Remove "_t" from
resulting column names (as the t-statistic may or may not be used) -
Select p-value adjustment method when performing correlation analyses
(Benjamini-Hochberg is set by default) - Documentation: - Fix
obsolete function calls in function documentation - Hide non-exported
functions from reference PDF manual

[cydar](https://bioconductor.org/packages/cydar)
-----

                        Changes in version 1.8.0                        

- Modified the plotSphere*() functions for more flexibility in colour
  choice.

- Added fix.zero= option in normalizeBatch() for range-based
  normalization.

[cytofast](https://bioconductor.org/packages/cytofast)
--------

                       Changes in version 0.99.0                        

- Submitted to Bioconductor (in development)

[dagLogo](https://bioconductor.org/packages/dagLogo)
-------

                       Changes in version 1.21.4                        

- Fix the bug if there are multiple position in one peptides for
  fetchSequence.

                       Changes in version 1.21.3                        

- Update to grImport2

                       Changes in version 1.21.2                        

- merge the change by Haibo.

                       Changes in version 1.21.1                        

- documentation generated by RoxygenNote.

- add function cleanPeptides.

[dcanr](https://bioconductor.org/packages/dcanr)
-----

                 Changes in version 0.99.0 (2019-03-25)                 

- Submitted to Bioconductor

[debrowser](https://bioconductor.org/packages/debrowser)
---------

                       Changes in version 1.10.7                        

- Better error reporting in data upload

- Better error reporting condition selection using metadata

- alldetected genes matrix downlad fixed in QC section

                       Changes in version 1.10.1                        

- LRT made default in DESeq2.

- Labels in main plots changed.

[deco](https://bioconductor.org/packages/deco)
----

                       Changes in version 0.99.53                       

- Bug fixes.

                       Changes in version 0.99.47                       

- Bug fixes.

- Annotation adapted to new "OrganismDbi" related packages.

- Three new diagnostic (plot) functions.

- Enlarged vignette.

                       Changes in version 0.99.42                       

- Bug fixes.

- Vignette converted into HTML format.

- Accepted in Bioconductor.

                       Changes in version 0.99.0                        

- Submitted to Bioconductor.

[decompTumor2Sig](https://bioconductor.org/packages/decompTumor2Sig)
---------------

                 Changes in version 1.99.1 (2019-04-07)                 

- Changed the procedure which converts lists of individual mutations
  from MPF files into tumor mutation profiles to reduce the memory
  footprint and allow for larger sets of tumors to be read from file.

- When genomes are read from files: remove genomes without SNVs (no
  mutation frequencies)

- Recognize indels for which either the REF or ALT base is specified as
  "-"

- Fix bugs: o Error when quadprog returned a mutation frequency of
  minimally larger than 1 (which is theoretically impossible but can
  probably happen due to the rounding of floating point numbers)

                 Changes in version 1.99.0 (2018-11-10)                 

- When transcription direction is taken into account: exclude mutations
  in regions with overlapping genes of opposing transcription
  directions by default! Previous versions of decompTumor2Sig took the
  approach of pmsignature, using the transcription direction of the
  first gene encountered in the transcript database (which is rather
  arbitrary); excluding these mutations appears more appropriate. The
  old approach can still be used with an additional function parameter.

- Removed pmsignature from the suggested packages (not allowed in
  Bioconductor)

- Reduced the number of tumor genome examples in extdata for faster
  processing and smaller package size (six out of 21 tumors from
  PMID:22608084)

                 Changes in version 1.3.3 (2018-09-24)                  

- Implemented own signature plotting to remove code dependency from
  pmsignature

- Decoupled the following data conversions functions from pmsignature
  code: o getGenomesFromMutFeatData() o getSignaturesFromEstParam()

- Moved some common code to a new internal function for ease of
  maintenance.

- Fixed bugs: o Error when some sequence names where not found in the
  reference genome, e.g., due to different names of decoy sequences. o
  Minor problems with unlikely signatures composed of only the mutated
  base (no flanking bases)

                 Changes in version 1.3.2 (2018-08-29)                  

- Fixed a minor bug in mapSignatureSets() that occurred when the two
  signature sets had the same size.

- Added functions to verify the format of signatures, genomes and
  exposures: o isAlexandrovSet(), isShiraishiSet(), isSignatureSet() o
  sameSignatureFormat() o isExposureSet()

- Moved description of deprecated BiocInstaller in vignette to
  BiocManager.

                 Changes in version 1.3.1 (2018-08-15)                  

- Updated DESCRIPTION file o replaced Author and Maintainer by
  Authors@R o added URL for BugReports

- Changed from manual NAMESPACE and *Rd files to a creation by roxygen2

- Changed function names for reading data from "load..." to "read..."

- Shortened the longest function names o
  getGenomesFromMutationFeatureData -> getGenomesFromMutFeatData o
  getSignatureListFromEstimatedParameters -> getSignaturesFromEstParam

- Used robust sequence creation with seq, seq_len, seq_along instead of
  1:N etc

- Consistent use of native class checking such as is.numeric()

- Implemented wrapper functions for external classes which do not
  provide the necessary accessor functions

- Replaced instances of sapply() and unlist(lapply()) by vapply()

- Updated vignette and made most code chunks runnable

- Added greedy search option to plotExplainedVariance (significant
  speed-up)

                 Changes in version 1.3.0 (2018-07-26)                  

- Submitted to Bioconductor

[deepSNV](https://bioconductor.org/packages/deepSNV)
-------

                 Changes in version 1.99.3 (2013-07-25)                 

Updates

- A few changes to shearwater vignette

- Renamed arguments pi.gene and pi.backgr in makePrior()

Bugfixes

- Fixed bug in bf2Vcf() when no variant is called

                 Changes in version 1.99.2 (2013-07-11)                 

Updates

- Updated CITATION

- Added verbose option to bam2R to suppress output

- Changed mode() to "integer" for value of loadAllData()

Bugfixes

- Fixed bug when only one variant is called in bf2Vcf()

                 Changes in version 1.99.1 (2013-06-25)                 

Updates

- Using knitr for prettier vignettes

- Including shearwater vignette

Bugfixes

- fixed issues with deletions in bf2Vcf()

- makePrior() adds background on all sites

                 Changes in version 1.99.0 (2013-04-30)                 

Updates

- New shearwater algorithm

- Including VCF output through summary(deepSNV, value="VCF")

[DEGreport](https://bioconductor.org/packages/DEGreport)
---------

                       Changes in version 1.19.2                        

- Fix: Error when metadata doesn't contain group column. Thanks to
  @hexaflexa.

- Fix: Update documentation link in degComb.

- Fix: Fix error when factor has extra-levels by @roryk.

- Fix: Fix cutoff summary in degPattern.

                       Changes in version 1.19.1                        

- Feature: Accept data.frame to plot custom clusters.

- Feature: Add cluster consistency plot to degPattern.

[DelayedArray](https://bioconductor.org/packages/DelayedArray)
------------

                       Changes in version 0.10.0                        

NEW FEATURES

- Many improvements to matrix multiplication (%*%) of DelayedMatrix
  objects by Aaron Lun. Also add limited support for (t)crossprod
  methods.

- Add rowsum() and colsum() methods for DelayedMatrix objects. These
  methods are block-processed operations.

- Many improvements to the RleArray() contructor (see messages for
  commits 582234a7 and 0a36ee01 for more info).

- Add seedApply()

- Add multGrids() utility (still a work-in-progress, not documented
  yet)

[DelayedMatrixStats](https://bioconductor.org/packages/DelayedMatrixStats)
------------------

                        Changes in version 1.5.1                        

- Move rowsum() and colsum() to DelayedArray package. Thanks @hpages
(https://github.com/PeteHaitch/DelayedMatrixStats/pull/56).

[DeMixT](https://bioconductor.org/packages/DeMixT)
------

                       Changes in version 0.99.0                        

- New package DeMixT, for cell type-specific deconvolution of
  heterogeneous tumor samples with two or three components using
  expression data from RNAseq or microarray platforms

[derfinder](https://bioconductor.org/packages/derfinder)
---------

                       Changes in version 1.17.3                        

NEW FEATURES

- Add ORCID's following changes at
  http://bioconductor.org/developers/package-guidelines/#description

                       Changes in version 1.17.2                        

BUG FIXES

- Use R's random seeds from version 3.5.0 for the test thanks to
  https://twitter.com/StrictlyStat/status/1103303028751372289

[derfinderHelper](https://bioconductor.org/packages/derfinderHelper)
---------------

                       Changes in version 1.17.3                        

NEW FEATURES

- Add ORCID's following changes at
  http://bioconductor.org/developers/package-guidelines/#description

                       Changes in version 1.17.2                        

BUG FIXES

- Use R's random seeds from version 3.5.0 for the test thanks to
  https://twitter.com/StrictlyStat/status/1103303028751372289

[derfinderPlot](https://bioconductor.org/packages/derfinderPlot)
-------------

                       Changes in version 1.17.2                        

NEW FEATURES

- Add ORCID's following changes at
  http://bioconductor.org/developers/package-guidelines/#description

[DEsubs](https://bioconductor.org/packages/DEsubs)
------

                        Changes in version 1.9.1                        

- Added biocViews tag.

[diffcyt](https://bioconductor.org/packages/diffcyt)
-------

                       Changes in version 1.3.17                        

- Add additional optional arguments for methods 'testDA_edgeR',
  'testDS_limma', and 'testDS_LMM'.

                       Changes in version 1.3.12                        

- Allow selecting columns of 'experiment_info' with character vector of
  column names, when creating design matrix or model formula.

                        Changes in version 1.3.6                        

- Renamed and extended summary table function 'topTable' (previously
  'topClusters').

[diffuStats](https://bioconductor.org/packages/diffuStats)
----------

                        Changes in version 1.3.1                        

- Fixed vignette title and citation - Small changes to documentation
- Version number not malformed anymore

[DiscoRhythm](https://bioconductor.org/packages/DiscoRhythm)
-----------

                 Changes in version 0.99.8 (2019-04-29)                 

- New URL for public server

                 Changes in version 0.99.7 (2019-04-29)                 

- Write intermediate report files to same directory as output report

                 Changes in version 0.99.6 (2019-04-26)                 

- Addition of ncores and port arguments to discoApp

                 Changes in version 0.99.4 (2019-04-24)                 

- Addition of more unit tests on inputs

- Bug fixes

- Addition of more unit tests on inputs

- Bug fixes

                 Changes in version 0.99.3 (2019-04-18)                 

- Bug fixes

                 Changes in version 0.99.2 (2019-04-18)                 

- Use match.arg for argument options

                 Changes in version 0.99.1 (2019-04-09)                 

- Added support for usage of SummarizedExperiment objects

                 Changes in version 0.99.0 (2019-04-04)                 

- Submitted to Bioconductor

[dmrseq](https://bioconductor.org/packages/dmrseq)
------

                 Changes in version 1.3.1 (2018-11-02)                  

- dmrseq now requires that input BSseq objects be ordered (as the
  BSseq() constructor no longer automatically orders loci).

[DNABarcodeCompatibility](https://bioconductor.org/packages/DNABarcodeCompatibility)
-----------------------

                 Changes in version 0.99.8 (2018-11-25)                 

- Notify that the package was approved by Bioconductor

- Fixed all Bioconductor reviewer's comments and updated documentation
  section accordingly. Passed BioCheck 1.18 locally

                 Changes in version 0.99.7 (2018-11-19)                 

- Updated documentation section. Passed BioCheck 1.18 locally

                 Changes in version 0.99.6 (2018-11-19)                 

- Change installation procedure to allow package install on R<3.6.
  Passed BioCheck 1.18 locally

                 Changes in version 0.99.5 (2018-11-19)                 

- Remote BiocCheck imposes R >=3.6. Passed BioCheck 1.18 locally

                 Changes in version 0.99.4 (2018-11-19)                 

- Improved error handling. Passed BioCheck 1.18 locally

                 Changes in version 0.99.3 (2018-11-13)                 

- Fixed R version to 3.5. Passed BioCheck 1.18 locally

                 Changes in version 0.99.2 (2018-11-13)                 

- New package features for the Shiny interface. Passed BioCheck 1.18
  locally

                 Changes in version 0.99.1 (2018-10-15)                 

- Passed BioCheck 1.17 locally

                 Changes in version 0.99.0 (2018-10-15)                 

- Prepared package for submission to Bioconductor repository (BioCheck
  1.16)

[DominoEffect](https://bioconductor.org/packages/DominoEffect)
------------

                        Changes in version 1.3.1                        

UDPATE

- change Ensembl version from 73 to 75

                        Changes in version 1.3.0                        

NEW FEATURES

- DominoEffect in BioC 3.9 development release

[DOSE](https://bioconductor.org/packages/DOSE)
----

                        Changes in version 3.9.4                        

- export parse_ratio (2019-03-29, Tue)

- bug fixed of get_enriched (2019-01-14, Mon) -
https://github.com/GuangchuangYu/clusterProfiler/issues/177

                        Changes in version 3.9.2                        

- mv enrichment vignettes to clusterProfiler-book (2019-01-10, Thu)

                        Changes in version 3.9.1                        

- asis parameter in [.enrichResult and [.gseaResult (2018-12-24, Mon)
- https://github.com/GuangchuangYu/enrichplot/issues/17

[doseR](https://bioconductor.org/packages/doseR)
-----

                       Changes in version 0.99.5                        

- Resolved biocCheck errors, warnings and notes.

- Completed SummarizedExperiment integration.

[DropletUtils](https://bioconductor.org/packages/DropletUtils)
------------

                        Changes in version 1.4.0                        

- Removed read10xMatrix().

- Supported CellRanger v3 output files in read10xMolInfo(),
  read10xCounts(), write10xCounts().

- Modified barcodeRanks() to return a DataFrame with knee/inflection
  estimates in metadata.

- Slight change to random number generation in emptyDrops() to be
  agnostic to number of cores.

[easyRNASeq](https://bioconductor.org/packages/easyRNASeq)
----------

                       Changes in version 2.19.6                        

- More fiddling with documentation link warnings on windows

                       Changes in version 2.19.5                        

- Fixed more documentation link warnings on windows

                       Changes in version 2.19.4                        

- Fixed link warnings on windows

                       Changes in version 2.19.3                        

- Changed to use BiocFileCache for the tests and examples

                       Changes in version 2.19.2                        

- Added the volume and page in the citation

- Added the ImmunoOncology field per request from Bioc core

- Fixed the documentation warnings on Windows

                       Changes in version 2.19.1                        

- Dropped the RangedData support as these are now deprecated.

                       Changes in version 2.18.4                        

- Changed from curl_download to download.file

- Bioc core changes

                       Changes in version 2.18.2                        

- Added the ImmunoOncology field per request from Bioc core

- Added the volume and page in the citation

- Fixed the documentation warnings on Windows

                       Changes in version 2.18.1                        

- Adapted an example as RangedData objects are being deprecated.

[edgeR](https://bioconductor.org/packages/edgeR)
-----

                       Changes in version 3.26.0                        

- read10X() now automatically detects files names from latest
  CellRanger version.

- glmTreat() now checks whether 'contrast' is a matrix with multiple
  columns and uses first column.

- The TMMwzp method has been renamed to TMMwsp, but calls to
  method="TMMwzp" will still be respected. calcNormFactors(object) now
  returns a named vector when 'object' is a matrix, with
  colnames(objects) as the names.

- New 'random' method for zscoreNBinom().

- Add arguments 'log' and 'prior.count' to cpmByGroup() and
  rpkmByGroup().

- Bug fix to filterByExpr().

[ENCODExplorer](https://bioconductor.org/packages/ENCODExplorer)
-------------

                        Changes in version 2.9.0                        

NEW FEATURES

- The ENCODExplorer was split into 2 packages: ENCODExplorer and
  ENCODExplorerData

- The encode_df object is now available through the AnnotationHub

- The functions to interact with the ENCODExplorer API (except the
  search function) were moved to the ENCODExplorerData

[EnhancedVolcano](https://bioconductor.org/packages/EnhancedVolcano)
---------------

                        Changes in version 1.1.4                        

- changed 'colOverride' to 'colCustom'

- added a new 'shape' parameter to control the shape of plotted points

- added a new 'shapeCustom' parameter, akin to functionality supplied
  by 'colCustom' (formerly 'colOverride')

- added functionality to add title, subtitle, and caption (parameters:
  title, subtitle, caption, titleLabSize, subtitleLabSize,
  captionLabSize = 12)

- added functionality to add extra vertical and horizontal lines
  (parameters: hline, hlineType, hlineCol, hlineWidth, vline,
  vlineType, vlineCol, vlineWidth)

- changed 'colConnectors' default from "black" to "grey10"

- added extra parameters for line connectors (parameters:
  typeConnectors, endsConnectors, lengthConnectors)

- 'labhjust' and 'labvjust' are now 'transcriptLabhjust' and
  'transcriptLabvjust', respectively

- user can now specify to draw transcript labels as text and text in
  boxes via 'boxedlabels' (TRUE/FALSE)

                        Changes in version 1.1.2                        

- added functionality to shade specific genes in the plot space
  (parameters: shade, shadeLabel, shadeAlpha, shadeFill, shadeSize,
  shadeBins)

- added new parameters 'labhjust' and 'labvjust' for fine tuning
  position of labels

- suppress warnings about removal of missing values

- now easier to hide the legend via legendVisible = TRUE/FALSE

[EnrichedHeatmap](https://bioconductor.org/packages/EnrichedHeatmap)
---------------

                       Changes in version 1.13.1                        

- update the package according to the new version of ComplexHeatmap (>=
  1.99.0)

- removed `EnrichedHeatmap` class and now the enriched heatmap is
  purely a `Heatmap` class object.

- add `flip_upstream` argument in `normalizeToMatrix()`.

- `pos_line` and `pos_line_gp` use the setting in `EnrichedHeatmap()`
  by default.

- `colSds` -> `matrixStats::colSds`

[enrichplot](https://bioconductor.org/packages/enrichplot)
----------

                        Changes in version 1.3.2                        

- dotplot supports setting x to other variable, e.g. NES (2019-01-10,
Thu) - mv vignette to clusterProfiler-book.

[enrichTF](https://bioconductor.org/packages/enrichTF)
--------

                 Changes in version 1.0.0 (2019-04-30)                  

- Transcription factors enriched in regions based on Bioconductor is
  released.

- Initial release version contains pipeline and all dependents for
  transcription factors enrichment analysis in given regions. User only
  need to provide a BED file and can obtain significant enriched
  transcription factors' p-values of three statistic test.

                 Changes in version 0.99.0 (2019-03-21)                 

- Submitted to Bioconductor

[ensembldb](https://bioconductor.org/packages/ensembldb)
---------

                       Changes in version 2.7.10                        

- Export getGenomeTwoBitFile (issue #95).

- Update vignette on using getGenomeTwoBitFile instead of
  getGenomeFaFile.

                        Changes in version 2.7.9                        

- Add intronsByTranscript method (issue #94).

                        Changes in version 2.7.7                        

- Performance improvement for proteinToTranscript.

                        Changes in version 2.7.6                        

- Increase performance of genomeToTranscript for input GRanges of
  length > 1.

                        Changes in version 2.7.5                        

- Fix class definitions for R >= 3.6.

                        Changes in version 2.7.4                        

- Update perl script to add columns tx_id_version and gene_id_version
  to the transcript and gene database tables.

                        Changes in version 2.7.3                        

- Implement mapIds multiVals = "CharacterList" option (issue #87).

                        Changes in version 2.7.2                        

- Add vignette with coordinate mapping use cases

                        Changes in version 2.7.1                        

- transcriptToGenome returns an empty GRanges if the input range is
  outside of the transcript's sequence instead of throwing an error.

[ensemblVEP](https://bioconductor.org/packages/ensemblVEP)
----------

                       Changes in version 1.26.0                        

- add support for Ensembl release 95/96

[EpiDISH](https://bioconductor.org/packages/EpiDISH)
-------

                        Changes in version 2.0.0                        

- Add CellDMC and hepidish functions.

[epihet](https://bioconductor.org/packages/epihet)
------

                       Changes in version 0.99.0                        

IMPROVEMENTS AND BUG FIXES

- updates for review

[epivizrServer](https://bioconductor.org/packages/epivizrServer)
-------------

                       Changes in version 999.999                       

- This NEWS file is only a placeholder. The version 999.999 does not
  really exist. Please read the NEWS on Github: <URL:
  https://github.com/epiviz/epivizrServer>

[evaluomeR](https://bioconductor.org/packages/evaluomeR)
---------

                Changes in version 0.99.81 (2019-02-26)                 

- Updated DESCRIPTION metadata

                 Changes in version 0.99.8 (2019-02-19)                 

- Removed dependency with orphaned 'fpc' CRAN package

                 Changes in version 0.99.7 (2019-02-19)                 

- The method 'correlations' is now 'metricsCorrelations'

                 Changes in version 0.99.6 (2019-02-19)                 

- Internal improvements according to Bioconductor reviews

                 Changes in version 0.99.5 (2019-02-19)                 

- Removing some @examples so that they do not exceed the time limit of
  Bioconductor check

                 Changes in version 0.99.4 (2019-02-16)                 

- The `correlations` method also uses `SummarizedExperiment` as output
  object.

                 Changes in version 0.99.3 (2019-02-16)                 

- All methods depend on `SummarizedExperiment` as input/output object.

                 Changes in version 0.99.2 (2019-02-15)                 

- SummarizedExperiment can be now processed via `seToDataFrame` method.
  Adding SummarizedExperiment and airway dependencies.

                 Changes in version 0.99.1 (2019-02-11)                 

- Submitted to Bioconductor

[ExperimentHub](https://bioconductor.org/packages/ExperimentHub)
-------------

                        Changes in version 1.9.0                        

SIGNIFICANT USER-VISIBLE CHANGES

- (1.9.2) The Hubs have been updated to use BiocFileCache as a backend
  utility for managing and downloading resources. This allows for file
  specific caching mechanism. This change is a significant user change
  as it changed the default directories for the Hubs
  (AnnotationHub/ExperimentHub) as well as the base directory no longer
  being hidden.

[ExperimentHubData](https://bioconductor.org/packages/ExperimentHubData)
-----------------

                        Changes in version 1.9.0                        

NEW FEATURES

- Added ability to have multiple RDataPaths associated with single hub
  id for strongly associated files (like bam and its bai index file)

[FamAgg](https://bioconductor.org/packages/FamAgg)
------

                       Changes in version 1.11.5                        

- Ensure same ordering of pedigree and provided labels in buildPed
  (issue #19).

                       Changes in version 1.11.4                        

- Fix bug in buildPed with prune = TRUE that resulted in either a
  missing mother or father in some complicated family structures.

                       Changes in version 1.11.3                        

- Fixed bug in buildPed missing out mates in some special cases.

- Fixed bug in buildPed causing differences between individual IDs and
  rownames of the pedigree data.frame.

                       Changes in version 1.11.2                        

- Fix corner case in generating plotting data pedigree.

- Add -family parameter for haplopainter, got lost during refactoring
  (issue #16 aftermath).

                       Changes in version 1.11.1                        

- Fix issue with ids in the pedigree being factors.

[FELLA](https://bioconductor.org/packages/FELLA)
-----

                        Changes in version 1.3.1                        

- Fixed main vignette title to match that of paper

- Fixed `Unicode char' (U+301)` error (had a tilde as `\'\i` in
  bibfile)

- Added citation to BMC article

[fgsea](https://bioconductor.org/packages/fgsea)
-----

                        Changes in version 1.9.6                        

- Fix in log2err calculation

- Small fixes for plotGseaTable

                        Changes in version 1.9.2                        

- Added fgseaMultilevel function for more accurate logPval estimation

[fishpond](https://bioconductor.org/packages/fishpond)
--------

                       Changes in version 0.99.30                       

- added two interaction tests, described in ?swish

- incorporate qvalue package for pvalue, locfdr and qvalue

- added plotMASwish() to facilitate plotting

- wilcoxP is removed, and the mean is used instead

                       Changes in version 0.99.0                        

- fishpond getting ready for submission to Bioc

[flowAI](https://bioconductor.org/packages/flowAI)
------

                       Changes in version 1.12.5                        

- increased pen_valueFS default value

                       Changes in version 1.12.2                        

- fix bug with render function

[flowCL](https://bioconductor.org/packages/flowCL)
------

                       Changes in version 1.21.1                        

- Changed endpoint in man pages, and added ImmunoOncology as a
  biocViews tag.

                       Changes in version 1.21.0                        

- Same as previous. Bumped by BioConductor.

[flowClust](https://bioconductor.org/packages/flowClust)
---------

                       Changes in version 3.21.2                        

- Fixed some tests and rewrote the vignette with Rmarkdown.

[FoldGO](https://bioconductor.org/packages/FoldGO)
------

                 Changes in version 1.1.2 (2019-04-12)                  

- Bug with missing set of background genes fixed

                        Changes in version 1.1.1                        

- Bug with p-value correction before filtering fixed

                 Changes in version 1.1.0 (2018-10-31)                  

- First release

[GAPGOM](https://bioconductor.org/packages/GAPGOM)
------

                 Changes in version 1.0.0 (2018-02-13)                  

Added

- Installation chapter to main vignette.

- Extra tests.

- Basic user input checks.

- Tests for checking precalculated matrices versions.

- Package added to Bioconductor

- Extra script for potential addition in the future (chromosome
  position calculations. File is in inst folder: "chrom_pos.R")

Changed

- Interfacing with GO.db to increase performance (gets converted to
  environment first).

- The way TopoICSim is calculated for genes/genelists, skipping certain
  unnecessary calculations, improving performance greatly.

- `apply` functions.

- How test results are stored -> they are now stored in a list instead
  of seperately.

- Examples, tests and vignette calculations to make sure they don't
  last too long.

- Readme, more verbose.

- Vignettes/Documentation - typo fixes and general small improvements.

Removed

- All global assignments.

- Class accessors (where possible).

Fixed

- "Fixed" news file to plain text format.

- Precalculated topoicsim matrix.

- Compliance with R CMD check.

- Compliance with R CMD BiocCheck.

- Dependency issues.

- Download function of fantom data.

- Fantom 5 functions.

- Calculation time output (actually in seconds now).

                 Changes in version 0.3.0 (2018-11-22)                  

Added

- Re-added data which lncRNApred's paper is based on.

- Support for all AnnotationDbi Key types.

- Vignette for GAPGOM.

- Vignette for benchmarks/performance measuring.

- Term algorithm as exported function.

- Set go data as exported function.

- Some frequently used params can now be set manually (go_data,
  All_Go_pairs).

- Seperate scoring function (If people want to do their own
  enrichment/only get semantic similarity scores).

- Some more arguments for the algorithms to make the user have more
  control.

- Custom gene interface for topoicsim.

Changed

- `DESCRIPTION` file, now adheres to some smaller requirements
  Bioconductor enforces.

- Data preparation functions, they are now updated and more clear.

- Topoicsim for gene and geneset are now 1 generic function
  `topo_ic_sim_genes`.

- Documentation and comments (again).

- Argument parsing for topoicsim.

- Some tests and their results.

- Used arguments in both algorithms.

- Precalculated values are deprecated for this version and thus turned
  off, warning gets shown if it is tried to be used anyway. The values
  will be updated in upcoming releases and probably adhere to only
  certain `org.db` package versions (with a warning shown if incorrect
  versions are used).

- Some `lapply` loops, now for loops, as they were using global
  assignments and this is forbidden/unfavorable in conventions.

- Updated data-raw files, they now work properly for current version
  and are better documented.

Removed

- All forms of parallelization support. It was buggy/inconsistent and
  where it could be implemented insignificant performance gains were
  made.

- Fantom5 example data

- Direct interface from fantom5 file to expset
  `fantom_load_expressionset()`.

Fixed

- Inconsistent result of lncrnapred compared to original algorithm. Now
  is exactly the same.

- Documentation of internal functions is now hidden.

- `data-raw` missing in `.Rbuildignore`.

- `R CMD check` failing because of different lncRNApred results in
  factor indices. Fixed by converting unnecesary factors to chars.

- `R CMD check` failing because of `T` and `F` usage instead of
  `TRUE`/`FALSE`.

- `fantom_to_expset()` not always loading correctly; mouse and human
  had different metadata.

                 Changes in version 0.2.0 (2018-10-20)                  

Added

- TopoICSim performance improvements.

- LncRNApred performance improvements.

- Parallelisation for combined method

- Some small extra options for similarity prediction.

- Precalculated similarity scores between frequent GO terms.

- Data preperation file, describing methods used for generating package
  data.

- Progress bars for TopoICSim.

- Basic tests (for main algorithms only).

- Some small tests for the main algorithms.

- LICENSE file.

Changed

- Return value of main TopoICSim between gene sets to better match
  return value of TopoICSim between two genes.

- All functions are now pre-compiled (except for data_gen.R).

- Imports (per function).

- Comments, a lot of them.

- Implementation of entrez -> goid lookup. (some ids get looked up
  twice)

- `data.frame` subsetting/ddply functions --> replaced with way faster
  data.table alternatives.

- Vectorisation of parts that weren't yet vectorized.

- Changed some loops back to for loops instead of apply --> some
  weren't a good use case for apply.

- Some matrixes now use the Matrix library.

Removed

- Nothing.

Fixed

- R depends version (3.4.4 --> 3.5.1).

- Performance bug in TopoICSim (#4).

- Performance bug in entrez -> goid lookup (#5).

- `fdr` bug (#7).

- Bug where id_select_vector was actually selected for rather than
  filtered for (#8).

- Topoicsim being a total memory hog, it left a lot of unused items in
  ram --> using `gc()` now as an optional argument.

- Newline loading bar bug (#9).

                 Changes in version 0.1.0 (2018-10-04)                  

Added

- Changelog (NEWS.md).

- Better input interface and refactoring towards ExpressionSet based
  system.

- Replacement for current example data by ExpressionSet class of
  fantom5 annotated data (small subset of fantom5).

- Documentation file for previously mentioned new data.

- EntrezID based system for looking up GO ids instead of hardcoded
  translation table without source.

Changed

- Small refactor to switch case in expression_prediction_function().

- A lot of refactoring in expression_prediction_function() and
  enrichment_analysis() to make it work with ExpressionSets.

- Documentation of lncRNApred related functions

Removed

- Old example/expression data and their documentation, this will be
  re-added later once proper source is known.

Fixed

- Bumped up y version number to correct Bioconductor version
  conventions.

- Small typo in zzz.R

- .gitignore mistake (still including .Rdata temp files... These got
  really big :( )

- File typo data_preperation.R --> data_preparation.R

                 Changes in version 0.0.1 (2018-09-25)                  

Added

- Expression metric prediction algorithm made by Rezvan Ehsani and Finn
  Drablos.

- TopoICSim algorithm/measure made by Rezvan Ehsani and Finn Drablos.

- Documentation to the previously denoted algorithms and metrics.

- Default datasets to compute some examples with.

[GateFinder](https://bioconductor.org/packages/GateFinder)
----------

                        Changes in version 1.3.2                        

- fixed conditional statement issue

                        Changes in version 1.3.0                        

- Bug fix re.removing unnecessary suggestion of flowUtils.

[gdsfmt](https://bioconductor.org/packages/gdsfmt)
------

                       Changes in version 1.20.0                        

UTILITIES

- optimize the C implementation of 'packedreal8' using a look-up table

NEW FEATURES

- new data types 'packedreal8u', 'packedreal16u', 'packedreal24u' and
  'packedreal32u'

BUG FIXES

- the compression method 'LZ4_RA.max' does not compress data

- `add.gdsn()` fails if a factor variable has no level

- `add.gdsn(, storage=index.gdsn())` accepts the additional parameters
  from `index.gdsn()`, e.g., 'offset' and 'scale' for packedreal8

                       Changes in version 1.18.1                        

BUG FIXES

- the node name should not contain a slash '/'

[GeneNetworkBuilder](https://bioconductor.org/packages/GeneNetworkBuilder)
------------------

                       Changes in version 1.25.1                        

- update author's email.

[GENESIS](https://bioconductor.org/packages/GENESIS)
-------

                       Changes in version 2.13.7                        

- Remove all monomorphic variants (including all heterozygotes) from
  association test results.

                       Changes in version 2.13.3                        

- Restore option to run pcair without kinship matrix.

                       Changes in version 2.13.2                        

- Add option to use imputed dosage in association tests.

[GENIE3](https://bioconductor.org/packages/GENIE3)
------

                        Changes in version 1.5.2                        

- Adding check for duplicated rownames (gene id/names)

- Bugfix in getLinkList: diag <- NA

[GenomeInfoDb](https://bioconductor.org/packages/GenomeInfoDb)
------------

                       Changes in version 1.20.0                        

SIGNIFICANT USER-VISIBLE CHANGES

- Update genomeMappingTbl.csv, the db used internally by genomeBuilds()
  and family.

BUG FIXES

- Fix bug in seqlevelsStyle() (see
  https://github.com/Bioconductor/GenomeInfoDb/issues/3 for more
  information)

- Fix fetchExtendedChromInfoFromUCSC("hg38"), which got broken by a
  recent change on the UCSC side (see
  https://support.bioconductor.org/p/117808/#117831 for more
  information)


[GenomicDataCommons](https://bioconductor.org/packages/GenomicDataCommons)
------------------

                        Changes in version 1.7.4                        

- slicing available for multiple regions

- added significant new query capabilities

- Simple Somatic Mutations (see ssms and ssm_occurrences)

- Copy Number Variations (see cnvs and cnv_occurrences)

- Gene details (see genes)

[GenomicFeatures](https://bioconductor.org/packages/GenomicFeatures)
---------------

                       Changes in version 1.36.0                        

NEW FEATURES

- Export and document helper functions tidyTranscripts(), tidyExons(),
  and tidyIntrons()

- Add support for mapToTranscripts() to map ranges that span introns

BUG FIXES

- Fix bug in makeTxDbFromGFF() when file is a GFF3File or GTFFile
  object

- Fix bug in makeFeatureDbFromUCSC() (see
  https://github.com/Bioconductor/GenomicFeatures/issues/15)

- Fix makeTxDbFromUCSC() on hg19/refGene and hg38/refGene tables (see
  message of commit ee575ee8 for more information)

[GenomicRanges](https://bioconductor.org/packages/GenomicRanges)
-------------

                       Changes in version 1.36.0                        

NEW FEATURES

- findOverlaps() now supports type="equal" on GRangesList objects

DEPRECATED AND DEFUNCT

- After being deprecated in BioC 3.8, the seqinfo() setter, seqnames(),
  and findOverlaps() are now defunct on RangedData objects.

[GenomicScores](https://bioconductor.org/packages/GenomicScores)
-------------

                        Changes in version 1.8.0                        

USER VISIBLE CHANGES

- Added support to latest release 2.1 of gnomAD MAF data lifted to
  GRCh38 by Ensembl, stored in packages MafDb.gnomAD.r2.1.GRCh38 and
  MafDb.gnomADex.r2.1.GRCh38.

BUG FIXES

- Bugfix in the support to TOPMED MAF data, to discard variants whose
  VCF FILTER is not set to PASS.

[GEOquery](https://bioconductor.org/packages/GEOquery)
--------

                       Changes in version 2.51.1                        

Improvements

- Add formal experimentData slot to GSE records. Of class MIAME. [from
  @vlakam]

[ggtree](https://bioconductor.org/packages/ggtree)
------

                       Changes in version 1.15.6                        

- remove getChild, getChild.df, getParent, getParent.df, getSibling,
getAncestor and getAncestor.df, instead use child, parent, sibling
and ancestor methods implemented in tidytree and treeio (2019-01-30,
Wed) - remove get.offspring.df and get.offspring.tip and instead use
tidytree::offspring (2019-01-28, Mon) - facet_widths function to set
relative widths of facet panels (2019-01-28, Mon) - the output is
ggplotify::as.ggplot(grid_object), so it is not the original ggtree
object.

                       Changes in version 1.15.5                        

- bug fixed of theme_tree2 (2019-01-14, Mon) -
https://github.com/GuangchuangYu/ggtree/issues/218 - mv rescale_tree
to treeio (2019-01-11, Fri)

                       Changes in version 1.15.4                        

- reimplement MRCA as a method inherited from tidytree (2019-01-10,
Thu) - mv vignettes to treedata-book

                       Changes in version 1.15.3                        

- move reroot method to treeio package and rename to root
(2018-12-28, Fri) - bug fixed for setting branch.length="none" in
unrooted layouts (2018-12-26, Wed) - bug introduced in
https://github.com/GuangchuangYu/ggtree/pull/201

                       Changes in version 1.15.2                        

- compatible with tibble v=2.0.0 (2018-11-29, Thu)

                       Changes in version 1.15.1                        

- now revts also reverse the branch column (2018-11-11, Sun) -
https://groups.google.com/d/msgid/bioc-ggtree/50765a34-53e5-44d2-bb8f-6d20a11fa890%40googlegroups.com
- better msg when taxa name not consistent in sequence and tree files
with msaplot. (2018-11-07, Wed) -
https://github.com/GuangchuangYu/ggtree/issues/172#issuecomment-436585370

[GladiaTOX](https://bioconductor.org/packages/GladiaTOX)
---------

                       Changes in version 0.99.0                        

- Submitted to Bioconductor

[GNET2](https://bioconductor.org/packages/GNET2)
-----

                       Changes in version 0.99.13                       

- Added heuristic split of regression trees by K-means.

                       Changes in version 0.99.6                        

- Improve the performance for module building with R code, now
  build_moduleR has same level of performance with its C++ version.

                       Changes in version 0.99.5                        

- Submitted to Bioconductor.

                       Changes in version 0.99.1                        

- Initial release.

[GOfuncR](https://bioconductor.org/packages/GOfuncR)
-------

                        Changes in version 1.3.4                        

USER-LEVEL CHANGES

- update GO-graph to weekly release (version 27-Mar-2019)

- fix issue with "~" as abbreviation for home directory in custom
  ontology path

                        Changes in version 1.3.3                        

USER-LEVEL CHANGES

- update GO-graph (version 26-Mar-2019)

[gpuMagic](https://bioconductor.org/packages/gpuMagic)
--------

                 Changes in version 0.99.6 (2019-04-19)                 

- Add more function documentation

                 Changes in version 0.99.5 (2019-04-18)                 

- Fix grammar issue

                       Changes in version 0.1.19                        

- Add "customized opencl code" vignette

                       Changes in version 0.1.16                        

- Implement memory pool

                 Changes in version 0.1.7 (2019-02-15)                  

- Fix the incorrect local thread number

                 Changes in version 0.1.3 (2019-02-15)                  

- Adding Mac builder flag

- Bugs fixing

                 Changes in version 0.1.2 (2019-02-13)                  

- Trying to fix the build error

                 Changes in version 0.1.1 (2019-02-13)                  

- Trying to fix the build error

                 Changes in version 0.1.0 (2019-02-13)                  

- Submitted to Bioconductor

[graper](https://bioconductor.org/packages/graper)
------

                 Changes in version 0.99.0 (2018-11-07)                 

- Submitted to Bioconductor

[graphite](https://bioconductor.org/packages/graphite)
--------

                 Changes in version 1.29.3 (2019-04-17)                 

- Updated all pathway data.

[gtrellis](https://bioconductor.org/packages/gtrellis)
--------

                       Changes in version 1.15.2                        

- fixed warnings in Rd files.

                       Changes in version 1.15.1                        

- Use new definition of `Legends-class`.

[Guitar](https://bioconductor.org/packages/Guitar)
------

                 Changes in version 1.99.0 (2019-04-24)                 

- The following changes have been made primarily in this release:

- 1. Add a confidence interval for the curve

- 2. Modify GuitarcoordinateTXDB to remove the generation of the bin
  interval of the guitar coordinate system.

- 3. Sampling the input site information file.

- 4. Change the coordinate structure and normalize the abscissa of the
  coordinate system to the interval 0~1.

- 5. Change the composition and calculation of weight in the density
  drawing method.

- 6. Open various parameters in the drawing process to the user for
  selection.

[Gviz](https://bioconductor.org/packages/Gviz)
----

                       Changes in version 1.28.0                        

BUG FIXES

- Fixed issue with hg19 genome version and BiomartGeneRegionTrack: user
  needs to provide biomart object pointing to grch37.ensembl.org

- Fixed IdeogramTrack to use user provided title

[HCABrowser](https://bioconductor.org/packages/HCABrowser)
----------

                        Changes in version 1.0.0                        

- HCABrowser released

[HDF5Array](https://bioconductor.org/packages/HDF5Array)
---------

                       Changes in version 1.12.0                        

NEW FEATURES

- Add 'prefix' arg to save/loadHDF5SummarizedExperiment()

- Add quickResaveHDF5SummarizedExperiment() for fast re-saving after
  initial saveHDF5SummarizedExperiment(). See
  ?quickResaveHDF5SummarizedExperiment for more information.

- Add h5mread() as a faster alternative to rhdf5::h5read(). It is now
  the workhorse behind the extract_array() method for HDF5ArraySeed
  objects. This change should significantly speed up block processing
  of HDF5ArraySeed-based DelayedArray objects (including HDF5Array
  objects).

[HiCBricks](https://bioconductor.org/packages/HiCBricks)
---------

                        Changes in version 1.1.5                        

- HDF file locking option has been removed.

- data.table warning is now disabled while reading entire matrices.

- Vignette documentation has been converted to Rmd and BiocStyle html
  output.

[HilbertCurve](https://bioconductor.org/packages/HilbertCurve)
------------

                       Changes in version 1.13.3                        

- set `interpolate = FALSE` in `grid.raster()`

- improved `hc_map()`

                       Changes in version 1.13.2                        

- fixed warnings in Rd files

                       Changes in version 1.13.1                        

- update with new definition of legends

[hipathia](https://bioconductor.org/packages/hipathia)
--------

                 Changes in version 1.5.1 (2019-02-21)                  

- Adding function node_color

- Including parameter adjust in node_color_per_de function. Set TRUE as
  default, previously considered as FALSE

- Allowing node_colors_per_de to accept any experimental design
  accepted by limma::makeContrasts.

- Removing non-visible function compute_difexp

- Adding GO IDs to GO annotations and making them the rownames of the
  go_vals matrix

- Adding get_go_names function, to translate GO Ids to GO function
  names

- Adapting to new RData names in AnnotationHub, which include the
  version of the data in hpAnnot files.

- Restricting GO names to 50 characters.

- Adding group.by slot to metaginfo objects.

[HIREewas](https://bioconductor.org/packages/HIREewas)
--------

                 Changes in version 1.0.1 (2018-11-18)                  

- Corrected the title of vignettes

[HPAanalyze](https://bioconductor.org/packages/HPAanalyze)
----------

                         Changes in version 1.1                         

- Changes in version 1.1.5 + Reformatted NEWS file.

- Changes in version 1.1.4 + Added the query vignette.

- Changes in version 1.1.3 + Fixed issues where hpaVis and hpaXml give
  faulty output. + All hpaVis functions as well as hpaListParam and
  hpaSubset now use the bundled dataset by defauft if no data argument
  is specifed. + All hpaVis functions now have default for all of their
  arguments and will create a warning message when the defaults are
  used. + The in-depth vignette was rewritten, and new vignettes added
  for ease of use. + Updated the themes of hpaVis functions for better
  clarity. + Updated documentations. + Hex sticker is now available.

- Changes in version 1.1.1 + HPAanalyze now uses openxlsx for
  hpaExport() instead of XLConnect.

- Changes in version 1.1.0 + Starting devel for Bioconductor 3.9

[hpar](https://bioconductor.org/packages/hpar)
----

                        Changes in version 1.25                         

Changes in version 1.25.1

- Update to HPA release 18.1 <2019-01-21 Mon>

Changes in version 1.25.0

- New release for Bioconductor devel 3.9

[HTSFilter](https://bioconductor.org/packages/HTSFilter)
---------

                       Changes in version 1.23.1                        

- -- Add ImmunoOncology to biocViews in DESCRIPTION

[ideal](https://bioconductor.org/packages/ideal)
-----

                        Changes in version 1.8.0                        

New features

- plot_ma gains an additional parameter, labels_repel, for better
placing the labels on the features to mark - The pairwise correlation
plots can now use logarithmic scale in the axes, use smaller subsets
of the data for quicker inspection, and resizes the correlation info
proportionally to its intensity - The id types can now be chosen
among the keytypes of the corresponding annotation packages (which
still need to be installed when ideal is launched). Other input
fields that specify id types also behave in a similar manner (e.g. in
the Signature Explorer tab). This caused a problem for scenarios
where common id types such as ENSEMBL are not available, like in
Arabidopsis (where ids are provided often as TAIR) - thanks to Marc
Galland for picking this up in
https://github.com/federicomarini/ideal/issues/1 - The Signatures
Explorer tab also has a fully fledged tour for first-time users,
together with a collapsible help panel to describe its functionality
in brief - The zoomed MA-plot gains a new widget to control the
labels size for the names of the genes - It is possible to export the
input data together with the results in a combined
SummarizedExperiment object, which can be seamlessly fed into iSEE
(http://bioconductor.org/packages/release/bioc/html/iSEE.html). This
leverages a new function, wrapup_for_iSEE, which is available and
exported from ideal.

Other notes

- An information box has been added to provide detailed information
on the required input formats - Added notification to specify how to
install the airway package for demonstration purposes if not already
available - The Signatures Explorer tab is now displayed in a
conditional panel, i.e. only when the required inputs are provided

[infercnv](https://bioconductor.org/packages/infercnv)
--------

                 Changes in version 0.99.8 (2019-04-20)                 

- Fix missing labels on heatmap due to change in how axis() handles
  overlaps.

- Improve auto thresholding of heatmap colors to still work when using
  stronger denoise settings.

- Updated some examples to use tempfile() for output.

- Fix duplicate piece of code that produced an error when using
  plot_steps.

                 Changes in version 0.99.0 (2019-03-15)                 

- Submitted to Bioconductor

[InPAS](https://bioconductor.org/packages/InPAS)
-----

                       Changes in version 1.15.2                        

- update author's email.

                       Changes in version 1.15.1                        

- add the new biocViews tag "ImmunoOncology"

[INSPEcT](https://bioconductor.org/packages/INSPEcT)
-------

                       Changes in version 1.12.1                        

- major update of INSPEcT, which now handles additional functions and
  analysis, such as: (i) extracting RNA dynamics from only total
  RNA-seq data, (ii) assessing differential regulation between steady
  states with an improved statistics, (iii) assessing the burden of the
  processing rate step in the responsiveness of mature RNA.

- minor updates regard the estimation of the variance from replicates,
  that now uses the package "DESeq2" (when read counts are available)
  or "plgem" (when only expression data is available) ######## the
  INSPEcT datasets older than this version are NOT COMPATIBLE anymore
  ############## ######## and must be updated using the function
  "convert_ids", available in the ############## ######## "inst" folder
  of the package. Usage: ############## >
  source(file.path(system.file(package='INSPEcT'), 'convert_ids.R')) >
  new_ids <- convert_ids(old_ids, degDuringPulse=c(FALSE,TRUE))

[IntEREst](https://bioconductor.org/packages/IntEREst)
--------

                        Changes in version 1.8.0                        

NEW FEATURES

- interest() and interest.sequential() functions support the
  possibility to define other parameter settings of BamFile() function
  of Rsamtools package e.g.  "qnamePrefixEnd" and "qnameSuffixStart".

- annotateU12() can return the PWM match scores (if requested). The
  hybrid AT-AG and GT-AC U12 types can be set to be ignored. The
  introns/exons coordinates can be filtered according to their map to
  another set of coordinates.

BUG FIXES

- referencePrepare() just gives warning and continue without annotating
  gene names (when annotateGeneIds=TRUE parameter is set) if annotating
  with gene names is not possible.

- interest() used to return Error if single end mapped reads did not
  existed at all (despite being requetsed). This has been corrected.

[IRanges](https://bioconductor.org/packages/IRanges)
-------

                       Changes in version 2.18.0                        

NEW FEATURES

- Add some methods for CharacterList derivatives (nchar, substring,
  substr, chartr, toupper, tolower, sub, gsub, grepl).

BUG FIXES

- Fix unlist() on a SimpleRleList object of length 0

- Fix drop() for FactorList derivatives

- Fix removed rownames upon replacing in a SplitDataFrameList

DEPRECATED AND DEFUNCT

- Deprecate RangedData objects.  The use of RangedData objects has been
  discouraged in favor of GRanges or GRangesList objects since BioC
  2.12, that is, since 2014. Developers are required to migrate their
  code to use GRanges or GRangesList instead of RangedData objects (the
  GRanges and GRangesList classes are defined in the GenomicRanges
  package).

- Several RangedData methods are now defunct (after being deprecated in
  BioC 3.8): - score, score<-, lapply, within, countOverlaps; -
  coercions from list, data.frame, DataTable, Rle, RleList,
  RleViewsList, IntegerRanges, or IntegerRangesList to RangedData.

[iSEE](https://bioconductor.org/packages/iSEE)
----

                        Changes in version 1.3.9                        

- Add ORCID identifiers.

- Fix subscript error when tables receive a selection.

- Fix panel names in panel organization selectize input.

- Control the application of panel organization updates using an action
  button.

- Fix child replotting upon lasso close.

- Fix code reporting for zero-length DataFrame

                        Changes in version 1.3.8                        

- Support multiple selections.

- Avoid Javascript error with check group conditional.

                        Changes in version 1.3.7                        

- Add ImmunoOncology in biocViews.

                        Changes in version 1.3.6                        

- Control point size.

                        Changes in version 1.3.5                        

- Additional information during the default tour.

                        Changes in version 1.3.4                        

- Fix panel organization selectize.

                        Changes in version 1.3.3                        

- Update default tour steps to match updated user interface.

- Parse quote symbols literally in default tour steps.

- Fix name-to-index conversion of feature names for heat map panel.

                        Changes in version 1.3.2                        

- Move panel organization to modal with selectize to control panel
  display and ordering, remove sidebar.

- Add control of width and height of new panels.

- Enable voice control.

- Refactor internal functions.

                        Changes in version 1.3.1                        

- Fix invalid row index sent from tables in RStudio browser.

- Fix initialization of search fields for tables that are initialized
  with an incoming selection.

- Fix constant field name.

[IsoCorrectoR](https://bioconductor.org/packages/IsoCorrectoR)
------------

                        Changes in version 1.0.3                        

- added automated code testing with testthat

- fixed typo in 'biocViews' field of DESCRIPTION

                        Changes in version 1.0.1                        

- Fixed an issue where correction would abort with an error message if
  chemical formulas in the molecule file contain Phosphorus (thanks to
  Mathieu Bousquet for reporting this)

- Fixed an issue that could lead to an error if measurement IDs in the
  measurement file are a suffix of other measurement IDs (e.g.,
  cysteine_1 and homocysteine_1)

- The display of error messages in the R console (they always appear in
  the log file) is now no longer suppressed if verbose is set to FALSE
  in the IsoCorrection() function call

[IsoCorrectoRGUI](https://bioconductor.org/packages/IsoCorrectoRGUI)
---------------

                        Changes in version 0.0.1                        

- prepare for Biocoductor submission

[IsoformSwitchAnalyzeR](https://bioconductor.org/packages/IsoformSwitchAnalyzeR)
---------------------

- NEWS file couldn't be formatted. Please see the link to the NEWS file on the package landing page for complete details 
 
[isomiRs](https://bioconductor.org/packages/isomiRs)
-------

                       Changes in version 1.11.3                        

FEATURES

- Combined samples and create new object. Useful for technical
  replicates.

- Add isoAnnotate function to get more information for each isomiR.

- Add isomiR naming and sequences table in metadata of object.

- Remove reference sequences from isoPlot figures.

- Support filtering isomiRs with low % of importance (statistically
  supported).

- Support isoPLot to use only certain isomiRs.

                       Changes in version 1.11.2                        

FIX

- Fix NEWS format.

- Fix documentation typos.

FEATURES

- Filter isomiRs by importance ratio.

- Allow to plot selected isomiRs with isoPlot function.

[karyoploteR](https://bioconductor.org/packages/karyoploteR)
-----------

                       Changes in version 1.10.0                        

NEW FEATURES

- Added new color managements functions including is.color and
  transparent and improved the existing ones.

- Added new functions to color data elements based on different
  features: chromosome, position, overlapping of regions...

- Improved gene and transcript plotting functions. Added functions to
  automatically get gene symbols from OrgDb objects. Improved
  transcript and gene positioning. Added a function to merge all
  transcripts of a gene into one.

- kpPlotBigWig can now adjust the ymax value based on the global
  maximum, per chromosome or taking into account only the visible
  region.

SIGNIFICANT USER-VISIBLE CHANGES

- It is now posible to add labels to the right-hand side of chromosomes
  with kpAddLabels

- Improved documentation and vignette

- makeGenesDataFromTxDb does not need a karyoplot anymore. Changed
  parameter order.

BUG-FIXES

- It is now possible to use cex in plotKaryotype to specify chromosome
  name sizes

- Multiple small bug fixes

[KEGGprofile](https://bioconductor.org/packages/KEGGprofile)
-----------

                       Changes in version 1.25.1                        

- Fix bugs in download_KEGGfile function.

[KinSwingR](https://bioconductor.org/packages/KinSwingR)
---------

                 Changes in version 1.0.4 (2019-04-24)                  

- correction to pseudo count in scorePWM and viewPWM

- added option of motif coloring by AA properties to viewPWM

- update vignette

                 Changes in version 1.0.3 (2018-11-12)                  

- Bug in network output from swing fix

                 Changes in version 1.0.2 (2018-11-09)                  

- Making viewPWM visible

                 Changes in version 1.0.1 (2018-11-05)                  

- Fixed bugs - matrix division in swing() and labels in buildPWM

- input_data in outputs of ScoreSequences - to trace trimmed sequences

[limma](https://bioconductor.org/packages/limma)
-----

                       Changes in version 3.40.0                        

- Major rewrite of arrayWeights() to improve speed and stability. The
  arrayWeightsSimple() function has been removed and its functionality
  incorporated into arrayWeights(). A new 'var.group' argument has been
  added to simplify specification of the variance design matrix. The
  weights are now squeezed slightly towards unit and a new argument
  'prior.n' has been added to control the prior weight with which the
  array weights are squeezed. arrayWeights() now chooses between the
  REML and gene-by-gene algorithms automatically by default. REML is
  chosen when there are no prior weights or missing values and
  gene-by-gene is used otherwise. arrayWeights() now checks for and
  skips over any genes with zero residual variances.

- voom() now checks for experiments that have no replication. In this
  case it now returns weights that are all 1, with an informative
  warning but no error. Two references have been added to the voom help
  page.

- classifyTests.Rd now focuses on nested F-tests. The function FStat()
  is now removed. It was a convenience wrapper for classifyTestsF()
  with fstat.only=TRUE but is not used often enough to justify itself.
  classifyTestsP() is no longer exported as it is not intended to be
  called by users. The function classifyTestsT() is now removed, having
  been superceded by more sophisticated multiple testing methods.

- contrasts.fit() now removes any test statistics or p-values found in
  the fit object. This avoids any potential mis-match between
  coefficients and test statistics if eBayes() has been run on the
  object previously.
  
  More details have been added to the warning about approximations in
  the contrasts.fit() help page.

- The old function ebayes(), which was deprecated a year ago in favor
  of eBayes(), is now removed.

- The fitted() and residuals() methods for MArrayLM objects now give an
  informative error message if the object contains contrasts instead of
  the original coefficients.

- Rewrite Filtering chapter of User's Guide.

- Add Research Fields to the biocViews entry in the package DESCRIPTION
  file.

[maftools](https://bioconductor.org/packages/maftools)
--------

                        Changes in version 2.0.0                        

SIGNIFICANT USER-LEVEL IMPROVEMENT

- Oncoplot, oncostrip and coOncoplot code has been migrated from
  ComplexHeatmap to base graphics. Default values for some arguments
  have been changed.

- Changed default value for mafObj argument in subsetMaf to TRUE.
  Issue: #235

- All ggplots are migrated to base graphics for consistancy

- TCGA mutations per sample used for tcgaCompare are updated to more
  stable MC3 cohort from deprecated broad firehose.

IMPROVEMENTS

- tcgaCompare now accpets comparing against user specifc cohorts.
  Issue: #229

- Add missing genes and samples to oncoplot even when they are not
  altered. Issue: #254

- Added MTRNR2L8 protein structure to pfam database. Issue: #255

- Added addtionalFeature argument to oncoplots. Issue: #270

- Added log scale option for summary and oncoplots. Issue: #273

- Made protein domain db compatible with R version < 3.5 Issue: #276

- Added draw `draw_titv`, and `exprsTbl` argument to oncoplot

- Additional edits with TMB visualization and additionalFeatures Issue:
  #289

- Cooncoplot improvements Issue: #296

BUG FIX

- Oncoplot annotationOrder bug fix. Issue: #293

- Update code for AnnovarToMaf. PR #288, #234

- fixed handling of Status in survival. PR #274

- Fix swapped confidance intervals in mafCompare function Issue: #269

- update docs, and vectorized code Issue: #266 #267

- Improve oncoplot annotations for numeric class Issue: #263 and #203

- In annovarToMaf read the .tsv file considering the header. PR #259

- Fix legend title order. Issue: #253

- GISTIC sample size issue fix. Issue: #249

- Bug fix for oncoplots with CNV data. Issue: #240

- Better error handling in plotApobecDiff with no SMGs. Issue: #232

- Bug fix that caused signature extraction to terminate in case of zero
  mutations. Issue: #228

- Bug fix due to missing levels contingency tables in clinical
  enrichment. Issue: #225

- Add showTumorSampleBarcodes to PlotOncogenicPathways. Issue: #214

- Cleanup docs for Oncoplot. Issue: #207

- Implemented `isNumeric` and `groupAnnotationBySize` to improve
  sorting. Issue: #207

- Fix ComplexHeatmap update. Issue: #205 #210

- plotmafSummary addStats bug fix Issue: #204

- Rainfall plot data.table null fix. Issue: #201

- Support non human BSgenome. Issue: #197

- SampleID bug fix. Issue: #153

[MAGeCKFlute](https://bioconductor.org/packages/MAGeCKFlute)
-----------

                        Changes in version 1.2.2                        

- Revise ReadRRA function.

- Add ReadsgRRA function to read sgrna_summary in MAGeCK RRA results.

- Add sgRankView function to visualize the rank of sgRNA targeting top
  selected genes.

- Remove bugs in FluteMLE and FluteRRA.

- Add additional visualization functions in FluteRRA.

- Add EnrichedGeneView to visualize the core enriched genes in enriched
  pathways.

- Integrate WikiPathways, PID terms and EHMN pathways into enrichment
  functions.

- Export all the hidden functions.

                        Changes in version 1.2.1                        

- Add VolcanoView to show positive and negative selected genes.

- Improve EnrichedView and EnrichedGSEView function.

- Add QC figures in vignettes.

- Merge MsigDB, KEGG, and GO genesets together and integrate them into
  package.

[matter](https://bioconductor.org/packages/matter)
------

                       Changes in version 1.9.11                        

BUG FIXES

- Fixed documentation link to 'base::lapply'

                 Changes in version 1.9.10 (2018-12-22)                 

SIGNIFICANT USER-VISIBLE CHANGES

- Vectors can be used as 'virtual_mat' rows/columns now

                 Changes in version 1.9.9 (2018-12-22)                  

SIGNIFICANT USER-VISIBLE CHANGES

- Constructor for 'drle' now allows forcing delta = 0

- Updated display descriptions in 'show' methods

- Check for NULL dimensions when subsetting 'virtual_mat' and
  'sparse_mat' classes

                 Changes in version 1.9.8 (2018-12-18)                  

BUG FIXES

- Check for NULL dimensions in 'matter_arr' subsetting

                 Changes in version 1.9.7 (2018-12-18)                  

BUG FIXES

- Fixed bug in atoms() where 'index_offset' and 'index_extent' would
  overflow when creating long 'matter_vec' objects

                 Changes in version 1.9.6 (2018-12-14)                  

SIGNIFICANT USER-VISIBLE CHANGES

- Subsetting matter matrices, arrays, and data frames now checks for
  the correct number of dimensions

- Linear indexing support for 'matter_mat' and 'matter_arr'

BUG FIXES

- Class checks now use R_check_slot_etc() in C so subclasses of
  'matter' objects should inherit correct behavior

- Fixed other indexing issues in subsetting + assignment

                 Changes in version 1.9.5 (2018-12-14)                  

SIGNIFICANT USER-VISIBLE CHANGES

- Updated documentation with `matter-datatypes`, size of data types are
  now explicitly defined

- Changed 'filemode' slot to be a factor with levels c('r', 'w', 'rw')
  instead of a string

- Data access now uses C++ fstream instead of C

BUG FIXES

- On Windows, accessing data elements in large files with byte offets
  larger than MAX_INT now works

- On Windows, data modes of 'long' and 'ulong' are now correctly 64-bit
  integer types

- Assignment of negative values now checked for 'ulong'

                        Changes in version 1.9.4                        

NEW FEATURES

- Added %% and %/% delayed operations from Arith group

- Added & and | delayed operations from Logic group

- All Ops group generics now supported as delayed operations

- Added 'lapply' and 'sapply' functions for 'matter_list'

- Support for BiocParallel in 'apply', 'lapply', and 'sapply'

BUG FIXES

- Fix floating point exception error when dividing by integer 0L in
  delayed division operations

                        Changes in version 1.9.3                        

NEW FEATURES

- Added matrix multiplication for 'sparse_mat' matrices

- Added matrix multiplication for 'virtual_mat' matrices

                 Changes in version 1.9.2 (2018-11-12)                  

BUG FIXES

- Renamed internal class 'bytes' to 'num_bytes' due to Rcpp

                 Changes in version 1.9.1 (2018-11-12)                  

BUG FIXES

- Removed BiocGenerics 'lengths' from NAMESPACE

                 Changes in version 1.8.2 (2018-12-13)                  

BUG FIXES

- Fixed large file offset bug on Windows (again)

                 Changes in version 1.8.1 (2018-12-11)                  

BUG FIXES

- Fixed bug causing large files to read incorrectly on Windows

[mbkmeans](https://bioconductor.org/packages/mbkmeans)
--------

                 Changes in version 0.99.0 (2019-03-08)                 

- This is the pre-submission version.

- Change default in mbkmeans to use a batch size equal to 5% of the
  data.

- Change default in mbkmeans to max 100 iterations.

- Fix blocksize() to work if impossible to calculate available ram.

[Melissa](https://bioconductor.org/packages/Melissa)
-------

                       Changes in version 0.99.0                        

- First package release. News and details of updates will be added in
  future releases.

[metagene2](https://bioconductor.org/packages/metagene2)
---------

                 Changes in version 1.1.0 (2019-04-05)                  

- Submitted package to Bioconductor.

[metagenomeSeq](https://bioconductor.org/packages/metagenomeSeq)
-------------

                        Changes in version 1.25                         

- Added 'wrenchNorm' function

- Added option to use IHW as p-value adustment method in 'MRcoefs'

- Modified 'expSummary' slot in 'MRexperiment' object to be of class
  'list' instead of 'environment'

[MetaNeighbor](https://bioconductor.org/packages/MetaNeighbor)
------------

                        Changes in version 1.3.0                        

- Added memory efficient version of MetaNeighbor and MetaNeighborUS

- Added supporting functions for memory efficient versions

[MetCirc](https://bioconductor.org/packages/MetCirc)
-------

                 Changes in version 1.13.1 (2019-04-04)                 

- check if package passes R CMD build and R CMD check without any error
  messages and vignette can be run without any errors

[methimpute](https://bioconductor.org/packages/methimpute)
----------

                        Changes in version 1.5.2                        

BUGFIXES

- Removed runnable example from exportMethylome() because of memory
  issues with data.table::fwrite() since Bioconductor 3.9.

[methylGSA](https://bioconductor.org/packages/methylGSA)
---------

                        Changes in version 1.1.5                        

- Parallel option is supported

                        Changes in version 1.1.1                        

- Fixed typos in vignettes - Included citation

[methylKit](https://bioconductor.org/packages/methylKit)
---------

                        Changes in version 1.9.4                        

NEW FUNCTIONS AND FEATURES

- calculateDiffMeth: introduce method "midPval" for fast testing of
  different methylation if there is one sample per group, contributed
  by @zellerivo

IMPROVEMENTS AND BUG FIXES

- calculateDiffMeth: - fix rare case issue with invalid p-values on
  calculateDiffMeth call, that caused "qvalue" p-value adjustment to
  fail https://github.com/al2na/methylKit/issues/90 - let user choose
  between new "midPval" or older "fast.fisher" implemtation for
  statistical test used to determine the methylation differences
  https://github.com/al2na/methylKit/issues/96 - introduce p-value
  lookup table to reduce multiple testing of identical configurations
  https://github.com/al2na/methylKit/issues/96 - reduce code
  duplication and overhead by introducing S3 functions to be called by
  S4 methods

- update vignette with section about usage of methylKit for analysis of
  affinity- and array-based methylation assays

- methSeg: if object with multiple chromosomes is used, remove chroms
  with less than two ranges

- unite: fix issue where digits were written in scientific notation
  https://github.com/al2na/methylKit/issues/130

- getCorrelation: fix issue where getCorrelation produces matrix with
  NA when using "min.per.group"
  https://github.com/al2na/methylKit/issues/137

                        Changes in version 1.9.3                        

IMPROVEMENTS AND BUG FIXES

- getCorrelation: reduce code duplication and overhead by introducing
  S3 functions to be called by S4 methods

- fread.gzipped: rewrite building of shell command with system2 to
  mimic R.Utils::gunzip

- improve validity checks for methylDB objects

- pool: check length of treatment and number of sample is same

- headTabix: use fread.gzipped only when user wants to get more than
  1e6 rows

- test-1-read: check if methylRawDB is same when directory is given or
  not, i.e. test wether directory affects object

- methylRaw: check for content in tabix file, break if not

- calculateDiffMeth: dealing with case of full model not being
  different from reduced model; print warning and set NaN to 1

- pool: check length of treatment and number of sample is same

- improve validity checks for methylDB objects

                        Changes in version 1.9.2                        

IMPROVEMENTS AND BUG FIXES

- fread.gzipped: update function to check compression based on file
  content and allow to decompress .bgz files

- headTabix: use fread.gzipped directly on the tabix file fixes issue
  https://github.com/al2na/methylKit/issues/141

                        Changes in version 1.9.1                        

IMPROVEMENTS AND BUG FIXES

- add methylRawListDB constructor fixes
  https://github.com/al2na/methylKit/issues/134

- update validation functions for methylRaw/methylRawDB: check for
  single sample id

- unite: write S3 functions for class methylRawList and methylRawListDB
  Fixes https://github.com/al2na/methylKit/issues/132

[methylPipe](https://bioconductor.org/packages/methylPipe)
----------

                       Changes in version 1.17.1                        

- updated or fixed the following functions: + getCposDensity : fixed
  the warning message in checking for NA in a list

[MetNet](https://bioconductor.org/packages/MetNet)
------

                 Changes in version 1.1.2 (2019-04-16)                  

- check that MetNet passes all builds without any errors or warnings

                 Changes in version 1.1.1 (2019-04-03)                  

- correct typo in vignette

[MGFR](https://bioconductor.org/packages/MGFR)
----

                        Changes in version 1.9.2                        

- Added an argument class.vec to the function getMarkerGenes.rnaseq()
  to provide the classes of samples in the data matrix

- Changed default of the argument annotate to FALSE

- Fixed an error caused when no markers are found

[microbiome](https://bioconductor.org/packages/microbiome)
----------

                 Changes in version 1.5.4 (2018-11-20)                  

- Fixed bug in merge_taxa2

- Fixed bug in Pielou's evenness

- New functions: readcount, bfratio, aggregate_top_taxa

- rewritten aggregate_taxa and moved the top argument in the new
  function aggregate_top_taxa

- plot_composition function: new options for sample.sort and otu.sort

- Added Phylum level to taxonomy tables in example data sets * New
  function: dominant

- The diversities function is now replaces by alpha function. The alpha
  is more generic and can return also other alpha diversity indices.

- plot_frequencies function now only returns the ggplot object

- Renamed the global function into alpha

- Renamed arguments rarity.detection and rarity.prevalence into
  detection and prevalence in the rarity function

- Added Chao1 index in richness function

- In atlas1006 data set, pseudocount of +1 in otu table has been
  removed to facilitate comparison with sequencing data sets and to
  avoid confusion

- In atlas1006 data set, only a single replicate per subject-time
  combination is chosen (at random)

- New function collapse_replicates has been added

- Abundance matrices (otu tables) for all example data sets now
  starting from 0 without pseudocount

- Changed the default for the detection argument in the richness
  function to detection=0

- removed rarity.threshold and rarity.prevalence options from the alpha
  function

- Various minor fixes; see github commits for many more details o Color
  order in plot_landscape legend now follows the factor levels of the
  col argument o Fixed example in remove_samples

[minfi](https://bioconductor.org/packages/minfi)
-----

                        Changes in version 1.29                         

- v1.29.4 Fixed a bug in detectionP where the function always returned
  NA for a 27k array. Reported by Laurenz Holcik.

- v1.29.4 Fixed a bug in preprocessNoob where the function would error
  when supplied an input data set with only 1 sample (when using
  default arguments). Reported by Matt Oldach.

[miRSM](https://bioconductor.org/packages/miRSM)
-----

                        Changes in version 1.1.4                        

- Update imports <2019-04-19, Fri>

                     Changes in version 1.1.1-1.1.3                     

- Update miRSM.R <2019-03-08, Fri>

[miRsponge](https://bioconductor.org/packages/miRsponge)
---------

                        Changes in version 1.9.3                        

- Update the Groundtruth <2019-03-13, Wed>.

                        Changes in version 1.9.2                        

- Rename the miRsponge package called miRspongeR, and add zzz.R file to
  points users to the new package ‘miRspongeR’ <2019-03-07, Thur>.


[Modstrings](https://bioconductor.org/packages/Modstrings)
----------

                 Changes in version 0.99.0 (2019-01-15)                 

- Submitted to Bioconductor

[MOFA](https://bioconductor.org/packages/MOFA)
----

                 Changes in version 0.99.0 (2019-01-25)                 

- Submitted to Bioconductor

[motifStack](https://bioconductor.org/packages/motifStack)
----------

                       Changes in version 1.27.5                        

- change plotMotifLogo to grob.

                       Changes in version 1.27.4                        

- Update authors information.

- change the d3-tip background color.

- adjust the fontsize to remove the gaps.

- change the title position.

                       Changes in version 1.27.3                        

- use grImport2 instead grImport to avoid the installation of
  ghostscript.

                       Changes in version 1.27.2                        

- add knitr in vignettes to pass window's builder

                       Changes in version 1.27.1                        

- try to pass window's builder

[mpra](https://bioconductor.org/packages/mpra)
----

                        Changes in version 1.5.3                        

- Update main paper citation.

- Update vignette with better examples for allelic studies.

- Add mpraSetAllelicExample to available data.

[MSnbase](https://bioconductor.org/packages/MSnbase)
-------

                         Changes in version 2.9                         

Changes in version 2.9.6

- remove plain NEWS file now that R supports NEWS.md

Changes in version 2.9.5

- Add isolationWindowLowerMz and isolationWindowUpperMz methods
<2019-04-15 Mon> - Add filterPrecursorMz method <2019-04-15 Mon> -
Add filterIsolationWindow method <2019-05-16 Tue>

Changes in version 2.9.4

- Remove NAnnotatedDataframe <2019-04-10 Wed>

Changes in version 2.9.3

- Fix bug in ms2df (order of fcols could be scambled) <2018-12-04
Tue> - Fix bug in MSnSet constructor (see issue #386) <2018-12-07
Fri> - Add filterPolarity method (see issue #388, feature requeste by
@zeehio) <2018-12-10 Mon>

Changes in version 2.9.2

- Fix problem that impute(x, method="knn") sets the seed (issue
#380).  Contributed by Constantin Ahlmann-Eltze <2018-11-16 Fri> -
Conserve spectra names in combineSpectra (see #379) <2018-11-15 Thu>
- Fix bug in writeMSData - see #383 <2018-11-22 Thu>

Changes in version 2.9.1

- Add clean,Chromatograms (issue #354) <2018-11-07 Wed> - Add
as,Spectra,list as,Spectra,MSnExp and as,MSnExp,Spectra (issue #370)
<2018-11-07 Wed> - Add filterMsLevel,Spectra <2018-11-07 Wed> - Fix
regression bug in plot,Spectrum1 <2018-11-14 Wed>

Changes in version 2.9.0

- New version for Bioc 3.9 devel

[MSnID](https://bioconductor.org/packages/MSnID)
-----

                       Changes in version 1.17.1                        

- Broadened the scope of the tool to ImmunoOncology - that is finding
  rare events that reflected in protein sequence.

[msPurity](https://bioconductor.org/packages/msPurity)
--------

                       Changes in version 1.9.12                        

- Troublshoot mac "[MSData::Spectrum::getMZIntensityPairs()] Sizes do
  not match" error

                       Changes in version 1.9.11                        

- Bug fix for createMSP - now handles metada with duplicate grpids

                       Changes in version 1.9.10                        

- Documentation updates

                        Changes in version 1.9.9                        

- NEW FUNCTION: createDatabase + Updated version of create_database
  that uses an updated schema

- NEW FUNCTION: spectralMatching + Updated version spectral_matching
  that allows for more flexibility

- Vigenettes and documentation update

                        Changes in version 1.9.8                        

- NEW FUNCTION: filterFragSpectra (for purityA objects) + LC-MS/MS
  filtering of spectra (prior to averaging)

- NEW FUNCTIONS: averageIntraFragSpectra, averageInterFragSpectra,
  averageAllFragSpectra (for purityA objects) + LC-MS/MS averaging and
  filtering functionality

- NEW FUNCTION: createMSP + Create msp files from purityA objects where
  XCMS features have been linked to fragmentation spectra

- Updated create_database and spectral matching to have the option to
  use averaged fragmentation spectra

                        Changes in version 1.9.2                        

- Bug fix for groupPeaks and groupPeaksEx (the ppm argument was not
  working and there was inconsistent behaviour with larger datasets).
  Thanks to Elliot for spotting (https://github.com/litepalmer)

- Updated documentation for spectral matching

[mzR](https://bioconductor.org/packages/mzR)
---

                       Changes in version 2.17.4                        

- Update documentation, pointing to MSnbase rather than directly use
  low-level classes. <2019-04-17 Wed>

                       Changes in version 2.17.3                        

- Extract isolation window from mzML files (issue #193): data.frame
  returned by header gains columns "isolationWindowTargetMZ",
  "isolationWindowLowerOffset", "isolationWindowUpperOffset".

                       Changes in version 2.17.2                        

- Fix bug #185: Error in R_nc4_open: Too many open files

[NADfinder](https://bioconductor.org/packages/NADfinder)
---------

                        Changes in version 1.7.4                        

- Fix the warning of documentation of computeLibSizeChrom.

                        Changes in version 1.7.1                        

- Update author email.

[NBAMSeq](https://bioconductor.org/packages/NBAMSeq)
-------

                 Changes in version 0.99.1 (2019-02-13)                 

- Package created

[NBSplice](https://bioconductor.org/packages/NBSplice)
--------

                        Changes in version 1.1.1                        

- DESCRIPTION: Adding ImmunoOncology as a BiocView tag

[netboost](https://bioconductor.org/packages/netboost)
--------

                  Changes in version 0.99 (2019-04-02)                  

- Final submission changes for Bioconductor.


[NetPathMiner](https://bioconductor.org/packages/NetPathMiner)
------------

                       Changes in version 1.19.1                        

- Added a new function to create a metabolite network.

- Added a new function to reindex networks using vertex attributes.

- Removed compiler warnings.

- Fixed various bugs in network manipulation.

[ngsReports](https://bioconductor.org/packages/ngsReports)
----------

                 Changes in version 0.99.8 (2019-04-03)                 

- **This is a breaking change!** Significant changes to most functions
  have been implemented

- Extracting modules is no longer performed by individual functions,
  but is now performed using the function getModule

- FastqcFileLists objects are no longer defined and the class
  FastqcFile has been made into the private class .FastqcFile

- FastqcDataList objects have names attributes and can now be subset
  using names

- The function fileName has been renamed as fqName to clarify that it
  refers to the underlying Fastq file for a Fastqc report

- All log file imports are now handled by the single function
  importNgsLogs

- Most plot functions have been renamed with shorter names, e.g.
  plotOverrepresentedSequences is now plotOverrep

- The FastQC version is now obtained using fqcVersion not Version

- The functions genomes() and transcriptomes() have been removed and
  this information is now obtained using gcAvail(object, type)

- The function getGcDistn() has been renamed as estGcDistn() to avoid
  any confusion with getGC() which works on TheoreticalGC objects.

                 Changes in version 0.99.1 (2019-02-03)                 

- Removed Remotes from DESCRIPTION

- Added getGcDistribution to enable calculation of GC Content
  Distributions from FastaFiles

                 Changes in version 0.99.0 (2019-02-01)                 

- Submitted to Bioconductor

- Shiny App has been moved to a separate package, located at
  https://github.com/UofABioinformaticsHub/shinyNgsReports

[OmaDB](https://bioconductor.org/packages/OmaDB)
-----

                 Changes in version 1.99.3 (2019-04-15)                 

- update format of NEWS file

                 Changes in version 1.99.2 (2019-03-13)                 

- re-license under GPLv3

- updated ipython notebook with examples from manuscript to new release
  of OMA browser

- improve package according to bioconductor guidelines: o improve
  README and DESCRIPTION o various cleanups of code and data

                 Changes in version 1.99.1 (2018-12-28)                 

- fixes bioconductor CI warnings on missing links in deprecated
  functions

                 Changes in version 1.99.0 (2018-12-21)                 

- major refactoring of codebase: o getData split into getProtein,
  getGenome, getOMAGroup o renamed getAnnotation -> annotateSequence o
  renamed getXref -> searchSequence

- getProtein allows to retrieve many protein ids at once

- improved documenation by a lot.

[OmicsLonDA](https://bioconductor.org/packages/OmicsLonDA)
----------

                       Changes in version 0.99.10                       

- Fix FeatureSpline plots in README and vignette

                       Changes in version 0.99.9                        

- Fix all CRAN and Bioconductor checks

                       Changes in version 0.99.1                        

- Submitted to Bioconductor

[OncoSimulR](https://bioconductor.org/packages/OncoSimulR)
----------

                 Changes in version 2.13.2 (2019-03-18)                 

- changes in behavior of sample (see NEWS and
  https://bugs.r-project.org/bugzilla/show_bug.cgi?id=17494) were
  leading to failures of some tests. Using RNGversion in some tests.

                 Changes in version 2.13.1 (2019-02-07)                 

- bumped to one over the BioC-3.9 version

[onlineFDR](https://bioconductor.org/packages/onlineFDR)
---------

                        Changes in version 1.1.3                        

MODIFICATIONS

- updated unit tests for LOND, bonfInfinite and LORDdep

                        Changes in version 1.1.2                        

MODIFICATIONS

- updated unit test for LORD

                        Changes in version 1.1.1                        

MODIFICATIONS

- added the LORD++ procedure for online FDR control

- updated references

- updated vignette

[oposSOM](https://bioconductor.org/packages/oposSOM)
-------

                        Changes in version 2.0.3                        

- Check for detected singleton spot modules implemented

                        Changes in version 2.0.2                        

- csv output for international spreadsheet format.

                        Changes in version 2.0.1                        

- Colored edges in sample correlation networks & minor fixes.

[ORFik](https://bioconductor.org/packages/ORFik)
-----

                        Changes in version 1.3.7                        

SIGNIFICANT USER-VISIBLE CHANGES

- mapToGRanges is now much faster and uses much less memory. A c++
  reimplementation of pmapFromTranscripts from GenomicFeatures.

[OVESEG](https://bioconductor.org/packages/OVESEG)
------

                 Changes in version 0.99.2 (2019-01-14)                 

- trigger version bump

                 Changes in version 0.99.1 (2019-01-14)                 

- modifiy vignette to fix TIMEOUT build in MAC

                 Changes in version 0.99.0 (2019-01-10)                 

- Submitted to Bioconductor

[PAST](https://bioconductor.org/packages/PAST)
----

                Changes in version 0.99.19 (2019-04-22)                 

- bug fixes

- improvements to speed

                Changes in version 0.99.18 (2019-04-22)                 

- documentation updates and minor code clean-up

                Changes in version 0.99.17 (2019-04-18)                 

- updated pathways functions to reflect changes across other methods

                Changes in version 0.99.16 (2019-04-18)                 

- updated SNP function and added rtracklayer dependency

                Changes in version 0.99.15 (2019-04-17)                 

- renamed LD function and changed added column name specification

                Changes in version 0.99.14 (2019-04-16)                 

- renamed merge function and changed method of describing input type

                Changes in version 0.99.13 (2019-04-16)                 

- addressed description issues

                Changes in version 0.99.12 (2019-04-15)                 

- updated manual

                Changes in version 0.99.11 (2019-04-15)                 

- updated manual

                Changes in version 0.99.10 (2019-04-15)                 

- added types of data

                 Changes in version 0.99.9 (2019-04-15)                 

- separated Shiny app

                 Changes in version 0.99.8 (2019-04-15)                 

- corrected author list format

                 Changes in version 0.99.7 (2019-04-15)                 

- corrected author list format

                 Changes in version 0.99.6 (2019-04-11)                 

- corrected documentation

                 Changes in version 0.99.4 (2019-04-08)                 

- updated vignette

- added filtering options to R Shiny app

                 Changes in version 0.99.3 (2019-04-03)                 

- Corrected NEWS

- added vignette description

                 Changes in version 0.99.2 (2019-02-28)                 

- Removed extra code

                 Changes in version 0.99.1 (2019-02-28)                 

- Closed connections after finishing with them

                 Changes in version 0.99.0 (2019-02-28)                 

- Submitted to Bioconductor

[PathoStat](https://bioconductor.org/packages/PathoStat)
---------

                        Changes in version 1.8.1                        

- Update data uploading

- Modify visualization

- Add biomarker tab

[pathview](https://bioconductor.org/packages/pathview)
--------

                       Changes in version 1.23.3                        

- fix error and warning cause by geneannot.map() function. change
  if(in.type == out.type) to if(all(in.type == out.type)). change
  biocLite installation to BiocManager::install. make similar changes
  in sim.mol.data() function.

- update NAMESPACE file on importFrom lines on AnnotationDbi and
  BiocManager.

[pathwayPCA](https://bioconductor.org/packages/pathwayPCA)
----------

                       Changes in version 0.99.1                        

2019-02-01

We are submitting to Bioconductor soon, so we are resolving as many
of the BiocCheck() ERRORs, WARNINGs, and NOTEs. See
https://github.com/gabrielodom/pathwayPCA/issues/64

                       Changes in version 0.98.0                        

2018-12-13

See the issues within the "Bioconductor Submission" and "Vignette
Work" milestones for detailed descriptions of these changes, their
discussion, and motivation:
https://github.com/gabrielodom/pathwayPCA/issues

- Standardized naming conventions to UpperCamelCase for consistency -
Changed class name pathwaySet -> pathwayCollection - Added functions
SubsetPathwayData, getPathPCLs, LoadOntoPCs, write_gmt, getSampleIDs,
CheckSampleIDs, getTrimPathwayCollection, getPathwayCollection,
CheckPwyColl, CheckAssay, and print and subset methods for
pathwayCollection objects - Added sample ID requirements for input
assays and phenotype data.  Changed the phenotype required class from
anything to a data frame.  - Made the lars implementation a touch
more robust. If the algorithm fails to converge, then we default back
to regular SVD - The AESPCA function can also return pathway-specific
vanilla PCA.  Also, we inspected the parametric p-values compared to
the permutation-based p-values. They congruence is nearly perfect for
all of the data sets we tested. Thus, we changed the default p-value
estimation method for AESPCA to be parametric.  - Method functions
(*_pVals()) now return a list with the p-values data frame, list of
PC vectors (in a data frame), and list of loading vectors (in a data
frame). The getPathPCLs function will subset this list to return the
PCs, loadings, and administrivia of the method PCA output for a
single pathway - added the June homo sapiens pathway collection from
WikiPathways: wikipwsHS_Entrez_pathwayCollection - Lily wrote a
vignette geared to show off the diverse functionality of the package,
so this is the new main vignette. I broke off the plots from chapter
4 into their own chapter. The five vignettes I wrote are now
supplemental chapters - updated vignettes and website

                     Changes in version 0.0.0.9000                      

2018-03-20

- Added a NEWS.md file to track changes to the package.  - Built
website.

[pcaExplorer](https://bioconductor.org/packages/pcaExplorer)
-----------

                       Changes in version 2.10.0                        

New features

- Added extra parameters to topGOtable to offer more control on the
method used, and the option to correct the p-values for multiple
testing (via the Benjamini-Yekutieli procedure) - pca2go has now an
option to return (early) a list with vectors of genes with high
loadings - Better preview of the uploaded data with modal dialog
windows triggered by buttons which appear once corresponding inputs
are available - Improved notification system: in addition to the
progress bar, info that all input is correctly specified, or suggest
optional annotation loading - Added flexibility to select separator
for each of the uploadable files - The pairwise correlation plots can
now use logarithmic scale in the axes, use smaller subsets of the
data for quicker inspection, and resizes the correlation info
proportionally to its intensity - The sample to sample heatmap
supports additionally manhattan and correlation-based distances -
There is a new vignette with a detailed quick start, "Up and running
with pcaExplorer", specifying how the workflow with pcaExplorer can
look like, demonstrated on the airway dataset - In the Instructions
panel, we added buttons to access the fully rendered documentation,
either local or online if e.g. deployed to a server. Related to this,
pcaExplorer has a new parameter, runLocal, to control this behavior -
An additional parameter, annopkg, has been added to pca2go() to
override the behavior with the organism parameter (this is useful
when the name of the annotation package is not conform to the
classical org.Xx.eg.db, e.g. for Arabidopsis Thaliana); a detailed
use case has been added in the main vignette

Other notes

- The computing of the required objects now requires the explicit
action on the dedicated button, and the tooltip informs the user on
what steps are taken (including normalization) - An information box
has been added to provide detailed information on the required input
formats - Added notification to specify how to install the airway
package for demonstration purposes if not already available - Added
startup message upon loading the package - The content in the
Instructions tab is now contained in collapsible elements - The file
formats accepted by pcaExplorer are now specified both in the
vignette text, as well as in the app at runtime - The content of the
Instructions tab is now more compact, containing the rendered "Up and
running with pcaExplorer" vignette. The full vignettes can be
accessed via buttons in the same panel - Added instructions to
install phantomJS via the webshot package - would raise an error when
previewing the report

[PCAtools](https://bioconductor.org/packages/PCAtools)
--------

                        Changes in version 1.0.0                        

- added new function getLoadings, which returns component loadings

[PepsNMR](https://bioconductor.org/packages/PepsNMR)
-------

                 Changes in version 1.1.3 (2019-03-11)                  

NEW FEATURES

- Added the ZeroFilling function

- Made the printed text for each pre-processing function optional

                 Changes in version 1.1.2 (2019-03-08)                  

BUG CORRECTION

- Debugged Bucketing and InternalReferencing for n = 1

[PGA](https://bioconductor.org/packages/PGA)
---

                       Changes in version 1.13.3                        

- Fix bug in function reportGear.R

                       Changes in version 1.13.2                        

- Fix bug in function reportGear.R

[phantasus](https://bioconductor.org/packages/phantasus)
---------

                        Changes in version 1.3.5                        

- FGSEA tool

- Cache + static files optimizations

- Dataset permanent link feature

                        Changes in version 1.3.4                        

- Switch to fgseaMultilevel

                        Changes in version 1.3.3                        

- Custom order for annotation

- Revamped Charts tool

- AnnotationDB optimization

                        Changes in version 1.3.2                        

- PCA plot, GSEA plot and Chart tool can be exported to SVG

- Group by in column context menu

- Dataset open history

- Annotate dataset by AnnotateDB

- Switch column datatype in context menu

- GSEA plot uses general palette to color heatmap\annotation

- Minor bug fixes

[piano](https://bioconductor.org/packages/piano)
-----

                        Changes in version 2.0.0                        

NEW FEATURES

- Added a function exploreGSAres() for interactively exploring the
  result object returned by runGSA(). This opens a Shiny app in a
  browser where users can view the different results tables and figures
  and get detailed information on specific gene-sets and genes.

- Added a function networkPlot2() that is an improvement to the old
  function networkPlot().

DOCUMENTATION

- Added a new vignette focused on gene-set analysis

[Pigengene](https://bioconductor.org/packages/Pigengene)
---------

                 Changes in version 1.9.26 (2019-04-24)                 

Bug Fixes

- In the compute.pigengene function, if a module has only one gene, its
  name is now not omitted in the csv file.

                 Changes in version 1.9.24 (2019-04-24)                 

Changes in existing functions

- An object of pigengene-class can now have a heavyToLow attribute.

                 Changes in version 1.9.20 (2019-04-12)                 

Changes in existing functions

- In the compute.pigengene function, the columns of Data and names of
  modules can now differ.

                 Changes in version 1.9.14 (2019-02-12)                 

Changes in existing functions

- The dOrderByW argument is now added for the compute.pigengene
  function.

                 Changes in version 1.9.8 (2018-11-22)                  

Changes in existing functions

- All of the learn.bn function can now be set from one.step.pigengene
  through bnArgs.

                 Changes in version 1.9.4 (2018-11-16)                  

Changes in existing functions

- gene.mapping can now output multiple conventions.

[pipeFrame](https://bioconductor.org/packages/pipeFrame)
---------

                 Changes in version 1.0.0 (2019-04-30)                  

- Bioinformatics R pipeline framework based on Bioconductor is
  released.

- Initial release version contains pipeline framework function elements
  including base class for each pipeline step, genome related data
  setting, global configuration, configuration initialization and
  pipeline flow chart management.

- Currently this framework is applied to package enrichTF.

                 Changes in version 0.99.0 (2019-03-21)                 

- Submitted to Bioconductor

[polyester](https://bioconductor.org/packages/polyester)
---------

                       Changes in version 1.99.3                        

- NB function now exported

- note that version 1.99.3 on GitHub was version 1.1.0 on Bioconductor.

                       Changes in version 1.99.2                        

- bug fix in fragment generation (last 2 bases of transcript were never
  sequenced)

[PoTRA](https://bioconductor.org/packages/PoTRA)
-----

                Changes in version 0.99.93 (2019-04-16)                 

- Updated the PoTRA algorithm

                Changes in version 0.99.92 (2019-04-16)                 

- Updated the PoTRA algorithm

                Changes in version 0.99.91 (2019-04-11)                 

- Removed examples section from the manual

                Changes in version 0.99.90 (2019-04-09)                 

- Added examples section to the manual

                Changes in version 0.99.89 (2019-04-09)                 

- Updated the usage section of the Rd document

                Changes in version 0.99.88 (2019-04-09)                 

- Updated the alias field in the Rd document

                Changes in version 0.99.87 (2019-04-09)                 

- Updated the lazy load field in the description file

                Changes in version 0.99.86 (2019-04-09)                 

- Edited the manual and updated the code in the .R files

                Changes in version 0.99.85 (2019-04-08)                 

- Edited the manual

                Changes in version 0.99.84 (2019-04-08)                 

- Edited documentation

                Changes in version 0.99.83 (2019-04-08)                 

- Edited script and exported function

                Changes in version 0.99.82 (2019-04-08)                 

- Removed R file

                Changes in version 0.99.81 (2019-04-08)                 

- Edited the manual

                Changes in version 0.99.80 (2019-04-07)                 

- Edited the manual

                Changes in version 0.99.79 (2019-04-07)                 

- Edited the manual

                Changes in version 0.99.78 (2019-04-07)                 

- Edited the manual

                Changes in version 0.99.77 (2019-04-06)                 

- Made minor edit to the PoTRA algorithm

                Changes in version 0.99.76 (2019-04-06)                 

- Made minor edits to the vignette

                Changes in version 0.99.75 (2019-04-06)                 

- Added a data object to the package

                Changes in version 0.99.74 (2019-04-03)                 

- Updated the package install instructions in the vignette

                Changes in version 0.99.73 (2019-04-03)                 

- Improved the code in the vignette and edited the manual

                Changes in version 0.99.72 (2019-04-02)                 

- Updated the landing page abstract

                Changes in version 0.99.71 (2019-04-02)                 

- Added devtools install script for repmis in vignette

                Changes in version 0.99.70 (2019-04-01)                 

- Accepted to Bioconductor

                 Changes in version 0.99.0 (2019-02-21)                 

- Submitted to Bioconductor

[pqsfinder](https://bioconductor.org/packages/pqsfinder)
---------

                         Changes in version 2.0                         

NEW FEATURES

- The qsfinder algorithm is ~100 times faster on real-world sequences
  than before. The main idea of the optimization is to stop the
  exhaustive search when we are sure there can't be found better PQS.
  The optimization is applied only in non-overlapping mode.

- New boolean argument called "deep". Once set to TRUE, pqsfinder will
  disable optimizations and run full exhaustive search.

SIGNIFICANT USER-VISIBLE CHANGES

- Default minimal PQS score was increased from 26 to 52. The score 52
  shows the best balanced accuracy on G4 sequencing data provided by
  Chambers et al. 2015.

BUG FIXES

- Fixed resolution of overlapping sequences. When multiple same-scoring
  PQS overlap, the shortest one is chosen.

[pram](https://bioconductor.org/packages/pram)
----

                       Changes in version 0.99.17                       

- pram is in Bioconductor

                       Changes in version 0.99.13                       

- First release

[profileplyr](https://bioconductor.org/packages/profileplyr)
-----------

                 Changes in version 0.99.0 (2019-03-29)                 

- Submitted to Bioconductor

[pRoloc](https://bioconductor.org/packages/pRoloc)
------

                        Changes in version 1.23                         

Changes in version 1.23.4

- Remove plain NEWS now that R supports NEWS.md

Changes in version 1.23.3

- Link to TAGM F1000 workflow <2019-04-11 Thu> - Add ref to TAGM
F1000 in CITATION and README <2019-04-11 Thu> - Update biomart
interface data and AnnotationParams-related code <2019-04-12 Fri>

Changes in version 1.23.2

- Add mcmc helper functions.  - New logPosteriors accessor for
MAPParams object.  - New plotConsProfiles function, contributed by
Tom Smith (see PR #131).  - Update TAGM vignette, refering to
workflow <2019-03-14 Thu>

Changes in version 1.23.1

- Add note about missing data in QSep man <2018-10-31 Wed> - Add
mcmc-helper documentation <2018-11-04 Sun> - Fixing numerical
instability in Cholesky decomposition (see #124) <2018-11-30 Fri> -
Add option to display or not display grid in plot2D <2018-12-12 Wed>
- mrkHClust now uses mrkConsProfiles and returns the hclust object -
see issue #130 for details and background <2018-12-19 Wed>

Changes in version 1.23.0

- New version for Bioc 3.9 devel

[PureCN](https://bioconductor.org/packages/PureCN)
------

                       Changes in version 1.14.0                        

NEW FEATURES

- support for copynumber package and its multisample segmentation

- beta support for PSCBS weighting

- support for gene symbol filtering in FilterCallableLoci.R (e.g.
  --exclude "^HLA")

- added segmentationHclust function that clusters provided segmentation
  using log2-ratio and B-allele frequencies

- min.target.width and small.targets in preprocessIntervals to
  automatically deal with too small targets

- calculate confidence intervals for cellular fractions

- throw additional warning when sample is flagged as NON-ABERRANT and
  pick the diploid solution with lowest purity as best

SIGNIFICANT USER-VISIBLE CHANGES

- significant runtime improvements

- callLOH now reports all segments, even if there are no informative
  SNPs since some users were not aware that segments are missing from
  this output. Use keep.no.snp.segments = FALSE to restore old
  behaviour.

- more detailed output of callLOH

- renamed num.snps.segment to num.snps in callAlterations output

BUGFIXES

- fixed crash in PureCN.R when gene symbols are missing from interval
  file

- fixed crash in runAbsoluteCN with matched normals and high
  test.purity minimum (#74)

[qckitfastq](https://bioconductor.org/packages/qckitfastq)
----------

                       Changes in version 0.99.12                       

- Addressed Bioconductor review points in:
  https://github.com/Bioconductor/Contributions/issues/753

- Added adapter content tabulation and plot implementation using RSeqAn
  and Rcpp

- Added checking for gzipped files (as opposed to raw fastqs) and
  automatic detection of quality score encoding (Sanger, Solexa,
  Illumina1.3, and Illumina1.5).

                       Changes in version 0.99.0                        

- First release

[qPLEXanalyzer](https://bioconductor.org/packages/qPLEXanalyzer)
-------------

                        Changes in version 1.1.4                        

- Fixed bug in coveragePlot - github Issue 15: Previously the function
  cleaned the beginning of each peptide sequence by
  gsub("^\[.\]\.([A-Z]+)\.\[.\]$", "\1", .).  However, it is not always
  the case that the terminal amino acids are surround by square
  brackets. I have put a more general regex in place that removes
  everything up to the first '.' and everything after the last '.'.

                        Changes in version 1.1.3                        

- 'ImmunoOncology' added to BiocViews.

                        Changes in version 1.1.2                        

- Updates to vignette to avoid build errors when attempting to connect
  to Uniprot

                        Changes in version 1.1.1                        

- Fixed bug in maVolPlot - github Issue 12: Previously the
  'selectedGenes' argument used GeneSymbols to match genes to highlight
  in the plot. Not all accessions are annotated with a GeneSymbol, and
  some GeneSymbols are linked to more than one Accession. This could
  lead to situations where incorrect genes were being highlighted due
  to duplicate entries or NAs. The function now accepts and matches to
  accessions instead - this is now noted in the man pages.

- Fixed bug in convertToMSnset - github Issue 11: The function
  previously threw and error if the metadata was a tibble as it needs
  to set row names. The function now tranforms the metadata object to a
  data frame.

                        Changes in version 1.1.0                        

- First Bioconductor Devel (Bioconductor 3.9)

[qsmooth](https://bioconductor.org/packages/qsmooth)
-------

                Changes in version 0.99.11 (2019-04-09)                 

- Fixed bug YAML in vignette causing an error on bioc nightly builds

                Changes in version 0.99.10 (2019-04-04)                 

- Fixed typo in ORCID

                 Changes in version 0.99.9 (2019-04-04)                 

- Adding ORCID for Stephanie Hicks, Hector Corrada Bravo and Rafael
  Irizarry

                 Changes in version 0.99.8 (2019-04-03)                 

- Fixed header of vignette .Rmd

                 Changes in version 0.99.7 (2019-03-31)                 

- Updating usage examples in qsmooth.R so R CMD check runs under 5 mins

                 Changes in version 0.99.6 (2019-03-30)                 

- Reducing vignette so R CMD check runs under 5 mins

                 Changes in version 0.99.5 (2019-03-30)                 

- Updating usage examples so R CMD check runs under 5 mins

                 Changes in version 0.99.4 (2019-03-29)                 

- Running `R CMD BiocCheck` gave me a warning unless I used R version
  3.7, but then this failed to pass in qsmooth version 0.99.3. So back
  to R version 3.6

                 Changes in version 0.99.3 (2019-03-29)                 

- Created a show method to show only first and last three values in the
  qsmoothWeights function and the head of the qsmoothData normalized
  matrix

- Updated all the examples and vignette to use the ExperimentHub
  bodymapRat data package

- Updated documentation in the plots.R file

- Included argument checking in the qsmooth.R file. Now checks if
  object is a matrix, data.frame or SummarizedExperiment.

- Included functionality for a user providing a SummarizedExperiment
  and extracting the counts from the assay slot (if it exists).

                 Changes in version 0.99.2 (2019-03-29)                 

- Removed hard-coded date in DESCRIPTION file

- Included Bioconductor installation instructions in README.md file

- Changed the name of the vignette so user can use
  `vigentte("qsmooth")`

- man. Changed the description of the return to reflect the actual
  output

                 Changes in version 0.99.0 (2019-02-07)                 

- Submitted to Bioconductor

[QuasR](https://bioconductor.org/packages/QuasR)
-----

                       Changes in version 1.24.0                        

NEW FEATURES

- added support for spliced alignments using HISAT2 (via Rhisat2
  package)

- migrated from Rsamtools to Rhtslib C library for operations on bam
  files

- migrated unit test from RUnit to testthat

- ignore reference deletions (BAM_CDEL) in qCount(..., reportLevel =
  "junction")

[r3Cseq](https://bioconductor.org/packages/r3Cseq)
------

                 Changes in version 1.29.0 (2018-12-08)                 

- migrated all codes to support the BioC3.9! This is because the
  BioC3.9 will not anymore support the 'RangedData' which is mainly
  used by r3Cseq in the previous versions.

[Rbowtie](https://bioconductor.org/packages/Rbowtie)
-------

                       Changes in version 1.23.2                        

NEW FEATURES

- updated bowtie to version 1.2.2.p1 + fixes from github (up to master
  branch commit 58c6ac9 from Sep 5, 2018, and unmerged pull request
  #74)

[RcisTarget](https://bioconductor.org/packages/RcisTarget)
----------

                         Changes in version 1.3                         

- Added function: getMotifAnnotation

[RCM](https://bioconductor.org/packages/RCM)
---

                        Changes in version 0.3.0                        

- Importance parameters *ψ* are enabled again when non-parametric
  response functions are used, but not used for plotting.

- 2D sample plots for constrained ordination with non-parametric
  response functions have been disabled, as they are not interpretable.
  Variable plots are the only 2D plots still allowed

- Explained deviance and inertia can be plotted on the axes rahter than
  the *ψ*'s using the "plotPsi" argument to the plot.RCM() function.

- Possibility to provide lower dimensional fits has been disabled.
  *RCM* is fast enough to fit the whole model.

                        Changes in version 0.2.0                        

- Importance parameters *ψ* are no longer calculated when
  non-parametric response functions are used.

[Rcpi](https://bioconductor.org/packages/Rcpi)
----

                 Changes in version 1.19.2 (2019-02-24)                 

Bug Fixes

- Fixed the issues in calcTwoProtGOSim() and calcParProtGOSim() to
use the latest GOSemSim API for computing GO based semantic
similarity.

                 Changes in version 1.19.1 (2018-11-21)                 

Bug Fixes

- Fixed API endpoints that are not working due to their changes
across time (#5).  - Fixed the vignette example for QSRR study (#6).

Improvements

- Reformat vignette code with tidyverse style.

[RcwlPipelines](https://bioconductor.org/packages/RcwlPipelines)
-------------

                Changes in version 0.99.11 (2019-03-27)                 

- 10 more tools were added.

                Changes in version 0.99.10 (2019-03-27)                 

- Submitted to Bioconductor

- 27 tools were collected.

[RCy3](https://bioconductor.org/packages/RCy3)
----

                        Changes in version 2.4.0                        

- New style setting functions - setEdgeFontFaceDefault -
  setEdgeFontFaceMapping - setEdgeFontSizeMapping - setEdgeLabelDefault
  - setEdgeLabelColorDefault - setEdgeLabelColorMapping -
  setEdgeLabelOpacityDefault - setEdgeLabelOpacityMapping -
  setEdgeOpacityDefault - setEdgeSourceArrowShapeMapping -
  setEdgeTargetArrowShapeMapping - setEdgeTooltipDefault -
  setNodeBorderOpacityDefault - setNodeBorderOpacityMapping -
  setNodeFillOpacityDefault - setNodeFillOpacityMapping -
  setNodeFontFaceDefault - setNodeFontFaceBypass -
  setNodeFontFaceMapping - setNodeFontSizeMapping -
  setNodeHeightDefault - setNodeHeightMapping - setNodeLabelDefault -
  setNodeLabelColorMapping - setNodeLabelOpacityDefault -
  setNodeLabelOpactiyMapping - setNodeTooltipDefault -
  setNodeTooltipBypass - setNodeWidthDefault - setNodeWidthMapping

- Bug Fixes - createSubnetwork -- #43 network suids i/o names -
  openSession works with current working directory -- #50 +Doc Fixes -
  improved visual.prop handling in mapVisualProperty -- #49,#53 user
  report - added file overwrite warnings to all export and save
  functions

                        Changes in version 2.2.7                        

- Doc Fixes - vignette for phylogenetic trees

                        Changes in version 2.2.6                        

- Bug Fixes - saveSession, exportXXX -- #39 default to working
  directory - createNetworkFromDataFrame and .edgeNameToSuids -- #41
  multigraph support - BiocCheck errors and warnings -- #42

- Doc Fixes - added tests for multigraphs

                        Changes in version 2.2.5                        

- Bug Fixes - createXXXFilter and applyFilter -- #40 network arg

- Doc Fixes - new filters vignette

                        Changes in version 2.2.4                        

- Doc Fixes - vignettes

                        Changes in version 2.2.3                        

- Doc Fixes - vignettes

                        Changes in version 2.2.2                        

- Doc Fixes - new custom graphics vignette

                        Changes in version 2.2.1                        

- Doc Fixes - vignettes, readme and test

[recount](https://bioconductor.org/packages/recount)
-------

                        Changes in version 1.9.7                        

SIGNIFICANT USER-VISIBLE CHANGES

- Cleaned up the documentation of add_metadata() and changed the
  default `source` from `recount_brain_v1` to `recount_brain_v2`.

                        Changes in version 1.9.6                        

NEW FEATURES

- citation('recount')[5] now lists the recount-brain bioRxiv pre-print
  citation information.

                        Changes in version 1.9.5                        

NEW FEATURES

- We have now released the FANTOM-CAT/recount2 RSE files which you can
  access now through download_study(type = 'rse-fc'). See Luidy-Imada
  E, Sanchez DF, et al, bioRxiv, 2019 for more information.

                        Changes in version 1.9.4                        

BUG FIXES

- I made the example code for geo_characteristics() more robust since
  currently rentrez can occasionally fails.

                        Changes in version 1.9.3                        

NEW FEATURES

- Add ORCID's following changes at
  http://bioconductor.org/developers/package-guidelines/#description

                        Changes in version 1.9.2                        

BUG FIXES

- Added the argument `async` to snaptron_query() which can be set to
  FALSE to address the issue reported at
  https://github.com/ChristopherWilks/snaptron/issues/11

[regionReport](https://bioconductor.org/packages/regionReport)
------------

                       Changes in version 1.17.2                        

NEW FEATURES

- Add ORCID's following changes at
  http://bioconductor.org/developers/package-guidelines/#description

[RepViz](https://bioconductor.org/packages/RepViz)
------

                       Changes in version 0.99.14                       

RELEASE

- Implementation of the last comments from reviewers

[rGREAT](https://bioconductor.org/packages/rGREAT)
------

                       Changes in version 1.15.1                        

- `plotRegionGeneAssociationGraphs()`: fixe a bug of wrong value of
  `par(mfrow)`.

- fixed warnings of Rd files.

[rgsepd](https://bioconductor.org/packages/rgsepd)
------

                        Changes in version 1.15                         

- new legends in scatterplots.

- custom point colors in PCA plot function. uses sampleMeta$CustomColor
  and disables the built in legend.

- custom genes in plots. See ExtractProjection() very useful, not yet
  in vignette - it calculates a projection among your custom gene set.

- Add lfcThreshold to the DESeq routines, so the p-values are adjusted
  correctly. ( implemented 1.15.1 ) This changes the behavior of most
  of the system. Expect a different gene set.

- New option to INIT() called renormalize, allowing user to specify
  preferred matrix normalization schema rather than the default
  DESeq2::varianceStabilizingTransform (when renormalize=FALSE).

- uses p.adjust(method='BH') on GOSeq's results to shorten the GO term
  list. use the 'padj' column.

- New option to DESeq , vst with blind=TRUE/FALSE at user discretion.

- added CITATION FILE in 1.15.13

- tweaking vignette in 1.15.15 to debug bioconductor OSX build-server
  LaTeX versions. And 1.15.16, switching from biblatex to natbib.

[rhdf5](https://bioconductor.org/packages/rhdf5)
-----

                       Changes in version 2.28.0                        

NEW FEATURES

- Functions to test and set file locking have added.  These can help
  identify and address problems on Lustre and ZFS file systems.

USER VISIBLE CHANGES

- Reading a dataset of rank one will now return an R vector, rather
  than a one dimensional array.

BUG FIXES

- Large improvements to performances when selecting subsets of datasets
  using the `index` argument.

- Resolved limitations where large datasets would exceed HDF5's 4GB
  maximum chunk size.  The default options will now check if this will
  occur and adjust chunking automatically.

- Single-rank datasets larger than 2^31-1 entries can now be read into
  R vectors.

[Rhisat2](https://bioconductor.org/packages/Rhisat2)
-------

                       Changes in version 0.99.0                        

- Initial version

[Rhtslib](https://bioconductor.org/packages/Rhtslib)
-------

                       Changes in version 1.16.0                        

SIGNIFICANT USER-VISIBLE CHANGES

- Include various additional files from Samtools 1.7: bam_aux.c,
  bam_endian.h, and bam_cat.c

- Add C function faidx_fetch_seq_forced_lower() to the HTSlib code.
  This is an alternative to HTSlib C function faidx_fetch_seq(), with
  the following difference: if coordinates are outside the actual
  sequence, write N's, rather than adjusting the start,end. Used in the
  seqbias package (and originally written by seqbias' author and
  maintainer, Daniel Jones).

BUG FIXES

- Fix long-standing kvsprintf() bug on Windows. This low-level bug was
  responsible for breaking "scan_bcf_header" .Call entry point in
  Rsamtools, which in turn was responsible for the infamous 'invalid
  class "VCFHeader" object' bug in VariantAnnotation::readVcf().

- Fix for Solaris: Do not build standalone executables (e.g. bgzip,
  tabix, etc...). They're not needed and seem to cause problems on
  Solaris.

[RMassBank](https://bioconductor.org/packages/RMassBank)
---------

                       Changes in version 2.11.2                        

- Avoid writing out empty PUBLICATIONS

                       Changes in version 2.11.1                        

- Fix error in workflow steps 2 and 7 with rcdk >= 3.4.9

[rmelting](https://bioconductor.org/packages/rmelting)
--------

                       Changes in version 0.99.7                        

- Fixed missing value in withWE.

                       Changes in version 0.99.6                        

- Removed vignette build products from git repository.  - Updated
installation instructions.  - Fixed all instances of
stop(paste(...)).  - Added stringsAsFactors = FALSE to the eopts data
frame.  - Expanded the nested functions in meltingBatch.  - Updated
error and/or warning(s) handling by implementing internal function
withWE.  - Fixed issue with single sequence input in meltingBatch.

                       Changes in version 0.99.5                        

- For first review.  - Changed R and rjava dependency.

                       Changes in version 0.99.4                        

- Converted T/F to TRUE/FALSE.  - Documented return for melting.

                       Changes in version 0.99.3                        

- Fixed minor BiocCheck warnings.  - Subscribed to mailing list.

- Fixed gitignore issue

                       Changes in version 0.99.2                        

- Fixed missing Tables.xlsx

                       Changes in version 0.99.1                        

- First BioC submission

[RNASeqR](https://bioconductor.org/packages/RNASeqR)
-------

                        Changes in version 1.1.3                        

- Resolve all version bumps; Add new biocView tag "ImmunoOncology"
  <2018-12-4 Thu>

                        Changes in version 1.0.2                        

- Add new biocView tag "ImmunoOncology" <2018-12-3 Thu>

                        Changes in version 1.0.1                        

- Update RNASeqR.Rmd. <2018-12-3 Thu>

[RnaSeqSampleSize](https://bioconductor.org/packages/RnaSeqSampleSize)
----------------

                 Changes in version 1.15.1 (2018-11-12)                 

BUG FIXES

- Fix a bug in power estiamtion (cumsumBorder.cpp), which make results
  incorrect when k not equal to one.

[RnBeads](https://bioconductor.org/packages/RnBeads)
-------

                        Changes in version 2.1.2                        

- Fix in as.RnBeadRawSet

- Added new paper to CITATION

                        Changes in version 2.1.1                        

- Added support for SLURM compute cluster architecture

[rols](https://bioconductor.org/packages/rols)
----

                        Changes in version 2.11                         

CHANGES IN VERSION 2.11.2

- Fix failing test <2019-04-16 Tue>

CHANGES IN VERSION 2.11.1

- Add ImmunoOncology biocView

CHANGES IN VERSION 2.11.0

- New version for Bioc 3.9 (devel)

[ropls](https://bioconductor.org/packages/ropls)
-----

                       Changes in version 1.15.26                       

NEW FEATURE

- plot method: dev.new() is not called internally any more;
  consequently the 'parDevNewL' argument has been removed; layout can
  be either 1x1 or 2x2, depending on the length of the 'typeVc'
  argument; 'fig.pdfC' and 'info.txtC' arguments have been added for
  compatibility reasons with other packages; they have the same role as
  'file.pdfC' and '.sinkC' which now generate a deprecated warning.

                       Changes in version 1.15.24                       

MINOR MODIFICATION

- plot method: turning off ellipse display in case of a single sample
  the class

                       Changes in version 1.15.22                       

INTERNAL MODIFICATION

- imageF function: minor modification of the log parameter

                       Changes in version 1.15.20                       

NEW FEATURE

- imageF function to visualize a numerical matrix

                       Changes in version 1.15.18                       

INTERNAL MODIFICATION

- minor internal modification

                       Changes in version 1.15.16                       

INTERNAL MODIFICATION

- modification of the way additional columns are added to the pData and
  fData of an ExpressionSet instance

- 'Biobase' package now as 'depends' (not 'import')

                       Changes in version 1.15.14                       

INTERNAL MODIFICATION

- minor internal modification

                       Changes in version 1.15.12                       

INTERNAL MODIFICATION

- minor modification in examples defining ExpressionSet instances

                       Changes in version 1.15.10                       

INTERNAL MODIFICATION

- 'eset' is now a slot of the 'opls' class (used when the 'opls' method
  is applied to an ExpressionSet instance)

                       Changes in version 1.15.8                        

NEW FEATURE

- 'plot' method: parPaletteVc argument to specify colors

                       Changes in version 1.15.6                        

NEW FEATURE

- Application to ExpressionSet: first 2 scores and loadings, as well as
  fitted values and VIP now returned in the pData and fData

INTERNAL MODIFICATION

- Unit testing switched from 'RUnit' to 'testthat'

                       Changes in version 1.15.4                        

INTERNAL MODIFICATION

- Checking that the data matrix does not contain any infinite or NaN
  value

                       Changes in version 1.15.2                        

INTERNAL MODIFICATION

- ImmunoOncology tag added to biocViews

- minor correction in the diagnostics graph labeling

[rpx](https://bioconductor.org/packages/rpx)
---

                        Changes in version 1.19                         

rpx 1.19.4

- Delete plain NEWS now that R supports NEWS.md

rpx 1.19.3

- Fix NEWS

rpx 1.19.2

- Mention setting method on Windows <2019-04-16 Tue>

rpx 1.19.1

- Add ImmunoOncology biocView

rpx 1.19.0

- New release for Bioc 3.9

[Rsamtools](https://bioconductor.org/packages/Rsamtools)
---------

                         Changes in version 2.0                         

SIGNIFICANT USER-VISIBLE CHANGES

- Migrate Rsamtools to Rhtslib. See Rsamtools/migration_notes.md for
  more information about this migration.

- Remove unused fields from BamRangeIterator

- Remove BAM header hash init for pileup (already memoized in Rhtslib)

[RSeqAn](https://bioconductor.org/packages/RSeqAn)
------

                        Changes in version 1.3.1                        

- CharString as/wrap wrappers and tests written.

[Rsubread](https://bioconductor.org/packages/Rsubread)
--------

                       Changes in version 1.34.0                        

- New functions: simReads() and scanFasta(). simReads() generates
  simulation RNA-seq reads for transcripts.

- align() and subjunc() estimate fragment length from mapped read pairs
  and use the estimated length to assist reporting the best alignment.

- align() and subjunc() prefer alignments with no indels included over
  indel-containing alignments when same number of matched bases are
  found.

- align() and subjunc() check if index files were successfully loaded
  before starting read mapping.

- align() and subjunc() detect indels arising from incorrect shifting
  of seed sequence when being mapped to low-complexity region and
  exclude such indels from read re-alignment and indel reporting.

- buildindex(), align() and subjunc() support gzipped FASTA format.

- featureCounts() allows mapped reads to have ‘*’ as their chromosome
  name.

- removeDupReads() supports BAM-format input and output.

[RTCA](https://bioconductor.org/packages/RTCA)
----

                     Changes in version 2009-07-13                      

- combineRTCA(list): Additional column is renamed into Plate. The vlues
  is evaluated from list item names. When the list has no name, an
  integer index beginning from 1 is used. Special attentions to list
  partially with names is noted in the documentation.

- parseRTCA(file, dec=".",phenoData, skipWell,...): Example is added in
  the documentation how to import pre-configured phenoData. Details
  section in the documentation is re-written to describe the process of
  parsing.

- RTCA-class: Experiment ID added to RTCA class

- Makefile: add Makefile to simplify common tasks like check and
  install

- plotGridEffect: takes 'column' instead of 'col' as mode parameter,
  and renders the mode as the title of the legend. Documentation
  updated.

- plotRTCA: is removed from the package and is substituted by the plot
  function.

[rWikiPathways](https://bioconductor.org/packages/rWikiPathways)
-------------

                        Changes in version 1.4.0                        

- Bug fixes - Fixed createSubnetwork -- #43 network suids i/o names

- Doc fixes - Fixed sign in subsetting function in vignette -- user
  issue on bioc forum - Added XML to Imports - Added the missing
  References section to one of the vignettes - Added BioSchemas
  annotation to the vignettes

[S4Vectors](https://bioconductor.org/packages/S4Vectors)
---------

                       Changes in version 0.22.0                        

NEW FEATURES

- Add recursive argument to expand() methods

- Support DataFrame (or any tabular object) in Pairs

- List derivatives now support x[i] <- NULL

- Some Vector derivatives now support appending with [<-

BUG FIXES

- [<-,DataFrame only makes rownames for new rows when rownames present

- DataFrame() lazily deparses arguments

[scAlign](https://bioconductor.org/packages/scAlign)
-------

                 Changes in version 0.99.0 (2019-02-05)                 

- Submitted to Bioconductor

[scater](https://bioconductor.org/packages/scater)
------

                       Changes in version 1.12.0                        

- Removed all functions deprecated in the last release.

- Added option in runTSNE() to perform an external nearest neighbors
  check with BiocNeighbors.  Removed the deprecated rand.seed=
  argument.

- Added text_by= and related options to plotReducedDim(), to overlay
  annotation labels onto the plot.

- Switched to BSPARAM= from BiocSingular for controlling SVD
  parameters.  Deprecated the approximate= argument for runPCA().

- Switched runUMAP() to use uwot::umap.  Added support for supplying a
  pre-defined set of nearest neighbors.

[scds](https://bioconductor.org/packages/scds)
----

                 Changes in version 0.99.4 (2019-03-21)                 

- Added NEWS file

- Expanded readme

- Expanded vignette

[scMerge](https://bioconductor.org/packages/scMerge)
-------

                        Changes in version 0.99                         

scMerge 0.99.24

- Updated ciation information due to PNAS acceptance.

scMerge 0.99.23

- Fixed assignment based on feedbacks

scMerge 0.99.21

- Increase code coverage to 85%.

scMerge 0.99.20

- Updated vignette on SEGs and manuals

scMerge 0.99.19

- Fixed spelling - Added verbose option - Code coverage at 75 percent
(more tests on error handling needed)

scMerge 0.99.17

- Fixed README install_github vignette issue.  - Fixed pkgdown
organisation issue.  - Major updates on the scReplicate function:
more informative messages.  - Using cross-product of matrix to
perform SVD to speed up calculations.  - Added testing scripts.  -
Fixed vignette text output issues.

scMerge 0.99.11

- Reduced data size in scMerge to pass BioC checks

[scran](https://bioconductor.org/packages/scran)
-----

                       Changes in version 1.12.0                        

- Added option in quickCluster() to cluster on log-expression profiles.
  Modified defaults to use graph-based clustering on
  log-expression-derived PCs.

- Modified default choice of ref.clust= in computeSumFactors(). Degrade
  quietly to library size factors when cluster is to small for all pool
  sizes.

- Minor change to cyclone() random number generation for consistency
  upon parallelization.

- Added BPPARAM= to correlateNull() for parallelization. Minor change
  in random number generation for consistency upon parallelization.

- Minor change to parallelPCA() random number generation for
  consistency upon parallelization.

- Created correlateGenes() function to compute per-gene correlation
  statistics.

- Modified correlatePairs() to compute expected rho after all possible
  tie-breaking permutations. Deprecated cache.size= as all ranks are
  now returned as in-memory representations. Deprecated per.gene= in
  favour of an external call to correlateGenes(). Deprecated tol= as
  ties are now directly determined by rowRanks().

- Switched to BiocSingular for PCA calculations across all functions.
  Deprecated approximate= and pc.approx= arguments in favour of
  BSPARAM=.

- Deprecated all batch correction functions to prepare for the
  migration to batchelor.

[scRecover](https://bioconductor.org/packages/scRecover)
---------

                 Changes in version 1.0.0 (2019-05-01)                  

- Package released

                Changes in version 0.99.15 (2019-04-20)                 

- Correct parallelization in scImpute (imputation_model.R)

- Resume use set.seed in scImpute (imputation_model.R)

- Update vignettes

                Changes in version 0.99.10 (2019-04-03)                 

- Add Unit Tests to functions

- Update vignettes

- Made the following significant changes o Change default outputDir to
  temporary directory o Add parameter verbose to scRecover function

                 Changes in version 0.99.0 (2019-02-21)                 

- Submitted to Bioconductor

                 Changes in version 0.1.0 (2018-05-30)                  

- Package created

[scTensor](https://bioconductor.org/packages/scTensor)
--------

                       Changes in version 0.99.22                       

- newCCSParams, getParam, setParam, and cellCellSimulate are added

- The hyperlinks to CMap (Connectivity Map) are embedded in the HTML
  report

                       Changes in version 0.99.21                       

- Some bugs are fixed

                       Changes in version 0.99.18                       

- Enrichment Analysis is added in cellCellReport()

                       Changes in version 0.99.15                       

- Accepted in BioC 3.9

                       Changes in version 0.99.6                        

- Revised and modified some parts

                       Changes in version 0.99.0                        

- Package released

[seq2pathway](https://bioconductor.org/packages/seq2pathway)
-----------

                       Changes in version 1.15.0                        

- Added function plotTop10

[SeqArray](https://bioconductor.org/packages/SeqArray)
--------

                       Changes in version 1.24.0                        

NEW FEATURES

- a new function `seqResetVariantID()`

- a new option in `seqRecompress(, compress="none")` to uncompress all
  data

- `seqGetData()` allows a GDS file name in the first argument

                    Changes in version 1.22.4-1.22.6                    

BUG FIXES

- `seqSetFilter(, sample.id=)` fails to correctly select samples in
  rare cases (since SeqArray>=v1.22.0 uses the distribution of selected
  samples to optimize the data access of genotypes, see
  https://github.com/zhengxwen/SeqArray/issues/48)

- the bgzf VCF file is truncated in `seqGDS2VCF()` since the file is
  not closed appropriately

- invalid chromosomes and position in the output of `seqMerge()` when
  merging different samples but same variants

                    Changes in version 1.22.1-1.22.3                    

UTILITIES

- more information in `seqDelete()`

NEW FEATURES

- a new option 'scenario' in `seqVCF2GDS()` and `seqBCF2GDS()`

BUG FIXES

- export a haploid VCF file using `seqGDS2VCF()`

- export VCF without any FORMAT data in `seqGDS2VCF()`

- export GDS without genotypes in `seqExport()`

- fix parallel file writing in seqVCF2GDS(), when no genotype

[seqCAT](https://bioconductor.org/packages/seqCAT)
------

                        Changes in version 1.6.0                        

FEATURES

- Make the variant caller-specific filtering optional (on by default)
  and rename the relevant parameter names for increased clarity

- Add a check to look for gVCF files as input, including the existance
  of <NON_REF> alleles (common for gVCF files)

- Add checks to see if input VCFs correctly contain DP, AD and GT data

[seqcombo](https://bioconductor.org/packages/seqcombo)
--------

                        Changes in version 1.5.1                        

- fixed R check by import dplyr::n (2019-01-02, Wed)

[SeqVarTools](https://bioconductor.org/packages/SeqVarTools)
-----------

                       Changes in version 1.21.5                        

- getGenotype and getGenotypeAlleles work for haploid genotypes.

                       Changes in version 1.21.4                        

- Add vignette describing iterators.

                       Changes in version 1.21.1                        

- Add method to return imputed dosage.

[sevenbridges](https://bioconductor.org/packages/sevenbridges)
------------

                       Changes in version 1.13.5                        

Improvements

- Added new platform options for the recently introduced
environments.  Now we can choose from "aws-us", "aws-eu", "ali-cn",
"cgc", "cavatica", and "f4c" in Auth() calls.

                       Changes in version 1.13.4                        

New Features

- Added support for Markers API (Advance Access feature). See the
Markers API section in the vignette for details.  - Added support for
Actions API. See the Actions API section in the vignette for details.

Improvements

- Added a new field description to the Files class following the
recent API improvements.  - Updated the API vignette to reflect the
platform default setting update for spot instances (spot is now
enabled by default).

                       Changes in version 1.13.3                        

New Features

- Added support for Enterprise API. See the Enterprise API section in
the vignette for details.

Improvements

- Removed unnecessary package dependencies to optimize the time
needed for package installation and loading.  - New look for the
documentation website with improved text readability.

                       Changes in version 1.13.2                        

New Features

- Added support for Folder API. See the Folders API section in the
vignette for details.

Improvements

- Added floating TOC and changed the vignette theme for the HTML
vignettes available on Bioconductor, to improve the browsing
experience for long vignettes.

Bug Fixes

- Fixed Docker image build issue and updated Bunny version.

                       Changes in version 1.13.1                        

New Features

- Added support for setting execution hints per task run when
drafting new tasks.

[signeR](https://bioconductor.org/packages/signeR)
------

                        Changes in version 1.9.3                        

- fix parsing issue in genCountMatrixFromVcf for some vcfs

                        Changes in version 1.9.2                        

- fix for genCountMatrixFromVcf fails with phased genotypes

- added one-vcf-per-sample example to vignette

                        Changes in version 1.9.1                        

- fix prototype mismatch issue

- fix plot error: non-finite ylim when pvalue=0

[singscore](https://bioconductor.org/packages/singscore)
---------

                        Changes in version 1.3.2                        

- allow continous and discrete annotations for plotDispersion and
projectScoreLandscape. Annotations can now be part of the score
data.frame and then specified as a column name - plot themes updated
- citation updated: cite the singscore manuscript - number of bins
for the hexbin plot in plotLandscape is determined from the data -
fixed bug in calculation of scores. Boundary calculation was
previously done with all genes in the gene-set. It should be done
with genes present in both the gene-set and the data (i.e. after
filtering out those not measured in the data).  - TotalDispersion now
estimated as the mean of dispersions from the up- and down-regulated
gene sets instead of the sum (previous estimate divided by 2)

                        Changes in version 1.2.2                        

- created a website for the package - added ORCID IDs for authors -
added sticker for package

[sitePath](https://bioconductor.org/packages/sitePath)
--------

                       Changes in version 0.99.28                       

- Bug fix: Internal indexing error in 'multiFixationSites'

- The default for 'makePlot' in 'sneakPeak' changed to False

- The default 'tolerance' in 'fixationSite' changed to 0.01

- Improve visualization of 'fixationSites' return

                       Changes in version 0.99.27                       

- Improve 'multiFixationSite' with a two-way greedy algorithm

- Use dash line for excluded 'lineagePath' in 'plotSingleSite'

                       Changes in version 0.99.26                       

- Bug fix: 'nodeAAsum' subscript out of bound in 'multiFixationSite'

                       Changes in version 0.99.25                       

- Use amino acid color for 'plotSingleSite.fixationSite'

- Add 'excluded' legend to 'plotSingleSite'

- Update DESCRIPTION and NEWS

- Use BiocStyle for vignettes

                       Changes in version 0.99.6                        

- New functionality 'multiFixationSites'

- Rewrite pruner functions

- Add 'minPath' argument for 'sneakPeek'

- 'sitePath' function's name changed to 'lineagePath'

- Minimum entropy

- The underlying data structure

                       Changes in version 0.99.5                        

- Add new functionality 'addMSA'

- Avoid 'root.phylo' function

[SNPRelate](https://bioconductor.org/packages/SNPRelate)
---------

                       Changes in version 1.18.0                        

- `snpgdsBED2GDS()` allows a single file name without the extended file
  names (.bed, .fam, .bim)

[SpatialCPie](https://bioconductor.org/packages/SpatialCPie)
-----------

                 Changes in version 1.0.0 (2018-08-17)                  

- Initial release

[specL](https://bioconductor.org/packages/specL)
-----

                Changes in version 1.17.22 (2019-01-24)                 

- code cosmetics to eliminate ERRORS and WARNINGS in R CMD BiocCheck of
  version 3.9.

[splatter](https://bioconductor.org/packages/splatter)
--------

                 Changes in version 1.8.0 (2018-04-18)                  

- Add a Splat parameters vignette

- Rename the Splat path.length parameter to path.nSteps

- Fix a bug with parameter order in setParams

- Fix a bug where Splat groups were being simulated in alphanumeric
  order

- Protect against integer overflow in simulation functions

[sRACIPE](https://bioconductor.org/packages/sRACIPE)
-------

                 Changes in version 0.99.0 (2019-04-01)                 

- submitted to Bioconductor

[ssrch](https://bioconductor.org/packages/ssrch)
-----

                       Changes in version 0.99.0                        

- added to Bioconductor 3.9

[statTarget](https://bioconductor.org/packages/statTarget)
----------

                         Changes in version 2.0                         

NEW FEATURES

- New GUI o Mouse Hover for help information o .log file

- New Signal correction o Combat for QC-free Signal correction o
  QC-RFSC methods for metabolomics and proteomics data

- New feature slection o Random Forest and the Permutation based
  variable importance measures o new MDSplot for Random Forest o
  P-value based importance plot

- New data preprocessing o PQN/SUM/none normalization o center/none
  Scaling method

                       Changes in version 1.13.7                        

- Results of Volcano plot (add the log2(FC) and -log10(p.adjust)
  values)

[Structstrings](https://bioconductor.org/packages/Structstrings)
-------------

                 Changes in version 0.99.0 (2019-03-01)                 

- Submitted to Bioconductor

[StructuralVariantAnnotation](https://bioconductor.org/packages/StructuralVariantAnnotation)
---------------------------

                 Changes in version 0.99.0 (2019-04-05)                 

- Submitted to BioConductor

[SWATH2stats](https://bioconductor.org/packages/SWATH2stats)
-----------

                       Changes in version 1.13.6                        

UPDATE

- aLFQ is deprecated on CRAN and thus Vignette cannot depend anymore on
  the aLFQ package.

                       Changes in version 1.13.5                        

NEW FEATURES

- Update plot_variation and plot_variation_vs_total function

                       Changes in version 1.13.4                        

NEW FEATURES

- Update version for JPP (new version of pyprophet)

                       Changes in version 1.13.3                        

UPDATE

- DESCRIPTION: move PECA to enhances

                       Changes in version 1.13.2                        

UPDATE

- Vignette: make PECA chunk eval=FALSE to prevent error when the PECA
  package is not available on BioC

                       Changes in version 1.13.1                        

NEW FEATURES

- Add ImmunoOncology to biocViews

                       Changes in version 1.13.0                        

NEW FEATURES

- SWATH2stats in BioC 3.9 development release

[SynMut](https://bioconductor.org/packages/SynMut)
------

                Changes in version 0.99.12 (2019-04-29)                 

- Important bug fix in function "dinu_to": fixed wrong result in sample
  methods.

                Changes in version 0.99.11 (2019-04-29)                 

- Important bug fix in function "get_du": fixed the step parameter in
  oligonucleotideFrequency.

                Changes in version 0.99.10 (2019-04-23)                 

- Update documents.

                 Changes in version 0.99.9 (2019-04-23)                 

- Reduced dependency on seqinr.

                 Changes in version 0.99.8 (2019-04-23)                 

- Updated DESCRIPTION.

                 Changes in version 0.99.7 (2019-04-23)                 

- Updated NAMESPACE.

                 Changes in version 0.99.6 (2019-04-23)                 

- Updated the Bioconductor installation instructions in readme and
  vignette.

                 Changes in version 0.99.5 (2019-04-23)                 

- Added bioconductor installation instructions to README and vignette.

- Added comments to vignette to make workflow more clear.

- Duplicated code in the following functions were converted to helper
  functions. o codon_random.R o codon_to.R o dinu_to.R o codon_mimic.R

- Added region_related.R to store the helper functions related to
  extracting or merging region.

- Eliminated 67 lines of duplicated code in dinu_to.R

- Eliminated duplicated code in input_seq.R

                 Changes in version 0.99.0 (2019-04-16)                 

- Submitted to Bioconductor

[TargetSearch](https://bioconductor.org/packages/TargetSearch)
------------

                       Changes in version 1.40.0                        

SIGNIFICANT USER-VISIBLE CHANGES

- The graphical user interface (TargetSearchGUI) is being deprecated.
  This is due to old source code, lack of time for maintenance, and
  lack of interest. The GUI it is still part of this release and it
  will be removed in the next release, ie, 1.42.0

BUG FIXES

- Fix NA handling in quantMass method.

[TCGAbiolinks](https://bioconductor.org/packages/TCGAbiolinks)
------------

                       Changes in version 2.11.2                        

- Add GISTIC2 case to GDC prepare and manuals

[TFEA.ChIP](https://bioconductor.org/packages/TFEA.ChIP)
---------

                        Changes in version 1.3.5                        

New Features

- Integrate enrichment analysis results by TF with rankTFs(). The
  function uses the output of getCMstats() to perform Wilcoxon rank-sum
  test or GSEA on the ChIPs ordered by distance score.

New default database

- The TF-Gene database included with TFEA.ChIP was built using ReMap's
  ChIP-seq collection (v. 2018) and GeneHancer's Double Elite
  regulatory regions (v. 4.8). Because of memory limits, the internal
  database included in TFEA.ChIP only stores ChIP-gene information from
  the 1060 ChIP-seqs from the ENCODE project included in ReMap 2018. To
  download the full database, as well as other ready-to-use databases
  generated for TFEA.ChIP, visit:
  https://github.com/LauraPS1/TFEA.ChIP_downloads

[tofsims](https://bioconductor.org/packages/tofsims)
-------

                        Changes in version 099.1                        

SIGNIFICANT USER-VISIBLE CHANGES

- changed function behvaiour in the whole package from call-by-ref to
  call-by value. Adjusted accordingly all examples and the vignette.

INTERNALS

- depends now on ProtGenerics from which it uses 'mz'

- exchanged various print() with message()

[topconfects](https://bioconductor.org/packages/topconfects)
-----------

                 Changes in version 1.1.4 (2018-12-29)                  

- Prepare for Bioconductor submission.

                 Changes in version 1.1.3 (2018-11-14)                  

- Add "full" output option for limma_confects and normal_confects,
  which adds se, df, and fdr_zero columns.

                 Changes in version 1.1.2 (2018-06-30)                  

- Add DESeq2 support.

- Code to do with non-linear effect sizes has been move to the "ql"
  branch on github, pending publication of the basic method and a
  possible rethink and use of a simpler method.

                 Changes in version 1.0.1 (2018-02-04)                  

- Initial release.

                 Changes in version 0.99.5 (2019-02-18)                 

- Include a vignette giving an overview of the package's functions.

                 Changes in version 0.99.4 (2019-02-06)                 

- Reduce build time of vignette.

                 Changes in version 0.99.3 (2019-02-05)                 

- Where :: is used, use requireNamespace first to give an informative
  error message.

- Use match.arg in edger_confects.

- More tests.

                 Changes in version 0.99.2 (2019-01-15)                 

- Version bumped to test web-hook and Bioconductor automated build
  system.

                 Changes in version 0.99.1 (2018-12-30)                 

- Switch to required version number for Bioconductor submission.

- Switch email to address subscribed to bioc-devel.

- Require R 3.6.

[topdownr](https://bioconductor.org/packages/topdownr)
--------

                         Changes in version 1.5                         

- New version for Bioc 3.9 (devel)

Changes in version 1.5.6

- Revert changes for NULL indices of DataFrame introduced in 1.5.4
(a419f59, c4bfc1c) because they are fixed upstream in S4Vectors.
Keep unit tests in place. [2019-03-27]

Changes in version 1.5.5

- Depends on R >= 3.5.0 now, because the seralization format changed
in R.

Changes in version 1.5.4

- Fix for internal .makeRowNames/.colsToLogical/.colsToRle on
DataFrame without any numeric/character columns.  - Fix unit test
that uses set.seed (order changed during R-devel upgrade).

Changes in version 1.5.3

- biocViews: ImmunoOncology added by Kayla-Morrell
kayla.morrell@roswellpark.org [2019-01-04].

Changes in version 1.5.2

- Add expandMs1Conditions, expandTms2Conditions,
createExperimentsFragmentOptimisation functions to allow more
flexibility in method creation; see also #77 [2018-12-07].  - Modify
interface/arguments of writeMethodXmls to adapt to new method
creation workflow (the old interface will be defunct in Bioconductor
3.10 and removed in 3.11) [2018-12-07].  - Adapt the data-generation
vignette to the new workflow [2018-12-07].  - Deprecated
defaultMs1Settings and defaultMs2Settings. They will be defunct in
Bioconductor 3.10 and removed in 3.11 [2018-12-07].

Changes in version 1.5.1

- readTopDownFiles gains the argument conditions to control wheter
"FilterStrings" or a given number of conditions is used to create
condition IDs; see #77 [2018-11-07].

[TPP2D](https://bioconductor.org/packages/TPP2D)
-----

                 Changes in version 0.99.0 (2019-03-12)                 

- Submitted to Bioconductor

[trackViewer](https://bioconductor.org/packages/trackViewer)
-----------

                       Changes in version 1.19.28                       

- fit the small size canvas.

                       Changes in version 1.19.27                       

- add safe color theme.

                       Changes in version 1.19.26                       

- update author info.

                       Changes in version 1.19.25                       

- change save and load name for browseTracks.

                       Changes in version 1.19.24                       

- add save and load for browseTracks.

- add swith dandelion height method (mean or count) for browseTracks.

                       Changes in version 1.19.23                       

- add color picker for pie.stack for browseTracks.

                       Changes in version 1.19.22                       

- fix the bug in legend for browseTracks.

                       Changes in version 1.19.21                       

- fix the bug of that can not set color for undefined for browseTracks.

                       Changes in version 1.19.20                       

- fix the bug of lable is not saved for browseTracks.

                       Changes in version 1.19.19                       

- fix the bug of output svg for browseTracks.

                       Changes in version 1.19.18                       

- add rotate 45 in color picker for browseTracks.

                       Changes in version 1.19.17                       

- add apply to all in color picker for browseTracks.

                       Changes in version 1.19.16                       

- fix the grammar in vignettes.

                       Changes in version 1.19.15                       

- update vignettes.

                       Changes in version 1.19.14                       

- update undo for browseTracks.

                       Changes in version 1.19.13                       

- add undo for browseTracks.

                       Changes in version 1.19.12                       

- add remove all guideline function.

                       Changes in version 1.19.11                       

- fix the fonts bug for export png in browseTracks

                       Changes in version 1.19.10                       

- remove all svgs before browseTracks

                       Changes in version 1.19.9                        

- Build shiny output objects for browseTracks

                       Changes in version 1.19.8                        

- add dandelion plots to browseTracks

                       Changes in version 1.19.7                        

- fix a bug in track class

- improve one-step visualization for browseTracks

                       Changes in version 1.19.6                        

- improve one-step visualization for browseTracks

- try to avoid the error caused by VariantAnnotation

                       Changes in version 1.19.5                        

- improve one-step visualization for browseTracks

                       Changes in version 1.19.4                        

- add one-step visualization for browseTracks

                       Changes in version 1.19.3                        

- adjust y label position for viewTracks

                       Changes in version 1.19.2                        

- adjust legent position for dandelion.plot

                       Changes in version 1.19.1                        

- add featureLayerID to dandelion.plot

[transcriptogramer](https://bioconductor.org/packages/transcriptogramer)
-----------------

                        Changes in version 1.5.6                        

- All datasets were updated to match data from STRINGdb release 11.

                        Changes in version 1.5.1                        

- Change on the differentiallyExpressed() method return: output now
  contains raw p-values.

- Change on the clusterEnrichment() method return: output now contains
  raw p-values.

- A section called "Frequently asked questions" has been added in the
  vignette.

[transite](https://bioconductor.org/packages/transite)
--------

                        Changes in version 1.0.2                        

- updated roxygen2

- updated license

- clarified argument description of SPMA

- fixed issue with multi-threaded cluster environment

                        Changes in version 1.0.1                        

- fixed typo in vignette

- added code example for drawVolcanoPlot function

- increased size of circles for motif-associated k-mers in
  drawVolcanoPlot function

[treeio](https://bioconductor.org/packages/treeio)
------

                        Changes in version 1.7.4                        

- update test according to the change of default RNG in the comming
R-3.6 (2019-04-02, Tue)

                        Changes in version 1.7.3                        

- rescale_tree from ggtree (2019-01-11, Fri)

                        Changes in version 1.7.2                        

- MRCA methods for phylo and treedata (2019-01-10, Thu) - mv
vignettes to treedata-book - root method from ggtree::reroot
(2018-12-28, Fri) - rename to root for importing ape::root generic

                        Changes in version 1.7.1                        

- compatible with tibble v=2.0.0 (2018-11-29, Thu)


[tRNA](https://bioconductor.org/packages/tRNA)
----

                 Changes in version 1.1.2 (2018-03-27)                  

- bugfix for padding unpaired region in stem sequences

- bugfix for getting empty sequences of non-existing variable loops

- fixed man page issue

                 Changes in version 1.1.1 (2018-12-01)                  

- added Structstrings dependency

- getBasePairing now used from Structstrings package

- optimizations

[tRNAdbImport](https://bioconductor.org/packages/tRNAdbImport)
------------

                 Changes in version 1.1.1 (2019-01-26)                  

- added Modstrings and Structstrings dependency

- depending on the database selected the tRNA sequences will be
  returned as DNAStringSet or ModRNAStringSet

- now uses the DotBracketStringSet from the Structstrings package to
  store the structure information

[tRNAscanImport](https://bioconductor.org/packages/tRNAscanImport)
--------------

                 Changes in version 1.3.2 (2018-02-16)                  

- now exits with error message, if empty lines as delimiter of tRNA
  entries is found

                 Changes in version 1.3.1 (2018-01-26)                  

- now uses the DotBracketStringSet from the Structstrings package to
  store the structure information

[TVTB](https://bioconductor.org/packages/TVTB)
----

                 Changes in version 1.9.2 (2019-02-20)                  

Bug fix

- Fixes to pass updated BiocCheck requirements.

[tximeta](https://bioconductor.org/packages/tximeta)
-------

                       Changes in version 1.1.18                        

- Specifying gene=TRUE in addIds() when rows are transcripts will
  attempt to use a gene_id column to map the IDs. This usually gives a
  better mapping rate.

[tximport](https://bioconductor.org/packages/tximport)
--------

                       Changes in version 1.11.1                        

- Added argument 'sparse' and 'sparseThreshold' to allow for sparse
  count import. For the initial implemenation: only works for
  txOut=TRUE; countsFromAbundance either "no" or "scaledTPM"; doesn't
  work with inferential replicates, and only imports counts (and
  abundances if countsFromAbundance="scaledTPM").

[Ularcirc](https://bioconductor.org/packages/Ularcirc)
--------

                        Changes in version 1.1.3                        

NEW FEATURES

- Added ability to upload CIRI2 and circExplorer2 output

- New analysis features: PCA plot, heatmap analysis, distribution
  analysis

BUG FIXES

- Amended TwoSzabo dataset

- Amended workflow to minimise system crashes

[Uniquorn](https://bioconductor.org/packages/Uniquorn)
--------

                 Changes in version 2.3.5 (2019-02-27)                  

Fixed another windows VCF parse error

- Fixed yet another VCF file parsing error associated with the
  VariantAnnotation R package parser.

                 Changes in version 2.3.4 (2019-02-25)                  

Fixed windows VCF parse error

- Uniquorn experienced an error when parsing the internal test vcf file
  on Windows systems.

                 Changes in version 2.3.3 (2019-02-22)                  

Unzip test vcf file

- Unzipped the tested vcf file since VarriantAnnotion parser failed to
  read zipped file in future R version 3.6

                 Changes in version 2.3.2 (2019-02-04)                  

Remove dependency

- Removed dependency on data.table

- Removed data.table import due to biocondcutor request

[universalmotif](https://bioconductor.org/packages/universalmotif)
--------------

                        Changes in version 1.2.0                        

NEW FEATURES

- New motif_peaks() function: test for significantly overrepresented
  peaks of motif sites in a set of sequences.

- New get_bkg() function: calculate background frequencies of sequence
  alphabets, including higher order backgrounds. Works for any sequence
  alphabet. Can also create MEME background format files.

- read_meme() has a new option, readites.meta, which allows for reading
  individual motif site positions and P-values, as well as combined
  sequence P-values.

- shuffle_sequences(..., method = "markov") works for any set of
  characters instead of just DNA/RNA.

- shuffle_sequences(): new shuffling method, 'euler'. This allows for
  k>1 shuffling that preserves exact letter counts, as opposed to
  'markov'. This method is set as the new default shuffling method.

- create_sequences() has a new option "freqs" which allows for
  generating sequence from higher order backgrounds and any sequence
  alphabet (options "monofreqs", "difreqs" and "trifreqs" are now
  deprecated).

- universalmotif objects can now hold onto higher order backgrounds.

- motif_pvalue() with use.freq > 1 can calculate P-values from provided
  higher-order backgrounds, instead of assuming a uniform background.

- add_multifreq() adds corresponding higher order background
  probabilities to motifs.

- New get_klets() utility function: generate all possible k-lets for
  any set of characters.

- New score_match() utility function: score a match for a particular
  motif.

- New get_matches() utility function: get all possible motif matches
  above a certain score.

- New count_klets() utility function: count all k-lets for any string
  of characters.

- New motif_score() utility function: calculate motif score from input
  thresholds.

- New shuffle_string() utility function: shuffle a string of character
  using one of three methods: euler, linear, and markov.

- The native write_motifs()/read_motifs() universalmotif format is now
  YAML based. Motifs written before v1.2.0 can still be read by
  read_motifs().

MINOR CHANGES

- Increased input security for character type parameters throughout.

- Expanded motif_pvalue(), scan_sequences(), motif_tree() examples
  sections.

- New vignette sections for motif_peaks() and get_bkg() added to
  SequenceSearches.Rmd.

- Various vignette tweaks.

- Fixed various spelling mistakes throughout, added Language field to
  Description, and added spell check to tests.

- Documentation for the "random" shuffling method has been removed and
  a warning is shown when used to tell the user that it will be removed
  in the next minor update.

- Generally increased test coverage.

- The "k=1", "linear" and "markov" shuffling methods are much faster.

- create_sequences() for higher order backgrounds is much faster.

- Faster add_multifreqs(): slight improvement for DNA motifs, big
  improvement for non-DNA motifs.

- sample_sites() has been rewritten for use.freq > 1: the probability
  of each letter in the site is now dependent on the previous letters
  (also faster and more memory efficient for any use.freq).

- Improvement to calculating motif scores from p-value input: no longer
  guesses different scores, instead estimating a normal distribution of
  scores. This new approach is much, much faster and more memory
  efficient. It does however assume a uniform background.

- The "score.pct" column in scan_sequences() results now represents the
  percent score based on the total possible score, not just the score
  between zero and the max possible score.

- summarise_motifs() is much faster.

- Objects in data/ are saved using serialization format version 3.

- convert_motifs(motif, class = "universalmotif-universalmotif"):
  performs a validObject() check if "motif" is a universalmotif object.

- The show() method for universalmotif objects performs a validObject()
  check first.

- motif_rc() has a new option "ignore.alphabet", used to turn on or off
  the alphabet check (checks for DNA/RNA motif).

- Added "overwrite" and "append" options to write_*() functions.

- enrich_motifs(..., return.scan.results = FALSE): uses a slimmed down
  version of scan_sequences() which skips construction of the complete
  results data.frame, saving a tiny bit of time on large jobs.

- compare_motifs() now includes log P-values. This way comparisons can
  still be properly ranked even if their P-values are below the machine
  limit.

- convert_motifs() from MotifList (MotifDb) carries over dataSource.

- If a MEME motif has two names, the second will be assigned as
  "altname" by read_meme().

- Utilities documentation has been split into two: ?utils-motif and
  ?utils-sequence.

BUG FIXES

- Fixed IC score calculation from character input in create_motif().

- The internal DNA consensus letter calculation previously did not
  assign ambiguous letters when one PPM position was >0.5 and another
  was >0.25. This was unintended behaviour and will now output the
  proper ambigous DNA letter.

[variancePartition](https://bioconductor.org/packages/variancePartition)
-----------------

                       Changes in version 1.13.5                        

- add plotContrasts() to dream vignette

                       Changes in version 1.13.4                        

- export classes to fix bug with class "varPartResults" not being
  defined

- Thanks Megan Behringer

                       Changes in version 1.13.3                        

- add plotContrasts()

                       Changes in version 1.13.2                        

- Enable random slope models in dream, but not for estimating variance
  fractions

- Thanks Jonas Zierer

                       Changes in version 1.13.1                        

- Add progress bar at ETA

[VennDetail](https://bioconductor.org/packages/VennDetail)
----------

                Changes in version 0.99.18 (2019-04-25)                 

- address warnings

                Changes in version 0.99.17 (2019-04-25)                 

- address warnings

                Changes in version 0.99.16 (2019-04-25)                 

- address issues

                Changes in version 0.99.15 (2019-04-23)                 

- rewrite the vennpie function

                Changes in version 0.99.14 (2019-04-22)                 

- rewrite the venndetail function

                Changes in version 0.99.13 (2019-04-20)                 

- update the vignette

                Changes in version 0.99.12 (2019-03-29)                 

- update the vignette

                Changes in version 0.99.11 (2019-03-27)                 

- revised function based on the comments

                Changes in version 0.99.10 (2019-02-25)                 

- add wide output format

                 Changes in version 0.99.9 (2019-02-20)                 

- add wide output format

                 Changes in version 0.99.8 (2019-02-10)                 

- fix error,warnings and notes

                 Changes in version 0.99.7 (2019-02-09)                 

- update man doc

                 Changes in version 0.99.6 (2019-02-09)                 

- add show as generic function

                 Changes in version 0.99.5 (2019-02-08)                 

- update Get function

                 Changes in version 0.99.4 (2019-02-07)                 

- update NAMESPACE

                 Changes in version 0.99.3 (2019-02-07)                 

- modify the get function to Get

                 Changes in version 0.99.2 (2019-02-07)                 

- revised the test function

                 Changes in version 0.99.1 (2019-02-07)                 

- delete Rproj file

                 Changes in version 0.99.0 (2019-02-05)                 

- Submitted to Bioconductor

[xcms](https://bioconductor.org/packages/xcms)
----

                        Changes in version 3.5.5                        

- Add dirname and dirname<- methods for OnDiskMSnExp to change the path
  to the raw data files.

- Add section "Subset-based alignment" to the xcms vignette to describe
  the alignment possibility to perform alignments based on a subset of
  samples (e.g. QC samples).

                        Changes in version 3.5.4                        

- Fix problem in featureChromatograms with include = "feature_only"
  that could return a non-valid object.

- Ensure that XCMSnExp objects are updated if necessary in all analysis
  methods.

                        Changes in version 3.5.3                        

- Fix unit tests.

                        Changes in version 3.5.2                        

- Small changes in fillChromPeaks,XCMSnExp to reduce memory demand.

- Fix issue #359.

- Fix issue #360: rawEIC skipped last scan/spectrum if rtrange was
  provided.

- filterMsLevel keeps now chromatographic peaks and feature definitions
  from the specified MS levels (issue #362).

- Fix bug in `xcmsRaw` that leads to a netCDF error message (issue
  #363).

- Add parameter msLevel to chromPeaks for XCMSnExp objects.

- Add chromPeakData to allow adding arbitrary annotation to
  chromatographic peaks.

- Change default of parameter value in featureValues from value =
  "index" to value = "into".

- Add parameter isFilledColumn to chromPeaks allowing the old behaviour
  to include the is_filled column in the chromatographic peak matrix.

                        Changes in version 3.5.1                        

- Fix issue #349.

- Add updateObject function for XCMSnExp objects (issue #347).

- Add dropFilledChromPeaks methods for XChromatogram and XChromatograms
  objects.

- Add parameter filled = FALSE to chromatogram and featureChromatograms
  functions.

- Fix matchedFilter peak detection problems with empty spectra (issue
  #325).

- featureChromatograms extracts by default only chromatographic peaks
  associated with a feature.

- chromatogram,XCMSnExp extracts an XChromatogram containing also
  chromatographic peaks and feature definitions.

- Add featureValues method for XChromatograms objects (issue #336).

- Add correspondence analysis (peak grouping) for chromatographic data
  (for now only with PeakDensity method; issue #336).

- Add featureDefinitions slot to XChromatograms object and related
  accessor methods.

- Add subset alignment option subsetAdjust = "average" to adjust
  left-out samples (blanks or simply non-subset samples) based on an
  interpolation from the results of the previous and subsequent subset
  sample.

- Add parameter subsetAdjust to PeakGroupsParam allowing to switch
  between different methods to adjust samples left out in the alignment
  process.

- Alignment based on a sample subset for the peak groups method (issue
  #335): sample subset can be defined with the subset parameter,
  samples not included in the subset will be aligned based on the
  adjusted retention times of the closest sample in the subset.

- Add findChromPeaks,XChromatograms (issue #332).

- Add processHistory,XChromatograms.

- Add plot,XChromatograms method with automatic peak highlighting
  (issue #334).

- Add hasChromPeaks,XChromatograms method.

- Add XChromatograms class with constructor function and coercing
  method.

- Add hasChromPeaks,XChromatogram method.

- Add filterRt,XChromatogram, filterMz,XChromatogram.

- Add plot,XChromatogram function supporting of highlighting/drawing
  identified chromatographic peaks.

- findChromPeaks,Chromatogram returns an XChromatogram object (issue
  #329).

- Add chromPeaks,XChromatogram (issue #329).

- Add XChromatogram object (issue #329).

- Fix higlightChromPeaks with type = "polygon": peak filling represents
  now the full detected peak and is no longer cut by the provided rt.

- Add argument peakIds to highlightChromPeaks allowing to specify the
  IDs of peaks to be highlighted.

- Add example on clustering of base peak chromatograms to the vignette
  (issue #328).

- Small fix in the vignette (issue #327).

- Add parameter groupval to exportMetaboAnalyst (issue #296).

- Fix bug in show,XCMSnExp that would throw an error if no process
  history is present.

[Xeva](https://bioconductor.org/packages/Xeva)
----

                       Changes in version 0.99.7                        

- This is the new version

[zinbwave](https://bioconductor.org/packages/zinbwave)
--------

                 Changes in version 1.5.3 (2019-04-19)                  

- Updated vignette to reflect new way of interacting with Seurat.

- Illustrate the use of approximate `zinbsurf` function.

                 Changes in version 1.4.2 (2019-03-11)                  

- Fix bug in computeObservationalWeights.


NEWS from new and existing Data Experiment Packages
===================================


[bodymapRat](https://bioconductor.org/packages/bodymapRat)
----------

                 Changes in version 0.99.8 (2019-04-04)                 

- Adding ORCID for Stephanie Hicks

                 Changes in version 0.99.7 (2019-04-03)                 

- Fixed error in header of vignette

- Adding citation to package

                 Changes in version 0.99.6 (2019-03-29)                 

- Added documentation for bodymapRat object with examples

                 Changes in version 0.99.5 (2019-03-27)                 

- Added .onLoad() function as noted in ExperimentHub documentation

- Converted vignette and unit tests to load data from ExperimentHub

                 Changes in version 0.99.4 (2019-03-27)                 

- Now have successfully removed .rda object from data/ folder

- BFG Repo-Cleaner to remove large data file and clean git tree

                 Changes in version 0.99.3 (2019-03-19)                 

- Convert data package to ExperimentHub package

- Removed .rda object from data/ folder

- Removed Maintainer from DESCRIPTION file

- Updated README.md to include bioconductor installation

- Moved and renamed file to create data object into
  inst/scripts/make-data.Rmd

- Created a make-metadata.R file

                 Changes in version 0.99.1 (2019-02-23)                 

- Compressed and re-saved data object

                 Changes in version 0.99.0 (2019-02-07)                 

- Submitted to Bioconductor

[CluMSIDdata](https://bioconductor.org/packages/CluMSIDdata)
-----------

                       Changes in version 0.99.5                        

- added some more data for CluMSID MTBLS vignette

                       Changes in version 0.99.4                        

- removed invalid character from GC_post.csv

                       Changes in version 0.99.3                        

- updated R version dependency to 3.6

                       Changes in version 0.99.2                        

- updated R version dependency to 3.5.1

                       Changes in version 0.99.1                        

- started ignoring CluMSIDdata.Rproj

- updated R version dependency

                       Changes in version 0.99.0                        

- Bioconductor submission

                        Changes in version 0.1.0                        

- migrated data needed for vignette and examples from CluMSID to
  CluMSIDdata

[curatedAdipoChIP](https://bioconductor.org/packages/curatedAdipoChIP)
----------------

                 Changes in version 0.99.0 (2019-03-26)                 

- Submitted to Bioconductor

[curatedAdipoRNA](https://bioconductor.org/packages/curatedAdipoRNA)
---------------

                 Changes in version 0.99.0 (2019-02-06)                 

- Submitted to Bioconductor

[dsQTL](https://bioconductor.org/packages/dsQTL)
-----

                        Changes in version 2.17                         

USER VISIBLE CHANGES

- ch2locs (retrievable via dsQTL::getSNPlocs) has been changed at about
  1850 locations where rs numbers had been associated with hg19
  addresses; the dsQTL regions are hg18 as are all the chr2... SNP
  addresses.  Previously the discoverable rs numbers used in the
  Chicago distribution from
  http://eqtl.uchicago.edu/dsQTL_data/GENOTYPES/ had be mapped via
  SNPlocs...20111119, but now they come directly from the Chicago text
  file.

[FlowSorted.CordBloodCombined.450k](https://bioconductor.org/packages/FlowSorted.CordBloodCombined.450k)
---------------------------------

                 Changes in version 0.99.0 (2019-01-21)                 

- Submitted to Bioconductor

- Added NEWS file

- Initial development release

[HCAData](https://bioconductor.org/packages/HCAData)
-------

                        Changes in version 0.1.0                        

New features

- Initial version of the package!

[HDCytoData](https://bioconductor.org/packages/HDCytoData)
----------

                        Changes in version 1.3.4                        

- Add Samusik_01, Samusik_all, Nilsson_rare, and Mosmann_rare datasets

- Add Krieg_Anti_PD_1 dataset

- Include additional information in 'description' slot for flowSet
  object for Bodenmiller_BCR_XL dataset

- Update documentation

                        Changes in version 1.3.2                        

- Add Levine_32dim and Levine_13dim datasets

- Include additional information in 'description' slot for flowSet
  object for Bodenmiller_BCR_XL dataset

                        Changes in version 1.3.1                        

- Update flowSet object for Bodenmiller_BCR_XL dataset to include raw
  channel names

[mcsurvdata](https://bioconductor.org/packages/mcsurvdata)
----------

                        Changes in version 1.0.1                        

- Package submission

[MOFAdata](https://bioconductor.org/packages/MOFAdata)
--------

                 Changes in version 0.99.0 (2019-01-25)                 

- Submitted to Bioconductor

[mtbls2](https://bioconductor.org/packages/mtbls2)
------

                       Changes in version 1.13.1                        

- filePaths of raw files are now patched to match the location of the
  installed mtbls2 package upon onAttach(), to be compliant with the
  staged-install approach

[NestLink](https://bioconductor.org/packages/NestLink)
--------

                Changes in version 0.99.96 (2019-04-24)                 

- added DOI 10.1038/s41592-019-0389-8 in citation.

                Changes in version 0.99.83 (2018-12-20)                 

- new Dockerfile.

- new vignette `make-data.Rmd`.

- use <URL:
  http://bioconductor.org/packages/devel/bioc/html/ExperimentHub.html>.

- moved all package data to aws s3. data are now available by using
  ExperimentHub.

                Changes in version 0.99.50 (2018-09-25)                 

- prepare for submission on bioconductor.

[PtH2O2lipids](https://bioconductor.org/packages/PtH2O2lipids)
------------

                     Changes in version 2016-04-21                      

- Initial release for Bioconductor

[qPLEXdata](https://bioconductor.org/packages/qPLEXdata)
---------

                        Changes in version 1.1.1                        

- Fixed broken code in Vignette


NEWS from new and existing Workflows
===================================


[BgeeCall](https://bioconductor.org/packages/BgeeCall)
--------

                        Changes in version 1.0.0                        

- Public release of BgeeCall package. Enjoy!!!

[recountWorkflow](https://bioconductor.org/packages/recountWorkflow)
---------------

                        Changes in version 1.7.2                        

BUG FIXES

- Fix the call to bumphunter::annotateTranscripts().

- See
  https://gist.github.com/lcolladotor/196dabeb1ac628c35656bfa94b5d9577
  for more details.

                        Changes in version 1.7.1                        

BUG FIXES

- Fix the ftp url for the Gencode v25 gtf file.

[SingscoreAMLMutations](https://bioconductor.org/packages/SingscoreAMLMutations)
---------------------

                 Changes in version 0.99.0 (2019-03-25)                 

- Submitted to Bioconductor

Deprecated and Defunct Packages
===============================

Thirteen software packages were removed from this release (after being deprecated
in Bioc 3.8): nudge, GeneSelector, BiocInstaller, RamiGO, prot2D, ampliQueso,
gaucho, mQTL.NMR, facopy, cytofkit, GoogleGenomics, pbcmc, IrisSpatialFeatures

Thirteen software are deprecated in this release and will be removed in Bioc 3.10:
flowQ, rMAT, TSSi, flowQB, rSFFreader, ProCoNA, spliceSites, GOTHiC, DOQTL,
NGScopy, SVM2CRM, miRsponge, htSeqTools

Two experimental data packages were removed in this release (after being
deprecated in BioC 3.8): iontreeData, MSBdata.

Two experimental data packages are deprecated in this release and will be
removed in Bioc 3.10: PGPC, flowQBData

Two annotation packages were removed this release: MafDb.gnomADex.r2.0.1.hs37d5,
MafDb.gnomAD.r2.0.1.hs37d5 (they were replaced with MafDb.gnomADex.r2.1.hs37d5,
MafDb.gnomAD.r2.1.hs37d5)

Two annotation packages are deprecated in this release and will be removed
in Bioc 3.10: MafDb.gnomADex.r2.0.1.GRCh38, MafDb.gnomAD.r2.0.1.GRCh38 (they
have been replaced with MafDb.gnomADex.r2.1.GRCh38, MafDb.gnomAD.r2.1.GRCh38)
