# BioC 2015: Where Software and Biology Connect

July 21-22, 2015 (Developer Day: July 20)<br />
Fred Hutchinson Cancer Research Center, Seattle, WA

This conference highlights current developments within and beyond
Bioconductor. Morning scientific talks and afternoon practicals
provide conference participants with insights and tools required for
the analysis and comprehension of high-throughput genomic
data. 'Developer Day' precedes the main conference on July 20,
providing developers and would-be developers an opportunity to gain
insights into project direction and software development best
practices.

- [Speakers and Workshops](#speakers-and-workshops)
- [Tentative Schedule](#tentative-schedule)
- [Accommodation](#accommodation)
<!-- - [Workshop Abstracts](#workshop-abstracts) -->

# Speakers and Workshops

## Distinguished Speakers

- [Artem Sokolov](https://users.soe.ucsc.edu/~sokolov/), University of
  California, Santa Cruz. Integrating prior biological knowledge using
  Generalized Elastic Nets.
- [Cole Trapnell](http://cole-trapnell-lab.github.io/), University of
  Washington. Single cell trajectory analysis with
  [Monocle](/packages/release/bioc/html/monocle.html).
- [Eric Franzosa](http://huttenhower.sph.harvard.edu/), Huttenhower
  Lab, Harvard School of Public Health. Integrated multi'omic analysis
  methods for epidemiological profiling of the human microbiome.
- [Karl Broman](http://kbroman.org/), Univeristy of Wisconsin,
  Madison.
- [Wenyi Wang](http://odin.mdacc.tmc.edu/~wwang7/), M. D. Anderson
  Cancer Center. DeMix-Bayes: A Bayesian Model for the Deconvolution
  of Mixed Cancer Transcriptomes.
- [Zhijin Wu](http://www.stat.brown.edu/zwu/), Brown University.

## Community Speakers

- Aaron Lun, Walter and Eliza Hall Institute of Medical Research,
  Detecting differentially bound regions in ChIP-seq data with
  [csaw](http://bioconductor.org/packages/release/bioc/html/csaw.html).
- Leonard Goldstein, Genentech,
  [SGSeq](http://bioconductor.org/packages/release/bioc/html/SGSeq.html)
  and alternative transcripts.
- Nicole Deflaux, Google,
  [GoogleGenomics](http://bioconductor.org/packages/devel/bioc/html/GoogleGenomics.html).
- Paul Pyl, University of Copenhagen (DK), Metagenomics in
  Bioconductor.
- Wolfgang Huber, European Molecular Biology Laboratory, Weighted
  Hypothesis Testing.
- Laurent Gatto. _Bioconductor_ software for proteomics: (past),
  present and future.

## Workshops

- Aaron Lun.  Detecting differentially bound regions in ChIP-seq data
  with
  [csaw](http://bioconductor.org/packages/release/bioc/html/csaw.html).
- Andrzej Oles, Wolfgang Huber. Basics of image data and spatial
  patterns analysis in _R_.
- _Bioconductor_ Core Team. An introduction to _R_ and _Bioconductor_.
- Dan Tenenbaum, Tengfei Yin. Working with _docker_ and _AMI_
  containers for _R_, _Biconductor_, and common workflow language /
  [Rabix](https://www.rabix.org) integration.
- Greg Finak. Gating Designed Experiments With
  [openCyto](/packages/openCyto/).
- Herv&eacute; Pag&egrave;s, Michael Lawrence. Practical introduction
  to Bioconductor foundational data structures for high throughput
  sequencing analysis.
- James MacDonald. FAQ Live! Common questions and expert solutions,
  delivered in person.
- Kasper Hansen. Methylation.
- Leonard Goldstein
  [SGSeq](http://bioconductor.org/packages/release/bioc/html/SGSeq.html)
  and alternative splicing.
- Levi Waldron, Tim Triche, Aedin Culhane. Challenges and
  opportunities from integrative analysis of multi-omic data sets.
- Marc Carlson, Sonali Arora. _Bioconductor_ annotation resources.
- Mike Love. Differential expression, manipulation, and visualization
  of RNA-seq reads.
- Nicole Deflaux,
  [GoogleGenomics](http://bioconductor.org/packages/devel/bioc/html/GoogleGenomics.html).
- Thomas Girke. Automated NGS workflows with
  [systemPipeR](/packages/systemPipeR/) running on clusters or single
  machines, with a focus on VAR-seq.
- Valerie Obenchain, Martin Morgan, et al. Approaches to management
  and analysis of large genomic data.

# ![](/images/icons/calendar.gif) Tentative Schedule

## Developer Day

### Monday, July 20

Morning

- Project updates
- Developer / group activities.  Fast-pitch Challenge: better end-user
  tools for learning and using _R_ / _Bioconductor_.
- Contributed 'lightning' talks

Afternoon

- Shiny demo derby -- 'demolition derby: a competition in which cars
  are driven into each other until only one is left running' -- come
  prepared to demo your shiny apps and see how they compare with
  others from the community!
- Developer workshops. _Tentative_ possibilities: C / C++ integration
  (Nate Hayden); code analysis and test coverage with
  [lintr](https://github.com/jimhester/lintr) and
  [covr](https://github.com/jimhester/covr) (Jim Hester).
- Invited address

Evening

- Independent social activities

## Main Conference

### Tuesday, July 21

Morning:

- 8:30 - 11:15: Invited scientific presentations (3 x 45 minutes; 30
  minute coffee break)
- 11:15 - 12:00: Invited community presentations (3 x 15 minutes)

Afternoon

- 1:00pm - 2:45: Concurrent workshops
- 3:15pm - 5:00: Concurrent workshops

Evening

- 5:15 - 6:30pm: Poster session; social hour

### Wednesday, July 22

Morning:

- 8:30 - 11:15: Invited scientific presentations (3 x 45 minutes; 30
  minute coffee break)
- 11:15 - 12:00: Invited community presentations (3 x 15 minutes)

Afternoon

- 1:00pm - 2:45: Concurrent workshops
- 3:15pm - 5:00: Concurrent workshops

Evening

- 5:15 - 6:30pm: Social hour / conference conclusion

Presentations are made available after the conference; see
[course material](/help/course-materials/).

<span id="accomodations"></span>

# Accommodation

Hotels near the conference site (group rates no longer available.)

- [Residence Inn by Marriott â€“ Lake Union](http://www.marriott.com/hotels/travel/sealu-residence-inn-seattle-downtown-lake-union/)
- [Silver Cloud Inn - Lake Union](http://www.silvercloud.com/seattlelakeunion/)

# Workshop Abstracts

- Aaron Lun.  Detecting differentially bound regions in ChIP-seq data
  with
  [csaw](http://bioconductor.org/packages/release/bioc/html/csaw.html).

- Andrzej Oles, Wolfgang Huber. Basics of image data and spatial
  patterns analysis in _R_.

- _Bioconductor_ Core Team. An introduction to _R_ and _Bioconductor_.

  This lab provides an introduction to _R_ / _Bioconductor_ for
  high-throughput sequence analysis. It is designed for those who have
  some but not a lot of familiarity with _R_ and _Bioconductor_. The
  first part of the lab focuses on _R_ data types, functions, classes,
  methods, the package and help systems, and the Bioconductor web
  site. The second part of the lab takes a quick tour of essential
  packages, classes, and methods for sequence analysis. We will make
  brief stops at essential _Bioconductor_ packages like
  [GenomicRanges](http://bioconductor.org/packages/GenomicRanges),
  [Biostrings](http://bioconductor.org/packages/Biostrings),
  [GenomicAlignments](http://bioconductor.org/packages/GenomicAlignments),
  [GenomicFeatures](http://bioconductor.org/packages/GenomicFeatures),
  and [AnnotationHub](http://bioconductor.org/packages/AnnotationHub).

- Dan Tenenbaum, Tengfei Yin. Working with _docker_ and _AMI_
  containers for _R_, _Biconductor_, and common workflow language /
  [Rabix](https://www.rabix.org) integration.

- Greg Finak. Gating Designed Experiments With
  [openCyto](/packages/openCyto/).

  This workshop will cover flow cytometry and CyTOF data analysis in
  _Bioconductor_ using the OpenCyto framework. Users will learn to read
  in raw data, perform compensation and transformation and gating, and
  generate figures using the new ggcyto visualization framework based
  on ggplot. The focus will be on designed experiments where multiple
  samples are matched (e.g. as treatment and control). Users will
  learn how to use OpenCyto to leverage the positive and negative
  control samples to derive data-driven gates.
 
- Herv&eacute; Pag&egrave;s, Michael Lawrence. Practical introduction
  to Bioconductor foundational data structures for high throughput
  sequencing analysis.

- James MacDonald. FAQ Live! Common questions and expert solutions,
  delivered in person.

  Mystified by design matrices? Confused by contrasts? Bugged by
  errors? Then this is the workshop for you! In this workshop we will
  cover (at the very least) how to specify and understand design
  matrices and contrasts (critical for analyzing data using limma,
  DESeq/DESeq2, edgeR, etc), and how to interpret coefficients of a
  linear model. We will also show how to debug your (and others')
  code. This is a critical skill, not only for ensuring that your own
  code runs correctly, but for being able to provide feedback to
  package authors.

  These are the two of the most frequently asked questions on the
  Bioconductor support site, but you might have questions of your
  own. If you plan on attending this workshop, and have a particular
  question you would like to discuss, please email me: jmacdon at uw
  dot edu, and I will add your question to the list. There will be
  opportunities to ask questions during the workshop as well, but
  giving me time to prepare will increase the number of questions we
  can cover.

- Kasper Hansen. Methylation.

- Leonard Goldstein
  [SGSeq](http://bioconductor.org/packages/release/bioc/html/SGSeq.html)
  and alternative splicing.

- Levi Waldron, Tim Triche, Aedin Culhane. Challenges and
  opportunities from integrative analysis of multi-omic data sets.

- Marc Carlson, Sonali Arora. _Bioconductor_ annotation resources.

  Annotation is the process of adding valuable contextual information
  to your experimental findings. And the Bioconductor project has
  always had a large number of resources to assist with this. There
  have long been hundreds of annotation packages, many web services
  like [biomaRt](http://bioconductor.org/packages/biomaRt), and now
  there are also thousands of other valuable resources in the
  [AnnotationHub](http://bioconductor.org/packages/AnnotationHub).

  The first part of this lab will provide an overview of these
  different kinds of resources, and include demonstrations and
  exerices to help students learn both what is available and how these
  resources are commonly used. Topics covered will include accessing
  resources from the AnnotationHub, how to make use of the select
  interface with AnnotationDb objects and how to get range based data
  from TxDb or OrganismDb objects. This part of the lab is primarily
  aimed at new users who need to learn what sort of annotations are
  available and how to make use of them. It may also be useful for old
  timers who are still using outdated ways of accessing data like
  Bimap objects.

  The second part of this lab will focus on the new AnnotationHub, and
  on how to write recipes so that new resources can be added to the
  hub. The AnnotationHub has changed dramatically over the past year
  and we are interested in members of the community who might feel
  motivated to create recipes that can be used to expand the available
  content. So this second part of the lab will be aimed at more senior
  users who are interested seeing newer resources made available in
  the hub.

- Mike Love. Differential expression, manipulation, and visualization
  of RNA-seq reads.

- Nicole Deflaux,
  [GoogleGenomics](http://bioconductor.org/packages/devel/bioc/html/GoogleGenomics.html).

- Thomas Girke. Automated NGS workflows with
  [systemPipeR](/packages/systemPipeR/) running on clusters or single
  machines, with a focus on VAR-seq.

  This tutorial introduces systemPipeR, an R package designed for
  building end-to-end analysis pipelines with automated report
  generation for next generation sequence (NGS) applications. The
  package also provides support for running command-line software,
  such as NGS aligners and variant callers, on both single machines
  and compute clusters. The first part of the tutorial introduces the
  basic design of the package, and the second part gives an overview
  of a typical VAR-Seq analysis workflow including read
  QC/preprocessing, variant aware read alignments, variant calling,
  and annotations of SNPs and indels with genomic context information.

- Valerie Obenchain, Martin Morgan, et al. Approaches to management
  and analysis of large genomic data.

  This lab will cover strategies for managing large genomic
  data. Scalable computing techniques such as iteration, data
  restriction, file management and parallel evaluation will be
  discussed and demonstrated in analysis examples.

  The parallel section will include an orientation to the
  [BiocParallel](http://bioconductor.org/packages/BiocParallel)
  package. BiocParallel offers a unified API to the parallel, snow,
  BatchJobs and foreach packages. Each a solid package in its own
  right, BiocParallel aims to provide easy and consistent access to
  different parallel back-ends while preserving the individual
  strengths. Topics covered will include logging, error handling,
  monitoring progress of long running jobs, and setting timeout limits
  on workers. We'll work interactively with MulticoreParam and
  SnowParam and go though the steps of setting up a config file for
  using BatchJobsParam on a cluster.
