# BioC 2015: Where Software and Biology Connect

July 21-22, 2015 (Developer Day: July 20)<br />
Fred Hutchinson Cancer Research Center, Seattle, WA<br />
Obtain Your Amazon Machine Instance: 
    [https://courses.bioconductor.org](https://courses.bioconductor.org)<br />
twitter: [#BioC2015](https://twitter.com/search?q=%23BioC2015)<br />
Developer Day [notes](https://docs.google.com/document/d/1wJvWTxuCFc71acfSAZDNhUr8rifKXhGW8-Ugo6DVRyg/edit?usp=sharing)

This conference highlights current developments within and beyond
Bioconductor. Morning scientific talks and afternoon practicals
provide conference participants with insights and tools required for
the analysis and comprehension of high-throughput genomic
data. 'Developer Day' precedes the main conference on July 20,
providing developers and would-be developers an opportunity to gain
insights into project direction and software development best
practices.

- [Schedule](#schedule)
- [Workshop Abstracts](#workshop-abstracts)
- [Posters](#posters)
- [Code of Conduct](#code-of-conduct)
- [Parking](#parking)
- [Accommodation](#accommodation)

# ![](/images/icons/calendar.gif) Schedule

## Developer Day

### Monday, July 20

Morning -- 8:30 - 12:00, Arnold Building M1-A303/5/7 (times approximate)

- 8:30 Registration
- 8:50 Welcome and orientation
- 9:00 Project updates
- 9:45 Developer / group activities.  Fast-pitch Challenge: better end-user
  tools for learning and using _R_ / _Bioconductor_.
- 11:00 Contributed 'lightning' talks

Afternoon -- 1:00 - 5:00, Arnold Building M1-A303/5/7

- 1:00 Demo derby -- 'demolition derby: a competition in which cars are
  driven into each other until only one is left running' -- come
  prepared to demo your shiny or other interactive apps and see how
  they compare with others from the community!
- 2:00 Developer workshops. _Tentative_ possibilities: C / C++ integration;
  code analysis and test coverage with
  [lintr](https://github.com/jimhester/lintr) and
  [covr](https://github.com/jimhester/covr) (Jim Hester); packages for
  first-time contributors; data base and web resources in R; ...
- 4:00 Future directions -- presentation & discussion

Evening

- Independent social activities

## Main Conference

### Tuesday, July 21

Morning: Invited Scientific and Community Presentations -- Pelton Auditorium

- 8:30 - 9:15: [Cole Trapnell](http://cole-trapnell-lab.github.io/),
  University of Washington. Single cell trajectory analysis with
  [Monocle](/packages/monocle).
- 9:15 - 10:00: [Wenyi Wang](http://odin.mdacc.tmc.edu/~wwang7/),
  M. D. Anderson Cancer Center. DeMix-Bayes: A Bayesian Model for the
  Deconvolution of Mixed Cancer Transcriptomes.

Coffee Break, 10:00 - 10:30

- 10:30 - 11:15: [Karl Broman](http://kbroman.org/), University of
  Wisconsin, Madison. Interactive graphics for genetic data.
- 11:15 - 11:30: Paul Pyl, University of Copenhagen (DK), Metagenomics
  in Bioconductor.
- 11:30 - 11:45: Leonard Goldstein, Genentech,
  [SGSeq](http://bioconductor.org/packages/SGSeq) and alternative
  transcripts.
- 11:45 - 12:00: Nicole Deflaux, Google,
  [GoogleGenomics](http://bioconductor.org/packages/devel/bioc/html/GoogleGenomics.html).

Lunch, 12:00 - 1:00

Afternoon: Concurrent Workshops -- Pelton Auditorium and Arnold
Building M1-A303, 5, 7.

Session 1, 1:00pm - 2:45

- _Bioconductor_ Core Team. An introduction to _R_ and _Bioconductor_. Pelton.
- Dan Tenenbaum, Tengfei Yin,  Nan Xiao. Working with _Docker_
  containers for _R_, _Bioconductor_, and common workflow language /
  [Rabix](https://www.rabix.org) integration. M1-A305.
- Greg Finak. Gating Designed Experiments With
  [openCyto](/packages/openCyto/). M1-A303.
- Mike Love. Differential expression, manipulation, and
  visualization of RNA-seq reads. M1-A307.

Coffee Break, 2:45 - 3:15

Session 2, 3:15pm - 5:00

- Andrzej Oles, Wolfgang Huber. Basics of image data and spatial
  patterns analysis in _R_. M1-A303.
- James MacDonald. FAQ Live! Common questions and expert solutions,
  delivered in person. Pelton.
- Kasper Hansen. Methylation and the analysis of Illumina 450k microarrays. M1-A307.
- Nicole Deflaux, Siddhartha Bagaria, and Craig Citro,
  [GoogleGenomics](http://bioconductor.org/packages/devel/bioc/html/GoogleGenomics.html). M1-A305.

Evening, 5:15 - 6:30: Poster session; social hour

### Wednesday, July 22

Morning: Invited Scientific and Community Presentations -- Pelton Auditorium

- 8:30 - 9:15: [Eric Franzosa](http://huttenhower.sph.harvard.edu/),
  Huttenhower Lab, Harvard School of Public Health. Identifying
  personal microbiomes using metagenomic codes.
- 9:15 - 10:00: [Zhijin Wu](http://www.stat.brown.edu/zwu/), Brown
  University. Gene set analysis for large and sparse effects.

Coffee Break, 10:00 - 10:30

- 10:30 - 11:15: [Artem Sokolov](https://users.soe.ucsc.edu/~sokolov/),
  University of California, Santa Cruz. Integrating prior biological
  knowledge using Generalized Elastic Nets.
- 11:15 - 11:30: Aaron Lun, Walter and Eliza Hall Institute of Medical
  Research, Detecting differentially bound regions in ChIP-seq data
  with [csaw](http://bioconductor.org/packages/csaw).
- 11:30 - 11:45: Wolfgang Huber, European Molecular Biology
  Laboratory, Weighted Hypothesis Testing.
- 11:45 - 12:00: Laurent Gatto. _Bioconductor_ software for
  proteomics: (past), present and future.

Lunch, 12:00 - 1:00

Afternoon: Concurrent Workshops -- Pelton Auditorium and Arnold
Building M1-A303, 5, 7.

Session 3, 1:00pm - 2:45

- Marc Carlson, Sonali Arora. _Bioconductor_ annotation
  resources. M1-A303.
- Aaron Lun.  Detecting differentially bound regions in ChIP-seq
  data with
  [csaw](http://bioconductor.org/packages/csaw). M1-A307.
- Levi Waldron, Tim Triche, Aedin Culhane. Challenges and
  opportunities from integrative analysis of multi-omic data
  sets. Pelton.
- Thomas Girke. Automated NGS workflows with
  [systemPipeR](/packages/systemPipeR/) running on clusters or
  single machines, with a focus on VAR-seq. M1-A305.

Coffee Break, 2:45 - 3:15

Session 4,  3:15 - 5:00

- Leonard Goldstein [SGSeq](http://bioconductor.org/packages/SGSeq)
  and alternative splicing. M1-A305.
- Herv&eacute; Pag&egrave;s, Michael Lawrence. Practical
  introduction to _Bioconductor_ foundational data structures for high
  throughput sequencing analysis. M1-A307.
- Valerie Obenchain, Martin Morgan. Approaches to management and
  analysis of large genomic data. M1-A303.

Evening (5:15 - 6:30): Poster session; social hour

Presentations are made available after the conference; see
[course material](/help/course-materials/).

# Workshop Abstracts

- Aaron Lun.  Detecting differentially bound regions in ChIP-seq data
  with [csaw](http://bioconductor.org/packages/csaw).

  This workshop will use the csaw package to perform a differential
  binding analysis of a public ChIP-seq data set. Participants will
  run through all steps of the _csaw_ analysis pipeline, including
  counting reads into windows, abundance-based filtering,
  normalization of library-specific biases and statistical
  modelling. A basic understanding of the ChIP-seq procedure is
  expected.

- Andrzej Oles, Wolfgang Huber. Basics of image data and spatial
  patterns analysis in _R_.

  We demonstrate how image data can be processed in R, how
  quantitative information can be extracted from images, and how
  statistical methods can be used to explore and understand the
  data. In particular, we use
  [EBImage](http://bioconductor.org/packages/EBImage) to load and
  display images in R, and to manipulate, transform and apply filters
  to them. Using microscopy of cellular assays as an example, we show
  how to perform image segmentation followed by the extraction of
  quantitative object characteristics. Such information can be then
  used for downstream analyses, for example, analysis of spatial point
  patterns with the help of spatstat.

  Presenter: Andrzej Oles

- _Bioconductor_ Core Team. An introduction to _R_ and _Bioconductor_.

  This lab provides an introduction to _R_ / _Bioconductor_ for
  high-throughput sequence analysis. It is designed for those who have
  some but not a lot of familiarity with _R_ and _Bioconductor_. The
  first part of the lab focuses on _R_ data types, functions, classes,
  methods, the package and help systems, and the Bioconductor web
  site. The second part of the lab takes a quick tour of essential
  packages, classes, and methods for sequence analysis. We will make
  brief stops at essential _Bioconductor_ packages like
  [GenomicRanges](http://bioconductor.org/packages/GenomicRanges),
  [Biostrings](http://bioconductor.org/packages/Biostrings),
  [GenomicAlignments](http://bioconductor.org/packages/GenomicAlignments),
  [GenomicFeatures](http://bioconductor.org/packages/GenomicFeatures),
  and [AnnotationHub](http://bioconductor.org/packages/AnnotationHub).

- Dan Tenenbaum, Tengfei Yin,  Nan Xiao. Working with _Docker_
  containers for _R_, _Bioconductor_, and common workflow language /
  [Rabix](https://www.rabix.org) integration.

  Part I (Dan Tenenbaum): An introduction to Docker and the _Bioconductor_ Docker
  containers

  What is Docker and how can it help you? We'll introduce Docker and
  its idea of containerized applications, and how they can help with
  reproducibility and aspects of development (identical testing
  environments, a blank slate, pre-installed
  dependencies). _Bioconductor_ provides pre-built containers which
  already contain sets of packages used for various workflows. These
  containers can be used directly or you can in turn build upon them
  to make your own containers.  We'll discuss and demonstrate a number
  of use cases, and end with an exercise where you create your own
  container.

  Part II (Tengfei Yin,  Nan Xiao): We will introduce [common workflow language](http://common-workflow-language.github.io/draft-2/) and R package _cwl_, the implementation with [Rabix](https://www.rabix.org/) , then a demo about how to write R command line tool with docopt, how to convert your R command line tool to CWL, how to use rabix R package's R interface to describe your tool,  and use Rabix to develop, deploy and run it on AWS cloud with SBG platform or run it locally. We will also demonstrate dockerizing R Markdown documents with Rabix support using the [liftr](http://liftr.me) package; automating a workflow from raw data uploading, pipeline running, and report retrieving with the [sbgr](https://github.com/road2stat/sbgr) API package.
  
- Greg Finak. Gating Designed Experiments With
  [openCyto](/packages/openCyto/).

  This workshop will cover flow cytometry and CyTOF data analysis in
  _Bioconductor_ using the OpenCyto framework. Users will learn to read
  in raw data, perform compensation and transformation and gating, and
  generate figures using the new ggcyto visualization framework based
  on ggplot. The focus will be on designed experiments where multiple
  samples are matched (e.g. as treatment and control). Users will
  learn how to use OpenCyto to leverage the positive and negative
  control samples to derive data-driven gates.
 
- Herv&eacute; Pag&egrave;s, Michael Lawrence. Practical introduction
  to Bioconductor foundational data structures for high throughput
  sequencing analysis.

- James MacDonald. FAQ Live! Common questions and expert solutions,
  delivered in person.

  Mystified by design matrices? Confused by contrasts? Bugged by
  errors? Then this is the workshop for you! In this workshop we will
  cover (at the very least) how to specify and understand design
  matrices and contrasts (critical for analyzing data using limma,
  DESeq/DESeq2, edgeR, etc), and how to interpret coefficients of a
  linear model. We will also show how to debug your (and others')
  code. This is a critical skill, not only for ensuring that your own
  code runs correctly, but for being able to provide feedback to
  package authors.

  These are the two of the most frequently asked questions on the
  Bioconductor support site, but you might have questions of your
  own. If you plan on attending this workshop, and have a particular
  question you would like to discuss, please email me: jmacdon at uw
  dot edu, and I will add your question to the list. There will be
  opportunities to ask questions during the workshop as well, but
  giving me time to prepare will increase the number of questions we
  can cover.

- Kasper Hansen. Methylation and the analysis of Illumina 450k
  microarrays. In this tutorial we will present tools for analysis of
  DNA methylation data.  We will focus on using the
  [minfi](http://bioconductor.org/packages/minfi) package for analysis
  of the Illumina 450k methylation microarray.  Time permitting, we
  will also discuss using the bsseq package for analysis of whole
  genome bisulfite sequencing.

- Leonard Goldstein [SGSeq](http://bioconductor.org/packages/SGSeq)
  and alternative splicing.

  The [SGSeq](http://bioconductor.org/packages/SGSeq) package provides
  a framework for analyzing annotated and novel splice events from
  RNA-seq data. SGSeq predicts exons and splice junctions from reads
  aligned against a reference genome and assembles them into a
  genome-wide splice graph. Splice events are identified from the
  graph and quantified using reads spanning event boundaries.  This
  workshop provides an introduction to SGSeq functionality, including
  splice event detection, quantification, annotation and
  visualization.

- Levi Waldron, Tim Triche, Aedin Culhane. Challenges and
  opportunities from integrative analysis of multi-omic data sets.

  The falling cost of genomic assays facilitates collection of
  multi-assay genomic data (e.g., gene and transcript expression,
  structural variation, copy number, methylation, and microRNA data)
  from a set of biological specimens. This workshop takes participants
  through acquisition of multi-assay -omics data from major cancer
  genomics projects, representation of these complex data in
  Bioconductor, and integrative analysis for quality control,
  visualization, and statistical inference spanning data types.

- Marc Carlson, Sonali Arora. _Bioconductor_ annotation resources.

  Annotation is the process of adding valuable contextual information
  to your experimental findings. And the Bioconductor project has
  always had a large number of resources to assist with this. There
  have long been hundreds of annotation packages, many web services
  like [biomaRt](http://bioconductor.org/packages/biomaRt), and now
  there are also thousands of other valuable resources in the
  [AnnotationHub](http://bioconductor.org/packages/AnnotationHub).

  The first part of this lab will provide an overview of these
  different kinds of resources, and include demonstrations and
  exercises to help students learn both what is available and how these
  resources are commonly used. Topics covered will include accessing
  resources from the AnnotationHub, how to make use of the select
  interface with AnnotationDb objects and how to get range based data
  from TxDb or OrganismDb objects. This part of the lab is primarily
  aimed at new users who need to learn what sort of annotations are
  available and how to make use of them. It may also be useful for old
  timers who are still using outdated ways of accessing data like
  Bimap objects.

  The second part of this lab will focus on the new AnnotationHub, and
  on how to write recipes so that new resources can be added to the
  hub. The AnnotationHub has changed dramatically over the past year
  and we are interested in members of the community who might feel
  motivated to create recipes that can be used to expand the available
  content. So this second part of the lab will be aimed at more senior
  users who are interested seeing newer resources made available in
  the hub.

- Mike Love. Differential expression, manipulation, and visualization
  of RNA-seq reads.

  We will cover basic steps in RNA-seq analysis using a variety of
  _Bioconductor packages_, including: loading gene annotations from a
  variety of sources
  ([GenomicFeatures](http://bioconductor.org/packages/GenomicFeatures)),
  creating a count table which can be used by a variety of statistical
  packages within _Bioconductor_
  ([GenomicAlignments](http://bioconductor.org/packages/GenomicAlignments),
  [Rsubread](http://bioconductor.org/packages/Rsubread)), exploratory
  analysis, visualization and differential expression testing
  ([DESeq2](http://bioconductor.org/packages/DESeq2)), annotation of
  results tables
  ([AnnotationDbi](http://bioconductor.org/packages/AnnotationDbi)),
  generation of HTML reports
  ([ReportingTools](http://bioconductor.org/packages/ReportingTools)),
  and tools for examining alignments (GenomicAlignments).

- Nicole Deflaux, Siddhartha Bagaria and Craig Citro,
  [GoogleGenomics](http://bioconductor.org/packages/devel/bioc/html/GoogleGenomics.html).

  Google has some pretty amazing big data computational "hammers" that
  they have been applying to search and video data for a long time.
  In this workshop we take those same hammers and apply them to whole
  genome sequences.

  We'll work with both the
  [1,000 Genomes](http://www.1000genomes.org/) reads and variants and
  also the
  [Illumina Platinum Genomes](http://www.illumina.com/platinumgenomes/)
  gVCF variants.

  We do this all from the comfort of the R prompt using common
  packages including
  [VariantAnnotation](http://bioconductor.org/packages/VariantAnnotation),
  [ggbio](http://bioconductor.org/packages/ggbio),
  [ggplot2](http://bioconductor.org/packages/ggplot2),
  [dplyr](http://bioconductor.org/packages/dplyr),
  [bigrquery](http://bioconductor.org/packages/bigrquery), and the new
  Bioconductor package
  [GoogleGenomics](http://bioconductor.org/packages/GoogleGenomics)
  which provides an R interface to Google's implementation of the
  [Global Alliance for Genomics and Health](http://ga4gh.org/#/ap)
  API.

  And we'll do this in a reproducible fashion running RMarkdown files
  via [Dockerized Bioconductor](http://bioconductor.org/help/docker/)
  running on
  [Google Compute Engine](https://cloud.google.com/compute/) VMs!

- Thomas Girke. Automated NGS workflows with
  [systemPipeR](/packages/systemPipeR/) running on clusters or single
  machines, with a focus on VAR-seq.

  This tutorial introduces systemPipeR, an R package designed for
  building end-to-end analysis pipelines with automated report
  generation for next generation sequence (NGS) applications. The
  package also provides support for running command-line software,
  such as NGS aligners and variant callers, on both single machines
  and compute clusters. The first part of the tutorial introduces the
  basic design of the package, and the second part gives an overview
  of a typical VAR-Seq analysis workflow including read
  QC/preprocessing, variant aware read alignments, variant calling,
  and annotations of SNPs and indels with genomic context information.

- Valerie Obenchain, Martin Morgan, et al. Management
  and analysis of large genomic data.

  This lab will cover strategies for managing large genomic
  data. Scalable computing techniques such as iteration, data
  restriction, file management and parallel evaluation will be
  discussed and demonstrated in analysis examples.

  The parallel section will include an orientation to the
  [BiocParallel](http://bioconductor.org/packages/BiocParallel)
  package. BiocParallel offers a unified API to the parallel, snow,
  BatchJobs and foreach packages. Each a solid package in its own
  right, BiocParallel aims to provide easy and consistent access to
  different parallel back-ends while preserving the individual
  strengths. Topics covered will include logging, error handling,
  monitoring progress of long running jobs, and setting timeout limits
  on workers.

# Posters

- Jenny Drnevich, University of Illinois. An online platform for NGS trainers
  to share teaching experience and materials.

  Introduction. Over the last decade, the exponential increase in the use of next
  generation sequencing (NGS) applications has led to a high demand for
  researchers who are capable of analysing such data. Consequently, demand for
  training in this area has increased; researchers with experience in this
  analysis are often tasked with training other scientists and have to dedicate a
  significant portion of their time to prepare lectures and practicals, time
  which is already limited by their research projects.
 
  Aim. To tackle this issue, we wanted to create a system for NGS trainers to
  exchange teaching experience and materials, therefore not only greatly reducing
  the time required to put together lectures and practicals, but also promoting
  exchange between NGS trainers across the globe in an effort to create a
  community of NGS trainers, to share best training practices and to improve
  their teaching in order to equip scientists with the skill set needed to
  analyse and interpret their data. 
 
  Results. We here present a platform for trainers to upload their course
  materials using the "Git" versioning software as a back-end, according to a
  common set of descriptors. The descriptions are uploaded as markdown files
  using a set of pre-defined keywords to categorise the materials. A simple user
  interface allows trainers to search for the descriptive files using such
  keywords and retrieve specific modules from the materials repository.
  Information about the repository is available 
  [here](https://bioinformatics.upsc.se/ngs-training) and content will be
  accessible through the Goblet training portal.

- Shilin Zhao, Vanderbilt University. An R package to identify 
  context-dependent functional transcription factor pairs.

  Background. Transcription factors (TFs) are fundamental regulators of gene
  expression and generally function in a complex and cooperative manner.
  Identifying context-dependent cooperative TFs is essential for understanding
  how cells respond to environmental change. The huge amount of various omics
  data currently available, providing genome-wide physical binding and functional
  effect information about transcription, holds a great opportunity to study TF
  cooperativity. 
 
  Results. Here we developed an R package, FunTFPair, which provides an easy and
  powerful way to identify condition-specific TF pairs by integrating
  transcription factor binding sites from ENCODE and gene expression profiles
  from GEO. Users only need to provide the GEO ID that they are interested in.
  FunTFPair will automatically retrieve the expression profiles of the input GEO
  ID, get TF targets from ENCODE, and identify TF pairs whose common targets show
  statistically significant differences under changed experimental conditions or
  exhibit coordinated transcription in a specific condition. The functional TF
  pairs and their relative importance will be reported in a TF cooperation
  network. Two datasets from GEO are used as examples to demonstrate the usage
  and reliability of the package. 
 
  Conclusions. FunTFPair provides a simple and powerful way to explore potential
  cooperative TFs under specific conditions that users are interested in. The
  package is under development and currently available in a
  [github repository](https://github.com/slzhao/FunTFPair). We plan to submit
  to Bioconductor in the near future.

- Ning  Leng, Morgridge Institute for Research. Oscope: a statistical 
  pipeline for identifying oscillatory genes in unsynchronized single cell 
  RNA-seq experiments

  Oscillatory gene expression is fundamental to mammalian development,
  but technologies to monitor expression oscillations are limited. We
  have developed a statistical approach called Oscope to identify and
  characterize the transcriptional dynamics of oscillating genes in
  single-cell RNA-seq data from an unsynchronized cell population.
  Applications to a number of data sets demonstrate the utility of the
  approach and also identify a potential artifact in the Fluidigm C1
  platform.

- Sebastiano Battaglia, Roswell Park Cancer Institute. Genomic profile of 
  nuclear receptors-mediated transcription in prostate cancer.

  Prostate cancer (PCa) is the most commonly diagnosed cancer in the US and
  second most common cause of cancer death. Initial therapeutic approaches aim
  to reduce tumor burden through androgen deprivation therapy (ADT) but about
  half of the patients will recur, developing castration recurrent PCa (CRPCa),
  often clinically lethal. The androgen receptor (AR) is a key mediator of PCa
  growth in physiological and malignant conditions and AR inhibition is proven
  to be effective as first line of therapy for PCa patients, however, it is
  insufficient for patients with CRPCa. The vitamin D receptor (VDR) mediates
  the intracellular effects of 1,25(OH)2D3 (vitamin D) and numerous studies
  evaluated the in-vitro and in-vivo effects of vitamin D, however, vitamin D
  antitumor effect is lost in advanced cancers. The retinoic acid receptors
  (RARs) modulate cell differentiation and proliferation and retinoid
  metabolism is altered in PCa. AR, VDR and RARs belong to the nuclear receptor
  (NR) superfamily and their activity is strictly regulated by coregulatory
  complexes. The lysine specific demethylase 1A (LSD1) is a transcriptional
  regulator whose expression correlates with cancer aggressiveness and LSD1
  inhibition is proposed as antineoplastic approach in the clinical settings.
  Here we demonstrate the effects of a novel combinatorial therapy by targeting
  LSD1 and multiple NR.  Furthermore, by integrating RNASeq and ChIPSeq data we
  describe the effects of LSD1 as dual coregulator for AR, VDR and RARs and
  show that LSD1 modulates unique NR-activated transcriptional pathways,
  including MTORC1 and MYC signaling networks. We conclude that LSD1 is a locus
  specific coactivator and corepressor and, by modulating key oncogenic
  pathways, it regulates cancer progression and therapeutic response.

- Mark Dane, Oregon Health and Science University. Computational Pipelines for 
  High Content Screening of Microenvironment Microarrays 

  The OHSU MEP-LINCS project is developing a dataset and computational strategy
  to elucidate how signals from the microenvironment affect observable cellular
  phenotypes and intracellular transcriptional and proteomic networks. We are
  using microenvironment microarrays (MEMAs) and immunofluorescence imaging to
  capture phenotypes of cells grown on pairwise combinations of extracellular
  matrix proteins and ligands. We have developed two computational pipelines that
  create fully annotated datasets from the pilot MEP-LINCs high throughput
  screening experiments. The first pipeline processes population-level data that
  can be acquired and analyzed within an hour of completing an experiment. This
  QA pipeline provides feedback on experiment processes immediately after they
  occur.    

  The cell-level pipeline is designed to handle high-content imaging
  data in an automated fashion.  The essential tasks are to pre-process,
  normalize, and assess quality of the image feature data. Finally, we have
  developed computational approaches that prioritize microenvironment
  perturbagens (MEPs) for follow-on validation experiments. The pipeline is based
  on open source R packages and will contribute a MEMA package for public use.
  The cell-level dataset and accompanying analysis will be available for public
  download through the 
  [LINCs consortia](http://www.lincsproject.org/).

- Dr. Yuriy Gusev, Georgetown University. Chromosome instability index CIN 
  package for Bioconductor.

  The CIN package calculates the chromosome instability (CIN) index that
  allows to quantitatively characterize genome-wide copy number alterations
  as a measure of chromosome instability. The algorithm for this method will
  be described in a paper (in preparation).

  Genomic instability is known to be a fundamental trait in the development of
  tumors; and most human tumors exhibit this instability in structural and
  numerical alterations: deletions, amplifications, inversions or even losses
  and gains of whole chromosomes or chromosomes arms. The chromosome
  instability indicated by these copy number alternations is associated with
  various events in the development or the severity of tumors in terms of
  clinical outcome.

  To mathematically and quantitatively describe these alternations we first
  locate their genomic positions and measure their ranges. Such algorithms are
  referred to as “segmentation algorithms”.

  The CIN module accepts these segmentation results and calculates the genomic
  instability across a chromosome for a global view (referred to as “Chromosome
  CIN”, or “Standard or Regular CIN”), and the genomic instability across
  cytobands regions for higher resolution (referred to as “Cytobands CIN”).
  This allows to assess the impacts of copy number alternations on various
  biological events or clinical outcomes by studying the association of CIN
  indices with those events.

  The CIN Bioconductor package allows the automated processing of the
  experimental copy number data generated by Affymetrix SNP 6.0 arrays or
  similar high throughput technologies. An older version of this algorithm that
  shows overall instability has been integrated into G-DOC web portal and made
  available for users as part of the G-DOC Plus analysis tools 
  [here](https://gdoc.georgetown.edu). The CIN Bioconductor package calculates
  not only overall instability, but also gains and losses at the chromosome and
  cytoband level.

- Allison Thompson, Pacific Northwest National Laboratory. Signatures of 
  Climate Change: Implementation of High-Throughput Sequence Data Analysis 
  Methods.

  Abstract: Analyzing high-throughput sequencing data is particularly
  challenging due to the nature of the data collected. Before analysis can be
  done across samples, normalization must be performed to account for
  differences in sequencing depths and consequently make samples comparable.
  Additionally, traditional statistical tests for differential expression are
  not appropriate with the count data that is produced by sequencing. While
  numerous normalization and differential abundance algorithms for sequencing
  data exist, there is no clear optimal method in either case. This research
  focuses on comparing and contrasting normalization and differential testing
  methods and the resulting biological inferences.  Here we focus on the
  methods of DESeq and edgeR for pairwise tests and use big data tools
  developed by the Tessera project to do exploratory data analysis and examine
  cases where the two tests differ.

- Andrew J. Bass, Princeton University. superSeq: Assessing the limits of 
  sequencing depth through read subsampling.

  RNA-­Seq is a standard gene expression profiling technology for differential
  expression analysis. In RNA-­Seq studies, the read depth strongly affects the
  power of the test statistics, such that larger read depths induce higher
  statistical power. After a certain read depth, the power of the test
  statistics begins to asymptote, at which point there are only marginal
  improvements in power. Although existing methods, such as subSeq, can help
  determine if the read depth of an experiment is saturated, they are limited
  in that they do not provide a way of estimating the appropriate read depth
  for under-­ saturated experiments. We provide a new method called superSeq
  that models and estimates the increase in statistical power that would result
  in increasing the read depth for a given experiment. We then apply the
  superSeq framework to 38 RNA-­Seq experiments in the Expression Atlas. In the
  majority of the studies, the method accurately predicts the relationship
  between the power of the test statistics and the read depth. Researchers can
  thus use this method, implemented in the forthcoming R package superSeq , to
  determine the appropriate read depth for a completed experiment in order to
  maximize the statistical power.

- Cunye Qiao, Health Canada. Identify potential critical windows of fetal
  development during the 8-19 week time frame.

# Code of Conduct

_Bioconductor_ is built on the free and open exchange of scientific
ideas, and the contributions of our diverse user community.  In this
spirit, _BioC 2015_ is dedicated to providing a harassment-free
conference experience for everyone.  Harassment of any form (verbal,
physical, sexual, or other) will not be tolerated in talks, workshops,
poster sessions, social activities, or online.

# Parking

Space is designated in front of the Thomas, Fairview and Yale
buildings and in the Arnold building garage for a limited number of
visitors on a first come, first serve basis. Free visitor parking is
limited to two hours. Street parking and paid lots are nearby. See the
[Fred Hutch visitor's website](https://www.fredhutch.org/en/contact-us/visit-us.html)
for more details.

# Accommodation

Hotels near the conference site (group rates no longer available.)

- [Residence Inn by Marriott - Lake Union](http://www.marriott.com/hotels/travel/sealu-residence-inn-seattle-downtown-lake-union/)
- [Silver Cloud Inn - Lake Union](http://www.silvercloud.com/seattlelakeunion/)
