# BioC 2015: Where Software and Biology Connect

July 21-22, 2015 (Developer Day: July 20)<br />
Fred Hutchinson Cancer Research Center, Seattle, WA

This conference highlights current developments within and beyond
Bioconductor. Morning scientific talks and afternoon practicals
provide conference participants with insights and tools required for
the analysis and comprehension of high-throughput genomic
data. 'Developer Day' precedes the main conference on July 20,
providing developers and would-be developers an opportunity to gain
insights into project direction and software development best
practices.

# ![](/images/icons/magnifier.gif) Speakers and Workshops

## Distinguished Speakers

- [Artem Sokolov](https://users.soe.ucsc.edu/~sokolov/), University of
  California, Santa Cruz. Integrating prior biological knowledge using
  Generalized Elastic Nets.
- [Cole Trapnell](http://cole-trapnell-lab.github.io/), University of
  Washington. Single cell trajectory analysis with
  [Monocle](/packages/release/bioc/html/monocle.html).
- [Eric Franzosa](http://huttenhower.sph.harvard.edu/), Huttenhower
  Lab, Harvard School of Public Health. Integrated multi'omic analysis
  methods for epidemiological profiling of the human microbiome.
- [Karl Broman](http://kbroman.org/), Univeristy of Wisconsin,
  Madison.
- [Wenyi Wang](http://odin.mdacc.tmc.edu/~wwang7/), M. D. Anderson
  Cancer Center. DeMix-Bayes: A Bayesian Model for the Deconvolution
  of Mixed Cancer Transcriptomes.
- [Zhijin Wu](http://www.stat.brown.edu/zwu/), Brown University.

## Community Speakers

- Aaron Lun, Walter and Eliza Hall Institute of Medical Research,
  Detecting differentially bound regions in ChIP-seq data with
  [csaw](http://bioconductor.org/packages/release/bioc/html/csaw.html).
- Leonard Goldstein, Genentech,
  [SGSeq](http://bioconductor.org/packages/release/bioc/html/SGSeq.html)
  and alternative transcripts.
- Nicole Deflaux, Google,
  [GoogleGenomics](http://bioconductor.org/packages/devel/bioc/html/GoogleGenomics.html).
- Paul Pyl, University of Copenhagen (DK), Metagenomics in
  Bioconductor.
- Wolfgang Huber, European Molecular Biology Laboratory, Weighted
  Hypothesis Testing.
- Additional TBA.

## Workshops

- Aaron Lun.  Detecting differentially bound regions in ChIP-seq data
  with
  [csaw](http://bioconductor.org/packages/release/bioc/html/csaw.html).
- Andrzej Oles, Wolfgang Huber. Basics of image data and spatial
  patterns analysis in _R_.
- _Bioconductor_ Core Team. An introduction to _R_ and _Bioconductor_.
- Dan Tenenbaum, Tengfei Yin. Working with _docker_ and _AMI_
  containers for _R_, _Biconductor_, and common workflow language /
  [Rabix](https://www.rabix.org) integration.
- Greg Finak. Gating Designed Experiments With
  [openCyto](/packages/openCyto/).
- Herv&eacute; Pag&egrave;s, Michael Lawrence. Practical introduction
  to Bioconductor foundational data structures for high throughput
  sequencing analysis.
- James MacDonald. FAQ Live! Common questions and expert solutions,
  delivered in person.
- Kasper Hansen. Methylation.
- Leonard Goldstein
  [SGSeq](http://bioconductor.org/packages/release/bioc/html/SGSeq.html)
  and alternative splicing.
- Levi Waldron, Tim Triche, Aedin Culhane. Challenges and
  opportunities from integrative analysis of multi-omic data sets.
- Marc Carlson, Sonali Arora. _Bioconductor_ annotation resources.
- Mike Love. Differential expression, manipulation, and visualization
  of RNA-seq reads.
- Nicole Deflaux,
  [GoogleGenomics](http://bioconductor.org/packages/devel/bioc/html/GoogleGenomics.html).
- Thomas Girke. Automated NGS workflows with
  [systemPipeR](/packages/systemPipeR/) running on clusters or single
  machines, with a focus on VAR-seq.
- Valerie Obenchain, Martin Morgan, et al. Approaches to management
  and analysis of large genomic data.

# ![](/images/icons/calendar.gif) Tentative Schedule

## Developer Day

### Monday, July 20

Morning

- Project updates
- Developer / group activities.  Fast-pitch Challenge: better end-user
  tools for learning and using _R_ / _Bioconductor_.
- Contributed 'lightning' talks

Afternoon

- Developer workshops. _Tentative_ possibilities: C / C++ integration
  (Nate Hayden); code analysis and test coverage with
  [lintr](https://github.com/jimhester/lintr) and
  [covr](https://github.com/jimhester/covr) (Jim Hester).
- Invited address

Evening

- Independent social activities

## Main Conference

### Tuesday, July 21

Morning:

- 8:30 - 11:15: Invited scientific presentations (3 x 45 minutes; 30
  minute coffee break)
- 11:15 - 12:00: Invited community presentations (3 x 15 minutes)

Afternoon

- 1:00pm - 2:45: Concurrent workshops
- 3:15pm - 5:00: Concurrent workshops

Evening

- 5:15 - 6:30pm: Poster session; social hour

### Wednesday, July 22

Morning:

- 8:30 - 11:15: Invited scientific presentations (3 x 45 minutes; 30
  minute coffee break)
- 11:15 - 12:00: Invited community presentations (3 x 15 minutes)

Afternoon

- 1:00pm - 2:45: Concurrent workshops
- 3:15pm - 5:00: Concurrent workshops

Evening

- 5:15 - 6:30pm: Social hour / conference conclusion

Presentations are made available after the conference; see
[course material](/help/course-materials/).

<span id="accomodations"></span>

## Hotels near the conference site

#### Residence Inn by Marriott â€“ Lake Union

[Group rate](http://www.marriott.com/meeting-event-hotels/group-corporate-travel/groupCorp.mi?resLinkData=BioC
Conference%5ESEALU%60BIOBIOA%7CBIOBIOB%60323-349%60USD%60false%607/19/15%607/22/15%606/4/15&app=resvlink&stop_mobi=yes)
for BioC Conference - available until June 4. 


#### [Silver Cloud Inn - Lake Union](http://www.silvercloud.com/seattlelakeunion/)

Call 1.800.330.5812 for group rate, available until
May 20. Use code "FHCRC Bio C Conference".

