

                    <a name="documentContent"></a>

                    

                    

                    
    


        <h1 class="documentFirstHeading">Milan 2003</h1>
        
        
    
        <div class="documentDescription"></div>

    
        
    
        <div class="plain">
            &nbsp;
<center>
<p><b><font color="#000099">A short course on</font></b><br />
<b><font color="#000099"><font size="+2"><a href="http://www.economia.unimi.it/marray/">Computational and
Statistical Aspects of Microarray Analysis</a></font></font></b>
<br /><br />University of Milan
<br />May 26-30, 2003
<br /><br />Lecturers:<br />
<a href="http://www-lmc.imag.fr/lmc-sms/Anestis.Antoniadis/">Anestis Antoniadis</a><br />
<a href="http://biosun1.harvard.edu/~rgentlem/HTML/home.html">Robert Gentleman</a>
</p></center>
<br />
<center><p><b><font color="#000099">Lecture notes</font></b></p></center>
<table border>
<tr>
<td><b>Monday</b></td>
<td><b>Tuesday</b></td>
<td><b>Wednesday</b></td>
<td><b>Thursday</b></td>
<td><b>Friday</b></td>
</tr>
<tr>
<td><a href="Lectures/GenBio.pdf">Introduction to Genome
Biology</a></td>
<td><a href="Lectures/OligoArrays.pdf">DNA Microarray Data,
Oligonucleotide Arrays</a></td>
<td><a href="Lectures/Filtering.pdf">Microarray Experiments</a></td>
<td><a href="Lectures/MachineLearning.pdf">Some Things Every Biologist
Should Know About Machine Learning</a></td>
<td><a href="Lectures/classif.pdf">Classification in Microarray
Experiments</a></td>
</tr>
<tr>
<td><a href="Lectures/MilanR.pdf">R/S Programming Techniques</a></td>
<td><a href="Lectures/MilanBioC.pdf">Introduction to Bioconductor</a></td>
<td><a href="Lectures/dist.pdf">Distances and Expression
Measures</a></td>
<td><a href="Lectures/anestisMilan3.pdf">Penalized logistic regression
and classification of microarray data</a></td>
<td><a href="Lectures/anestisMilan4.pdf">Dimension Reduction
Techniques For Classification</a></td>
</tr>
<tr>
<td>&nbsp</td>
<td>&nbsp</td>
<td><a href="Lectures/anestisMilan1.pdf">Wavelets and Gene Selection
by Multiple Testing</a></td>
<td><a href="Lectures/anestisMilan3.pdf">Penalized Logistic Regression
and Classification of Microarray Data</a></td>
<td><a href="Lectures/anestisMilan4.pdf">Dimension Reduction
Techniques For Classification</a></td>
</tr>
<tr>
<td>&nbsp</td>
<td>&nbsp</td>
<td>&nbsp</td>
<td>&nbsp</td>
<td><a href="Lectures/anestisMilan2.pdf">Some Statistical Methods for
the Identification of Differentially Expressed Genes</a></td>
</tr>
</table>


<p><b><font color="#000099">Lab materials</font></b>
<br />&nbsp;
</p><table border nosave width="75%">
<tr nosave>
<td nowrap nosave>Lab1&nbsp;</td>

<td>Bioconductor Basics</td>

<td nosave>
<center><a href="PDF/Lab1.pdf">.pdf</a></center>
</td>

<td>
<center><a href="Rnw/Lab1.Rnw">.Rnw</a></center>
</td>

<td>
<center><a href="RCode/Lab1.R">.R</a></center>
</td>
</tr>

<tr nosave>
<td nowrap nosave>Lab2a</td>

<td>Bioinformatics (anotation package)</td>

<td>
<center><a href="PDF/Lab2a.pdf">.pdf</a></center>
</td>

<td>
<center><a href="Rnw/Lab2a.Rnw">.Rnw</a></center>
</td>

<td>
<center><a href="RCode/Lab2a.R">.R</a></center>
</td>
</tr>

<tr nosave>
<td nowrap nosave>Lab2b</td>

<td>An Introduction to Some Graphics in Bioconductor</td>

<td>
<center><a href="PDF/Lab2b.pdf">.pdf</a></center>
</td>

<td>
<center><a href="Rnw/Lab2b.Rnw">.Rnw</a></center>
</td>

<td>
<center><a href="RCode/Lab2b.R">.R</a></center>
</td>
</tr>

<tr>
<td>Lab3a&nbsp;</td>

<td>Introduction to Bioconductor's marray Packages</td>

<td>
<center><a href="PDF/Lab3a.pdf">.pdf</a></center>
</td>

<td>
<center><a href="Rnw/Lab3a.Rnw">.Rnw</a></center>
</td>

<td>
<center><a href="RCode/Lab3a.R">.R</a></center>
</td>
</tr>

<tr>
<td>Lab3b&nbsp;</td>

<td>Introduction to the  affy package</td>

<td>
<center><a href="PDF/Lab3b.pdf">.pdf</a></center>
</td>

<td>
<center><a href="Rnw/Lab3b.Rnw">.Rnw</a></center>
</td>

<td>
<center><a href="RCode/Lab3b.R">.R</a></center>
</td>
</tr>

<tr>
<td>Lab4&nbsp;</td>

<td>Differential Gene Expression</td>

<td>
<center><a href="PDF/Lab4.pdf">.pdf</a></center>
</td>

<td>
<center><a href="Rnw/Lab4.Rnw">.Rnw</a></center>
</td>

<td>
<center><a href="RCode/Lab4.R">.R</a></center>
</td>
</tr>

<tr>
<td>Lab5&nbsp;</td>

<td>Cluster Analysis Using R and Bioconductor</td>

<td>
<center><a href="PDF/Lab5.pdf">.pdf</a></center>
</td>

<td>
<center><a href="Rnw/Lab5.Rnw">.Rnw</a></center>
</td>

<td>
<center><a href="RCode/Lab5.R">.R</a></center>
</td>
</tr>

<tr>
<td>Lab6&nbsp;</td>

<td>Classification Using R and Bioconductor</td>

<td>
<center><a href="PDF/Lab6.pdf">.pdf</a></center>
</td>

<td>
<center><a href="Rnw/Lab6.Rnw">.Rnw</a></center>
</td>

<td>
<center><a href="RCode/Lab6.R">.R</a></center>
</td>
</tr>

<tr>
<td>Lab7&nbsp;</td>

<td>Analyzing Microarray Data: From Images to List of Candidate Genes</td>

<td>
<center><a href="PDF/Lab7.pdf">.pdf</a></center>
</td>

<td>
<center><a href="Rnw/Lab7.Rnw">.Rnw</a></center>
</td>

<td>
<center><a href="RCode/Lab7.R">.R</a></center>
</td>
</tr>

<tr>
<td>Lab8&nbsp;</td>

<td>Application of Machine Learning to Microarray Data, SVM and friends</td>

<td>
<center><a href="PDF/SVM.pdf">.pdf</a></center>
</td>

<td>
<center><a href="Rnw/SVM.Rnw">.Rnw</a></center>
</td>

<td>
<center><a href="RCode/SVM.R">.R</a></center>
</td>
</tr>

<tr>
<td>Lab9&nbsp;</td>
<td>Lab 9: An Introduction to Wavelets</td>
<td>
<center><a href="PDF/Lab9.pdf">.pdf</a></center>
</td>

<td>
<center><a href="Rnw/Lab9.rnw">.Rnw</a></center>
</td>

<td>
<center><a href="RCode/Lab9.R">.R</a></center>
</td>
</tr>

<tr>
<td>Lab10&nbsp;</td>
<td>EFDR: Some Statistical Methods for the Identification of
Differentially Expressed Genes</td>
<td>
<center><a href="PDF/EFDR.pdf">.pdf</a></center>
</td>
<td>
<center><a href="Rnw/EFDR.Rnw">.Rnw</a></center>
</td>
<td>
<center><a href="RCode/EFDR.R">.R</a></center>
</td>
</tr>

<tr>
<td>Lab11&nbsp;</td>
<td>Lab 11: Penalized Logistic Regression</td>
<td>
<center><a href="PDF/Lab11.pdf">.pdf</a></center>
</td>
<td>
<center><a href="Rnw/Lab11.rnw">.Rnw</a></center>
</td>
<td>
<center><a href="RCode/Lab11.R">.R</a></center>
</td>
</tr>

<tr>
<td>Lab12&nbsp;</td>
<td>Lab 12: Dimension reduction in R</td>
<td>
<center><a href="PDF/Lab12.pdf">.pdf</a></center>
</td>
<td>
<center><a href="Rnw/Lab12.rnw">.Rnw</a></center>
</td>
<td>
<center><a href="RCode/Lab12.R">.R</a><center>
</center></center></td>
</tr>
</table>

<p>Milan course package (<a href="Milan_1.0.0.tar.gz">Linux/Unix</a>)
(<a href="Milan_1.0.zip">Windows</a>).
</p><p>You can step through the labs interactively using the vExplorer function
and GUI from the tkWidgets package. At the R prompt type
<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
library(tkWidgets)
<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
library(Milan)
<br />&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
vExplorer()
<br />and select the package Milan and the vignette for a given lab, e.g.,
lab1.Rnw.
</p><p><b><font color="#000099">Installing software</font></b>
</p><p>The packages used for this course require the use of <b><a href="http://www.bioconductor.org/downloadingR.html">R version
1.7.0-patched</a></b> or newer.
</p><p>A <b><a href="getMilan.R">download script</a></b> has been provided.  To
use, type
<b>'source("http://www.bioconductor.org/workshops/Milan/getMilan.R")'</b> in
your R session and then <b>'getMilan()'</b>.  There are a few arguments to
getMilan(), tho most users will not find them necessary:<br />
</p><ul>
<li> getAllDeps: If this is TRUE, will automatically attempt to
download any package dependencies, otherwise the user will be
prompted.  Default is TRUE.</li>
<li> destdir: This defines where to install the packages.  Defaults to
.libPaths()[1]</li>
<li>force: If TRUE, will install a package even if not all of its
dependencies were available.  Default is TRUE.</li>
<li>versForce: If TRUE, will install a binary package even if it was
built on a different version of R.  Default is TRUE.</li>
<li>method: The download method to use, same as in
download.packages().  Default is 'auto'.</li>
</ul>

<p>
<b><font color="#000099">Documentation</font></b>
</p><p><font color="#000000">Main <a href="http://www.r-project.org/">R</a>
website: R manuals, R News.</font>
<br /><font color="#000000">Main <a href="http://www.bioconductor.org">Bioconductor</a>
website: Bioconductor short course, vignettes, talks, publications.</font>
<br />&nbsp;
<br />&nbsp;
<br />&nbsp;

        </p></div>
    
        
        <div class="documentByLine">

  

  <span>
    Created by
  </span>

  <a href="http://cobra.fhcrc.org:12080/Plones/rgentlem/bioconductor/Members/admin">admin</a>

  

  <br />

  

  

  <span>
    Last modified
  </span>
  2005-07-01 01:37 PM

  

</div>
        

    


                    
                        <div class="discussion">

    

</div>
                    

                