

                    <a name="documentContent"></a>

                    

                    

                    
    


        <h1 class="documentFirstHeading">Practical DNA Microarray Analysis</h1>
        
        
    
        <div class="documentDescription"></div>

    
        
    
        <div class="plain">
            <div class="c2">
Heidelberg and Berlin 2003<br />
<p><a href="http://www.dkfz.de/biostatistics/benner.html">Axel
Benner</a><sup>1</sup>, <a href="http://www.dkfz-heidelberg.de/ibios_old/people/brors">Benedikt
Brors</a><sup>2</sup>, <a href="http://www.molgen.mpg.de/%7Eheydebre">Anja von
Heydebreck</a><sup>3</sup>, <a href="http://www.dkfz.de/abt0840/whuber">Wolfgang Huber</a><sup>4</sup>,
<a href="http://www.biometrie.uni-heidelberg.de/mitarbeiter/MansmannUlrich/home.htm">
Ulrich Mansmann</a><sup>5</sup>, <a href="http://www.molgen.mpg.de/%7Emarkowet">Florian
Markowetz</a><sup>3</sup>, <a href="http:///www.mpi-sb.mpg.de/~rahnenfj/">J&ouml;rg
Rahnenf&uuml;hrer</a><sup>6</sup>, <a href="http://www.molgen.mpg.de/%7Espang">Rainer
Spang</a><sup>3</sup><br />
<br /></p>
<p class="c1"><sup>1</sup> <a href="http://www.dkfz.de/biostatistics">German Cancer Research Center,
Heidelberg, Central Unit Biostatistics</a>;<br />
<sup>2</sup> <a href="http://www.dkfz.de/ibios">German Cancer
Research Center, Heidelberg, Dep. Intelligent Bioinformatics
Systems</a>;<br />
<sup>3</sup> <a href="http://cmb.molgen.mpg.de/">Max-Planck-Institute for Molecular
Genetics, Berlin, Dep. Computational Molecular Biology</a>;<br />
<sup>4</sup> <a href="http://www.dkfz.de/abt0840">German Cancer
Research Center, Heidelberg, Dep. Molecular Genome
Analysis</a>;<br />
<sup>5</sup> <a href="http://www.biometrie.uni-heidelberg.de/">Univ. of Heidelberg,
Institute for Medical Biometry</a>;<br />
<sup>6</sup> <a href="http://www.mpi-sb.mpg.de/">Max-Plack-Institute for Informatics,
Saarbr&uuml;cken</a>.</p>
<table cellspacing="5" width="650">
<tr>
<td class="bar" colspan="3">&nbsp;Monday</td>
</tr>
<tr>
<td width="100">09.00-10.30</td>
<td nowrap><a href="qcnorm.pdf">First analysis steps</a></td>
<td nowrap>Wolfgang Huber</td>
</tr>
<tr>
<td>11.00-12.30</td>
<td nowrap><a href="MultipleTesting.pdf">Differential gene
expression</a></td>
<td nowrap>Anja von Heydebreck</td>
</tr>
<tr>
<td>13.30-14.30</td>
<td nowrap><a href="R-firststeps.pdf">First steps in R</a></td>
<td nowrap>Florian Markowetz</td>
</tr>
<tr>
<td valign="top">14.45-17.00</td>
<td>Exercises: <a href="lymphoma.pdf">Exploring cDNA data</a></td>
<td nowrap valign="top">Wolfgang Huber</td>
</tr>
<tr>
<td class="bar" colspan="3">&nbsp;Tuesday</td>
</tr>
<tr>
<td>09.00-10.30</td>
<td nowrap><a href="Exploratory.pdf">Exploratory data
analysis</a></td>
<td nowrap>J&ouml;rg Rahnenf&uuml;hrer</td>
</tr>
<tr>
<td>10.30-12.00</td>
<td nowrap><a href="annotation_brors.pdf">Annotation</a></td>
<td nowrap>Benedikt Brors</td>
</tr>
<tr>
<td>13.30-14.30</td>
<td nowrap><a href="bioconductor.pdf">BioConductor</a></td>
<td nowrap>Wolfgang Huber</td>
</tr>
<tr>
<td>14.45-17.00</td>
<td nowrap>Exercises: <a href="estrogen.pdf">2x2 factorial
design</a></td>
<td nowrap>Wolfgang Huber</td>
</tr>
<tr>
<td class="bar" colspan="3">&nbsp;Wednesday</td>
</tr>
<tr>
<td>09.00-10.15</td>
<td nowrap><a href="MolecularDiagnosis.pdf">Molecular
Diagnosis</a></td>
<td nowrap>Rainer Spang</td>
</tr>
<tr>
<td nowrap>10.15-11.00</td>
<td nowrap><a href="svm.pdf">Classification by Support Vector
Machines</a></td>
<td nowrap>Florian Markowetz</td>
</tr>
<tr>
<td>11.00-12.00</td>
<td nowrap><a href="resampling.pdf">Resampling and the
bootstrap</a></td>
<td nowrap>Axel Benner</td>
</tr>
<tr>
<td valign="top">13.30-17.00</td>
<td nowrap valign="top">Exercises: <a href="classif-exercises.pdf">classification</a></td>
<td nowrap valign="top">Florian Markowetz</td>
</tr>
<tr>
<td class="bar" colspan="3">&nbsp;Thursday</td>
</tr>
<tr>
<td>09.00-11.00</td>
<td nowrap><a href="experimental-design.pdf">Design of microarray
experiments</a></td>
<td nowrap>Ulrich Mansmann</td>
</tr>
<tr>
<td>11.00-12.00</td>
<td nowrap><a href="model-selection.pdf">Model Assessment and
Selection</a></td>
<td nowrap>Axel Benner</td>
</tr>
<tr>
<td>12.00-12.30</td>
<td nowrap><a href="roc.pdf">Screening by ROC curve
analysis</a></td>
<td nowrap>Axel Benner</td>
</tr>
<tr>
<td valign="top">13.30-17.00</td>
<td nowrap valign="top">Exercises: <a href="modelsel-exercise.pdf">Model selection</a>, <a href="Exercises_Bootstrap.pdf">Bootstrap</a></td>
<td nowrap valign="top">Ulrich Mansmann<br />
Axel Benner</td>
</tr>
</table>
<p>&copy; Copyright 2003, All rights reserved</p>
</div>
<p><b>Datasets</b> used in the afternoon exercises can be found in
the Lab Materials section of www.bioconductor.org.</p>
<p><b>Course format:</b><br />
ca. 15 students from laboratories that work with microarrays. In
the mornings, theory lectures; in the afternoons,
hands-on-experience in computer-classes with real data sets (public
data, or students' own data), using R/Bioconductor.</p>
<p><b>Course dates:</b><br />
Berlin, Jan 27-30; Heidelberg Mar 24-27; Heidelberg Oct 07-10;
Berlin Nov 10-13.</p>
<p><b>Course homepage:</b><br />
at <a href="http://compdiag.molgen.mpg.de/ngfn/">http://compdiag.molgen.mpg.de/ngfn/</a>.
There will be further courses in 2004 which can be found on the
same website.</p>


        </div>
    
        
        <div class="documentByLine">

  

  <span>
    Created by
  </span>

  <a href="http://cobra.fhcrc.org:12080/Plones/rgentlem/bioconductor/Members/admin">admin</a>

  

  <br />

  

  

  <span>
    Last modified
  </span>
  2005-07-01 11:37 AM

  

</div>
        

    


                    
                        <div class="discussion">

    

</div>
                    

                