

                    <a name="documentContent"></a>

                    

                    

                    
    


        <h1 class="documentFirstHeading">Bioconductor short course </h1>
        
        
    
        <div class="documentDescription"></div>

    
        
    
        <div class="plain">
            <p>&nbsp;</p>
<div class="c3">
<p><b><span class="c1">A short course on</span><br /></b>
<span class="c1"><span class="c2">Computational and Statistical
Aspects of Microarray Analysis</span></span><br />
<br />
Kitasato University<br />
July 14-25, 2003<br />
<br />
Lecturer:<br />
<a href="http://gentleman.fhcrc.org">Robert
Gentleman</a></p>
</div>
<p><br /></p>
<div class="c3">
<p class="c4"><span class="c1">Schedule of Topics</span></p>
</div>
<table border>
<tr>
<td></td>
<td><b>Monday, July 14</b></td>
<td><b>Tuesday, July 15</b></td>
<td><b>Wednesday, July 16</b></td>
<td><b>Thursday, July 17</b></td>
<td><b>Friday, July 18</b></td>
</tr>
<tr>
<td><b>Lecture</b></td>
<td><a href="../Milan/Lectures/GenBio.pdf">Introduction to Genome
Biology</a></td>
<td><a href="../Milan/Lectures/MilanR.pdf">R/S Programming
Techniques</a></td>
<td><a href="../Milan/Lectures/OligoArrays.pdf">DNA Microarray
Data, Oligonucleotide Arrays</a></td>
<td><a href="../Milan/Lectures/MilanBioC.pdf">Introduction to
Bioconductor</a></td>
<td><a href="Lectures/ExpDesign.pdf">Experimental Design</a> and
<a href="Lectures/ExpDesPaper.ps">Experimental Design
Paper</a></td>
</tr>
<tr>
<td><b>Labs</b></td>
<td><b>Lecture Con't</b></td>
<td><b>Lab1, Lab2a</b></td>
<td><b>Lab2b</b></td>
<td><b>Lab3b</b></td>
<td><b>Estrogen</b></td>
</tr>
</table>
<table border>
<tr>
<td></td>
<td><b>Monday, July 21</b></td>
<td><b>Tuesday, July 22</b></td>
<td><b>Wednesday, July 23</b></td>
<td><b>Thursday, July 24</b></td>
<td><b>Friday, July 25</b></td>
</tr>
<tr>
<td><b>Lecture</b></td>
<td><b>Holiday, No Lecture</b></td>
<td><a href="../Milan/Lectures/Filtering.pdf">Microarray
Experiments</a> and <a href="../Milan/Lectures/dist.pdf">Distances
and Expression Measures</a></td>
<td><a href="../Milan/Lectures/MachineLearning.pdf">Some Things
Every Biologist Should Know About Machine Learning</a></td>
<td><a href="../Milan/Lectures/classif.pdf">Classification in
Microarray Experiments</a></td>
<td><a href="../Milan/Lectures/cluster.pdf">Clustering in
Microarray Experiments</a></td>
</tr>
<tr>
<td><b>Labs</b></td>
<td><b>Holiday</b></td>
<td><b>Lab4</b></td>
<td><b>Lab5</b></td>
<td><b>Lab6, Lab7</b></td>
<td><b>Lab8</b></td>
</tr>
</table>
<p><b><span class="c1">Lab materials</span></b><br />
&nbsp;</p>
<table border nosave="" width="75%">
<tr nosave="">
<td nowrap nosave="">Lab1&nbsp;</td>
<td>Bioconductor Basics</td>
<td nosave="">
<div class="c3"><a href="../Milan/PDF/Lab1.pdf">.pdf</a></div>
</td>
<td>
<div class="c3"><a href="../Milan/Rnw/Lab1.Rnw">.Rnw</a></div>
</td>
<td>
<div class="c3"><a href="../Milan/RCode/Lab1.R">.R</a></div>
</td>
</tr>
<tr nosave="">
<td nowrap nosave="">Lab2a</td>
<td>Bioinformatics (anotation package)</td>
<td>
<div class="c3"><a href="../Milan/PDF/Lab2a.pdf">.pdf</a></div>
</td>
<td>
<div class="c3"><a href="../Milan/Rnw/Lab2a.Rnw">.Rnw</a></div>
</td>
<td>
<div class="c3"><a href="../Milan/RCode/Lab2a.R">.R</a></div>
</td>
</tr>
<tr nosave="">
<td nowrap nosave="">Lab2b</td>
<td>An Introduction to Some Graphics in Bioconductor</td>
<td>
<div class="c3"><a href="../Milan/PDF/Lab2b.pdf">.pdf</a></div>
</td>
<td>
<div class="c3"><a href="../Milan/Rnw/Lab2b.Rnw">.Rnw</a></div>
</td>
<td>
<div class="c3"><a href="../Milan/RCode/Lab2b.R">.R</a></div>
</td>
</tr>
<tr>
<td>Lab3a&nbsp;</td>
<td>Introduction to Bioconductor's marray Packages</td>
<td>
<div class="c3"><a href="../Milan/PDF/Lab3a.pdf">.pdf</a></div>
</td>
<td>
<div class="c3"><a href="../Milan/Rnw/Lab3a.Rnw">.Rnw</a></div>
</td>
<td>
<div class="c3"><a href="../Milan/RCode/Lab3a.R">.R</a></div>
</td>
</tr>
<tr>
<td>Lab3b&nbsp;</td>
<td>Introduction to the affy package</td>
<td>
<div class="c3"><a href="../Milan/PDF/Lab3b.pdf">.pdf</a></div>
</td>
<td>
<div class="c3"><a href="../Milan/Rnw/Lab3b.Rnw">.Rnw</a></div>
</td>
<td>
<div class="c3"><a href="../Milan/RCode/Lab3b.R">.R</a></div>
</td>
</tr>
<tr>
<td>Lab4&nbsp;</td>
<td>Differential Gene Expression</td>
<td>
<div class="c3"><a href="../Milan/PDF/Lab4.pdf">.pdf</a></div>
</td>
<td>
<div class="c3"><a href="../Milan/Rnw/Lab4.Rnw">.Rnw</a></div>
</td>
<td>
<div class="c3"><a href="../Milan/RCode/Lab4.R">.R</a></div>
</td>
</tr>
<tr>
<td>Lab5&nbsp;</td>
<td>Cluster Analysis Using R and Bioconductor</td>
<td>
<div class="c3"><a href="../Milan/PDF/Lab5.pdf">.pdf</a></div>
</td>
<td>
<div class="c3"><a href="../Milan/Rnw/Lab5.Rnw">.Rnw</a></div>
</td>
<td>
<div class="c3"><a href="../Milan/RCode/Lab5.R">.R</a></div>
</td>
</tr>
<tr>
<td>Lab6&nbsp;</td>
<td>Classification Using R and Bioconductor</td>
<td>
<div class="c3"><a href="../Milan/PDF/Lab6.pdf">.pdf</a></div>
</td>
<td>
<div class="c3"><a href="../Milan/Rnw/Lab6.Rnw">.Rnw</a></div>
</td>
<td>
<div class="c3"><a href="../Milan/RCode/Lab6.R">.R</a></div>
</td>
</tr>
<tr>
<td>Lab7&nbsp;</td>
<td>Analyzing Microarray Data: From Images to List of Candidate
Genes</td>
<td>
<div class="c3"><a href="../Milan/PDF/Lab7.pdf">.pdf</a></div>
</td>
<td>
<div class="c3"><a href="../Milan/Rnw/Lab7.Rnw">.Rnw</a></div>
</td>
<td>
<div class="c3"><a href="../Milan/RCode/Lab7.R">.R</a></div>
</td>
</tr>
<tr>
<td>Lab8&nbsp;</td>
<td>Application of Machine Learning to Microarray Data, SVM and
friends</td>
<td>
<div class="c3"><a href="../Milan/PDF/SVM.pdf">.pdf</a></div>
</td>
<td>
<div class="c3"><a href="../Milan/Rnw/SVM.Rnw">.Rnw</a></div>
</td>
<td>
<div class="c3"><a href="../Milan/RCode/SVM.R">.R</a></div>
</td>
</tr>
</table>
<p>Milan course package (<a href="../Milan/Milan_1.0.0.tar.gz">Linux/Unix</a>) (<a href="../Milan/Milan_1.0.zip">Windows</a>).</p>
<p>You can step through the labs interactively using the vExplorer
function and GUI from the tkWidgets package. At the R prompt
type<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
library(tkWidgets)<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
library(Milan)<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
vExplorer()<br />
and select the package Milan and the vignette for a given lab,
e.g., lab1.Rnw.</p>
<p class="c4"><span class="c1">Installing software</span></p>
<p>The packages used for this course require the use of <b><a href="/downloadingR.html">R version 1.7.1</a></b> or newer.</p>
<p>A <b><a href="../Milan/getMilan.R">download script</a></b> has
been provided. To use, type
<b>'source("/workshops/Milan/getMilan.R")'</b> in your R session
and then <b>'getMilan()'</b>. There are a few arguments to
getMilan(), tho most users will not find them necessary:<br /></p>
<ul>
<li>getAllDeps: If this is TRUE, will automatically attempt to
download any package dependencies, otherwise the user will be
prompted. Default is TRUE.</li>
<li>destdir: This defines where to install the packages. Defaults
to .libPaths()[1]</li>
<li>force: If TRUE, will install a package even if not all of its
dependencies were available. Default is TRUE.</li>
<li>versForce: If TRUE, will install a binary package even if it
was built on a different version of R. Default is TRUE.</li>
<li>method: The download method to use, same as in
download.packages(). Default is 'auto'.</li>
</ul>
<p>Alternatively, you can download the individual packages
here:<br /></p>
<ul>
<li>affy: <a href="/repository/release1.2/package/Source/affy_1.2.27.tar.gz">Source</a>,
<a href="/repository/release1.2/package/Win32/affy_1.2.27.zip">Win32</a></li>
<li>affydata: <a href="/repository/release1.2/package/Source/affydata_1.0.tar.gz">Source</a>,
<a href="/repository/release1.2/package/Win32/affydata_1.0.zip">Win32</a></li>
<li>Annotate: <a href="/repository/release1.2/package/Source/annotate_1.0.21.tar.gz">Source</a>,
<a href="/repository/release1.2/package/Win32/annotate_1.0.21.zip">Win32</a></li>
<li>Biobase: <a href="/repository/release1.2/package/Source/Biobase_1.3.22.tar.gz">Source</a>,
<a href="/repository/release1.2/package/Win32/Biobase_1.3.22.zip">Win32</a></li>
<li>bioclabs: <a href="/repository/Courses/bioclabs_0.1.tar.gz">Source</a>, <a href="/repository/Courses/bioclabs_0.1.zip">Win32</a></li>
<li>cluster: <a href="/CRANrepository/cluster_1.7.2.tar.gz">Source</a>, <a href="/CRANrepository/cluster.zip">Win32</a></li>
<li>DynDoc: <a href="/repository/release1.2/package/Source/DynDoc_1.0.tar.gz">Source</a>,
<a href="/repository/release1.2/package/Win32/DynDoc_1.0.zip">Win32</a></li>
<li>e1071: <a href="/CRANrepository/e1071_1.3-11.tar.gz">Source</a>, <a href="/CRANrepository/e1071.zip">Win32</a></li>
<li>edd: <a href="/repository/release1.2/package/Source/edd_1.2.0.tar.gz">Source</a>,
<a href="/repository/release1.2/package/Win32/edd_1.2.0.zip">Win32</a></li>
<li>ellipse: <a href="/CRANrepository/ellipse_0.2-10.tar.gz">Source</a>, <a href="/CRANrepository/ellipse.zip">Win32</a></li>
<li>genefilter: <a href="/repository/release1.2/package/Source/genefilter_1.0.5.tar.gz">Source</a>,
<a href="/repository/release1.2/package/Win32/gene_1.0.5.zip">Win32</a></li>
<li>geneplotter: <a href="/repository/release1.2/package/Source/geneplotter_1.0.7.tar.gz">Source</a>,
<a href="/repository/release1.2/package/Win32/geneplotter_1.0.7.zip">Win32</a></li>
<li>GO: <a href="/data/dataRepos/GO_1.2.6.tar.gz">Source</a>,
<a href="/data/dataRepos/GO_1.2.6.zip">Win32</a></li>
<li>golubEsets: <a href="/repository/Courses/golubEsets_1.0.tar.gz">Source</a>, <a href="/repository/Courses/golubEsets_1.0.zip">Win32</a></li>
<li>hgu95av2: <a href="/data/dataRepos/hgu95av2_1.2.6.tar.gz">Source</a>, <a href="/data/dataRepos/hgu95av2_1.2.6.zip">Win32</a></li>
<li>hu6800: <a href="/data/dataRepos/hu6800_1.2.6.tar.gz">Source</a>, <a href="/data/dataRepos/hu6800_1.2.6.zip">Win32</a></li>
<li>KEGG: <a href="/data/dataRepos/KEGG_1.2.6.tar.gz">Source</a>,
<a href="/data/dataRepos/KEGG_1.2.6.zip">Win32</a></li>
<li>mAffyu95a: <a href="/repository/Courses/mAffyu95A_1.0.tar.gz">Source</a></li>
<li>marrayNorm: <a href="/repository/release1.2/package/Source/marrayNorm_1.1.3.tar.gz">Source</a>,
<a href="/repository/release1.2/package/Win32/marrayNorm_1.1.3.zip">Win32</a></li>
<li>Milan: <a href="/repository/Courses/Milan_1.0.tar.gz">Source</a>, <a href="/repository/Courses/Milan_1.0.zip">Win32</a></li>
<li>multtest: <a href="/repository/release1.2/package/Source/multtest_1.0.2.tar.gz">Source</a>,
<a href="/repository/release1.2/package/Win32/multtest_1.0.2.zip">Win32</a></li>
<li>randomForest: <a href="/CRANrepository/randomForest_3.9-6.tar.gz">Source</a>, <a href="/CRANrepository/randomForest.zip">Win32</a></li>
<li>tkWidgets: <a href="/repository/release1.2/package/Source/tkWidgets_1.2.20.tar.gz">Source</a>,
<a href="/repository/release1.2/package/Win32/tkWidgets_1.2.20.zip">Win32</a></li>
</ul>
<p class="c4"><span class="c1">Documentation</span></p>
<p><span class="c5">Main <a href="http://www.r-project.org/">R</a>
website: R manuals, R News.</span><br />
<span class="c5">Main <a href="/">Bioconductor</a> website:
Bioconductor short course, vignettes, talks,
publications.</span><br />
&nbsp;<br />
&nbsp;<br />
&nbsp;</p>


        </div>
    
        
        <div class="documentByLine">

  

  <span>
    Created by
  </span>

  <a href="http://cobra.fhcrc.org:12080/Plones/rgentlem/bioconductor/Members/admin">admin</a>

  

  <br />

  

  

  <span>
    Last modified
  </span>
  2005-07-01 04:36 PM

  

</div>
        

    


                    
                        <div class="discussion">

    

</div>
                    

                