<!--
%\VignetteEngine{knitr::knitr}
%\VignetteIndexEntry{}
-->

<!--
To deploy this to the website, do the following:
library(knitr)
render_markdown(strict=TRUE)
knit("2014_July.Rmd")

This will produce an .md file that can be used 
on the website. The {:toc} and {:no_toc} tags
will be rendered by nanoc, the tool that produces
the website.
-->

{::options parse_block_html="true" /}

# *Bioconductor* Newsletter
{:.no_toc}

posted by [Valerie Obenchain](mailto:vobencha@fhcrc.org), July 2014

## Contents
{:.no_toc}

* Table of contents will replace this text.
{:toc}


## Contributed Article: Kasper Hansen TBD 


## Software Infrastructure

### `S4 Vectors` package

In April Herv&eacute; started to split out the low level functions 
from `IRanges` and move them to the new `S4Vectors` package.
`IRanges` had grown to 90 classes, 157 generics and 844 methods
and was becoming difficult to maintain. The plan is to move
code that does not involve ranges which includes the `Vector` and
`List` virtual classes, and `DataFrame`, `Rle` and `Hits` 
concrete classes. This is a work in progress and estimated to be 
about 30% done.

### `BSgenome` packages 2bit conversion

This past Spring the BSgenome data packages were regenerated to use 
the UCSC 2bit format to store the sequences on disk. The previous on 
disk storage format was 1 .rda file per chromosome which was well 
suited for working one chromosomes at a time. With the onset of 
range-based queries (e.g., extract all exons in a genome) there was 
a need to access many smaller regions across all chromosomes. This was 
problematic for the .rda format as it did not allow for random access;
the full chromosome must be loaded into memory to extract any region, 
regardless of size. 

We needed a new format that allowed fast random-access to disk 
files. Several options were considered and ultimately the 2bit 
format was chosen. For genomes that contain letters other than As, 
Cs, Gs, Ts, or Ns (e.g. hg17, hg18, GRCh38, Ecoli, TAIR.04232008, and 
TAIR.TAIR9), the 2bit format cannot be used out-of-the-box. These 
packages were generated with the old .rda storage format.

The 2bit format is available in devel only (Bioconductor 3.0). 
Herv&eacute; and Martin worked on this project with thanks to 
Michael for supporting the 2bit format in `rtracklayer`.

### Git-SVN Bridge

The `Bioconductor` project uses a Subversion (SVN) source control
system. SVN is effective for version control but does not offer 
social coding features such as GitHub's issue tracking, pull 
requests or ease in granting permissions.

In response to popular request, Dan created the Git-SVN Bridge 
which allows Github repositories to sync with the `Bioconductor` 
SVN repository. Commits made in SVN are propagated to GitHub and 
vice versa. The service has been well recieved with 73 bridges 
created as of June 2014.

To create a bridge see the Git-SVN Bridge 
[HOWTO](http://bioconductor.org/developers/how-to/git-svn/).

### pileups

Nate has been working on a `pileup()` function which computes
pileup statistics in BAM files. Design goals were versatile 
record filtering and flexible presentation of results for 
downstream analyses. Filtering is achieved through the 
`ScanBamParam` and `PileupParam` objects; output is a 
`data.frame` with variable columns based on the filtering applied. 

`pileup()` is available in the `Rsamtools` package in devel. Other 
`Bioconductor` packages that offer pileup-like functions include `gmapR` 
(bam_tally), `deepSNV` (bam2R) and `Rsubread` (featureCounts). All 
have slightly different input requirements and output formats, see 
the man pages for details.


## Education and Outreach

### Web site re-design

The `Bioconductor` landing page was updated with what we hope
is a more intuitive and user friendly interface. The 'Install', 
'Learn', 'Use' and 'Develop' fields organize resources for
the novice through the advanced developer.

Have a look at the new 
[design](http://www.bioconductor.org/).

### biocViews

Sonali continued her work on biocViews this quarter and added a new
function, `recommendBiocViews()`, to the `biocViews` package. 
Based on words in the DESCRIPTION, man pages and vignette, 
the function returns recommended terms and identifies invalid 
ones. 

`recommendBiocViews()` has been incorporated into the Single Package 
Builder and new package authors are encouraged to run it before submitting 
a package. Remember that biocViews are case-sensitive and branch-sensitive 
(i.e., terms for an annotation data package must come from the AnnotationData 
branch of biocViews). 

Sonali distributed recommended views for all devel software packages
to the mailing list in June. For the complete list see this
[post](https://stat.ethz.ch/pipermail/bioc-devel/2014-June/005854.html).

### Instructional videos

We are looking into short, single-topic videos as an interactive complement 
to the traditional vignettes and workflows.

The plan is to create a series of 5 minute videos that encapsulate a 
HOWTO skill or overview a project aspect. You can tour the website with 
Dan or do a 'quick start' with Martin's overview of key packages and classes.
Watch Marc slice and dice an AnnotationDb object or read Bam and VCF files 
with Sonali and Valerie. 

Sonali and Martin have led this effort and plan to unveil the first videos 
at BioC 2014 in Boston.


## Build System

### Branching the experimental data repository

Historically, the `Bioconductor` software packages were the only repository 
with a distinct branch for each release. Experimental data and annotation packages 
had a trunk with no branches.

Starting with the Spring 2014 release a branch was created for the 
experimental data. The motivation was to allow software and experimental 
data packages to evolve together in release and devel build environments. 
It was often the case that updates to a software package broke the companion 
experimental data. Changes made to the experimental data were committed to
trunk and propagated to both release and devel builds creating 
incompatibilities in one place or the other.

A consequence of creating the new branch is the need to bump y of the x.y.z 
version at release time (as we do for software). Annotations remain the same 
with the single trunk repository and no y bump. 

### New Mac OS X Mavericks build machines

An `R` 3.1.0 binary for Mac OS X 10.9 (Mavericks) is now available from
R Core. This `R` as been built with Xcode 5 to leverage new compilers and 
functionalities in Mavericks not available in earlier OS X versions.

To provide compatible `Bioconductor` package binaries we needed new
build machines. Dan has configured two new Mavericks, 
one in [release](http://www.bioconductor.org/checkResults/devel/bioc-LATEST/)
(morelia) and one in 
[devel](http://www.bioconductor.org/checkResults/devel/bioc-LATEST/)
(oaxaca).

The introduction of Xcode and the CLANG compiler resulted in new 
errors for packages with C and C++ code. Nate and Dan spent many
hours troubleshooting with package authors and came up with a list 
of common problems and solutions. Lessons learned were 
distilled into the [C++/Mavericks Best Practices](**TODO**)
document.


## Quarterly Project Statistics

There were 86953 distinct IP downloads of `Bioconductor` software packages
over the past quarter (April - June). During this time 41 new software 
packages were accepted. A full summary of package download stats is
available [here](http://www.bioconductor.org/packages/stats/).

The web site saw approximately 119,000 visitors (26% increase from the 
previous year) from 180 countries, with the US, China, United Kingdom, 
Germany, and Canada at the head of the pack. 


## Resources, Courses and Conferences

### Data Analysis for Genome Biology (CSAMA)

This one week intensive course is offered each year in Brixen-Bressanone,
Italy and focuses on statistical and computational analysis of large-scale 
biological experiments. The course is intended for researchers with basic 
familiarity with the experimental technologies and who are interested in 
developing their own advanced data analyses.

Topics this year included RNASeq differential expression, variant calling 
and ChIP-Seq as well as the essentials of statistical testing, machine 
learning and visualization. Michael Lawrence presented a `Scalable 
Genomics` lab which covered topics of limiting resource consumption,
using iteration when appropriate, and scaling genome graphics. Much of
the material is currently press for the Journal of Statistical Science. 
**TODO** pre-print reference?

Materials from the current course are available on the 
[web](http://bioconductor.org/help/course-materials/2014/CSAMA2014).

### Community Resources

The new [Community Resources](http://bioconductor.org/help/community/) link
includes the book and lab from  MOOC: PH525x Data Analysis for Genomics.
This online course was offered by Rafael Irizarry and Michael Love through
the Department of Biostatistics at the Dana Farber Cancer Institute and 
Harvard School of Public Health. Course goals were to enable students to 
analyze and interpret data generated by modern genomics technology, 
specifically microarray and next generation sequencing. Applications 
included gene expression, association of genomic variants to disease, and 
measuring epigenetic marks.

Also on the Community page are links to YouTube videos made by community
members, tips on getting started with R/Bioconductor by Thomas Girke, analysis 
of 23andme data by Vince Buffalo and Sean Davis' R/Bioconductor blog. 

### BioC 2014

The annual meeting is in Boston this year (July 30 - August 1). See the
[web site](http://www.bioconductor.org/help/course-materials/2014/BioC2014/)
for a list of speakers and workshops.

Please send comments or questions to Valerie at 
[vobencha@fhcrc.org](vobencha@fhcrc.org).
